<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_75) on Wed Apr 15 07:10:57 AEST 2015 -->
<title>TranscriberAdaptor</title>
<meta name="date" content="2015-04-15">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="TranscriberAdaptor";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TranscriberAdaptor.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html" title="class in com.ojcoleman.ahni.transcriber"><span class="strong">Prev Class</span></a></li>
<li>Next Class</li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" target="_top">Frames</a></li>
<li><a href="TranscriberAdaptor.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">com.ojcoleman.ahni.transcriber</div>
<h2 title="Class TranscriberAdaptor" class="title">Class TranscriberAdaptor&lt;T extends <a href="../../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&gt;</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>com.ojcoleman.ahni.transcriber.TranscriberAdaptor&lt;T&gt;</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../com/anji/integration/Transcriber.html" title="interface in com.anji.integration">Transcriber</a>&lt;T&gt;</dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a>, <a href="../../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html" title="class in com.ojcoleman.ahni.transcriber">NEATTranscriberBain</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="strong">TranscriberAdaptor&lt;T extends <a href="../../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&gt;</span>
extends java.lang.Object
implements <a href="../../../../com/anji/integration/Transcriber.html" title="interface in com.anji.integration">Transcriber</a>&lt;T&gt;</pre>
<div class="block">A base class for neural network transcribers that may define neuron or synapse model parameters. Some settings are
 specific to HyperNEAT type encoding schemes.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#connectionWeightMax">connectionWeightMax</a></strong></code>
<div class="block">The maximum connection weight in the substrate.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#connectionWeightMin">connectionWeightMin</a></strong></code>
<div class="block">The minimum connection weight in the substrate.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#HYPERNEAT_CONNECTION_WEIGHT_MAX">HYPERNEAT_CONNECTION_WEIGHT_MAX</a></strong></code>
<div class="block">The maximum weight values in the substrate network.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#HYPERNEAT_CONNECTION_WEIGHT_MIN">HYPERNEAT_CONNECTION_WEIGHT_MIN</a></strong></code>
<div class="block">The minimum weight values in the substrate network.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelParamClassCount">neuronModelParamClassCount</a></strong></code>
<div class="block">The number of parameter collections defining neuron model classes/types.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelParamsMax">neuronModelParamsMax</a></strong></code>
<div class="block">The maximum values for each neuron model parameter.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelParamsMin">neuronModelParamsMin</a></strong></code>
<div class="block">The minimum values for each neuron model parameter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelParamsRange">neuronModelParamsRange</a></strong></code>
<div class="block">The value range for each neuron model parameter (neuronModelParamsMax - neuronModelParamsMin).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelParamsThreshold">neuronModelParamsThreshold</a></strong></code>
<div class="block">the minimum (possibly CPPN) output required to produce a non-zero parameter value.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelTypeCount">neuronModelTypeCount</a></strong></code>
<div class="block">The number of available neuron types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelTypeParam">neuronModelTypeParam</a></strong></code>
<div class="block">The name of the parameter in the neuron model that specifies the type of a neuron.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelTypes">neuronModelTypes</a></strong></code>
<div class="block">The allowable values for the <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelTypeParam"><code>neuronModelTypeParam</code></a> model parameter that specify the type of a neuron.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronParamNames">neuronParamNames</a></strong></code>
<div class="block">The names or labels for parameters in the neuron model.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronParamsEnabled">neuronParamsEnabled</a></strong></code>
<div class="block">Subclasses may use this to know whether neuron model parameters are enabled and available.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronTypesEnabled">neuronTypesEnabled</a></strong></code>
<div class="block">Subclasses may use this to know whether multiple neuron types are enabled and available.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL">SUBSTRATE_NEURON_MODEL</a></strong></code>
<div class="block">Property to specify which neuron model to use.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS">SUBSTRATE_NEURON_MODEL_PARAMS</a></strong></code>
<div class="block">Property to specify the available parameters for neurons.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES">SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</a></strong></code>
<div class="block">Optional property to allow specifying that instead of parameter values being defined individually for each
 neuron, there will be a set of parameter value collections which can be thought of as defining a class of neuron
 as determined by the specific parameter values (which will determine the behaviour of that class of neuron).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_MAX">SUBSTRATE_NEURON_MODEL_PARAMS_MAX</a></strong></code>
<div class="block">Property to specify maximum values for the parameters for each neuron.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_MIN">SUBSTRATE_NEURON_MODEL_PARAMS_MIN</a></strong></code>
<div class="block">Property to specify minimum values for the parameters for each neuron.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD">SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD</a></strong></code>
<div class="block">Property to specify the minimum CPPN output required to produce a non-zero parameter value.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_TYPES">SUBSTRATE_NEURON_MODEL_TYPES</a></strong></code>
<div class="block">Property to specify allowable neuron model types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL">SUBSTRATE_SYNAPSE_MODEL</a></strong></code>
<div class="block">Property to specify which synapse model to use.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM">SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</a></strong></code>
<div class="block">Property to specify which parameter in the synapse model will be set to 0 if the connection should not be enabled
 or plastic.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS">SUBSTRATE_SYNAPSE_MODEL_PARAMS</a></strong></code>
<div class="block">Property to specify the available parameters for synapses.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES">SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</a></strong></code>
<div class="block">Optional property to allow specifying that instead of parameter values being defined individually for each
 synapse, there will be a set of parameter value collections which can be thought of as defining a class of
 synapse as determined by the specific parameter values (which will determine the behaviour of that class of
 synapse).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX">SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX</a></strong></code>
<div class="block">Property to specify maximum values for the parameters for each synapse.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN">SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN</a></strong></code>
<div class="block">Property to specify minimum values for the parameters for each synapse.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD">SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD</a></strong></code>
<div class="block">Property to specify the minimum CPPN output required to produce a non-zero parameter value.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_TYPES">SUBSTRATE_SYNAPSE_MODEL_TYPES</a></strong></code>
<div class="block">Property to specify allowable synapse model types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseDisableParamName">synapseDisableParamName</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamClassCount">synapseModelParamClassCount</a></strong></code>
<div class="block">The number of parameter collections defining synapse model classes/types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsMax">synapseModelParamsMax</a></strong></code>
<div class="block">The maximum values for each synapse model parameter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsMin">synapseModelParamsMin</a></strong></code>
<div class="block">The minimum values for each synapse model parameter.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsRange">synapseModelParamsRange</a></strong></code>
<div class="block">The value range for each synapse model parameter (synapseModelParamsMax[i] - synapseModelParamsMin[i]).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsThreshold">synapseModelParamsThreshold</a></strong></code>
<div class="block">the minimum (possibly CPPN) output required to produce a non-zero parameter value.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeCount">synapseModelTypeCount</a></strong></code>
<div class="block">The number of available synapse types.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam">synapseModelTypeParam</a></strong></code>
<div class="block">The name of the parameter in the synapse model that specifies the type of a synapse.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypes">synapseModelTypes</a></strong></code>
<div class="block">The allowable values for the <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam"><code>synapseModelTypeParam</code></a> model parameter that specify the type of a synapse.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.lang.String[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamNames">synapseParamNames</a></strong></code>
<div class="block">The names or labels for parameters in the synapse model.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamsEnabled">synapseParamsEnabled</a></strong></code>
<div class="block">Subclasses may use this to know whether synapse model parameters are enabled and available.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseTypesEnabled">synapseTypesEnabled</a></strong></code>
<div class="block">Subclasses may use this to know whether multiple synapse types are enabled and available.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#TranscriberAdaptor()">TranscriberAdaptor</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getChromosomeInputNeuronCount()">getChromosomeInputNeuronCount</a></strong>()</code>
<div class="block">May be overridden to specify how many input neurons the Chromosome should encode.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getChromosomeOutputNeuronCount()">getChromosomeOutputNeuronCount</a></strong>()</code>
<div class="block">May be overridden to specify how many output neurons the Chromosome should encode.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getExtraGenesForInitialChromosome(com.anji.neat.NeatConfiguration)">getExtraGenesForInitialChromosome</a></strong>(<a href="../../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config)</code>
<div class="block">Returns a list of <a href="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised"><code>Allele</code></a>s, or more importantly <a href="../../../../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised"><code>Gene</code></a>s referenced by the Alleles, that this
 transcriber knows what to do with.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronModelParamsMax()">getNeuronModelParamsMax</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronModelParamsMax(int)">getNeuronModelParamsMax</a></strong>(int&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronModelParamsMin()">getNeuronModelParamsMin</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronModelParamsMin(int)">getNeuronModelParamsMin</a></strong>(int&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronParamClassCount()">getNeuronParamClassCount</a></strong>()</code>
<div class="block">Get the number of available neuron types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronParameterCount()">getNeuronParameterCount</a></strong>()</code>
<div class="block">Get the number of available neuron model parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronParameterName(int)">getNeuronParameterName</a></strong>(int&nbsp;index)</code>
<div class="block">Get the name of the specified neuron model parameter.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronParameterValues(org.jgapcustomised.Chromosome)">getNeuronParameterValues</a></strong>(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom)</code>
<div class="block">Gets the neuron parameter values for each parameter in each parameter class for the specified chromosome, in the
 format [class][param] (param order same as <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronParamNames"><code>neuronParamNames</code></a>).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronTypeCount()">getNeuronTypeCount</a></strong>()</code>
<div class="block">Get the number of available neuron types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getParameterValuesForCollection(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.integration.ParamCollection,%20double[][])">getParameterValuesForCollection</a></strong>(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom,
                               <a href="../../../../com/ojcoleman/ahni/integration/ParamCollection.html" title="class in com.ojcoleman.ahni.integration">ParamCollection</a>&nbsp;collection,
                               double[][]&nbsp;values)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseModelParamsMax()">getSynapseModelParamsMax</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseModelParamsMax(int)">getSynapseModelParamsMax</a></strong>(int&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseModelParamsMin()">getSynapseModelParamsMin</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseModelParamsMin(int)">getSynapseModelParamsMin</a></strong>(int&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseParamClassCount()">getSynapseParamClassCount</a></strong>()</code>
<div class="block">Get the number of available synapse types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseParameterCount()">getSynapseParameterCount</a></strong>()</code>
<div class="block">Get the number of available synapse model parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseParameterName(int)">getSynapseParameterName</a></strong>(int&nbsp;index)</code>
<div class="block">Get the name of the specified synapse model parameter.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseParameterValues(org.jgapcustomised.Chromosome)">getSynapseParameterValues</a></strong>(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom)</code>
<div class="block">Gets the synapse parameter values for each parameter in each parameter class the specified chromosome, in the
 format [class][param] (param order same as <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamNames"><code>synapseParamNames</code></a>).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseTypeCount()">getSynapseTypeCount</a></strong>()</code>
<div class="block">Get the number of available synapse types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#init(com.ojcoleman.ahni.hyperneat.Properties)">init</a></strong>(<a href="../../../../com/ojcoleman/ahni/hyperneat/Properties.html" title="class in com.ojcoleman.ahni.hyperneat">Properties</a>&nbsp;props)</code>
<div class="block">This method should be called from overriding methods.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</a></code></td>
<td class="colLast"><code><strong><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#transcribe(org.jgapcustomised.Chromosome,%20T,%20java.util.Map)">transcribe</a></strong>(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c,
          <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</a>&nbsp;substrate,
          java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</code>
<div class="block">Sub-classes can override this method to convert the genotype to a phenotype, using an existing substrate object
 to allow performance gains by avoiding destroying and creating large objects or arrays, and optionally providing
 additional options.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_com.anji.integration.Transcriber">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;com.anji.integration.<a href="../../../../com/anji/integration/Transcriber.html" title="interface in com.anji.integration">Transcriber</a></h3>
<code><a href="../../../../com/anji/integration/Transcriber.html#getPhenotypeClass()">getPhenotypeClass</a>, <a href="../../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a>, <a href="../../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome,%20T)">transcribe</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="HYPERNEAT_CONNECTION_WEIGHT_MIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HYPERNEAT_CONNECTION_WEIGHT_MIN</h4>
<pre>public static final&nbsp;java.lang.String HYPERNEAT_CONNECTION_WEIGHT_MIN</pre>
<div class="block">The minimum weight values in the substrate network. If this is not specified then the negated value of
 HYPERNEAT_CONNECTION_WEIGHT_MAX will be used.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.HYPERNEAT_CONNECTION_WEIGHT_MIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="HYPERNEAT_CONNECTION_WEIGHT_MAX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>HYPERNEAT_CONNECTION_WEIGHT_MAX</h4>
<pre>public static final&nbsp;java.lang.String HYPERNEAT_CONNECTION_WEIGHT_MAX</pre>
<div class="block">The maximum weight values in the substrate network.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.HYPERNEAT_CONNECTION_WEIGHT_MAX">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL</pre>
<div class="block">Property to specify which neuron model to use. Allowable values are implementation dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL_PARAMS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL_PARAMS</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL_PARAMS</pre>
<div class="block">Property to specify the available parameters for neurons. Allowable values are implementation dependent. If
 HyperNEAT is being used then the CPPN may have an output for each parameter.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL_PARAMS">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</pre>
<div class="block">Optional property to allow specifying that instead of parameter values being defined individually for each
 neuron, there will be a set of parameter value collections which can be thought of as defining a class of neuron
 as determined by the specific parameter values (which will determine the behaviour of that class of neuron). Each
 neuron should then reference one of these collections/classes and have its parameters set accordingly. If this is
 set to 0 or not specified then each neuron should be assigned parameters values directly (e.g. via an output for
 each parameter from a CPPN in a HyperNEAT encoding scheme), if this is set to a value greater than 0 then it
 should specify the number of classes.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL_PARAMS_MIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL_PARAMS_MIN</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL_PARAMS_MIN</pre>
<div class="block">Property to specify minimum values for the parameters for each neuron. Allowable values are implementation
 dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL_PARAMS_MIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL_PARAMS_MAX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL_PARAMS_MAX</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL_PARAMS_MAX</pre>
<div class="block">Property to specify maximum values for the parameters for each neuron. Allowable values are implementation
 dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL_PARAMS_MAX">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD</pre>
<div class="block">Property to specify the minimum CPPN output required to produce a non-zero parameter value. Actual parameter
 values will scale from [0, ann.transcriber.neuron.model.params.max] or [ann.transcriber.neuron.model.params.min,
 0].</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_NEURON_MODEL_TYPES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_NEURON_MODEL_TYPES</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_NEURON_MODEL_TYPES</pre>
<div class="block">Property to specify allowable neuron model types. Typically: the CPPN only needs to specify the type to use if
 there is more than one type; if there are only two neuron types then a single CPPN output will be used to specify
 which to use, otherwise a "whichever CPPN output has the highest output value" encoding will be used to specify
 the type. Allowable values are implementation dependent, however by default it is assumed that a parameter name
 followed by one or more integer values to specify the allowable types is used (this assumes that the model is
 set-up with a parameter that can take on integer values to decide what type the model is).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_NEURON_MODEL_TYPES">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL</pre>
<div class="block">Property to specify which synapse model to use. Allowable values are implementation dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_PARAMS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_PARAMS</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_PARAMS</pre>
<div class="block">Property to specify the available parameters for synapses. Allowable values are implementation dependent. If
 HyperNEAT is being used then the CPPN may have an output for each parameter.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_PARAMS">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</pre>
<div class="block">Optional property to allow specifying that instead of parameter values being defined individually for each
 synapse, there will be a set of parameter value collections which can be thought of as defining a class of
 synapse as determined by the specific parameter values (which will determine the behaviour of that class of
 synapse). Each synapse should then reference one of these collections/classes and have its parameters set
 accordingly. If this is set to 0 or not specified then each synapse should be assigned parameters values directly
 (e.g. via an output for each parameter from a CPPN in a HyperNEAT encoding scheme), if this is set to a value
 greater than 0 then it should specify the number of classes.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN</pre>
<div class="block">Property to specify minimum values for the parameters for each synapse. Allowable values are implementation
 dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX</pre>
<div class="block">Property to specify maximum values for the parameters for each synapse. Allowable values are implementation
 dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD</pre>
<div class="block">Property to specify the minimum CPPN output required to produce a non-zero parameter value. Actual parameter
 values will scale from [0, ann.transcriber.neuron.model.params.max] or [ann.transcriber.neuron.model.params.min,
 0].</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</pre>
<div class="block">Property to specify which parameter in the synapse model will be set to 0 if the connection should not be enabled
 or plastic. This is typically applied to a "learning rate" parameter. Usage and allowable values are
 implementation dependent. The criteria for whether a synapse should be enabled is implementation dependent.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBSTRATE_SYNAPSE_MODEL_TYPES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBSTRATE_SYNAPSE_MODEL_TYPES</h4>
<pre>public static final&nbsp;java.lang.String SUBSTRATE_SYNAPSE_MODEL_TYPES</pre>
<div class="block">Property to specify allowable synapse model types. Typically: the CPPN only needs to specify the type to use if
 there is more than one type; if there are only two synapse types then a single CPPN output will be used to
 specify which to use, otherwise a "whichever CPPN output has the highest output value" encoding will be used to
 specify the type. Allowable values are implementation dependent, however by default it is assumed that a
 parameter name followed by one or more integer values to specify the allowable types is used (this assumes that
 the model is set-up with a parameter that can take on integer values to decide what type the model is).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#com.ojcoleman.ahni.transcriber.TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_TYPES">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="connectionWeightMin">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>connectionWeightMin</h4>
<pre>protected&nbsp;double connectionWeightMin</pre>
<div class="block">The minimum connection weight in the substrate.</div>
</li>
</ul>
<a name="connectionWeightMax">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>connectionWeightMax</h4>
<pre>protected&nbsp;double connectionWeightMax</pre>
<div class="block">The maximum connection weight in the substrate.</div>
</li>
</ul>
<a name="neuronTypesEnabled">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronTypesEnabled</h4>
<pre>protected&nbsp;boolean neuronTypesEnabled</pre>
<div class="block">Subclasses may use this to know whether multiple neuron types are enabled and available.</div>
</li>
</ul>
<a name="synapseTypesEnabled">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseTypesEnabled</h4>
<pre>protected&nbsp;boolean synapseTypesEnabled</pre>
<div class="block">Subclasses may use this to know whether multiple synapse types are enabled and available.</div>
</li>
</ul>
<a name="neuronParamsEnabled">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronParamsEnabled</h4>
<pre>protected&nbsp;boolean neuronParamsEnabled</pre>
<div class="block">Subclasses may use this to know whether neuron model parameters are enabled and available.</div>
</li>
</ul>
<a name="synapseParamsEnabled">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseParamsEnabled</h4>
<pre>protected&nbsp;boolean synapseParamsEnabled</pre>
<div class="block">Subclasses may use this to know whether synapse model parameters are enabled and available.</div>
</li>
</ul>
<a name="neuronParamNames">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronParamNames</h4>
<pre>protected&nbsp;java.lang.String[] neuronParamNames</pre>
<div class="block">The names or labels for parameters in the neuron model.</div>
</li>
</ul>
<a name="synapseParamNames">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseParamNames</h4>
<pre>protected&nbsp;java.lang.String[] synapseParamNames</pre>
<div class="block">The names or labels for parameters in the synapse model.</div>
</li>
</ul>
<a name="neuronModelParamClassCount">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelParamClassCount</h4>
<pre>protected&nbsp;int neuronModelParamClassCount</pre>
<div class="block">The number of parameter collections defining neuron model classes/types.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES"><code>SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</code></a></dd></dl>
</li>
</ul>
<a name="synapseModelParamClassCount">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelParamClassCount</h4>
<pre>protected&nbsp;int synapseModelParamClassCount</pre>
<div class="block">The number of parameter collections defining synapse model classes/types.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES"><code>SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</code></a></dd></dl>
</li>
</ul>
<a name="neuronModelParamsMin">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelParamsMin</h4>
<pre>protected&nbsp;double[] neuronModelParamsMin</pre>
<div class="block">The minimum values for each neuron model parameter.</div>
</li>
</ul>
<a name="neuronModelParamsMax">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelParamsMax</h4>
<pre>protected&nbsp;double[] neuronModelParamsMax</pre>
<div class="block">The maximum values for each neuron model parameter.</div>
</li>
</ul>
<a name="neuronModelParamsRange">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelParamsRange</h4>
<pre>protected&nbsp;double[] neuronModelParamsRange</pre>
<div class="block">The value range for each neuron model parameter (neuronModelParamsMax - neuronModelParamsMin).</div>
</li>
</ul>
<a name="neuronModelParamsThreshold">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelParamsThreshold</h4>
<pre>protected&nbsp;double[] neuronModelParamsThreshold</pre>
<div class="block">the minimum (possibly CPPN) output required to produce a non-zero parameter value. Actual parameter values are
 typically scaled from [0, ann.transcriber.neuron.model.params.max] or [ann.transcriber.neuron.model.params.min,
 0]. Not always implemented.</div>
</li>
</ul>
<a name="synapseModelParamsMin">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelParamsMin</h4>
<pre>protected&nbsp;double[] synapseModelParamsMin</pre>
<div class="block">The minimum values for each synapse model parameter.</div>
</li>
</ul>
<a name="synapseModelParamsMax">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelParamsMax</h4>
<pre>protected&nbsp;double[] synapseModelParamsMax</pre>
<div class="block">The maximum values for each synapse model parameter.</div>
</li>
</ul>
<a name="synapseModelParamsRange">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelParamsRange</h4>
<pre>protected&nbsp;double[] synapseModelParamsRange</pre>
<div class="block">The value range for each synapse model parameter (synapseModelParamsMax[i] - synapseModelParamsMin[i]).</div>
</li>
</ul>
<a name="synapseModelParamsThreshold">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelParamsThreshold</h4>
<pre>protected&nbsp;double[] synapseModelParamsThreshold</pre>
<div class="block">the minimum (possibly CPPN) output required to produce a non-zero parameter value. Actual parameter values are
 typically scaled from [0, ann.transcriber.neuron.model.params.max] or [ann.transcriber.neuron.model.params.min,
 0]. Not always implemented.</div>
</li>
</ul>
<a name="synapseDisableParamName">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseDisableParamName</h4>
<pre>protected&nbsp;java.lang.String synapseDisableParamName</pre>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM"><code>SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</code></a></dd></dl>
</li>
</ul>
<a name="neuronModelTypeParam">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelTypeParam</h4>
<pre>protected&nbsp;java.lang.String neuronModelTypeParam</pre>
<div class="block">The name of the parameter in the neuron model that specifies the type of a neuron.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_TYPES"><code>SUBSTRATE_NEURON_MODEL_TYPES</code></a>, 
<a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS"><code>SUBSTRATE_NEURON_MODEL_PARAMS</code></a></dd></dl>
</li>
</ul>
<a name="synapseModelTypeParam">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelTypeParam</h4>
<pre>protected&nbsp;java.lang.String synapseModelTypeParam</pre>
<div class="block">The name of the parameter in the synapse model that specifies the type of a synapse.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_TYPES"><code>SUBSTRATE_SYNAPSE_MODEL_TYPES</code></a>, 
<a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS"><code>SUBSTRATE_SYNAPSE_MODEL_PARAMS</code></a></dd></dl>
</li>
</ul>
<a name="neuronModelTypes">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelTypes</h4>
<pre>protected&nbsp;double[] neuronModelTypes</pre>
<div class="block">The allowable values for the <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronModelTypeParam"><code>neuronModelTypeParam</code></a> model parameter that specify the type of a neuron.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_TYPES"><code>SUBSTRATE_NEURON_MODEL_TYPES</code></a>, 
<a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS"><code>SUBSTRATE_NEURON_MODEL_PARAMS</code></a></dd></dl>
</li>
</ul>
<a name="synapseModelTypes">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>synapseModelTypes</h4>
<pre>protected&nbsp;double[] synapseModelTypes</pre>
<div class="block">The allowable values for the <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam"><code>synapseModelTypeParam</code></a> model parameter that specify the type of a synapse.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_TYPES"><code>SUBSTRATE_SYNAPSE_MODEL_TYPES</code></a>, 
<a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS"><code>SUBSTRATE_SYNAPSE_MODEL_PARAMS</code></a></dd></dl>
</li>
</ul>
<a name="neuronModelTypeCount">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronModelTypeCount</h4>
<pre>protected&nbsp;int neuronModelTypeCount</pre>
<div class="block">The number of available neuron types.</div>
</li>
</ul>
<a name="synapseModelTypeCount">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>synapseModelTypeCount</h4>
<pre>protected&nbsp;int synapseModelTypeCount</pre>
<div class="block">The number of available synapse types.</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="TranscriberAdaptor()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TranscriberAdaptor</h4>
<pre>public&nbsp;TranscriberAdaptor()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="init(com.ojcoleman.ahni.hyperneat.Properties)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>init</h4>
<pre>public&nbsp;void&nbsp;init(<a href="../../../../com/ojcoleman/ahni/hyperneat/Properties.html" title="class in com.ojcoleman.ahni.hyperneat">Properties</a>&nbsp;props)</pre>
<div class="block">This method should be called from overriding methods.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/hyperneat/Configurable.html#init(com.ojcoleman.ahni.hyperneat.Properties)"><code>Configurable.init(Properties)</code></a></dd></dl>
</li>
</ul>
<a name="transcribe(org.jgapcustomised.Chromosome,com.anji.integration.Activator,java.util.Map)">
<!--   -->
</a><a name="transcribe(org.jgapcustomised.Chromosome, T, java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>transcribe</h4>
<pre>public&nbsp;<a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</a>&nbsp;transcribe(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c,
           <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</a>&nbsp;substrate,
           java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)
                               throws <a href="../../../../com/anji/integration/TranscriberException.html" title="class in com.anji.integration">TranscriberException</a></pre>
<div class="block">Sub-classes can override this method to convert the genotype to a phenotype, using an existing substrate object
 to allow performance gains by avoiding destroying and creating large objects or arrays, and optionally providing
 additional options.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>c</code> - chromosome to transcribe</dd><dd><code>substrate</code> - An existing phenotype substrate to reuse</dd><dd><code>options</code> - A set of transcriber specific options. May be null.</dd>
<dt><span class="strong">Returns:</span></dt><dd>phenotype</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../com/anji/integration/TranscriberException.html" title="class in com.anji.integration">TranscriberException</a></code></dd></dl>
</li>
</ul>
<a name="getNeuronTypeCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronTypeCount</h4>
<pre>public&nbsp;int&nbsp;getNeuronTypeCount()</pre>
<div class="block">Get the number of available neuron types.</div>
</li>
</ul>
<a name="getSynapseTypeCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseTypeCount</h4>
<pre>public&nbsp;int&nbsp;getSynapseTypeCount()</pre>
<div class="block">Get the number of available synapse types.</div>
</li>
</ul>
<a name="getNeuronParamClassCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronParamClassCount</h4>
<pre>public&nbsp;int&nbsp;getNeuronParamClassCount()</pre>
<div class="block">Get the number of available neuron types.</div>
</li>
</ul>
<a name="getSynapseParamClassCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseParamClassCount</h4>
<pre>public&nbsp;int&nbsp;getSynapseParamClassCount()</pre>
<div class="block">Get the number of available synapse types.</div>
</li>
</ul>
<a name="getNeuronParameterCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronParameterCount</h4>
<pre>public&nbsp;int&nbsp;getNeuronParameterCount()</pre>
<div class="block">Get the number of available neuron model parameters.</div>
</li>
</ul>
<a name="getNeuronParameterName(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronParameterName</h4>
<pre>public&nbsp;java.lang.String&nbsp;getNeuronParameterName(int&nbsp;index)</pre>
<div class="block">Get the name of the specified neuron model parameter.</div>
</li>
</ul>
<a name="getSynapseParameterCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseParameterCount</h4>
<pre>public&nbsp;int&nbsp;getSynapseParameterCount()</pre>
<div class="block">Get the number of available synapse model parameters.</div>
</li>
</ul>
<a name="getSynapseParameterName(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseParameterName</h4>
<pre>public&nbsp;java.lang.String&nbsp;getSynapseParameterName(int&nbsp;index)</pre>
<div class="block">Get the name of the specified synapse model parameter.</div>
</li>
</ul>
<a name="getNeuronModelParamsMin()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronModelParamsMin</h4>
<pre>public&nbsp;double[]&nbsp;getNeuronModelParamsMin()</pre>
</li>
</ul>
<a name="getNeuronModelParamsMax()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronModelParamsMax</h4>
<pre>public&nbsp;double[]&nbsp;getNeuronModelParamsMax()</pre>
</li>
</ul>
<a name="getSynapseModelParamsMin()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseModelParamsMin</h4>
<pre>public&nbsp;double[]&nbsp;getSynapseModelParamsMin()</pre>
</li>
</ul>
<a name="getSynapseModelParamsMax()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseModelParamsMax</h4>
<pre>public&nbsp;double[]&nbsp;getSynapseModelParamsMax()</pre>
</li>
</ul>
<a name="getNeuronModelParamsMin(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronModelParamsMin</h4>
<pre>public&nbsp;double&nbsp;getNeuronModelParamsMin(int&nbsp;index)</pre>
</li>
</ul>
<a name="getNeuronModelParamsMax(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronModelParamsMax</h4>
<pre>public&nbsp;double&nbsp;getNeuronModelParamsMax(int&nbsp;index)</pre>
</li>
</ul>
<a name="getSynapseModelParamsMin(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseModelParamsMin</h4>
<pre>public&nbsp;double&nbsp;getSynapseModelParamsMin(int&nbsp;index)</pre>
</li>
</ul>
<a name="getSynapseModelParamsMax(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseModelParamsMax</h4>
<pre>public&nbsp;double&nbsp;getSynapseModelParamsMax(int&nbsp;index)</pre>
</li>
</ul>
<a name="getChromosomeInputNeuronCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeInputNeuronCount</h4>
<pre>public&nbsp;int&nbsp;getChromosomeInputNeuronCount()</pre>
<div class="block">May be overridden to specify how many input neurons the Chromosome should encode. This default implementation
 returns the value specified by the property <a href="../../../../com/anji/neat/NeatConfiguration.html#STIMULUS_SIZE_KEY"><code>NeatConfiguration.STIMULUS_SIZE_KEY</code></a> if given,
 otherwise <a href="../../../../com/anji/neat/NeatConfiguration.html#DEFAULT_STIMULUS_SIZE"><code>NeatConfiguration.DEFAULT_STIMULUS_SIZE</code></a>.</div>
</li>
</ul>
<a name="getChromosomeOutputNeuronCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeOutputNeuronCount</h4>
<pre>public&nbsp;int&nbsp;getChromosomeOutputNeuronCount()</pre>
<div class="block">May be overridden to specify how many output neurons the Chromosome should encode. This default implementation
 returns the value specified by the property <a href="../../../../com/anji/neat/NeatConfiguration.html#RESPONSE_SIZE_KEY"><code>NeatConfiguration.RESPONSE_SIZE_KEY</code></a> if given,
 otherwise <a href="../../../../com/anji/neat/NeatConfiguration.html#DEFAULT_RESPONSE_SIZE"><code>NeatConfiguration.DEFAULT_RESPONSE_SIZE</code></a>.</div>
</li>
</ul>
<a name="getNeuronParameterValues(org.jgapcustomised.Chromosome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronParameterValues</h4>
<pre>public&nbsp;double[][]&nbsp;getNeuronParameterValues(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom)</pre>
<div class="block">Gets the neuron parameter values for each parameter in each parameter class for the specified chromosome, in the
 format [class][param] (param order same as <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronParamNames"><code>neuronParamNames</code></a>).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES"><code>SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</code></a></dd></dl>
</li>
</ul>
<a name="getSynapseParameterValues(org.jgapcustomised.Chromosome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSynapseParameterValues</h4>
<pre>public&nbsp;double[][]&nbsp;getSynapseParameterValues(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom)</pre>
<div class="block">Gets the synapse parameter values for each parameter in each parameter class the specified chromosome, in the
 format [class][param] (param order same as <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamNames"><code>synapseParamNames</code></a>).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES"><code>SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</code></a></dd></dl>
</li>
</ul>
<a name="getParameterValuesForCollection(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.integration.ParamCollection, double[][])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getParameterValuesForCollection</h4>
<pre>protected&nbsp;int&nbsp;getParameterValuesForCollection(<a href="../../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom,
                                  <a href="../../../../com/ojcoleman/ahni/integration/ParamCollection.html" title="class in com.ojcoleman.ahni.integration">ParamCollection</a>&nbsp;collection,
                                  double[][]&nbsp;values)</pre>
</li>
</ul>
<a name="getExtraGenesForInitialChromosome(com.anji.neat.NeatConfiguration)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getExtraGenesForInitialChromosome</h4>
<pre>public&nbsp;java.util.List&lt;<a href="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;getExtraGenesForInitialChromosome(<a href="../../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config)
                                                         throws <a href="../../../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</a></pre>
<div class="block">Returns a list of <a href="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised"><code>Allele</code></a>s, or more importantly <a href="../../../../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised"><code>Gene</code></a>s referenced by the Alleles, that this
 transcriber knows what to do with. This default implementation adds <a href="../../../../com/ojcoleman/ahni/integration/ParamAllele.html" title="class in com.ojcoleman.ahni.integration"><code>ParamAllele</code></a>s for each neuron and
 synapse model parameter class (see <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES"><code>SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</code></a> and
 <a href="../../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES"><code>SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</code></a>). It also adds the ParamMutationOperator to the list of mutation
 operators for the given configuration if it hasn't already been added. Sub-classes may override this method to
 add alternate or additional Alleles/Genes.</div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</a></code></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TranscriberAdaptor.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html" title="class in com.ojcoleman.ahni.transcriber"><span class="strong">Prev Class</span></a></li>
<li>Next Class</li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" target="_top">Frames</a></li>
<li><a href="TranscriberAdaptor.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
