<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Mar 22 08:31:00 EST 2013 -->
<TITLE>
Uses of Class com.anji.neat.ConnectionAllele
</TITLE>

<META NAME="date" CONTENT="2013-03-22">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class com.anji.neat.ConnectionAllele";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?com/anji/neat//class-useConnectionAllele.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ConnectionAllele.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Class<br>com.anji.neat.ConnectionAllele</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.integration"><B>com.anji.integration</B></A></TD>
<TD><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.neat"><B>com.anji.neat</B></A></TD>
<TD><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.integration"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A> in <A HREF="../../../../com/anji/integration/package-summary.html">com.anji.integration</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../com/anji/integration/package-summary.html">com.anji.integration</A> that return <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></CODE></FONT></TD>
<TD><CODE><B>XmlPersistableAllele.</B><B><A HREF="../../../../com/anji/integration/XmlPersistableAllele.html#connectionFromXml(org.w3c.dom.Node)">connectionFromXml</A></B>(org.w3c.dom.Node&nbsp;node)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convert from XML to <code>ConnectionGene</code> object</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.neat"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A> in <A HREF="../../../../com/anji/neat/package-summary.html">com.anji.neat</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../com/anji/neat/package-summary.html">com.anji.neat</A> that return <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></CODE></FONT></TD>
<TD><CODE><B>NeatConfiguration.</B><B><A HREF="../../../../com/anji/neat/NeatConfiguration.html#newConnectionAllele(java.lang.Long, java.lang.Long)">newConnectionAllele</A></B>(java.lang.Long&nbsp;srcNeuronId,
                    java.lang.Long&nbsp;destNeuronId)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;factory method to construct new connection allele from neuron <code>srcNeuronId</code> to neuron
 <code>destNeuronId</code> according to NEAT add connection mutation; if a previous mutation has occurred adding a
 connection between srcNeuronId and destNeuronId, returns connection with that id; otherwise, new innovation id</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></CODE></FONT></TD>
<TD><CODE><B>NeatConfiguration.</B><B><A HREF="../../../../com/anji/neat/NeatConfiguration.html#newConnectionAllele(java.lang.Long, java.lang.Long, double)">newConnectionAllele</A></B>(java.lang.Long&nbsp;srcNeuronId,
                    java.lang.Long&nbsp;destNeuronId,
                    double&nbsp;weight)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;factory method to construct new connection allele from neuron <code>srcNeuronId</code> to neuron
 <code>destNeuronId</code> according to NEAT add connection mutation; if a previous mutation has occurred adding a
 connection between srcNeuronId and destNeuronId, returns connection with that id; otherwise, new innovation id</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../com/anji/neat/package-summary.html">com.anji.neat</A> that return types with arguments of type <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForDestNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForDestNeurons</A></B>(java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connAlleles,
                                       java.util.Collection&nbsp;destNeuronInnovationIds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForSrcNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForSrcNeurons</A></B>(java.util.Collection&nbsp;connAlleles,
                                      java.util.Collection&nbsp;srcNeuronInnovationIds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionList(java.util.Collection)">getConnectionList</A></B>(java.util.Collection&lt;<A HREF="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns all connection genes in <code>genes</code> as <code>List</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionMap(java.util.Set)">getConnectionMap</A></B>(java.util.Set&lt;<A HREF="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns all connections in <code>alleles</code> as <code>SortedMap</code></TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../com/anji/neat/package-summary.html">com.anji.neat</A> with parameters of type <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AddNeuronMutationOperator.</B><B><A HREF="../../../../com/anji/neat/AddNeuronMutationOperator.html#addNeuronAtConnection(com.anji.neat.NeatConfiguration, java.util.Map, com.anji.neat.ConnectionAllele, java.util.Set, java.util.Set)">addNeuronAtConnection</A></B>(<A HREF="../../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
                      java.util.Map&lt;java.lang.Long,<A HREF="../../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;&nbsp;neurons,
                      <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&nbsp;oldConnectAllele,
                      java.util.Set&lt;<A HREF="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
                      java.util.Set&lt;<A HREF="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../../com/anji/neat/package-summary.html">com.anji.neat</A> with type arguments of type <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronAnywhereMutationOperator.</B><B><A HREF="../../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#addNeuron(com.anji.neat.NeatConfiguration, java.util.List, java.util.List, java.util.Set, java.util.Set)">addNeuron</A></B>(<A HREF="../../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
          java.util.List&lt;<A HREF="../../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;&nbsp;neurons,
          java.util.List&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connections,
          java.util.Set&lt;<A HREF="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
          java.util.Set&lt;<A HREF="../../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AddConnectionMutationOperator.</B><B><A HREF="../../../../com/anji/neat/AddConnectionMutationOperator.html#connectionAllowed(com.anji.neat.NeuronAllele, com.anji.neat.NeuronAllele, java.util.Collection)">connectionAllowed</A></B>(<A HREF="../../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&nbsp;src,
                  <A HREF="../../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&nbsp;dest,
                  java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;conns)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForDestNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForDestNeurons</A></B>(java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connAlleles,
                                       java.util.Collection&nbsp;destNeuronInnovationIds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)">neuronsAreConnected</A></B>(java.lang.Long&nbsp;srcNeuronId,
                    java.lang.Long&nbsp;destNeuronId,
                    java.util.Collection&lt;<A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;non-recursive starting point for recursive search</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../index.html?com/anji/neat//class-useConnectionAllele.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ConnectionAllele.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
