<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.6.0_27) on Sun Nov 24 09:23:38 EST 2013 -->
<title>Uses of Class org.jgapcustomised.Allele</title>
<meta name="date" content="2013-11-24">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.jgapcustomised.Allele";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>PREV</li>
<li>NEXT</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised//class-useAllele.html" target="_top">FRAMES</a></li>
<li><a href="Allele.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.jgapcustomised.Allele" class="title">Uses of Class<br>org.jgapcustomised.Allele</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#com.anji.integration">com.anji.integration</a></td>
<td class="colLast">
<div class="block"><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.anji.neat">com.anji.neat</a></td>
<td class="colLast">
<div class="block"><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.integration">com.ojcoleman.ahni.integration</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.transcriber">com.ojcoleman.ahni.transcriber</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.jgapcustomised">org.jgapcustomised</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.jgapcustomised.impl">org.jgapcustomised.impl</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="com.anji.integration">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a> with parameters of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../com/anji/integration/XmlPersistableAllele.html#XmlPersistableAllele(org.jgapcustomised.Allele)">XmlPersistableAllele</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;aAllele)</code>
<div class="block">ctor</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.anji.neat">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a></strong></code>
<div class="block">Gene corresponding to NEAT connection gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a></strong></code>
<div class="block">Gene corresponding to NEAT node gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a></div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> that return <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">NeuronAllele.</span><code><strong><a href="../../../com/anji/neat/NeuronAllele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">ConnectionAllele.</span><code><strong><a href="../../../com/anji/neat/ConnectionAllele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><span class="strong">PruneMutationOperator.</span><code><strong><a href="../../../com/anji/neat/PruneMutationOperator.html#getAllelesToRemove(org.jgapcustomised.ChromosomeMaterial)">getAllelesToRemove</a></strong>(<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><span class="strong">NeatChromosomeUtility.</span><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#initAlleles(short, short, short, com.anji.neat.NeatConfiguration, boolean)">initAlleles</a></strong>(short&nbsp;numInputs,
           short&nbsp;numHidden,
           short&nbsp;numOutputs,
           <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
           boolean&nbsp;fullyConnected)</code>
<div class="block">constructs genes for neural net with specified input and output dimension, JGAP/NEAT configuration, and amount of
 connectivity</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> with parameters of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">NeuronAllele.</span><code><strong><a href="../../../com/anji/neat/NeuronAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;otherAllele)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">ConnectionAllele.</span><code><strong><a href="../../../com/anji/neat/ConnectionAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;otherAllele)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> with type arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AddConnectionMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddConnectionMutationOperator.html#addConnections(int, com.anji.neat.NeatConfiguration, java.util.List, java.util.SortedMap, java.util.Set, java.util.Set)">addConnections</a></strong>(int&nbsp;numConnectionsToAdd,
              <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
              java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;neuronList,
              java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;conns,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Given the collections of neurons and connections, returns the new connections that should be added, up to a max
 of numConnectionsToAdd.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AddConnectionMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddConnectionMutationOperator.html#addConnections(int, com.anji.neat.NeatConfiguration, java.util.List, java.util.SortedMap, java.util.Set, java.util.Set)">addConnections</a></strong>(int&nbsp;numConnectionsToAdd,
              <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
              java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;neuronList,
              java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;conns,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Given the collections of neurons and connections, returns the new connections that should be added, up to a max
 of numConnectionsToAdd.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AddNeuronAnywhereMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#addNeuron(com.anji.neat.NeatConfiguration, java.util.List, java.util.List, java.util.Set, java.util.Set)">addNeuron</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
         java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;neurons,
         java.util.List&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;connections,
         java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
         java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AddNeuronAnywhereMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#addNeuron(com.anji.neat.NeatConfiguration, java.util.List, java.util.List, java.util.Set, java.util.Set)">addNeuron</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
         java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;neurons,
         java.util.List&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;connections,
         java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
         java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">AddNeuronMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronMutationOperator.html#addNeuronAtConnection(com.anji.neat.NeatConfiguration, java.util.Map, com.anji.neat.ConnectionAllele, java.util.Set, java.util.Set)">addNeuronAtConnection</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                     java.util.Map&lt;java.lang.Long,<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;neurons,
                     <a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&nbsp;oldConnectAllele,
                     java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
                     java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">AddNeuronMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronMutationOperator.html#addNeuronAtConnection(com.anji.neat.NeatConfiguration, java.util.Map, com.anji.neat.ConnectionAllele, java.util.Set, java.util.Set)">addNeuronAtConnection</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                     java.util.Map&lt;java.lang.Long,<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;neurons,
                     <a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&nbsp;oldConnectAllele,
                     java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
                     java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;</code></td>
<td class="colLast"><span class="strong">NeatChromosomeUtility.</span><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionList(java.util.Collection)">getConnectionList</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</code>
<div class="block">returns all connection genes in <code>genes</code> as <code>List</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;</code></td>
<td class="colLast"><span class="strong">NeatChromosomeUtility.</span><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionMap(java.util.Set)">getConnectionMap</a></strong>(java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</code>
<div class="block">returns all connections in <code>alleles</code> as <code>SortedMap</code></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><span class="strong">NeatChromosomeUtility.</span><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection)">getNeuronList</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</code>
<div class="block">return all neurons in <code>genes</code> as <code>List</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><span class="strong">NeatChromosomeUtility.</span><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection, com.anji.neat.NeuronType)">getNeuronList</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles,
             <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>&nbsp;type)</code>
<div class="block">if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><span class="strong">NeatChromosomeUtility.</span><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronListForTypes(java.util.Collection, com.anji.neat.NeuronType[])">getNeuronListForTypes</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles,
                     <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>[]&nbsp;types)</code>
<div class="block">if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SingleTopologicalMutationOperator.</span><code><strong><a href="../../../com/anji/neat/SingleTopologicalMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SingleTopologicalMutationOperator.</span><code><strong><a href="../../../com/anji/neat/SingleTopologicalMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">AddConnectionMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddConnectionMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 connection mutation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">AddConnectionMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddConnectionMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 connection mutation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">WeightMutationOperator.</span><code><strong><a href="../../../com/anji/neat/WeightMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToRemove)</code>
<div class="block">Removes from <code>genesToAdd</code> and adds to <code>genesToRemove</code> all connection genes that are
 modified.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">WeightMutationOperator.</span><code><strong><a href="../../../com/anji/neat/WeightMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToRemove)</code>
<div class="block">Removes from <code>genesToAdd</code> and adds to <code>genesToRemove</code> all connection genes that are
 modified.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">PruneMutationOperator.</span><code><strong><a href="../../../com/anji/neat/PruneMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToRemove)</code>
<div class="block">Traverse network flowing forward and backward to identify unvisited connections and neurons.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">PruneMutationOperator.</span><code><strong><a href="../../../com/anji/neat/PruneMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;genesToRemove)</code>
<div class="block">Traverse network flowing forward and backward to identify unvisited connections and neurons.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">AddNeuronMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 node mutation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">AddNeuronMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 node mutation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">AddNeuronAnywhereMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">AddNeuronAnywhereMutationOperator.</span><code><strong><a href="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.integration">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../com/ojcoleman/ahni/integration/package-summary.html">com.ojcoleman.ahni.integration</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../com/ojcoleman/ahni/integration/package-summary.html">com.ojcoleman.ahni.integration</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../com/ojcoleman/ahni/integration/ParamAllele.html" title="class in com.ojcoleman.ahni.integration">ParamAllele</a></strong></code>
<div class="block">Generic parameter Allele that defines a single double floating point value.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/integration/package-summary.html">com.ojcoleman.ahni.integration</a> that return <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">ParamAllele.</span><code><strong><a href="../../../com/ojcoleman/ahni/integration/ParamAllele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/integration/package-summary.html">com.ojcoleman.ahni.integration</a> with parameters of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">ParamAllele.</span><code><strong><a href="../../../com/ojcoleman/ahni/integration/ParamAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;otherAllele)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../com/ojcoleman/ahni/integration/package-summary.html">com.ojcoleman.ahni.integration</a> with type arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">ParamMutationOperator.</span><code><strong><a href="../../../com/ojcoleman/ahni/integration/ParamMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Mutates (some of) the <a href="../../../com/ojcoleman/ahni/integration/ParamAllele.html" title="class in com.ojcoleman.ahni.integration"><code>ParamAllele</code></a>s in the given target.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">ParamMutationOperator.</span><code><strong><a href="../../../com/ojcoleman/ahni/integration/ParamMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Mutates (some of) the <a href="../../../com/ojcoleman/ahni/integration/ParamAllele.html" title="class in com.ojcoleman.ahni.integration"><code>ParamAllele</code></a>s in the given target.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.transcriber">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><span class="strong">TranscriberAdaptor.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getExtraGenesForInitialChromosome(com.anji.neat.NeatConfiguration)">getExtraGenesForInitialChromosome</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config)</code>
<div class="block">Returns a list of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised"><code>Allele</code></a>s, or more importantly <a href="../../../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised"><code>Gene</code></a>s referenced by the Alleles, that this
 transcriber knows what to do with.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.jgapcustomised">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> that return <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">Allele.</span><code><strong><a href="../../../org/jgapcustomised/Allele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;alleleToMatch)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.SortedSet&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><span class="strong">ChromosomeMaterial.</span><code><strong><a href="../../../org/jgapcustomised/ChromosomeMaterial.html#getAlleles()">getAlleles</a></strong>()</code>
<div class="block">Retrieves the set of genes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.SortedSet&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#getAlleles()">getAlleles</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with parameters of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">Allele.</span><code><strong><a href="../../../org/jgapcustomised/Allele.html#distance(org.jgapcustomised.Allele)">distance</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;target)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;alleleToMatch)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract boolean</code></td>
<td class="colLast"><span class="strong">Allele.</span><code><strong><a href="../../../org/jgapcustomised/Allele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;otherAllele)</code>
<div class="block">Return true iff all the parameters of this allele match those of the given allele.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected abstract void</code></td>
<td class="colLast"><span class="strong">MutationOperator.</span><code><strong><a href="../../../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Leaves <code>material</code> unmodified, but updates <code>allelesToAdd</code> and <code>allelesToRemove</code>
 with modifications.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract void</code></td>
<td class="colLast"><span class="strong">MutationOperator.</span><code><strong><a href="../../../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
      <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;target,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
      java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">Leaves <code>material</code> unmodified, but updates <code>allelesToAdd</code> and <code>allelesToRemove</code>
 with modifications.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ChromosomeMaterial.</span><code><strong><a href="../../../org/jgapcustomised/ChromosomeMaterial.html#setAlleles(java.util.SortedSet)">setAlleles</a></strong>(java.util.SortedSet&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;aAlleles)</code>
<div class="block">for hibernate</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected static void</code></td>
<td class="colLast"><span class="strong">MutationOperator.</span><code><strong><a href="../../../org/jgapcustomised/MutationOperator.html#updateMaterial(org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">updateMaterial</a></strong>(<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;material,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">updates <code>material</code> with specified sets of alleles; alleles present in both lists will be added (or
 replaced if the gene existed on original material)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static void</code></td>
<td class="colLast"><span class="strong">MutationOperator.</span><code><strong><a href="../../../org/jgapcustomised/MutationOperator.html#updateMaterial(org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">updateMaterial</a></strong>(<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;material,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToAdd,
              java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;allelesToRemove)</code>
<div class="block">updates <code>material</code> with specified sets of alleles; alleles present in both lists will be added (or
 replaced if the gene existed on original material)</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type arguments of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection)">ChromosomeMaterial</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;a_initialAlleles)</code>
<div class="block">Create chromosome with no parents.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long)">ChromosomeMaterial</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;a_initialGenes,
                  java.lang.Long&nbsp;aPrimaryParentId)</code>
<div class="block">Create chromosome with one parents.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)">ChromosomeMaterial</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;a_initialAlleles,
                  java.lang.Long&nbsp;aPrimaryParentId,
                  java.lang.Long&nbsp;aSecondaryParentId)</code>
<div class="block">Create chromosome with two parents.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.jgapcustomised.impl">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a> in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></strong></code>
<div class="block">A Gene implementation that supports two possible values (alleles) for each gene: true and false.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></strong></code>
<div class="block">A Gene implementation that supports a integer values for its allele.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> that return <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">BooleanAllele.</span><code><strong><a href="../../../org/jgapcustomised/impl/BooleanAllele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">IntegerAllele.</span><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><span class="strong">IntegerAllele.</span><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#newAllele(org.jgapcustomised.Configuration)">newAllele</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration)</code>
<div class="block">Provides an implementation-independent means for creating new Gene instances.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> with parameters of type <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">BooleanAllele.</span><code><strong><a href="../../../org/jgapcustomised/impl/BooleanAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;otherAllele)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">IntegerAllele.</span><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;otherAllele)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>PREV</li>
<li>NEXT</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised//class-useAllele.html" target="_top">FRAMES</a></li>
<li><a href="Allele.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
