<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Jul 05 16:56:47 EST 2013 -->
<TITLE>
Uses of Class org.jgapcustomised.Allele
</TITLE>

<META NAME="date" CONTENT="2013-07-05">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.jgapcustomised.Allele";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?org/jgapcustomised//class-useAllele.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Allele.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Class<br>org.jgapcustomised.Allele</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.integration"><B>com.anji.integration</B></A></TD>
<TD><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.neat"><B>com.anji.neat</B></A></TD>
<TD><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.jgapcustomised"><B>org.jgapcustomised</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.jgapcustomised.impl"><B>org.jgapcustomised.impl</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.integration"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> in <A HREF="../../../com/anji/integration/package-summary.html">com.anji.integration</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../com/anji/integration/package-summary.html">com.anji.integration</A> with parameters of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../com/anji/integration/XmlPersistableAllele.html#XmlPersistableAllele(org.jgapcustomised.Allele)">XmlPersistableAllele</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;aAllele)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ctor</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.neat"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subclasses of <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gene corresponding to NEAT connection gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Gene corresponding to NEAT node gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a></TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A> that return <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>NeuronAllele.</B><B><A HREF="../../../com/anji/neat/NeuronAllele.html#cloneAllele()">cloneAllele</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>ConnectionAllele.</B><B><A HREF="../../../com/anji/neat/ConnectionAllele.html#cloneAllele()">cloneAllele</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A> that return types with arguments of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#initAlleles(short, short, short, com.anji.neat.NeatConfiguration, boolean)">initAlleles</A></B>(short&nbsp;numInputs,
            short&nbsp;numHidden,
            short&nbsp;numOutputs,
            <A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
            boolean&nbsp;fullyConnected)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;constructs genes for neural net with specified input and output dimension, JGAP/NEAT configuration, and amount of
 connectivity</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A> with parameters of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>NeuronAllele.</B><B><A HREF="../../../com/anji/neat/NeuronAllele.html#distance(org.jgapcustomised.Allele)">distance</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>ConnectionAllele.</B><B><A HREF="../../../com/anji/neat/ConnectionAllele.html#distance(org.jgapcustomised.Allele)">distance</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>NeuronAllele.</B><B><A HREF="../../../com/anji/neat/NeuronAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;otherAllele)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>ConnectionAllele.</B><B><A HREF="../../../com/anji/neat/ConnectionAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;otherAllele)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronAnywhereMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#addNeuron(com.anji.neat.NeatConfiguration, java.util.List, java.util.List, java.util.Set, java.util.Set)">addNeuron</A></B>(<A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
          java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;&nbsp;neurons,
          java.util.List&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connections,
          java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
          java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronAnywhereMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#addNeuron(com.anji.neat.NeatConfiguration, java.util.List, java.util.List, java.util.Set, java.util.Set)">addNeuron</A></B>(<A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
          java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;&nbsp;neurons,
          java.util.List&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connections,
          java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
          java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AddNeuronMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronMutationOperator.html#addNeuronAtConnection(com.anji.neat.NeatConfiguration, java.util.Map, com.anji.neat.ConnectionAllele, java.util.Set, java.util.Set)">addNeuronAtConnection</A></B>(<A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
                      java.util.Map&lt;java.lang.Long,<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;&nbsp;neurons,
                      <A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&nbsp;oldConnectAllele,
                      java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
                      java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>AddNeuronMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronMutationOperator.html#addNeuronAtConnection(com.anji.neat.NeatConfiguration, java.util.Map, com.anji.neat.ConnectionAllele, java.util.Set, java.util.Set)">addNeuronAtConnection</A></B>(<A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
                      java.util.Map&lt;java.lang.Long,<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;&nbsp;neurons,
                      <A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&nbsp;oldConnectAllele,
                      java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
                      java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionList(java.util.Collection)">getConnectionList</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns all connection genes in <code>genes</code> as <code>List</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionMap(java.util.Set)">getConnectionMap</A></B>(java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns all connections in <code>alleles</code> as <code>SortedMap</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection)">getNeuronList</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return all neurons in <code>genes</code> as <code>List</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NeatChromosomeUtility.</B><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection, com.anji.neat.NeuronType)">getNeuronList</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles,
              <A HREF="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</A>&nbsp;type)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>WeightMutationOperator.</B><B><A HREF="../../../com/anji/neat/WeightMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;jgapConfig,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;genesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;genesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Removes from <code>genesToAdd</code> and adds to <code>genesToRemove</code> all connection genes that are
 modified.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>WeightMutationOperator.</B><B><A HREF="../../../com/anji/neat/WeightMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;jgapConfig,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;genesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;genesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Removes from <code>genesToAdd</code> and adds to <code>genesToRemove</code> all connection genes that are
 modified.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;jgapConfig,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 node mutation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;jgapConfig,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 node mutation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronAnywhereMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;jgapConfig,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>AddNeuronAnywhereMutationOperator.</B><B><A HREF="../../../com/anji/neat/AddNeuronAnywhereMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;jgapConfig,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.jgapcustomised"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> that return <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>abstract &nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>Allele.</B><B><A HREF="../../../org/jgapcustomised/Allele.html#cloneAllele()">cloneAllele</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>Chromosome.</B><B><A HREF="../../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;alleleToMatch)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> that return types with arguments of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.SortedSet&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>ChromosomeMaterial.</B><B><A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html#getAlleles()">getAlleles</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the set of genes.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with parameters of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>abstract &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>Allele.</B><B><A HREF="../../../org/jgapcustomised/Allele.html#distance(org.jgapcustomised.Allele)">distance</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>Chromosome.</B><B><A HREF="../../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;alleleToMatch)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>abstract &nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Allele.</B><B><A HREF="../../../org/jgapcustomised/Allele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;otherAllele)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Return true iff all the parameters of this allele match those of the given allele.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>MutationOperator.</B><B><A HREF="../../../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Leaves <code>material</code> unmodified, but updates <code>allelesToAdd</code> and <code>allelesToRemove</code>
 with modifications.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>MutationOperator.</B><B><A HREF="../../../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)">mutate</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
       <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToAdd,
       java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;allelesToRemove)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Leaves <code>material</code> unmodified, but updates <code>allelesToAdd</code> and <code>allelesToRemove</code>
 with modifications.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ChromosomeMaterial.</B><B><A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html#setAlleles(java.util.SortedSet)">setAlleles</A></B>(java.util.SortedSet&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;aAlleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;for hibernate</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection)">ChromosomeMaterial</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialAlleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create chromosome with no parents.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long)">ChromosomeMaterial</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialGenes,
                   java.lang.Long&nbsp;aPrimaryParentId)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create chromosome with one parents.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)">ChromosomeMaterial</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialAlleles,
                   java.lang.Long&nbsp;aPrimaryParentId,
                   java.lang.Long&nbsp;aSecondaryParentId)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create chromosome with two parents.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.jgapcustomised.impl"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Subclasses of <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A Gene implementation that supports two possible values (alleles) for each gene: true and false.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;class</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A Gene implementation that supports a integer values for its allele.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A> that return <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>BooleanAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/BooleanAllele.html#cloneAllele()">cloneAllele</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>IntegerAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/IntegerAllele.html#cloneAllele()">cloneAllele</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B>IntegerAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/IntegerAllele.html#newAllele(org.jgapcustomised.Configuration)">newAllele</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Provides an implementation-independent means for creating new Gene instances.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A> with parameters of type <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>BooleanAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/BooleanAllele.html#distance(org.jgapcustomised.Allele)">distance</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>IntegerAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/IntegerAllele.html#distance(org.jgapcustomised.Allele)">distance</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;target)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>BooleanAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/BooleanAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;otherAllele)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>IntegerAllele.</B><B><A HREF="../../../org/jgapcustomised/impl/IntegerAllele.html#isEquivalent(org.jgapcustomised.Allele)">isEquivalent</A></B>(<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;otherAllele)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?org/jgapcustomised//class-useAllele.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Allele.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
