<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_07) on Mon Oct 15 14:08:27 EST 2012 -->
<title>Uses of Class org.jgapcustomised.Chromosome</title>
<meta name="date" content="2012-10-15">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.jgapcustomised.Chromosome";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised/class-use/Chromosome.html" target="_top">Frames</a></li>
<li><a href="Chromosome.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.jgapcustomised.Chromosome" class="title">Uses of Class<br>org.jgapcustomised.Chromosome</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#com.anji.integration">com.anji.integration</a></td>
<td class="colLast">
<div class="block"><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.anji.neat">com.anji.neat</a></td>
<td class="colLast">
<div class="block"><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.anji.persistence">com.anji.persistence</a></td>
<td class="colLast">
<div class="block"><code>com.anji.persistence</code> contains classes implementing a simple persistence interface.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#ojc.ahni.experiments.objectrecognition">ojc.ahni.experiments.objectrecognition</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#ojc.ahni.hyperneat">ojc.ahni.hyperneat</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.jgapcustomised">org.jgapcustomised</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.jgapcustomised.impl">org.jgapcustomised.impl</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="com.anji.integration">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SimpleSelector.</span><code><strong><a href="../../../com/anji/integration/SimpleSelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfigurator,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_chromosomeToAdd)</code>
<div class="block">Add <code>a_chromosomeToAdd</code> to set of chromosomes to be evaluated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a></code></td>
<td class="colLast"><span class="strong">ActivatorTranscriber.</span><code><strong><a href="../../../com/anji/integration/ActivatorTranscriber.html#newActivator(org.jgapcustomised.Chromosome)">newActivator</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;ch)</code>
<div class="block">Constructs <code>Activator</code> phenotype from <code>Chromosome</code>
 genotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</a></code></td>
<td class="colLast"><span class="strong">AnjiNetTranscriber.</span><code><strong><a href="../../../com/anji/integration/AnjiNetTranscriber.html#newAnjiNet(org.jgapcustomised.Chromosome)">newAnjiNet</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>
<div class="block">create new <code>AnjiNet</code> from <code>genotype</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></code></td>
<td class="colLast"><span class="strong">AnjiNetTranscriber.</span><code><strong><a href="../../../com/anji/integration/AnjiNetTranscriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</a></code></td>
<td class="colLast"><span class="strong">Transcriber.</span><code><strong><a href="../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>
<div class="block">Sub-classes must implement this method to convert the genotype to a phenotype.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></code></td>
<td class="colLast"><span class="strong">AnjiNetTranscriber.</span><code><strong><a href="../../../com/anji/integration/AnjiNetTranscriber.html#transcribe(org.jgapcustomised.Chromosome, com.anji.integration.AnjiActivator)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</a></code></td>
<td class="colLast"><span class="strong">Transcriber.</span><code><strong><a href="../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome, T)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c,
          <a href="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</a>&nbsp;substrate)</code>
<div class="block">Sub-classes must implement this method to convert the genotype to a phenotype,
 using an existing substrate object to allow performance gains by avoiding
 destroying and creating large objects or arrays.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../com/anji/integration/XmlPersistableChromosome.html#XmlPersistableChromosome(org.jgapcustomised.Chromosome)">XmlPersistableChromosome</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.anji.neat">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Evolver.</span><code><strong><a href="../../../com/anji/neat/Evolver.html#getChamp()">getChamp</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><span class="strong">NeatCrossoverReproductionOperator.</span><code><strong><a href="../../../com/anji/neat/NeatCrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome, org.jgapcustomised.Chromosome)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;dominantChrom,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;recessiveChrom)</code>
<div class="block">Crossover according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> crossover
 methodology.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.anji.persistence">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/persistence/package-summary.html">com.anji.persistence</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/persistence/package-summary.html">com.anji.persistence</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, org.w3c.dom.Node)">chromosomeFromXml</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
                 org.w3c.dom.Node&nbsp;chromNode)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, java.lang.String)">chromosomeFromXml</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
                 java.lang.String&nbsp;xml)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#loadChromosome(java.lang.String, org.jgapcustomised.Configuration)">loadChromosome</a></strong>(java.lang.String&nbsp;id,
              <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Persistence.</span><code><strong><a href="../../../com/anji/persistence/Persistence.html#loadChromosome(java.lang.String, org.jgapcustomised.Configuration)">loadChromosome</a></strong>(java.lang.String&nbsp;id,
              <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HibernatePersistence.</span><code><strong><a href="../../../com/anji/persistence/HibernatePersistence.html#loadChromosome(java.lang.String, org.jgapcustomised.Configuration)">loadChromosome</a></strong>(java.lang.String&nbsp;id,
              <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/persistence/package-summary.html">com.anji.persistence</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#store(org.jgapcustomised.Chromosome)">store</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Persistence.</span><code><strong><a href="../../../com/anji/persistence/Persistence.html#store(org.jgapcustomised.Chromosome)">store</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HibernatePersistence.</span><code><strong><a href="../../../com/anji/persistence/HibernatePersistence.html#store(org.jgapcustomised.Chromosome)">store</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="ojc.ahni.experiments.objectrecognition">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../ojc/ahni/experiments/objectrecognition/package-summary.html">ojc.ahni.experiments.objectrecognition</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../ojc/ahni/experiments/objectrecognition/package-summary.html">ojc.ahni.experiments.objectrecognition</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><span class="strong">ObjectRecognitionFitnessFunction2.</span><code><strong><a href="../../../ojc/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;activator,
        int&nbsp;threadIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><span class="strong">ObjectRecognitionFitnessFunction3.</span><code><strong><a href="../../../ojc/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;activator,
        int&nbsp;threadIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><span class="strong">ObjectRecognitionFitnessFunction4.</span><code><strong><a href="../../../ojc/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;activator,
        int&nbsp;threadIndex)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="ojc.ahni.hyperneat">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../ojc/ahni/hyperneat/package-summary.html">ojc.ahni.hyperneat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../ojc/ahni/hyperneat/package-summary.html">ojc.ahni.hyperneat</a> declared as <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATFitnessFunction.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATFitnessFunction.html#lastBestChrom">lastBestChrom</a></strong></code>
<div class="block">The chromosome with the best performance in the previous generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATFitnessFunction.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATFitnessFunction.html#newBestChrom">newBestChrom</a></strong></code>
<div class="block">The chromosome with the current best performance.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../ojc/ahni/hyperneat/package-summary.html">ojc.ahni.hyperneat</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATEvolver.html#getChamp()">getChamp</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../ojc/ahni/hyperneat/package-summary.html">ojc.ahni.hyperneat</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><span class="strong">HyperNEATTargetFitnessFunction.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTargetFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract int</code></td>
<td class="colLast"><span class="strong">HyperNEATFitnessFunction.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>
<div class="block">Evaluate an individual genotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../ojc/ahni/hyperneat/BainNN.html" title="class in ojc.ahni.hyperneat">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTranscriberBain.html#newBainNN(org.jgapcustomised.Chromosome, ojc.ahni.hyperneat.BainNN)">newBainNN</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../ojc/ahni/hyperneat/BainNN.html" title="class in ojc.ahni.hyperneat">BainNN</a>&nbsp;phenotype)</code>
<div class="block">create a new neural network from the a genotype.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../ojc/ahni/hyperneat/GridNet.html" title="class in ojc.ahni.hyperneat">GridNet</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberGridNet.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTranscriberGridNet.html#newGridNet(org.jgapcustomised.Chromosome, ojc.ahni.hyperneat.GridNet)">newGridNet</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../ojc/ahni/hyperneat/GridNet.html" title="class in ojc.ahni.hyperneat">GridNet</a>&nbsp;phenotype)</code>
<div class="block">Create a new neural network from the a genotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../ojc/ahni/hyperneat/BainNN.html" title="class in ojc.ahni.hyperneat">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../ojc/ahni/hyperneat/GridNet.html" title="class in ojc.ahni.hyperneat">GridNet</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberGridNet.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTranscriberGridNet.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../ojc/ahni/hyperneat/GridNet.html" title="class in ojc.ahni.hyperneat">GridNet</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberGridNet.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTranscriberGridNet.html#transcribe(org.jgapcustomised.Chromosome, com.anji.integration.Activator)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../ojc/ahni/hyperneat/BainNN.html" title="class in ojc.ahni.hyperneat">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome, ojc.ahni.hyperneat.BainNN)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../ojc/ahni/hyperneat/BainNN.html" title="class in ojc.ahni.hyperneat">BainNN</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../ojc/ahni/hyperneat/package-summary.html">ojc.ahni.hyperneat</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HyperNEATFitnessFunction.</span><code><strong><a href="../../../ojc/ahni/hyperneat/HyperNEATFitnessFunction.html#evaluate(java.util.List)">evaluate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genotypes)</code>
<div class="block">Evaluate each chromosome in genotypes.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.jgapcustomised">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> declared as <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#bestPerforming">bestPerforming</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#fittest">fittest</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#m_chromosomes">m_chromosomes</a></strong></code>
<div class="block">Chromosomes that makeup thie Genotype's population.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#evolve()">evolve</a></strong>()</code>
<div class="block">Performs one generation cycle, evaluating fitness, selecting survivors, repopulting with
 offspring, and mutating new population.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getBestPerforming()">getBestPerforming</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getFittest()">getFittest</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getFittest()">getFittest</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getFittestChromosome()">getFittestChromosome</a></strong>()</code>
<div class="block">Retrieves the Chromosome in the population with the highest fitness value.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getPreviousFittest()">getPreviousFittest</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getRepresentative()">getRepresentative</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getChromosomes()">getChromosomes</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getChromosomes()">getChromosomes</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getElite(double, int)">getElite</a></strong>(double&nbsp;proportion,
        int&nbsp;minToSelect)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>
<div class="block">Select a given number of Chromosomes from the pool that will move on
 to the next generation population.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration, int)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
      int&nbsp;numToSurvive)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#add(org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract void</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#cullToElites(org.jgapcustomised.Chromosome)">cullToElites</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;popFittest)</code>
<div class="block">remove all non-elite chromosomes from this species, except for population-wide fittest</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">distance</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;target,
        <a href="../../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;parms)</code>
<div class="block">Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation
 methodology.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract int</code></td>
<td class="colLast"><span class="strong">FitnessFunction.</span><code><strong><a href="../../../org/jgapcustomised/FitnessFunction.html#evaluate(org.jgapcustomised.Chromosome)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_subject)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Determine the fitness of the given Chromosome instance.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getChromosomeFitnessValue(org.jgapcustomised.Chromosome)">getChromosomeFitnessValue</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">FitnessFunction.</span><code><strong><a href="../../../org/jgapcustomised/FitnessFunction.html#getFitnessValue(org.jgapcustomised.Chromosome)">getFitnessValue</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_subject)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Retrieves the fitness value of the given Chromosome.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#match(org.jgapcustomised.Chromosome)">match</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#remove(org.jgapcustomised.Chromosome)">remove</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><span class="strong">CrossoverReproductionOperator.</span><code><strong><a href="../../../org/jgapcustomised/CrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome, org.jgapcustomised.Chromosome)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;dominantChrom,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;recessiveChrom)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#setPreviousFittest(org.jgapcustomised.Chromosome)">setPreviousFittest</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;previousFittest)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;species,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chroms)</code>
<div class="block">If elitism is enabled, places appropriate chromosomes in <code>elite</code> list.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#addChromosomes(java.util.Collection)">addChromosomes</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chromosomes)</code>
<div class="block">Add the specified chromosomes to this Genotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#cull(java.util.Collection)">cull</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;keepers)</code>
<div class="block">Remove all chromosomes from this specie except <code>keepers</code>
 NOTE: this method does not update the species field in the removed Chromosomes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunction.</span><code><strong><a href="../../../org/jgapcustomised/BulkFitnessFunction.html#evaluate(java.util.List)">evaluate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;subjects)</code>
<div class="block">Calculates and sets the fitness values on each of the given Chromosomes via their
 setFitnessValue() method.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/Species.html#Species(org.jgapcustomised.SpeciationParms, org.jgapcustomised.Chromosome)">Species</a></strong>(<a href="../../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;aSpeciationParms,
       <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aRepresentative)</code>
<div class="block">Create new specie from representative.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/Genotype.html#Genotype(com.anji.util.Properties, org.jgapcustomised.Configuration, java.util.List)">Genotype</a></strong>(<a href="../../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a>&nbsp;props,
        <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
        java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;a_initialChromosomes)</code>
<div class="block">This constructor is used for random initial Genotypes.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.jgapcustomised.impl">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">ChromosomePool.</span><code><strong><a href="../../../org/jgapcustomised/impl/ChromosomePool.html#acquireChromosome()">acquireChromosome</a></strong>()</code>
<div class="block">Attempts to acquire an Chromosome instance from the chromosome pool.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">WeightedRouletteSelector.</span><code><strong><a href="../../../org/jgapcustomised/impl/WeightedRouletteSelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfigurator,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_chromosomeToAdd)</code>
<div class="block">Add a Chromosome instance to this selector's working pool of Chromosomes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ChromosomePool.</span><code><strong><a href="../../../org/jgapcustomised/impl/ChromosomePool.html#releaseChromosome(org.jgapcustomised.Chromosome)">releaseChromosome</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_chromosome)</code>
<div class="block">Releases a Chromosome to the pool.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised/class-use/Chromosome.html" target="_top">Frames</a></li>
<li><a href="Chromosome.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
