<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_75) on Sun Mar 15 11:01:18 AEDT 2015 -->
<title>Uses of Class org.jgapcustomised.Chromosome</title>
<meta name="date" content="2015-03-15">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.jgapcustomised.Chromosome";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised/class-use/Chromosome.html" target="_top">Frames</a></li>
<li><a href="Chromosome.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.jgapcustomised.Chromosome" class="title">Uses of Class<br>org.jgapcustomised.Chromosome</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#com.anji.integration">com.anji.integration</a></td>
<td class="colLast">
<div class="block"><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.anji.neat">com.anji.neat</a></td>
<td class="colLast">
<div class="block"><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.anji.persistence">com.anji.persistence</a></td>
<td class="colLast">
<div class="block"><code>com.anji.persistence</code> contains classes implementing a simple persistence interface.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.anji.util">com.anji.util</a></td>
<td class="colLast">
<div class="block"><code>com.anji.util</code> contains utility classes used in the ANJI framework.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.evaluation">com.ojcoleman.ahni.evaluation</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.evaluation.mocostfunctions">com.ojcoleman.ahni.evaluation.mocostfunctions</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.evaluation.novelty">com.ojcoleman.ahni.evaluation.novelty</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.experiments">com.ojcoleman.ahni.experiments</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.experiments.csb">com.ojcoleman.ahni.experiments.csb</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.experiments.doublepolebalancing">com.ojcoleman.ahni.experiments.doublepolebalancing</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.experiments.mr2d">com.ojcoleman.ahni.experiments.mr2d</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.experiments.objectrecognition">com.ojcoleman.ahni.experiments.objectrecognition</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.hyperneat">com.ojcoleman.ahni.hyperneat</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.misc">com.ojcoleman.ahni.misc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#com.ojcoleman.ahni.transcriber">com.ojcoleman.ahni.transcriber</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.jgapcustomised">org.jgapcustomised</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.jgapcustomised.impl">org.jgapcustomised.impl</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="com.anji.integration">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleSelector.</span><code><strong><a href="../../../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleSelector.</span><code><strong><a href="../../../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration,%20int)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
      int&nbsp;a_howManyToSelect)</code>
<div class="block">Returns the <code>a_howManyToSelect</code> chromosomes with highest fitness.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SimpleSelector.</span><code><strong><a href="../../../com/anji/integration/SimpleSelector.html#add(org.jgapcustomised.Configuration,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfigurator,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_chromosomeToAdd)</code>
<div class="block">Add <code>a_chromosomeToAdd</code> to set of chromosomes to be evaluated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a></code></td>
<td class="colLast"><span class="strong">ActivatorTranscriber.</span><code><strong><a href="../../../com/anji/integration/ActivatorTranscriber.html#newActivator(org.jgapcustomised.Chromosome)">newActivator</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;ch)</code>
<div class="block">Constructs an <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration"><code>Activator</code></a> phenotype from a <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised"><code>Chromosome</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</a></code></td>
<td class="colLast"><span class="strong">AnjiNetTranscriber.</span><code><strong><a href="../../../com/anji/integration/AnjiNetTranscriber.html#newAnjiNet(org.jgapcustomised.Chromosome)">newAnjiNet</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>
<div class="block">create new <code>AnjiNet</code> from <code>genotype</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</a></code></td>
<td class="colLast"><span class="strong">Transcriber.</span><code><strong><a href="../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>
<div class="block">Sub-classes must implement this method to convert the genotype to a phenotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></code></td>
<td class="colLast"><span class="strong">AnjiNetTranscriber.</span><code><strong><a href="../../../com/anji/integration/AnjiNetTranscriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></code></td>
<td class="colLast"><span class="strong">AnjiNetTranscriber.</span><code><strong><a href="../../../com/anji/integration/AnjiNetTranscriber.html#transcribe(org.jgapcustomised.Chromosome,%20com.anji.integration.AnjiActivator)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</a></code></td>
<td class="colLast"><span class="strong">Transcriber.</span><code><strong><a href="../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome,%20T)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c,
          <a href="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</a>&nbsp;substrate)</code>
<div class="block">Sub-classes must implement this method to convert the genotype to a phenotype, using an existing substrate object
 to allow performance gains by avoiding destroying and creating large objects or arrays.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../com/anji/integration/package-summary.html">com.anji.integration</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../com/anji/integration/XmlPersistableChromosome.html#XmlPersistableChromosome(org.jgapcustomised.Chromosome)">XmlPersistableChromosome</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.anji.neat">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Evolver.</span><code><strong><a href="../../../com/anji/neat/Evolver.html#getChamp()">getChamp</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><span class="strong">NeatCrossoverReproductionOperator.</span><code><strong><a href="../../../com/anji/neat/NeatCrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration,%20org.jgapcustomised.Chromosome,%20org.jgapcustomised.Chromosome)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;jgapConfig,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;parent1,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;parent2)</code>
<div class="block">Crossover according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> crossover
 methodology.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SpeciationStrategyKMeans.</span><code><strong><a href="../../../com/anji/neat/SpeciationStrategyKMeans.html#respeciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">respeciate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genomeList,
          java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;speciesList,
          <a href="../../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SpeciationStrategyOriginal.</span><code><strong><a href="../../../com/anji/neat/SpeciationStrategyOriginal.html#respeciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">respeciate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genomeList,
          java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;speciesList,
          <a href="../../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SpeciationStrategyKMeans.</span><code><strong><a href="../../../com/anji/neat/SpeciationStrategyKMeans.html#speciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">speciate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genomeList,
        java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;speciesList,
        <a href="../../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SpeciationStrategyOriginal.</span><code><strong><a href="../../../com/anji/neat/SpeciationStrategyOriginal.html#speciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">speciate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genomeList,
        java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;speciesList,
        <a href="../../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../com/anji/neat/package-summary.html">com.anji.neat</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NEATGenotype.html#NEATGenotype(com.anji.util.Properties,%20org.jgapcustomised.Configuration,%20java.util.List)">NEATGenotype</a></strong>(<a href="../../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a>&nbsp;props,
            <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
            java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;a_initialChromosomes)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.anji.persistence">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/persistence/package-summary.html">com.anji.persistence</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/persistence/package-summary.html">com.anji.persistence</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration,%20org.w3c.dom.Node)">chromosomeFromXml</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
                 org.w3c.dom.Node&nbsp;chromNode)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration,%20java.lang.String)">chromosomeFromXml</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
                 java.lang.String&nbsp;xml)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#loadChromosome(java.lang.String,%20org.jgapcustomised.Configuration)">loadChromosome</a></strong>(java.lang.String&nbsp;id,
              <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HibernatePersistence.</span><code><strong><a href="../../../com/anji/persistence/HibernatePersistence.html#loadChromosome(java.lang.String,%20org.jgapcustomised.Configuration)">loadChromosome</a></strong>(java.lang.String&nbsp;id,
              <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Persistence.</span><code><strong><a href="../../../com/anji/persistence/Persistence.html#loadChromosome(java.lang.String,%20org.jgapcustomised.Configuration)">loadChromosome</a></strong>(java.lang.String&nbsp;id,
              <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/anji/persistence/package-summary.html">com.anji.persistence</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FilePersistence.</span><code><strong><a href="../../../com/anji/persistence/FilePersistence.html#store(org.jgapcustomised.Chromosome)">store</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HibernatePersistence.</span><code><strong><a href="../../../com/anji/persistence/HibernatePersistence.html#store(org.jgapcustomised.Chromosome)">store</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Persistence.</span><code><strong><a href="../../../com/anji/persistence/Persistence.html#store(org.jgapcustomised.Chromosome)">store</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.anji.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/anji/util/package-summary.html">com.anji.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../com/anji/util/package-summary.html">com.anji.util</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DummyBulkFitnessFunction.</span><code><strong><a href="../../../com/anji/util/DummyBulkFitnessFunction.html#evaluate(java.util.List)">evaluate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;aSubjects)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.evaluation">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</a> declared as <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#lastBestChrom">lastBestChrom</a></strong></code>
<div class="block">The chromosome with the best performance in the previous generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#newBestChrom">newBestChrom</a></strong></code>
<div class="block">The chromosome with the current best performance.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</a> with type parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected java.util.Iterator&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#chromosomesIterator">chromosomesIterator</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">HyperNEATTargetFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">TargetFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">HyperNEATTargetFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">TargetFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>
<div class="block">
 Evaluate an individual genotype on one fitness objective (if more than one fitness objective is defined by this
 fitness function and/or it defines a behaviour for novelty search then
 <a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])"><code>BulkFitnessFunctionMT.evaluate(Chromosome, Activator, int, double[], Behaviour[])</code></a> must be overridden instead).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>
<div class="block">
 For fitness functions that define multiple fitness objectives (see <a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#fitnessObjectivesCount()"><code>BulkFitnessFunctionMT.fitnessObjectivesCount()</code></a>) and/or
 behaviours for novelty search (see <a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#noveltyObjectiveCount()"><code>BulkFitnessFunctionMT.noveltyObjectiveCount()</code></a>), evaluate an individual genotype over all of
 these objectives.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HyperNEATTargetFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">TargetFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>
<div class="block">This method may optionally be overridden to perform an evaluation of an individual genotype and output the results to a log.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AHNIFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/AHNIFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>
<div class="block">This method may optionally be overridden to perform an evaluation of an individual genotype and output the results to a log.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluateGeneralisation</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
                      <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
                      java.lang.String&nbsp;baseFileName,
                      boolean&nbsp;logText,
                      boolean&nbsp;logImage)</code>
<div class="block">This implementation of
 <a href="../../../com/ojcoleman/ahni/evaluation/AHNIFitnessFunction.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)"><code>AHNIFitnessFunction.evaluateGeneralisation(Chromosome, Activator, String, boolean, boolean)</code></a> is marked as
 final to allow this class to handle multiple fitness functions, some of which may define multiple objectives.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">AHNIFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/AHNIFitnessFunction.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluateGeneralisation</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
                      <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
                      java.lang.String&nbsp;baseFileName,
                      boolean&nbsp;logText,
                      boolean&nbsp;logImage)</code>
<div class="block">This method may optionally be overridden to perform an evaluation that performs a generalisation test of an 
 individual genotype and output the results to a log.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[])">evaluateGeneralisation</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
                      <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
                      java.lang.String&nbsp;baseFileName,
                      boolean&nbsp;logText,
                      boolean&nbsp;logImage,
                      double[]&nbsp;fitnessValues)</code>
<div class="block">This method is akin to
 <a href="../../../com/ojcoleman/ahni/evaluation/AHNIFitnessFunction.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)"><code>AHNIFitnessFunction.evaluateGeneralisation(Chromosome, Activator, String, boolean, boolean)</code></a> but allows
 for handling multiple fitness functions, some of which may define multiple objectives.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a></code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#generateSubstrate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator)">generateSubstrate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom,
                 <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate)</code>
<div class="block">A convenience method to generate a substrate from the given Chromosome using the configured transcriber for this
 fitness function.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">HyperNEATFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#postEvaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">postEvaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
            <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
            int&nbsp;evalThreadIndex)</code>
<div class="block">Sub-classes may override this method to perform operations after a Chromosome has been evaluated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#postEvaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">postEvaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
            <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
            int&nbsp;evalThreadIndex)</code>
<div class="block">Sub-classes may override this method to perform operations after a Chromosome has been evaluated.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HyperNEATFitnessFunction.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#evaluate(java.util.List)">evaluate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genotypes)</code>
<div class="block">Evaluate a set of chromosomes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunctionMT.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(java.util.List)">evaluate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genotypes)</code>
<div class="block">Evaluate a set of chromosomes.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.evaluation.mocostfunctions">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">CPPNSizeCost.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">BainNNConnectionLengthCost.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/BainNNConnectionLengthCost.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">BainNNConnectionCountCost.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/BainNNConnectionCountCost.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.evaluation.novelty">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/evaluation/novelty/package-summary.html">com.ojcoleman.ahni.evaluation.novelty</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/evaluation/novelty/package-summary.html">com.ojcoleman.ahni.evaluation.novelty</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">GenericBehaviourEvaluator.</span><code><strong><a href="../../../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.experiments">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/experiments/package-summary.html">com.ojcoleman.ahni.experiments</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/experiments/package-summary.html">com.ojcoleman.ahni.experiments</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">TMaze.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/TMaze.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MDP.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/MDP.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[],%20java.util.ArrayList)">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage,
         double[]&nbsp;fitnessValues,
         <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours,
         java.util.ArrayList&lt;com.ojcoleman.ahni.experiments.MDP.Environment&gt;&nbsp;environments)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RLRecurrentNetworkBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[],%20com.ojcoleman.ahni.experiments.RLRecurrentNetworkBased.Environment[])">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage,
         double[]&nbsp;fitnessValues,
         <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours,
         com.ojcoleman.ahni.experiments.RLRecurrentNetworkBased.Environment[]&nbsp;envs)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">TMaze.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/TMaze.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">RLRecurrentNetworkBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">MDP.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/MDP.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RLRecurrentNetworkBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MDP.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/MDP.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">TMaze.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/TMaze.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">RLRecurrentNetworkBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[])">evaluateGeneralisation</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
                      <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
                      java.lang.String&nbsp;baseFileName,
                      boolean&nbsp;logText,
                      boolean&nbsp;logImage,
                      double[]&nbsp;fitnessValues)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">MDP.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/MDP.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[])">evaluateGeneralisation</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
                      <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
                      java.lang.String&nbsp;baseFileName,
                      boolean&nbsp;logText,
                      boolean&nbsp;logImage,
                      double[]&nbsp;fitnessValues)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.experiments.csb">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RLContinuousStateBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage,
         double[]&nbsp;fitnessValues,
         <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">RLContinuousStateBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RLContinuousStateBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RLContinuousStateBased.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#postEvaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">postEvaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
            <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
            int&nbsp;evalThreadIndex)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.experiments.doublepolebalancing">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/experiments/doublepolebalancing/package-summary.html">com.ojcoleman.ahni.experiments.doublepolebalancing</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/experiments/doublepolebalancing/package-summary.html">com.ojcoleman.ahni.experiments.doublepolebalancing</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DoublePoleBalancing.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage,
         double[]&nbsp;fitnessValues,
         <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DoublePoleBalancingNVAntiWiggle.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancingNVAntiWiggle.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage,
         double[]&nbsp;fitnessValues,
         <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">DoublePoleBalancing.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DoublePoleBalancing.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.experiments.mr2d">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/experiments/mr2d/package-summary.html">com.ojcoleman.ahni.experiments.mr2d</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/experiments/mr2d/package-summary.html">com.ojcoleman.ahni.experiments.mr2d</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MobileRobot2D.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/mr2d/MobileRobot2D.html#_evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[],%20com.ojcoleman.ahni.experiments.mr2d.EnvironmentDescription[])">_evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage,
         double[]&nbsp;fitnessValues,
         <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours,
         <a href="../../../com/ojcoleman/ahni/experiments/mr2d/EnvironmentDescription.html" title="class in com.ojcoleman.ahni.experiments.mr2d">EnvironmentDescription</a>[]&nbsp;environmentDesc)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">MobileRobot2D.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/mr2d/MobileRobot2D.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int,%20double[],%20com.ojcoleman.ahni.evaluation.novelty.Behaviour[])">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        int&nbsp;evalThreadIndex,
        double[]&nbsp;fitnessValues,
        <a href="../../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]&nbsp;behaviours)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MobileRobot2D.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/mr2d/MobileRobot2D.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
        java.lang.String&nbsp;baseFileName,
        boolean&nbsp;logText,
        boolean&nbsp;logImage)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">MobileRobot2D.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/mr2d/MobileRobot2D.html#evaluateGeneralisation(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20java.lang.String,%20boolean,%20boolean,%20double[])">evaluateGeneralisation</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
                      <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate,
                      java.lang.String&nbsp;baseFileName,
                      boolean&nbsp;logText,
                      boolean&nbsp;logImage,
                      double[]&nbsp;fitnessValues)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.experiments.objectrecognition">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/experiments/objectrecognition/package-summary.html">com.ojcoleman.ahni.experiments.objectrecognition</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/experiments/objectrecognition/package-summary.html">com.ojcoleman.ahni.experiments.objectrecognition</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">ObjectRecognitionFitnessFunction3.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;activator,
        int&nbsp;threadIndex)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">ObjectRecognitionFitnessFunction2.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;activator,
        int&nbsp;threadIndex)</code>
<div class="block">Evaluate given individual by presenting the stimuli to the network in a random order to ensure the underlying
 network is not memorising the sequence of inputs.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><span class="strong">ObjectRecognitionFitnessFunction4.</span><code><strong><a href="../../../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#evaluate(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator,%20int)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
        <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;activator,
        int&nbsp;threadIndex)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.hyperneat">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</a> declared as <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#bestPerforming">bestPerforming</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>[]</code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#bestPerformingChromosomes">bestPerformingChromosomes</a></strong></code>
<div class="block">The best performing Chromosome from each generation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#fittest">fittest</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>[]</code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#fittestChromosomes">fittestChromosomes</a></strong></code>
<div class="block">The fittest Chromosome from each generation.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getBestPerformingFromLastGen()">getBestPerformingFromLastGen</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>[]</code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getBesttPerformingChromosomesForEachGen()">getBesttPerformingChromosomesForEachGen</a></strong>()</code>
<div class="block">Returns an array containing the best performing Chromosome from each generation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>[]</code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getFittestChromosomesForEachGen()">getFittestChromosomesForEachGen</a></strong>()</code>
<div class="block">Returns an array containing the fittest Chromosome from each generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getFittestFromLastGen()">getFittestFromLastGen</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getProportionalFitness(org.jgapcustomised.Chromosome)">getProportionalFitness</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>
<div class="block">Fitness of given Chromosome in proportion to maximum possible fitness.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HyperNEATEvolver.</span><code><strong><a href="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#logChamp(org.jgapcustomised.Chromosome,%20boolean,%20java.lang.String)">logChamp</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;champ,
        boolean&nbsp;force,
        java.lang.String&nbsp;label)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.misc">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</a> with type parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NSGAIISelector.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#population">population</a></strong></code>
<div class="block">The population to select from, set in <a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)"><code>NSGAIISelector.add(Configuration, List, List, Chromosome)</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">NSGAII.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAII.html#fastNonDominatedSort(java.util.List)">fastNonDominatedSort</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;individuals)</code>
<div class="block">Performs a fast non-domination sort of the specified individuals.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NSGAII.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAII.html#getTop(java.util.List,%20int)">getTop</a></strong>(java.util.List&lt;java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&gt;&nbsp;fronts,
      int&nbsp;numToSelect)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NSGAIISelector.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>
<div class="block">Modified version of <a href="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)"><code>NaturalSelector.select(Configuration)</code></a> that selects a number of
 elites and parents for each species based on the species fitness and the non-dominated sorting.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NSGAIISelector.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration,%20int)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
      int&nbsp;a_howManyToSelect)</code>
<div class="block">This method should not be used, only the <a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration)"><code>NSGAIISelector.select(Configuration)</code></a> method should be used.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">NSGAIISelector.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chroms)</code>
<div class="block">This method should not be used, only the <a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)"><code>NSGAIISelector.add(Configuration, List, List, Chromosome)</code></a> method
 should be used.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NSGAIISelector.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;species,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chroms,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;bestPerforming)</code>
<div class="block">Modified version of <a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)"><code>NaturalSelector.add(Configuration, List, List, Chromosome)</code></a> to prevent selecting elites from each
 species based on only a single objective.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NSGAIISelector.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;species,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chroms,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;bestPerforming)</code>
<div class="block">Modified version of <a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)"><code>NaturalSelector.add(Configuration, List, List, Chromosome)</code></a> to prevent selecting elites from each
 species based on only a single objective.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">NSGAII.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAII.html#fastNonDominatedSort(java.util.List)">fastNonDominatedSort</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;individuals)</code>
<div class="block">Performs a fast non-domination sort of the specified individuals.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NSGAII.</span><code><strong><a href="../../../com/ojcoleman/ahni/misc/NSGAII.html#getTop(java.util.List,%20int)">getTop</a></strong>(java.util.List&lt;java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&gt;&nbsp;fronts,
      int&nbsp;numToSelect)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="com.ojcoleman.ahni.transcriber">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">ESHyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.html#generateSubstrate(org.jgapcustomised.Chromosome)">generateSubstrate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>
<div class="block">Generate a substrate from a Chromosome according to the ES-HyperNEAT algorithm.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriber.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#getCPPN(org.jgapcustomised.Chromosome)">getCPPN</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><span class="strong">TranscriberAdaptor.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getNeuronParameterValues(org.jgapcustomised.Chromosome)">getNeuronParameterValues</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom)</code>
<div class="block">Gets the neuron parameter values for each parameter in each parameter class for the specified chromosome, in the
 format [class][param] (param order same as <a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#neuronParamNames"><code>TranscriberAdaptor.neuronParamNames</code></a>).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><span class="strong">TranscriberAdaptor.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getParameterValuesForCollection(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.integration.ParamCollection,%20double[][])">getParameterValuesForCollection</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom,
                               <a href="../../../com/ojcoleman/ahni/integration/ParamCollection.html" title="class in com.ojcoleman.ahni.integration">ParamCollection</a>&nbsp;collection,
                               double[][]&nbsp;values)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><span class="strong">TranscriberAdaptor.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#getSynapseParameterValues(org.jgapcustomised.Chromosome)">getSynapseParameterValues</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;chrom)</code>
<div class="block">Gets the synapse parameter values for each parameter in each parameter class the specified chromosome, in the
 format [class][param] (param order same as <a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamNames"><code>TranscriberAdaptor.synapseParamNames</code></a>).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">NEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html#newBainNN(org.jgapcustomised.Chromosome)">newBainNN</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>
<div class="block">create new <code>AnjiNet</code> from <code>genotype</code></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#newBainNN(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.nn.BainNN,%20java.util.Map)">newBainNN</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
         <a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a>&nbsp;substrate,
         java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</code>
<div class="block">Create a new neural network from a genotype.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberGridNet.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberGridNet.html#newGridNet(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.nn.GridNet)">newGridNet</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a>&nbsp;phenotype)</code>
<div class="block">Create a new neural network from the a genotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberGridNet.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberGridNet.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">NEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">ESHyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberGridNet.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberGridNet.html#transcribe(org.jgapcustomised.Chromosome,%20com.anji.integration.Activator)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.nn.BainNN)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">NEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.nn.BainNN)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">ESHyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.nn.BainNN)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a>&nbsp;substrate)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></code></td>
<td class="colLast"><span class="strong">HyperNEATTranscriberBain.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome,%20com.ojcoleman.ahni.nn.BainNN,%20java.util.Map)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype,
          <a href="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a>&nbsp;substrate,
          java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</a></code></td>
<td class="colLast"><span class="strong">TranscriberAdaptor.</span><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#transcribe(org.jgapcustomised.Chromosome,%20T,%20java.util.Map)">transcribe</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c,
          <a href="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</a>&nbsp;substrate,
          java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</code>
<div class="block">Sub-classes can override this method to convert the genotype to a phenotype, using an existing substrate object
 to allow performance gains by avoiding destroying and creating large objects or arrays, and optionally providing
 additional options.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#HyperNEATTranscriber.CPPN(org.jgapcustomised.Chromosome)">HyperNEATTranscriber.CPPN</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;genotype)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.jgapcustomised">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> declared as <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#bestPerforming">bestPerforming</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#bestPerforming">bestPerforming</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#fittest">fittest</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#elite">elite</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#m_chromosomes">m_chromosomes</a></strong></code>
<div class="block">Chromosomes that makeup thie Genotype's population.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#evolve()">evolve</a></strong>()</code>
<div class="block">Performs one generation cycle, evaluating fitness, selecting survivors, repopulting with offspring, and mutating
 new population.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getBestPerforming()">getBestPerforming</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getBestPerforming()">getBestPerforming</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getFittest()">getFittest</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getFittestChromosome()">getFittestChromosome</a></strong>()</code>
<div class="block">Retrieves the Chromosome in the population with the highest fitness value.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getPreviousBestPerforming()">getPreviousBestPerforming</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> that return types with arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#cullClones()">cullClones</a></strong>()</code>
<div class="block">Remove the clones from this species, as determined by <a href="../../../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)"><code>isEquivalent(Chromosome)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getChromosomes()">getChromosomes</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#getChromosomes()">getChromosomes</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getElite(double,%20int,%20org.jgapcustomised.Chromosome)">getElite</a></strong>(double&nbsp;proportion,
        int&nbsp;minToSelect,
        <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;bestPerforming)</code>
<div class="block">Determine and return the top specified proportion of chromosomes in this species.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getTop(int,%20boolean)">getTop</a></strong>(int&nbsp;numToSelect,
      boolean&nbsp;includeElites)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config)</code>
<div class="block">Select a given number of Chromosomes from the pool that will move on to the next generation population.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration,%20int)">select</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
      int&nbsp;numToSurvive)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#add(org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>
<div class="block">Add a chromosome to this species.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract void</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;species,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chroms,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;bestPerforming)</code>
<div class="block">If elitism is enabled, places appropriate chromosomes in <code>elite</code> list.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#addOrMoveFromCurrentSpecies(org.jgapcustomised.Chromosome)">addOrMoveFromCurrentSpecies</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>
<div class="block">Add the given Chromosome to this species, removing it from it's current species if necessary.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">ChromosomeFitnessComparator.</span><code><strong><a href="../../../org/jgapcustomised/ChromosomeFitnessComparator.html#compare(org.jgapcustomised.Chromosome,%20org.jgapcustomised.Chromosome)">compare</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c1,
       <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#cullToElites(org.jgapcustomised.Chromosome)">cullToElites</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;popBestPerforming)</code>
<div class="block">remove all non-elite chromosomes from this species, except for population-wide bestPerforming</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#distance(org.jgapcustomised.Chromosome,%20org.jgapcustomised.SpeciationParms)">distance</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;target,
        <a href="../../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;parms)</code>
<div class="block">Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#dominates(org.jgapcustomised.Chromosome)">dominates</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;otherChromosome)</code>
<div class="block">Checks whether this individual dominates the specified other individual, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract int</code></td>
<td class="colLast"><span class="strong">FitnessFunction.</span><code><strong><a href="../../../org/jgapcustomised/FitnessFunction.html#evaluate(org.jgapcustomised.Chromosome)">evaluate</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_subject)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Determine the fitness of the given Chromosome instance.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getChromosomeSpeciatedFitnessValue(org.jgapcustomised.Chromosome)">getChromosomeSpeciatedFitnessValue</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#getElite(double,%20int,%20org.jgapcustomised.Chromosome)">getElite</a></strong>(double&nbsp;proportion,
        int&nbsp;minToSelect,
        <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;bestPerforming)</code>
<div class="block">Determine and return the top specified proportion of chromosomes in this species.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">FitnessFunction.</span><code><strong><a href="../../../org/jgapcustomised/FitnessFunction.html#getFitnessValue(org.jgapcustomised.Chromosome)">getFitnessValue</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_subject)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Retrieves the fitness value of the given Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Chromosome.</span><code><strong><a href="../../../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)">isEquivalent</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c2)</code>
<div class="block">Compares this Chromosome against the specified object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#moveFromCurrentSpecies(org.jgapcustomised.Chromosome)">moveFromCurrentSpecies</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c)</code>
<div class="block">Move the given Chromosome from it's current species to this species.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#remove(org.jgapcustomised.Chromosome)">remove</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;aChromosome)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><span class="strong">CrossoverReproductionOperator.</span><code><strong><a href="../../../org/jgapcustomised/CrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration,%20org.jgapcustomised.Chromosome,%20org.jgapcustomised.Chromosome)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;dominantChrom,
         <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;recessiveChrom)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#setPreviousBestPerforming(org.jgapcustomised.Chromosome)">setPreviousBestPerforming</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;previousBestPerforming)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration,%20java.util.List,%20java.util.List,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;species,
   java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chroms,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;bestPerforming)</code>
<div class="block">If elitism is enabled, places appropriate chromosomes in <code>elite</code> list.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">Genotype.</span><code><strong><a href="../../../org/jgapcustomised/Genotype.html#addChromosomes(java.util.Collection)">addChromosomes</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;chromosomes)</code>
<div class="block">Add the specified chromosomes to this Genotype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#cull(java.util.Collection)">cull</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;keepers)</code>
<div class="block">Remove all chromosomes from this specie except <code>keepers</code> NOTE: this method does not update the species
 field in the removed Chromosomes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">NaturalSelector.</span><code><strong><a href="../../../org/jgapcustomised/NaturalSelector.html#ensureHighestPerformingIncluded(org.jgapcustomised.Species,%20java.util.List)">ensureHighestPerformingIncluded</a></strong>(<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&nbsp;s,
                               java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;list)</code>
<div class="block">Make sure population-wide best performing is included in list.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract void</code></td>
<td class="colLast"><span class="strong">BulkFitnessFunction.</span><code><strong><a href="../../../org/jgapcustomised/BulkFitnessFunction.html#evaluate(java.util.List)">evaluate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;subjects)</code>
<div class="block">Calculates and sets the fitness values on each of the given Chromosomes via their setFitnessValue() method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract void</code></td>
<td class="colLast"><span class="strong">ReproductionOperator.</span><code><strong><a href="../../../org/jgapcustomised/ReproductionOperator.html#reproduce(org.jgapcustomised.Configuration,%20java.util.List,%20int,%20java.util.List)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
         java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;parents,
         int&nbsp;numOffspring,
         java.util.List&lt;<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&gt;&nbsp;offspring)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">CrossoverReproductionOperator.</span><code><strong><a href="../../../org/jgapcustomised/CrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration,%20java.util.List,%20int,%20java.util.List)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
         java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;parentChroms,
         int&nbsp;numOffspring,
         java.util.List&lt;<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&gt;&nbsp;offspring)</code>
<div class="block">Adds new children of <code>parents</code> to <code>offspring</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SpeciationStrategy.</span><code><strong><a href="../../../org/jgapcustomised/SpeciationStrategy.html#respeciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">respeciate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genomeList,
          java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;specieList,
          <a href="../../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a>&nbsp;genotype)</code>
<div class="block">Speciates the genomes in genomeList into the provided species, starting from scratch.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Species.</span><code><strong><a href="../../../org/jgapcustomised/Species.html#setElites(java.util.List)">setElites</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;elites)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SpeciationStrategy.</span><code><strong><a href="../../../org/jgapcustomised/SpeciationStrategy.html#speciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">speciate</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;genomeList,
        java.util.List&lt;<a href="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&gt;&nbsp;specieList,
        <a href="../../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a>&nbsp;genotype)</code>
<div class="block">Speciates the genomes in genomeList into the provided species.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/Species.html#Species(org.jgapcustomised.SpeciationParms,%20org.jgapcustomised.Chromosome)">Species</a></strong>(<a href="../../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;aSpeciationParms,
       <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;first)</code>
<div class="block">Create new species with given Chromosome as first member of species.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/Genotype.html#Genotype(com.anji.util.Properties,%20org.jgapcustomised.Configuration,%20java.util.List)">Genotype</a></strong>(<a href="../../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a>&nbsp;props,
        <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
        java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;a_initialChromosomes)</code>
<div class="block">This constructor is used for random initial Genotypes.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.jgapcustomised.impl">
<!--   -->
</a>
<h3>Uses of <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a> in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> that return <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></code></td>
<td class="colLast"><span class="strong">ChromosomePool.</span><code><strong><a href="../../../org/jgapcustomised/impl/ChromosomePool.html#acquireChromosome()">acquireChromosome</a></strong>()</code>
<div class="block">Attempts to acquire an Chromosome instance from the chromosome pool.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> with parameters of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">WeightedRouletteSelector.</span><code><strong><a href="../../../org/jgapcustomised/impl/WeightedRouletteSelector.html#add(org.jgapcustomised.Configuration,%20org.jgapcustomised.Chromosome)">add</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfigurator,
   <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_chromosomeToAdd)</code>
<div class="block">Add a Chromosome instance to this selector's working pool of Chromosomes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ChromosomePool.</span><code><strong><a href="../../../org/jgapcustomised/impl/ChromosomePool.html#releaseChromosome(org.jgapcustomised.Chromosome)">releaseChromosome</a></strong>(<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;a_chromosome)</code>
<div class="block">Releases a Chromosome to the pool.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a> with type arguments of type <a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">CloneReproductionOperator.</span><code><strong><a href="../../../org/jgapcustomised/impl/CloneReproductionOperator.html#reproduce(org.jgapcustomised.Configuration,%20java.util.List,%20int,%20java.util.List)">reproduce</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;config,
         java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;parents,
         int&nbsp;numOffspring,
         java.util.List&lt;<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&gt;&nbsp;offspring)</code>
<div class="block">Adds new children of <code>parents</code> to <code>offspring</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">CloneReproductionOperator.</span><code><strong><a href="../../../org/jgapcustomised/impl/CloneReproductionOperator.html#reproduce(java.util.List,%20int,%20java.util.List)">reproduce</a></strong>(java.util.List&lt;<a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&gt;&nbsp;parents,
         int&nbsp;numOffspring,
         java.util.List&lt;<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&gt;&nbsp;offspring)</code>
<div class="block">Adds new children of <code>parents</code> to <code>offspring</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised/class-use/Chromosome.html" target="_top">Frames</a></li>
<li><a href="Chromosome.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
