<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Jul 05 16:56:47 EST 2013 -->
<TITLE>
Uses of Class org.jgapcustomised.Chromosome
</TITLE>

<META NAME="date" CONTENT="2013-07-05">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.jgapcustomised.Chromosome";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?org/jgapcustomised//class-useChromosome.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Chromosome.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Class<br>org.jgapcustomised.Chromosome</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.integration"><B>com.anji.integration</B></A></TD>
<TD><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.neat"><B>com.anji.neat</B></A></TD>
<TD><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.persistence"><B>com.anji.persistence</B></A></TD>
<TD><code>com.anji.persistence</code> contains classes implementing a simple persistence interface.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.anji.util"><B>com.anji.util</B></A></TD>
<TD><code>com.anji.util</code> contains utility classes used in the ANJI framework.&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.evaluation"><B>com.ojcoleman.ahni.evaluation</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.evaluation.mocostfunctions"><B>com.ojcoleman.ahni.evaluation.mocostfunctions</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.evaluation.novelty"><B>com.ojcoleman.ahni.evaluation.novelty</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.experiments"><B>com.ojcoleman.ahni.experiments</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.experiments.csb"><B>com.ojcoleman.ahni.experiments.csb</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.experiments.objectrecognition"><B>com.ojcoleman.ahni.experiments.objectrecognition</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.hyperneat"><B>com.ojcoleman.ahni.hyperneat</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.misc"><B>com.ojcoleman.ahni.misc</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#com.ojcoleman.ahni.transcriber"><B>com.ojcoleman.ahni.transcriber</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.jgapcustomised"><B>org.jgapcustomised</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#org.jgapcustomised.impl"><B>org.jgapcustomised.impl</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.integration"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/anji/integration/package-summary.html">com.anji.integration</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/integration/package-summary.html">com.anji.integration</A> that return types with arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleSelector.</B><B><A HREF="../../../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration)">select</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>SimpleSelector.</B><B><A HREF="../../../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration, int)">select</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration,
       int&nbsp;a_howManyToSelect)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the <code>a_howManyToSelect</code> chromosomes with highest fitness.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/integration/package-summary.html">com.anji.integration</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>SimpleSelector.</B><B><A HREF="../../../com/anji/integration/SimpleSelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfigurator,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;a_chromosomeToAdd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add <code>a_chromosomeToAdd</code> to set of chromosomes to be evaluated.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A></CODE></FONT></TD>
<TD><CODE><B>ActivatorTranscriber.</B><B><A HREF="../../../com/anji/integration/ActivatorTranscriber.html#newActivator(org.jgapcustomised.Chromosome)">newActivator</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;ch)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Constructs an <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration"><CODE>Activator</CODE></A> phenotype from a <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised"><CODE>Chromosome</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</A></CODE></FONT></TD>
<TD><CODE><B>AnjiNetTranscriber.</B><B><A HREF="../../../com/anji/integration/AnjiNetTranscriber.html#newAnjiNet(org.jgapcustomised.Chromosome)">newAnjiNet</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;create new <code>AnjiNet</code> from <code>genotype</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</A></CODE></FONT></TD>
<TD><CODE><B>AnjiNetTranscriber.</B><B><A HREF="../../../com/anji/integration/AnjiNetTranscriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</A></CODE></FONT></TD>
<TD><CODE><B>Transcriber.</B><B><A HREF="../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sub-classes must implement this method to convert the genotype to a phenotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</A></CODE></FONT></TD>
<TD><CODE><B>AnjiNetTranscriber.</B><B><A HREF="../../../com/anji/integration/AnjiNetTranscriber.html#transcribe(org.jgapcustomised.Chromosome, com.anji.integration.AnjiActivator)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</A></CODE></FONT></TD>
<TD><CODE><B>Transcriber.</B><B><A HREF="../../../com/anji/integration/Transcriber.html#transcribe(org.jgapcustomised.Chromosome, T)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c,
           <A HREF="../../../com/anji/integration/Transcriber.html" title="type parameter in Transcriber">T</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sub-classes must implement this method to convert the genotype to a phenotype, using an existing substrate object
 to allow performance gains by avoiding destroying and creating large objects or arrays.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../com/anji/integration/package-summary.html">com.anji.integration</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../com/anji/integration/XmlPersistableChromosome.html#XmlPersistableChromosome(org.jgapcustomised.Chromosome)">XmlPersistableChromosome</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;aChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.neat"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A> that return <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Evolver.</B><B><A HREF="../../../com/anji/neat/Evolver.html#getChamp()">getChamp</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/neat/package-summary.html">com.anji.neat</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A></CODE></FONT></TD>
<TD><CODE><B>NeatCrossoverReproductionOperator.</B><B><A HREF="../../../com/anji/neat/NeatCrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome, org.jgapcustomised.Chromosome)">reproduce</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
          <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;dominantChrom,
          <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;recessiveChrom)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Crossover according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> crossover
 methodology.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.persistence"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/anji/persistence/package-summary.html">com.anji.persistence</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/persistence/package-summary.html">com.anji.persistence</A> that return <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>FilePersistence.</B><B><A HREF="../../../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, org.w3c.dom.Node)">chromosomeFromXml</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
                  org.w3c.dom.Node&nbsp;chromNode)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>FilePersistence.</B><B><A HREF="../../../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, java.lang.String)">chromosomeFromXml</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
                  java.lang.String&nbsp;xml)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>FilePersistence.</B><B><A HREF="../../../com/anji/persistence/FilePersistence.html#loadChromosome(java.lang.String, org.jgapcustomised.Configuration)">loadChromosome</A></B>(java.lang.String&nbsp;id,
               <A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Persistence.</B><B><A HREF="../../../com/anji/persistence/Persistence.html#loadChromosome(java.lang.String, org.jgapcustomised.Configuration)">loadChromosome</A></B>(java.lang.String&nbsp;id,
               <A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>HibernatePersistence.</B><B><A HREF="../../../com/anji/persistence/HibernatePersistence.html#loadChromosome(java.lang.String, org.jgapcustomised.Configuration)">loadChromosome</A></B>(java.lang.String&nbsp;id,
               <A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/anji/persistence/package-summary.html">com.anji.persistence</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>FilePersistence.</B><B><A HREF="../../../com/anji/persistence/FilePersistence.html#store(org.jgapcustomised.Chromosome)">store</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Persistence.</B><B><A HREF="../../../com/anji/persistence/Persistence.html#store(org.jgapcustomised.Chromosome)">store</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>HibernatePersistence.</B><B><A HREF="../../../com/anji/persistence/HibernatePersistence.html#store(org.jgapcustomised.Chromosome)">store</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.anji.util"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/anji/util/package-summary.html">com.anji.util</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../com/anji/util/package-summary.html">com.anji.util</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>DummyBulkFitnessFunction.</B><B><A HREF="../../../com/anji/util/DummyBulkFitnessFunction.html#evaluate(java.util.List)">evaluate</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;aSubjects)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.evaluation"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Fields in <A HREF="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</A> declared as <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#lastBestChrom">lastBestChrom</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The chromosome with the best performance in the previous generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#newBestChrom">newBestChrom</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The chromosome with the current best performance.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Fields in <A HREF="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</A> with type parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.Iterator&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#chromosomesIterator">chromosomesIterator</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>TargetFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Evaluate an individual genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>HyperNEATTargetFitnessFunction.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>TargetFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method may optionally be overridden to perform an evaluation of an individual genotype and output the results to a log.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>HyperNEATTargetFitnessFunction.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>AHNIFitnessFunction.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/AHNIFitnessFunction.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method may optionally be overridden to perform an evaluation of an individual genotype and output the results to a log.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A></CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#generateSubstrate(org.jgapcustomised.Chromosome, com.anji.integration.Activator)">generateSubstrate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;chrom,
                  <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;A convenience method to generate a substrate from the given Chromosome using the configured transcriber for this
 fitness function.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#postEvaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">postEvaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
             <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
             int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sub-classes may override this method to perform operations after a Chromosome has been evaluated.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>HyperNEATFitnessFunction.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#postEvaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">postEvaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
             <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
             int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sub-classes may override this method to perform operations after a Chromosome has been evaluated.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunctionMT.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#evaluate(java.util.List)">evaluate</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;genotypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Evaluate a set of chromosomes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>HyperNEATFitnessFunction.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#evaluate(java.util.List)">evaluate</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;genotypes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Evaluate a set of chromosomes.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.evaluation.mocostfunctions"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>CPPNSizeCost.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>BainNNConnectionCountCost.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/mocostfunctions/BainNNConnectionCountCost.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.evaluation.novelty"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/evaluation/novelty/package-summary.html">com.ojcoleman.ahni.evaluation.novelty</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/evaluation/novelty/package-summary.html">com.ojcoleman.ahni.evaluation.novelty</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>GenericBehaviourEvaluator.</B><B><A HREF="../../../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.experiments"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/experiments/package-summary.html">com.ojcoleman.ahni.experiments</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/experiments/package-summary.html">com.ojcoleman.ahni.experiments</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>TMaze.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/TMaze.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>RLRecurrentNetworkBased.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>RLStateBased.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/RLStateBased.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>TMaze.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/TMaze.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>RLRecurrentNetworkBased.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>RLStateBased.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/RLStateBased.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.experiments.csb"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>RLContinuousStateBased.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         int&nbsp;evalThreadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>RLContinuousStateBased.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, java.lang.String, boolean, boolean)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate,
         java.lang.String&nbsp;baseFileName,
         boolean&nbsp;logText,
         boolean&nbsp;logImage)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.experiments.objectrecognition"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/experiments/objectrecognition/package-summary.html">com.ojcoleman.ahni.experiments.objectrecognition</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/experiments/objectrecognition/package-summary.html">com.ojcoleman.ahni.experiments.objectrecognition</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>ObjectRecognitionFitnessFunction2.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;activator,
         int&nbsp;threadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>ObjectRecognitionFitnessFunction3.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;activator,
         int&nbsp;threadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B>ObjectRecognitionFitnessFunction4.</B><B><A HREF="../../../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#evaluate(org.jgapcustomised.Chromosome, com.anji.integration.Activator, int)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
         <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;activator,
         int&nbsp;threadIndex)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.hyperneat"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Fields in <A HREF="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</A> declared as <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#bestPerforming">bestPerforming</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>[]</CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#bestPerformingChromosomes">bestPerformingChromosomes</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The best performing Chromosome from each generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#fittest">fittest</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>[]</CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#fittestChromosomes">fittestChromosomes</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The fittest Chromosome from each generation.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</A> that return <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getBestPerformingFromLastGen()">getBestPerformingFromLastGen</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>[]</CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getBesttPerformingChromosomesForEachGen()">getBesttPerformingChromosomesForEachGen</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns an array containing the best performing Chromosome from each generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>[]</CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getFittestChromosomesForEachGen()">getFittestChromosomesForEachGen</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns an array containing the fittest Chromosome from each generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getFittestFromLastGen()">getFittestFromLastGen</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#getProportionalFitness(org.jgapcustomised.Chromosome)">getProportionalFitness</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Fitness of given Chromosome in proportion to maximum possible fitness.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>HyperNEATEvolver.</B><B><A HREF="../../../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#logChamp(org.jgapcustomised.Chromosome, boolean)">logChamp</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;champ,
         boolean&nbsp;force)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.misc"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Fields in <A HREF="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</A> with type parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAIISelector.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#population">population</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The population to select from, set in <A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)"><CODE>NSGAIISelector.add(Configuration, List, List, Chromosome)</CODE></A>.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</A> that return types with arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAII.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAII.html#fastNonDominatedSort(java.util.List)">fastNonDominatedSort</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;individuals)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a fast non-domination sort of the specified individuals.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAII.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAII.html#getTop(java.util.List, int)">getTop</A></B>(java.util.List&lt;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&gt;&nbsp;fronts,
       int&nbsp;numToSelect)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAIISelector.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration)">select</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Modified version of <A HREF="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)"><CODE>NaturalSelector.select(Configuration)</CODE></A> that selects a number of
 elites and parents for each species based on the species fitness and the non-dominated sorting.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAIISelector.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration, int)">select</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration,
       int&nbsp;a_howManyToSelect)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method should not be used, only the <A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration)"><CODE>NSGAIISelector.select(Configuration)</CODE></A> method should be used.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>NSGAIISelector.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;chroms)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This method should not be used, only the <A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)"><CODE>NSGAIISelector.add(Configuration, List, List, Chromosome)</CODE></A> method
 should be used.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>NSGAIISelector.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt;&nbsp;species,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chroms,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;bestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Modified version of <A HREF="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)"><CODE>NaturalSelector.add(Configuration, List, List, Chromosome)</CODE></A> to prevent selecting elites from each
 species based on only a single objective.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>NSGAIISelector.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAIISelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt;&nbsp;species,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chroms,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;bestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Modified version of <A HREF="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)"><CODE>NaturalSelector.add(Configuration, List, List, Chromosome)</CODE></A> to prevent selecting elites from each
 species based on only a single objective.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAII.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAII.html#fastNonDominatedSort(java.util.List)">fastNonDominatedSort</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;individuals)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs a fast non-domination sort of the specified individuals.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NSGAII.</B><B><A HREF="../../../com/ojcoleman/ahni/misc/NSGAII.html#getTop(java.util.List, int)">getTop</A></B>(java.util.List&lt;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&gt;&nbsp;fronts,
       int&nbsp;numToSelect)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="com.ojcoleman.ahni.transcriber"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>ESHyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.html#generateSubstrate(org.jgapcustomised.Chromosome)">generateSubstrate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Generate a substrate from a Chromosome according to the ES-HyperNEAT algorithm.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>NEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html#newBainNN(org.jgapcustomised.Chromosome)">newBainNN</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;create new <code>AnjiNet</code> from <code>genotype</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#newBainNN(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.nn.BainNN, java.util.Map)">newBainNN</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
          <A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>&nbsp;substrate,
          java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a new neural network from a genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberGridNet.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberGridNet.html#newGridNet(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.nn.GridNet)">newGridNet</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>&nbsp;phenotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a new neural network from the a genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>ESHyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberGridNet.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberGridNet.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>NEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberGridNet.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberGridNet.html#transcribe(org.jgapcustomised.Chromosome, com.anji.integration.Activator)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/anji/integration/Activator.html" title="interface in com.anji.integration">Activator</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>ESHyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.nn.BainNN)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.nn.BainNN)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>NEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/NEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.nn.BainNN)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>&nbsp;substrate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A></CODE></FONT></TD>
<TD><CODE><B>HyperNEATTranscriberBain.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#transcribe(org.jgapcustomised.Chromosome, com.ojcoleman.ahni.nn.BainNN, java.util.Map)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype,
           <A HREF="../../../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>&nbsp;substrate,
           java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</A></CODE></FONT></TD>
<TD><CODE><B>TranscriberAdaptor.</B><B><A HREF="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#transcribe(org.jgapcustomised.Chromosome, T, java.util.Map)">transcribe</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c,
           <A HREF="../../../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="type parameter in TranscriberAdaptor">T</A>&nbsp;substrate,
           java.util.Map&lt;java.lang.String,java.lang.Object&gt;&nbsp;options)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sub-classes can override this method to convert the genotype to a phenotype, using an existing substrate object
 to allow performance gains by avoiding destroying and creating large objects or arrays, and optionally providing
 additional options.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#HyperNEATTranscriber.CPPN(org.jgapcustomised.Chromosome)">HyperNEATTranscriber.CPPN</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;genotype)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.jgapcustomised"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Fields in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> declared as <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#bestPerforming">bestPerforming</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#bestPerforming">bestPerforming</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#fittest">fittest</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Fields in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with type parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#elite">elite</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#m_chromosomes">m_chromosomes</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Chromosomes that makeup thie Genotype's population.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> that return <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#evolve()">evolve</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs one generation cycle, evaluating fitness, selecting survivors, repopulting with offspring, and mutating
 new population.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getBestPerforming()">getBestPerforming</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#getBestPerforming()">getBestPerforming</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#getFittest()">getFittest</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#getFittestChromosome()">getFittestChromosome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the Chromosome in the population with the highest fitness value.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getPreviousBestPerforming()">getPreviousBestPerforming</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getRepresentative()">getRepresentative</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> that return types with arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#cullClones()">cullClones</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remove the clones from this species, as determined by <A HREF="../../../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)"><CODE>isEquivalent(Chromosome)</CODE></A>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getChromosomes()">getChromosomes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#getChromosomes()">getChromosomes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getElite(double, int, org.jgapcustomised.Chromosome)">getElite</A></B>(double&nbsp;proportion,
         int&nbsp;minToSelect,
         <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;bestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Determine and return the top specified proportion of chromosomes in this species.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getTop(int, boolean)">getTop</A></B>(int&nbsp;numToSelect,
       boolean&nbsp;includeElites)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)">select</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Select a given number of Chromosomes from the pool that will move on to the next generation population.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration, int)">select</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
       int&nbsp;numToSurvive)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#add(org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;aChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt;&nbsp;species,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chroms,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;bestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;If elitism is enabled, places appropriate chromosomes in <code>elite</code> list.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#cullToElites(org.jgapcustomised.Chromosome)">cullToElites</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;popBestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;remove all non-elite chromosomes from this species, except for population-wide bestPerforming</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>Chromosome.</B><B><A HREF="../../../org/jgapcustomised/Chromosome.html#distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">distance</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;target,
         <A HREF="../../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>&nbsp;parms)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Chromosome.</B><B><A HREF="../../../org/jgapcustomised/Chromosome.html#dominates(org.jgapcustomised.Chromosome)">dominates</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;otherChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Checks whether this individual dominates the specified other individual, i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;int</CODE></FONT></TD>
<TD><CODE><B>FitnessFunction.</B><B><A HREF="../../../org/jgapcustomised/FitnessFunction.html#evaluate(org.jgapcustomised.Chromosome)">evaluate</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;a_subject)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;Determine the fitness of the given Chromosome instance.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getChromosomeSpeciatedFitnessValue(org.jgapcustomised.Chromosome)">getChromosomeSpeciatedFitnessValue</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;aChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#getElite(double, int, org.jgapcustomised.Chromosome)">getElite</A></B>(double&nbsp;proportion,
         int&nbsp;minToSelect,
         <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;bestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Determine and return the top specified proportion of chromosomes in this species.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>FitnessFunction.</B><B><A HREF="../../../org/jgapcustomised/FitnessFunction.html#getFitnessValue(org.jgapcustomised.Chromosome)">getFitnessValue</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;a_subject)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<B>Deprecated.</B>&nbsp;Retrieves the fitness value of the given Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Chromosome.</B><B><A HREF="../../../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)">isEquivalent</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Chromosome against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#match(org.jgapcustomised.Chromosome)">match</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;aChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#remove(org.jgapcustomised.Chromosome)">remove</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;aChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A></CODE></FONT></TD>
<TD><CODE><B>CrossoverReproductionOperator.</B><B><A HREF="../../../org/jgapcustomised/CrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome, org.jgapcustomised.Chromosome)">reproduce</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
          <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;dominantChrom,
          <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;recessiveChrom)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#setPreviousBestPerforming(org.jgapcustomised.Chromosome)">setPreviousBestPerforming</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;previousBestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>NaturalSelector.</B><B><A HREF="../../../org/jgapcustomised/NaturalSelector.html#add(org.jgapcustomised.Configuration, java.util.List, java.util.List, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt;&nbsp;species,
    java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chroms,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;bestPerforming)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;If elitism is enabled, places appropriate chromosomes in <code>elite</code> list.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Genotype.</B><B><A HREF="../../../org/jgapcustomised/Genotype.html#addChromosomes(java.util.Collection)">addChromosomes</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chromosomes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add the specified chromosomes to this Genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#cull(java.util.Collection)">cull</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;keepers)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Remove all chromosomes from this specie except <code>keepers</code> NOTE: this method does not update the species
 field in the removed Chromosomes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>abstract &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>BulkFitnessFunction.</B><B><A HREF="../../../org/jgapcustomised/BulkFitnessFunction.html#evaluate(java.util.List)">evaluate</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;subjects)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates and sets the fitness values on each of the given Chromosomes via their setFitnessValue() method.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>CrossoverReproductionOperator.</B><B><A HREF="../../../org/jgapcustomised/CrossoverReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, java.util.List, int, java.util.List)">reproduce</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;parentChroms,
          int&nbsp;numOffspring,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&gt;&nbsp;offspring)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds new children of <code>parents</code> to <code>offspring</code>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ReproductionOperator.</B><B><A HREF="../../../org/jgapcustomised/ReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, java.util.List, int, java.util.List)">reproduce</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;parents,
          int&nbsp;numOffspring,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&gt;&nbsp;offspring)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>Species.</B><B><A HREF="../../../org/jgapcustomised/Species.html#setElites(java.util.List)">setElites</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;elites)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructors in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/jgapcustomised/Species.html#Species(org.jgapcustomised.SpeciationParms, org.jgapcustomised.Chromosome)">Species</A></B>(<A HREF="../../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>&nbsp;aSpeciationParms,
        <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;aRepresentative)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create new specie from representative.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Constructor parameters in <A HREF="../../../org/jgapcustomised/package-summary.html">org.jgapcustomised</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../org/jgapcustomised/Genotype.html#Genotype(com.anji.util.Properties, org.jgapcustomised.Configuration, java.util.List)">Genotype</A></B>(<A HREF="../../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>&nbsp;props,
         <A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration,
         java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;a_initialChromosomes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This constructor is used for random initial Genotypes.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="org.jgapcustomised.impl"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A> that return <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B>ChromosomePool.</B><B><A HREF="../../../org/jgapcustomised/impl/ChromosomePool.html#acquireChromosome()">acquireChromosome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Attempts to acquire an Chromosome instance from the chromosome pool.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A> with parameters of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>WeightedRouletteSelector.</B><B><A HREF="../../../org/jgapcustomised/impl/WeightedRouletteSelector.html#add(org.jgapcustomised.Configuration, org.jgapcustomised.Chromosome)">add</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfigurator,
    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;a_chromosomeToAdd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add a Chromosome instance to this selector's working pool of Chromosomes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>ChromosomePool.</B><B><A HREF="../../../org/jgapcustomised/impl/ChromosomePool.html#releaseChromosome(org.jgapcustomised.Chromosome)">releaseChromosome</A></B>(<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;a_chromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Releases a Chromosome to the pool.</TD>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Method parameters in <A HREF="../../../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A> with type arguments of type <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>CloneReproductionOperator.</B><B><A HREF="../../../org/jgapcustomised/impl/CloneReproductionOperator.html#reproduce(org.jgapcustomised.Configuration, java.util.List, int, java.util.List)">reproduce</A></B>(<A HREF="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;config,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;parents,
          int&nbsp;numOffspring,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&gt;&nbsp;offspring)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds new children of <code>parents</code> to <code>offspring</code>.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>CloneReproductionOperator.</B><B><A HREF="../../../org/jgapcustomised/impl/CloneReproductionOperator.html#reproduce(java.util.List, int, java.util.List)">reproduce</A></B>(java.util.List&lt;<A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;parents,
          int&nbsp;numOffspring,
          java.util.List&lt;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&gt;&nbsp;offspring)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Adds new children of <code>parents</code> to <code>offspring</code>.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?org/jgapcustomised//class-useChromosome.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Chromosome.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
