<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Mar 22 11:50:05 EST 2013 -->
<TITLE>
M-Index
</TITLE>

<META NAME="date" CONTENT="2013-03-22">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="M-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-11.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-13.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-12.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-12.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">K</A> <A HREF="index-11.html">L</A> <A HREF="index-12.html">M</A> <A HREF="index-13.html">N</A> <A HREF="index-14.html">O</A> <A HREF="index-15.html">P</A> <A HREF="index-16.html">Q</A> <A HREF="index-17.html">R</A> <A HREF="index-18.html">S</A> <A HREF="index-19.html">T</A> <A HREF="index-20.html">U</A> <A HREF="index-21.html">V</A> <A HREF="index-22.html">W</A> <A HREF="index-23.html">X</A> <A HREF="index-24.html">Y</A> <A HREF="index-25.html">Z</A> <HR>
<A NAME="_M_"><!-- --></A><H2>
<B>M</B></H2>
<DL>
<DT><A HREF="../org/jgapcustomised/Genotype.html#m_activeConfiguration"><B>m_activeConfiguration</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>The current active Configuration instance.
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#m_activeConfiguration"><B>m_activeConfiguration</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>The current active configuration that is in use.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#m_activeConfiguration"><B>m_activeConfiguration</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>The current active configuration that is in use.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#m_boundsUnitsToIntegerUnits"><B>m_boundsUnitsToIntegerUnits</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Stores the number of integer range units that a single bounds-range unit represents.
<DT><A HREF="../org/jgapcustomised/Genotype.html#m_chromosomes"><B>m_chromosomes</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>Chromosomes that makeup thie Genotype's population.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#m_fitnessValue"><B>m_fitnessValue</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>Stores the fitness value of this Chromosome as determined by the active fitness function.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#m_isSelectedForNextGeneration"><B>m_isSelectedForNextGeneration</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#m_lowerBounds"><B>m_lowerBounds</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>The lower bounds of values represented by this Gene.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#m_performanceValue"><B>m_performanceValue</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Genotype.html#m_species"><B>m_species</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>Species that makeup this Genotype's population.
<DT><A HREF="../org/jgapcustomised/Genotype.html#m_specParms"><B>m_specParms</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#m_upperBounds"><B>m_upperBounds</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>The upper bounds of values represented by this Gene.
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#m_value"><B>m_value</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>References the internal boolean value of this Gene.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#m_value"><B>m_value</B></A> - 
Variable in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>References the internal integer value (allele) of this Gene.
<DT><A HREF="../com/anji/integration/ChromosomeToNetworkXml.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.anji.integration.<A HREF="../com/anji/integration/ChromosomeToNetworkXml.html" title="class in com.anji.integration">ChromosomeToNetworkXml</A>
<DD>Saves to persistence network XML representation of chromosomes.
<DT><A HREF="../com/anji/integration/Evaluator.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.anji.integration.<A HREF="../com/anji/integration/Evaluator.html" title="class in com.anji.integration">Evaluator</A>
<DD>main method
<DT><A HREF="../com/anji/neat/NeatActivator.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.anji.neat.<A HREF="../com/anji/neat/NeatActivator.html" title="class in com.anji.neat">NeatActivator</A>
<DD>Loads chromosome from persistence, transcribes it into activator, and displays activation.
<DT><A HREF="../com/anji/neat/Run.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.anji.neat.<A HREF="../com/anji/neat/Run.html" title="class in com.anji.neat">Run</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SigmoidActivationFunction.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/SigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">SigmoidActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/util/Reset.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.anji.util.<A HREF="../com/anji/util/Reset.html" title="class in com.anji.util">Reset</A>
<DD>Interface to execute via command line.
<DT><A HREF="../com/ojcoleman/ahni/hyperneat/Run.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.ojcoleman.ahni.hyperneat.<A HREF="../com/ojcoleman/ahni/hyperneat/Run.html" title="class in com.ojcoleman.ahni.hyperneat">Run</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/GridNet.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/util/ParameterTuner.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/ParameterTuner.html" title="class in com.ojcoleman.ahni.util">ParameterTuner</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/util/PostProcess.html#main(java.lang.String[])"><B>main(String[])</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</A>
<DD>Utility method to process command line arguments, set-up file IO, and call required post-processing method.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#mapValueToWithinBounds()"><B>mapValueToWithinBounds()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Maps the value of this IntegerGene to within the bounds specified by
 
 the m_upperBounds and m_lowerBounds instance variables.
<DT><A HREF="../org/jgapcustomised/Species.html#match(org.jgapcustomised.Chromosome)"><B>match(Chromosome)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#MAX_ROTATE_KEY"><B>MAX_ROTATE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#MAX_STAGNANT_GENERATIONS_KEY"><B>MAX_STAGNANT_GENERATIONS_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#MAX_STAGNANT_MAINTAIN_FITTEST_GENERATIONS_KEY"><B>MAX_STAGNANT_MAINTAIN_FITTEST_GENERATIONS_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#MAX_THREADS_KEY"><B>MAX_THREADS_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>Property key for maximum number of threads to use for fitness evaluation (including transcription of
 genotype/cppn to phenotype/substrate).
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#maxConnectionWeight"><B>maxConnectionWeight</B></A> - 
Variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Genotype.html#maxSpeciesSize"><B>maxSpeciesSize</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#MIN_SCALE_KEY"><B>MIN_SCALE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#MIN_THREADS_KEY"><B>MIN_THREADS_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>Property key for minimum number of threads to use for fitness evaluation (including transcription of
 genotype/cppn to phenotype/substrate).
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#minConnectionWeight"><B>minConnectionWeight</B></A> - 
Variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#MINIMUM_AGE_KEY"><B>MINIMUM_AGE_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Genotype.html#minSpeciesSize"><B>minSpeciesSize</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/util/Misc.html" title="class in com.anji.util"><B>Misc</B></A> - Class in <A HREF="../com/anji/util/package-summary.html">com.anji.util</A><DD>&nbsp;<DT><A HREF="../com/anji/util/Misc.html#Misc()"><B>Misc()</B></A> - 
Constructor for class com.anji.util.<A HREF="../com/anji/util/Misc.html" title="class in com.anji.util">Misc</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#MULTI_KEY"><B>MULTI_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>Property key for specifying additional fitness function classes in
 a multi-objective evaluation.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#MULTI_PROBABILITY_KEY"><B>MULTI_PROBABILITY_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>Property key for specifying the probability of including the additional fitness functions in the 
 evaluations each generation.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#MULTI_WEIGHTING_KEY"><B>MULTI_WEIGHTING_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>Property key for specifying the relative weighting of additional fitness functions in
 a multi-objective evaluation.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#multiFitnessFunctionProbability"><B>multiFitnessFunctionProbability</B></A> - 
Variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#multiFitnessFunctions"><B>multiFitnessFunctions</B></A> - 
Variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#multiFitnessFunctionsWeights"><B>multiFitnessFunctionsWeights</B></A> - 
Variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/AddConnectionMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set, Set)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/AddConnectionMutationOperator.html" title="class in com.anji.neat">AddConnectionMutationOperator</A>
<DD>Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 connection mutation.
<DT><A HREF="../com/anji/neat/AddNeuronAnywhereMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set&lt;Allele&gt;, Set&lt;Allele&gt;)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/AddNeuronAnywhereMutationOperator.html" title="class in com.anji.neat">AddNeuronAnywhereMutationOperator</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/AddNeuronMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set&lt;Allele&gt;, Set&lt;Allele&gt;)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/AddNeuronMutationOperator.html" title="class in com.anji.neat">AddNeuronMutationOperator</A>
<DD>Adds connections according to <a href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> add
 node mutation.
<DT><A HREF="../com/anji/neat/PruneMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set, Set)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/PruneMutationOperator.html" title="class in com.anji.neat">PruneMutationOperator</A>
<DD>Traverse network flowing forward and backward to identify unvisited connections and neurons.
<DT><A HREF="../com/anji/neat/RemoveConnectionMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set, Set)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/RemoveConnectionMutationOperator.html" title="class in com.anji.neat">RemoveConnectionMutationOperator</A>
<DD>Removes, in ascending order of weight magnitude, those connections whose weight magnitude is less than the
 maximum weight to be removed.
<DT><A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set, Set)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat">SingleTopologicalMutationOperator</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/WeightMutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set&lt;Allele&gt;, Set&lt;Allele&gt;)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/WeightMutationOperator.html" title="class in com.anji.neat">WeightMutationOperator</A>
<DD>Removes from <code>genesToAdd</code> and adds to <code>genesToRemove</code> all connection genes that are
 modified.
<DT><A HREF="../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><B>mutate(Configuration, ChromosomeMaterial, Set&lt;Allele&gt;, Set&lt;Allele&gt;)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/MutationOperator.html" title="class in org.jgapcustomised">MutationOperator</A>
<DD>Leaves <code>material</code> unmodified, but updates <code>allelesToAdd</code> and <code>allelesToRemove</code>
 with modifications.
<DT><A HREF="../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, java.util.List)"><B>mutate(Configuration, List&lt;ChromosomeMaterial&gt;)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/MutationOperator.html" title="class in org.jgapcustomised">MutationOperator</A>
<DD>The <A HREF="../org/jgapcustomised/MutationOperator.html#mutate(org.jgapcustomised.Configuration, org.jgapcustomised.ChromosomeMaterial, java.util.Set, java.util.Set)"><CODE>MutationOperator.mutate(Configuration, ChromosomeMaterial, Set, Set)</CODE></A> method will be invoked on each of the given individuals.
<DT><A HREF="../org/jgapcustomised/MutationOperator.html" title="class in org.jgapcustomised"><B>MutationOperator</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>Abstract class for mutation operators.<DT><A HREF="../org/jgapcustomised/MutationOperator.html#MutationOperator(double)"><B>MutationOperator(double)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/MutationOperator.html" title="class in org.jgapcustomised">MutationOperator</A>
<DD>&nbsp;
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-11.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-13.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-12.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-12.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">K</A> <A HREF="index-11.html">L</A> <A HREF="index-12.html">M</A> <A HREF="index-13.html">N</A> <A HREF="index-14.html">O</A> <A HREF="index-15.html">P</A> <A HREF="index-16.html">Q</A> <A HREF="index-17.html">R</A> <A HREF="index-18.html">S</A> <A HREF="index-19.html">T</A> <A HREF="index-20.html">U</A> <A HREF="index-21.html">V</A> <A HREF="index-22.html">W</A> <A HREF="index-23.html">X</A> <A HREF="index-24.html">Y</A> <A HREF="index-25.html">Z</A> <HR>

</BODY>
</HTML>
