<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Mar 22 11:50:05 EST 2013 -->
<TITLE>
S-Index
</TITLE>

<META NAME="date" CONTENT="2013-03-22">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="S-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-17.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-19.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-18.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-18.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">K</A> <A HREF="index-11.html">L</A> <A HREF="index-12.html">M</A> <A HREF="index-13.html">N</A> <A HREF="index-14.html">O</A> <A HREF="index-15.html">P</A> <A HREF="index-16.html">Q</A> <A HREF="index-17.html">R</A> <A HREF="index-18.html">S</A> <A HREF="index-19.html">T</A> <A HREF="index-20.html">U</A> <A HREF="index-21.html">V</A> <A HREF="index-22.html">W</A> <A HREF="index-23.html">X</A> <A HREF="index-24.html">Y</A> <A HREF="index-25.html">Z</A> <HR>
<A NAME="_S_"><!-- --></A><H2>
<B>S</B></H2>
<DL>
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#SAVE_IMAGES_KEY"><B>SAVE_IMAGES_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#SAVE_IMAGES_KEY"><B>SAVE_IMAGES_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#scale(int, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)"><B>scale(int, int, HyperNEATTranscriber)</B></A> - 
Method in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>Allow a sub-class to make the necessary changes when a substrate scale occurs.
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html#scale(int, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)"><B>scale(int, int, HyperNEATTranscriber)</B></A> - 
Method in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction2</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#scale(int, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)"><B>scale(int, int, HyperNEATTranscriber)</B></A> - 
Method in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#scale(int, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)"><B>scale(int, int, HyperNEATTranscriber)</B></A> - 
Method in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/TestTargetFitnessFunction.html#scale(int, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)"><B>scale(int, int, HyperNEATTranscriber)</B></A> - 
Method in class com.ojcoleman.ahni.experiments.<A HREF="../com/ojcoleman/ahni/experiments/TestTargetFitnessFunction.html" title="class in com.ojcoleman.ahni.experiments">TestTargetFitnessFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_COUNT_KEY"><B>SCALE_COUNT_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>Property key for the maximum number of rescalings to perform.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_FACTOR_KEY"><B>SCALE_FACTOR_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>Property key for multiplicative amount to scale substrate by when the performance of the best individual in the
 population has reached the specified level.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_PERFORMANCE_KEY"><B>SCALE_PERFORMANCE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>Property key for the performance level required before a scaling is performed.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_RIP_KEY"><B>SCALE_RIP_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>Property key for whether the performance values should be recorded before the final scaling has been performed.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#scaleCount"><B>scaleCount</B></A> - 
Variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>The maximum number of rescalings to perform.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#scalePerformance"><B>scalePerformance</B></A> - 
Variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</A>
<DD>The performance level required before a scaling is performed.
<DT><A HREF="../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration, int)"><B>select(Configuration, int)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration">SimpleSelector</A>
<DD>Returns the <code>a_howManyToSelect</code> chromosomes with highest fitness.
<DT><A HREF="../org/jgapcustomised/impl/WeightedRouletteSelector.html#select(org.jgapcustomised.Configuration, int)"><B>select(Configuration, int)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/WeightedRouletteSelector.html" title="class in org.jgapcustomised.impl">WeightedRouletteSelector</A>
<DD>Select a given number of Chromosomes from the pool that will move on to the next generation population.
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)"><B>select(Configuration)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>Select a given number of Chromosomes from the pool that will move on to the next generation population.
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration, int)"><B>select(Configuration, int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#setAlleles(java.util.SortedSet)"><B>setAlleles(SortedSet&lt;Allele&gt;)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>for hibernate
<DT><A HREF="../org/jgapcustomised/Configuration.html#setBulkFitnessFunction(org.jgapcustomised.BulkFitnessFunction)"><B>setBulkFitnessFunction(BulkFitnessFunction)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>Sets the bulk fitness function to be used for this genetic algorithm.
<DT><A HREF="../com/ojcoleman/ahni/util/Point.html#setCoordinates(double, double, double)"><B>setCoordinates(double, double, double)</B></A> - 
Method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/Point.html" title="class in com.ojcoleman.ahni.util">Point</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html#setCoords(int, double, double)"><B>setCoords(int, double, double)</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</A>
<DD>Sets the coordinate of the specified neuron.
<DT><A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html#setCoords(int, double, double, double)"><B>setCoords(int, double, double, double)</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</A>
<DD>Sets the coordinate of the specified neuron.
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setElitismMinSpeciesSize(int)"><B>setElitismMinSpeciesSize(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setElitismMinToSelect(int)"><B>setElitismMinToSelect(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setElitismProportion(double)"><B>setElitismProportion(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Configuration.html#setEventManager(org.jgapcustomised.event.EventManager)"><B>setEventManager(EventManager)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>Sets the EventManager that is to be associated with this configuration.
<DT><A HREF="../com/ojcoleman/ahni/hyperneat/Properties.html#setEvolver(com.ojcoleman.ahni.hyperneat.HyperNEATEvolver)"><B>setEvolver(HyperNEATEvolver)</B></A> - 
Method in class com.ojcoleman.ahni.hyperneat.<A HREF="../com/ojcoleman/ahni/hyperneat/Properties.html" title="class in com.ojcoleman.ahni.hyperneat">Properties</A>
<DD>Set the HyperNEATEvolver that is handling this run.
<DT><A HREF="../org/jgapcustomised/Configuration.html#setFitnessFunction(org.jgapcustomised.FitnessFunction)"><B>setFitnessFunction(FitnessFunction)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>Sets the fitness function to be used for this genetic algorithm.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#setFitnessValue(int)"><B>setFitnessValue(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>Sets the fitness value of this Chromosome.
<DT><A HREF="../com/anji/nn/Neuron.html#setId(long)"><B>setId(long)</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/NeuronConnection.html#setId(long)"><B>setId(long)</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/NeuronConnection.html" title="class in com.anji.nn">NeuronConnection</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Configuration.html#setIdFactory(org.jgapcustomised.IdFactory)"><B>setIdFactory(IdFactory)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Gene.html#setInnovationId(java.lang.Long)"><B>setInnovationId(Long)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised">Gene</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/AnjiActivator.html#setInputPattern(com.anji.nn.Pattern)"><B>setInputPattern(Pattern)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/GridNet.html#setInputs(double[][])"><B>setInputs(double[][])</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>
<DD>Set array to use as the input layer.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#setIsSelectedForNextGeneration(boolean)"><B>setIsSelectedForNextGeneration(boolean)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.
<DT><A HREF="../com/anji/integration/TargetFitnessFunction.html#setMaxFitnessValue(int)"><B>setMaxFitnessValue(int)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/TargetFitnessFunction.html" title="class in com.anji.integration">TargetFitnessFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#setMaxFitnessValue(int)"><B>setMaxFitnessValue(int)</B></A> - 
Method in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATTargetFitnessFunction</A>
<DD>Sets the maximum possible fitness value this function will return.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.html#setMaxFitnessValue(int)"><B>setMaxFitnessValue(int)</B></A> - 
Method in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.html" title="class in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator</A>
<DD>Sets the maximum possible fitness value this function will return.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#setMaxFitnessValue(int)"><B>setMaxFitnessValue(int)</B></A> - 
Method in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">TargetFitnessFunctionMT</A>
<DD>Sets the maximum possible fitness value this function will return.
<DT><A HREF="../org/jgapcustomised/FitnessFunction.html#setMaxFitnessValue(int)"><B>setMaxFitnessValue(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/FitnessFunction.html" title="class in org.jgapcustomised">FitnessFunction</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setMaxStagnantGenerations(int)"><B>setMaxStagnantGenerations(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setMaxStagnantGenerationsMaintainFittest(boolean)"><B>setMaxStagnantGenerationsMaintainFittest(boolean)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setMinAge(int)"><B>setMinAge(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ReproductionOperator.html#setMutateRate(double)"><B>setMutateRate(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ReproductionOperator.html" title="class in org.jgapcustomised">ReproductionOperator</A>
<DD>Set the probability that individuals produced by this operator will be a candidate for having mutations applied to them (independent of the mutation rates of MutationOperators).
<DT><A HREF="../org/jgapcustomised/MutationOperator.html#setMutationRate(double)"><B>setMutationRate(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/MutationOperator.html" title="class in org.jgapcustomised">MutationOperator</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/util/Properties.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class com.anji.util.<A HREF="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/GridNet.html#setName(java.lang.String)"><B>setName(String)</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Configuration.html#setNaturalSelector(org.jgapcustomised.NaturalSelector)"><B>setNaturalSelector(NaturalSelector)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>set selector
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#setNeuronDisabled(int, boolean)"><B>setNeuronDisabled(int, boolean)</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>Allows keeping a record of which neurons are disabled.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(double, double, double, com.ojcoleman.bain.base.NeuronCollection, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN, boolean)"><B>setNeuronParameters(double, double, double, NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</A>
<DD>Set the parameters for a neuron, specifying the neuron coordinates directly (see
 <A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(double, double, double)"><CODE>HyperNEATTranscriber.CPPN.setTargetCoordinates(double, double, double)</CODE></A>).
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(int, int, int, com.ojcoleman.bain.base.NeuronCollection, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN, boolean)"><B>setNeuronParameters(int, int, int, NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</A>
<DD>Set the parameters for a neuron, specifying the neuron coordinates with grid indices into the substrate (see
 <A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinatesFromGridIndices(int, int, int)"><CODE>HyperNEATTranscriber.CPPN.setTargetCoordinatesFromGridIndices(int, int, int)</CODE></A>).
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(com.ojcoleman.ahni.util.Point, com.ojcoleman.bain.base.NeuronCollection, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN, boolean)"><B>setNeuronParameters(Point, NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</A>
<DD>Set the parameters for a neuron, specifying the neuron coordinates with a Point (see
 <A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(com.ojcoleman.ahni.util.Point)"><CODE>HyperNEATTranscriber.CPPN.setTargetCoordinates(Point)</CODE></A>).
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(com.ojcoleman.bain.base.NeuronCollection, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN, boolean)"><B>setNeuronParameters(NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</A>
<DD>Set the parameters for a neuron.
<DT><A HREF="../com/anji/integration/AnjiActivator.html#setNumCycles(int)"><B>setNumCycles(int)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#setPatterns(double[][][], double[][][], double, double)"><B>setPatterns(double[][][], double[][][], double, double)</B></A> - 
Method in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATTargetFitnessFunction</A>
<DD>Set the input and target output pattern pairs to use for evaluations.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#setPatterns(double[][], double[][], double, double)"><B>setPatterns(double[][], double[][], double, double)</B></A> - 
Method in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">TargetFitnessFunctionMT</A>
<DD>Set the input and target output pattern pairs to use for evaluations.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#setPerformanceValue(double)"><B>setPerformanceValue(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Configuration.html#setPopulationSize(int)"><B>setPopulationSize(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>Sets the population size to be used for this genetic algorithm.
<DT><A HREF="../org/jgapcustomised/Species.html#setPreviousFittest(org.jgapcustomised.Chromosome)"><B>setPreviousFittest(Chromosome)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/RandomConnection.html#setRand(java.util.Random)"><B>setRand(Random)</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/RandomConnection.html" title="class in com.anji.nn">RandomConnection</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Configuration.html#setRandomGenerator(java.util.Random)"><B>setRandomGenerator(Random)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>Sets the random generator to be used for this genetic algorithm.
<DT><A HREF="../org/jgapcustomised/Configuration.html#setSampleChromosomeMaterial(org.jgapcustomised.ChromosomeMaterial)"><B>setSampleChromosomeMaterial(ChromosomeMaterial)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>Sets sample ChromosomeMaterial that is to be used as a guide for the construction of other Chromosomes.
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#setSecondaryParentId(java.lang.Long)"><B>setSecondaryParentId(Long)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#setShouldMutate(boolean)"><B>setShouldMutate(boolean)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>May be used by reproduction operators to indicate that this individual is a candidate for mutations.
<DT><A HREF="../org/jgapcustomised/ReproductionOperator.html#setSlice(double)"><B>setSlice(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ReproductionOperator.html" title="class in org.jgapcustomised">ReproductionOperator</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinates(double, double)"><B>setSourceCoordinates(double, double)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the source neuron in a two-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinates(double, double, double)"><B>setSourceCoordinates(double, double, double)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the source neuron in a three-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinates(com.ojcoleman.ahni.util.Point)"><B>setSourceCoordinates(Point)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the source neuron in a substrate with dimensions with range [0, 1] (coordinates are
 translated to a user-specified range if necessary, e.g.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinatesFromGridIndices(int, int, int)"><B>setSourceCoordinatesFromGridIndices(int, int, int)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the source neuron specified by the given indices into a grid-based substrate.
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setSpeciatedFitness(boolean)"><B>setSpeciatedFitness(boolean)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#setSpeciationTarget(int)"><B>setSpeciationTarget(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#setSpeciationThreshold(double)"><B>setSpeciationThreshold(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#setSpeciationThresholdChange(double)"><B>setSpeciationThresholdChange(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatCommonCoeff(double)"><B>setSpecieCompatCommonCoeff(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatDisjointCoeff(double)"><B>setSpecieCompatDisjointCoeff(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatExcessCoeff(double)"><B>setSpecieCompatExcessCoeff(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Species.html#setStagnantGenerationsCount(int)"><B>setStagnantGenerationsCount(int)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#setStepsPerStepForNonLayeredFF()"><B>setStepsPerStepForNonLayeredFF()</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>(Re)calculates the number of steps required to fully activate a non-layered feed-forward network.
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#setSurvivalRate(double)"><B>setSurvivalRate(double)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setSynapseParameters(com.ojcoleman.bain.base.SynapseCollection, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN, boolean, boolean)"><B>setSynapseParameters(SynapseCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean, boolean)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</A>
<DD>Set the parameters for a synapse.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(double, double)"><B>setTargetCoordinates(double, double)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the source neuron in a two-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(double, double, double)"><B>setTargetCoordinates(double, double, double)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the source neuron in a three-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(com.ojcoleman.ahni.util.Point)"><B>setTargetCoordinates(Point)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the target neuron in a substrate with dimensions with range [0, 1] (coordinates are
 translated to a user-specified range if necessary, e.g.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinatesFromGridIndices(int, int, int)"><B>setTargetCoordinatesFromGridIndices(int, int, int)</B></A> - 
Method in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>Set the coordinates of the target neuron specified by the given indices into a grid-based substrate.
<DT><A HREF="../com/anji/neat/ConnectionAllele.html#setToRandomValue(java.util.Random, boolean)"><B>setToRandomValue(Random, boolean)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>
<DD>Set weight to random value from a Gaussian distribution determined by <A HREF="../com/anji/neat/ConnectionAllele.html#RANDOM_STD_DEV"><CODE>ConnectionAllele.RANDOM_STD_DEV</CODE></A>
<DT><A HREF="../com/anji/neat/NeuronAllele.html#setToRandomValue(java.util.Random, boolean)"><B>setToRandomValue(Random, boolean)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Allele.html#setToRandomValue(java.util.Random, boolean)"><B>setToRandomValue(Random, boolean)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>
<DD>Sets the value of this Gene to a random legal value for the implementation.
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#setToRandomValue(java.util.Random, boolean)"><B>setToRandomValue(Random, boolean)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>Sets the value (allele) of this Gene to a random legal value.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#setToRandomValue(java.util.Random, boolean)"><B>setToRandomValue(Random, boolean)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Sets the value (allele) of this Gene to a random Integer value between
 
 the lower and upper bounds (if any) of this Gene.
<DT><A HREF="../com/anji/util/Reset.html#setUserInteraction(boolean)"><B>setUserInteraction(boolean)</B></A> - 
Method in class com.anji.util.<A HREF="../com/anji/util/Reset.html" title="class in com.anji.util">Reset</A>
<DD>default is false; set to true if you want it to delete everything without prompting user
<DT><A HREF="../com/anji/nn/Pattern.html#setValue(int, double)"><B>setValue(int, double)</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/Pattern.html" title="class in com.anji.nn">Pattern</A>
<DD>Set stimulus value at <code>idx</code> to <code>value</code>.
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#setValue(java.lang.Object)"><B>setValue(Object)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>Sets the value of this Gene to the new given value.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#setValue(java.lang.Integer)"><B>setValue(Integer)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Sets the value (allele) of this Gene to the new given value.
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#setValueFromPersistentRepresentation(java.lang.String)"><B>setValueFromPersistentRepresentation(String)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>Sets the value and internal state of this Gene from the string
 
 representation returned by a previous invocation of the
 
 getPersistentRepresentation() method.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#setValueFromPersistentRepresentation(java.lang.String)"><B>setValueFromPersistentRepresentation(String)</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Sets the value and internal state of this Gene from the string
 
 representation returned by a previous invocation of the
 
 getPersistentRepresentation() method.
<DT><A HREF="../com/anji/nn/Pattern.html#setValues(double[])"><B>setValues(double[])</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/Pattern.html" title="class in com.anji.nn">Pattern</A>
<DD>Set stimulus values.
<DT><A HREF="../com/anji/neat/ConnectionAllele.html#setWeight(double)"><B>setWeight(double)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/NeuronConnection.html#setWeight(double)"><B>setWeight(double)</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/NeuronConnection.html" title="class in com.anji.nn">NeuronConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#SHAPE_SIZE_KEY"><B>SHAPE_SIZE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#SHAPE_SIZE_KEY"><B>SHAPE_SIZE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#SHAPE_TYPE_KEY"><B>SHAPE_TYPE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#SHAPE_TYPE_KEY"><B>SHAPE_TYPE_KEY</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#shouldMutate()"><B>shouldMutate()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>Indicates whether this individual is a candidate for mutations (after reproduction).
<DT><A HREF="../com/anji/nn/activationfunction/SigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>SigmoidActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>Modified classic sigmoid.<DT><A HREF="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>SignedClampedLinearActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>&nbsp;<DT><A HREF="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html#SignedClampedLinearActivationFunction()"><B>SignedClampedLinearActivationFunction()</B></A> - 
Constructor for class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">SignedClampedLinearActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SignedStepActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>SignedStepActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>Step activation function.<DT><A HREF="../com/ojcoleman/ahni/util/ParameterTuner.html#significanceFactor"><B>significanceFactor</B></A> - 
Static variable in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/ParameterTuner.html" title="class in com.ojcoleman.ahni.util">ParameterTuner</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration"><B>SimpleSelector</B></A> - Class in <A HREF="../com/anji/integration/package-summary.html">com.anji.integration</A><DD>Selects chromosomes based directly on fitness value, as opposed to a statistical probability.<DT><A HREF="../com/anji/integration/SimpleSelector.html#SimpleSelector()"><B>SimpleSelector()</B></A> - 
Constructor for class com.anji.integration.<A HREF="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration">SimpleSelector</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SineActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>SineActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>Sine activation function.<DT><A HREF="../com/anji/util/Properties.html#singletonObjectProperty(java.lang.String)"><B>singletonObjectProperty(String)</B></A> - 
Method in class com.anji.util.<A HREF="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/util/Properties.html#singletonObjectProperty(java.lang.Class)"><B>singletonObjectProperty(Class&lt;T&gt;)</B></A> - 
Method in class com.anji.util.<A HREF="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat"><B>SingleTopologicalMutationOperator</B></A> - Class in <A HREF="../com/anji/neat/package-summary.html">com.anji.neat</A><DD>&nbsp;<DT><A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html#SingleTopologicalMutationOperator()"><B>SingleTopologicalMutationOperator()</B></A> - 
Constructor for class com.anji.neat.<A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat">SingleTopologicalMutationOperator</A>
<DD>should call <code>init()</code> after this constructor
<DT><A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html#SingleTopologicalMutationOperator(double, double, com.anji.nn.RecurrencyPolicy)"><B>SingleTopologicalMutationOperator(double, double, RecurrencyPolicy)</B></A> - 
Constructor for class com.anji.neat.<A HREF="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat">SingleTopologicalMutationOperator</A>
<DD>ctor
<DT><A HREF="../com/ojcoleman/ahni/util/CircularFifoBuffer.html#size()"><B>size()</B></A> - 
Method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="class in com.ojcoleman.ahni.util">CircularFifoBuffer</A>
<DD>Returns the number of elements in the buffer.
<DT><A HREF="../org/jgapcustomised/Chromosome.html#size()"><B>size()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>Returns the size of this Chromosome (the number of alleles it contains).
<DT><A HREF="../org/jgapcustomised/impl/Pool.html#size()"><B>size()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/Pool.html" title="class in org.jgapcustomised.impl">Pool</A>
<DD>Retrieves the number of objects currently available in this pool.
<DT><A HREF="../org/jgapcustomised/Species.html#size()"><B>size()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html#SKEWED"><B>SKEWED</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html" title="class in com.anji.neat">RemoveConnectionMutationOperator.Strategy</A>
<DD>likelihood of connection being removed is inversely proprtional to weight
<DT><A HREF="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html#SMALL"><B>SMALL</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html" title="class in com.anji.neat">RemoveConnectionMutationOperator.Strategy</A>
<DD>number of connections removed (n) is calculated based on total number of connections, and the n smallest
 connections are removed, excepting any whose weight is larger than <code>maxWeightRemoved</code>
<DT><A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.Connection.html#source"><B>source</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.Connection.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.Connection</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.TempConnection.html#sourcePoint"><B>sourcePoint</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.TempConnection.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.TempConnection</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Genotype.html#speciate()"><B>speciate()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A>
<DD>(Re)assigns all chromosomes to a species.
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#SPECIATED_FITNESS_KEY"><B>SPECIATED_FITNESS_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/NaturalSelector.html#speciatedFitness"><B>speciatedFitness</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#SPECIATION_TARGET_KEY"><B>SPECIATION_TARGET_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, speciation target (# species)
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#SPECIATION_THRESHOLD_CHANGE_KEY"><B>SPECIATION_THRESHOLD_CHANGE_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, amount to change speciation threshold to maintain speciation target
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#SPECIATION_THRESHOLD_KEY"><B>SPECIATION_THRESHOLD_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, speciation threshold
<DT><A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised"><B>SpeciationParms</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>&nbsp;<DT><A HREF="../org/jgapcustomised/SpeciationParms.html#SpeciationParms()"><B>SpeciationParms()</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>
<DD>default ctor
<DT><A HREF="../org/jgapcustomised/Species.html#SPECIE_TAG"><B>SPECIE_TAG</B></A> - 
Static variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>XML base tag
<DT><A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised"><B>Species</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>Species are reproductively isolated segments of a population.<DT><A HREF="../org/jgapcustomised/Species.html#Species(org.jgapcustomised.SpeciationParms, org.jgapcustomised.Chromosome)"><B>Species(SpeciationParms, Chromosome)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>Create new specie from representative.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html#squareErrors()"><B>squareErrors()</B></A> - 
Method in enum com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html" title="enum in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator.ErrorType</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html#squareTotalError()"><B>squareTotalError()</B></A> - 
Method in enum com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html" title="enum in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator.ErrorType</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatIdMap.html#SRC_NEURON_ID_XML_TAG"><B>SRC_NEURON_ID_XML_TAG</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</A>
<DD>XML tag for NEAT ID mapping, source neuron ID
<DT><A HREF="../com/anji/persistence/FilePersistence.html#startRun(java.lang.String)"><B>startRun(String)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/HibernatePersistence.html#startRun(java.lang.String)"><B>startRun(String)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/Persistence.html#startRun(java.lang.String)"><B>startRun(String)</B></A> - 
Method in interface com.anji.persistence.<A HREF="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</A>
<DD>Begin run.
<DT><A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html#STATE_COUNT_INCREASE_DELTA"><B>STATE_COUNT_INCREASE_DELTA</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.<A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html" title="class in com.ojcoleman.ahni.experiments">RLStateBased</A>
<DD>The amount to increase the number of states in the generated environments when the current size has been sufficiently mastered (see <A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html#DIFFICULTY_INCREASE_PERFORMANCE"><CODE>RLStateBased.DIFFICULTY_INCREASE_PERFORMANCE</CODE></A>.
<DT><A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html#STATE_COUNT_INITIAL"><B>STATE_COUNT_INITIAL</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.<A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html" title="class in com.ojcoleman.ahni.experiments">RLStateBased</A>
<DD>The initial number of states in the generated environments.
<DT><A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html#STATE_COUNT_MAX"><B>STATE_COUNT_MAX</B></A> - 
Static variable in class com.ojcoleman.ahni.experiments.<A HREF="../com/ojcoleman/ahni/experiments/RLStateBased.html" title="class in com.ojcoleman.ahni.experiments">RLStateBased</A>
<DD>The maximum amount to increase the number of states in the generated environments to.
<DT><A HREF="../com/ojcoleman/ahni/util/Statistics.html" title="class in com.ojcoleman.ahni.util"><B>Statistics</B></A> - Class in <A HREF="../com/ojcoleman/ahni/util/package-summary.html">com.ojcoleman.ahni.util</A><DD>Provides methods to perform statistical analysis on a set of results.<DT><A HREF="../com/ojcoleman/ahni/util/Statistics.html#Statistics(int)"><B>Statistics(int)</B></A> - 
Constructor for class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/Statistics.html" title="class in com.ojcoleman.ahni.util">Statistics</A>
<DD>Creates a new empty Statistics for run results consisting of the given number of generations.
<DT><A HREF="../com/ojcoleman/ahni/util/Statistics.html#Statistics(com.ojcoleman.ahni.util.Results)"><B>Statistics(Results)</B></A> - 
Constructor for class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/Statistics.html" title="class in com.ojcoleman.ahni.util">Statistics</A>
<DD>Creates a new Statistics based on the given results.
<DT><A HREF="../com/anji/nn/AnjiNet.html#step()"><B>step()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</A>
<DD>indicates a time step has passed
<DT><A HREF="../com/anji/nn/CacheNeuronConnection.html#step()"><B>step()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</A>
<DD>update value
<DT><A HREF="../com/anji/nn/Neuron.html#step()"><B>step()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</A>
<DD>indicates a time step has passed
<DT><A HREF="../com/anji/nn/StepHourglassConnection.html#step()"><B>step()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn">StepHourglassConnection</A>
<DD>proceed one step closer to completion
<DT><A HREF="../com/ojcoleman/ahni/nn/GridNet.html#step()"><B>step()</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>
<DD>Perform one activation step, consisting of cyclesPerStep steps, calculating new activation for all neurons.
<DT><A HREF="../com/anji/nn/activationfunction/StepActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>StepActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>Step activation function.<DT><A HREF="../com/ojcoleman/ahni/nn/GridNet.html#stepFF()"><B>stepFF()</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>
<DD>Perform one complete cycle for a feed forward network, propogating signal from input layer to output layer, only
 activating each layer once in sequence (connections can only exist from layer n to layer n+1).
<DT><A HREF="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn"><B>StepHourglassConnection</B></A> - Class in <A HREF="../com/anji/nn/package-summary.html">com.anji.nn</A><DD>StepHourglassConnection Connection returning a value between 0 and 1, where 1 indicates the fully allotted amount of
 time remaining and 0 indicates no time remainnig.<DT><A HREF="../com/anji/nn/StepHourglassConnection.html#StepHourglassConnection()"><B>StepHourglassConnection()</B></A> - 
Constructor for class com.anji.nn.<A HREF="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn">StepHourglassConnection</A>
<DD>ctor
<DT><A HREF="../com/anji/integration/TargetFitnessFunction.html#STIMULI_FILE_NAME_KEY"><B>STIMULI_FILE_NAME_KEY</B></A> - 
Static variable in class com.anji.integration.<A HREF="../com/anji/integration/TargetFitnessFunction.html" title="class in com.anji.integration">TargetFitnessFunction</A>
<DD>properties key, file containing strimuli
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#STIMULUS_SIZE_KEY"><B>STIMULUS_SIZE_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, dimension of neural net stimulus
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#store()"><B>store()</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>Store to persistence.
<DT><A HREF="../com/anji/neat/NeatIdMap.html#store()"><B>store()</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</A>
<DD>Persist ID factories and maps.
<DT><A HREF="../com/anji/persistence/FilePersistence.html#store(org.jgapcustomised.Chromosome)"><B>store(Chromosome)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/FilePersistence.html#store(com.anji.integration.Activator)"><B>store(Activator)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/FilePersistence.html#store(com.anji.run.Run)"><B>store(Run)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/HibernatePersistence.html#store(org.jgapcustomised.Chromosome)"><B>store(Chromosome)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/HibernatePersistence.html#store(com.anji.integration.Activator)"><B>store(Activator)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/HibernatePersistence.html#store(com.anji.run.Run)"><B>store(Run)</B></A> - 
Method in class com.anji.persistence.<A HREF="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/Persistence.html#store(org.jgapcustomised.Chromosome)"><B>store(Chromosome)</B></A> - 
Method in interface com.anji.persistence.<A HREF="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/Persistence.html#store(com.anji.integration.Activator)"><B>store(Activator)</B></A> - 
Method in interface com.anji.persistence.<A HREF="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/Persistence.html#store(com.anji.run.Run)"><B>store(Run)</B></A> - 
Method in interface com.anji.persistence.<A HREF="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/IdFactory.html#store()"><B>store()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/IdFactory.html" title="class in org.jgapcustomised">IdFactory</A>
<DD>persist object to file
<DT><A HREF="../com/anji/integration/PresentationEventListener.html#storeRun(boolean)"><B>storeRun(boolean)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/PresentationEventListener.html" title="class in com.anji.integration">PresentationEventListener</A>
<DD>Store/update run presentation data based on <code>genotype</code>.
<DT><A HREF="../com/anji/Copyright.html#STRING"><B>STRING</B></A> - 
Static variable in class com.anji.<A HREF="../com/anji/Copyright.html" title="class in com.anji">Copyright</A>
<DD>copyright string
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_CYCLES_PER_STEP"><B>SUBSTRATE_CYCLES_PER_STEP</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>For recurrent networks, the number of activation cycles to perform each time the substrate network is presented
 with new input and queried for its output.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_DEPTH"><B>SUBSTRATE_DEPTH</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>The number of layers in the substrate network.
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#SUBSTRATE_EXECUTION_MODE"><B>SUBSTRATE_EXECUTION_MODE</B></A> - 
Static variable in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_HEIGHT"><B>SUBSTRATE_HEIGHT</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>Comma separated list of the height of each layer in the network, including input and output layers and starting
 with the input layer.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#SUBSTRATE_MAX_RECURRENT_CYCLE"><B>SUBSTRATE_MAX_RECURRENT_CYCLE</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBain</A>
<DD>When determining if the substrate network is recurrent, the maximum cycle length to search for before the network
 is considered recurrent.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL"><B>SUBSTRATE_NEURON_MODEL</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify which neuron model to use.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS"><B>SUBSTRATE_NEURON_MODEL_PARAMS</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify CPPN outputs that set the parameters for each neuron.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_MAX"><B>SUBSTRATE_NEURON_MODEL_PARAMS_MAX</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify maximum values for the parameters for each neuron.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_MIN"><B>SUBSTRATE_NEURON_MODEL_PARAMS_MIN</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify minimum values for the parameters for each neuron.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD"><B>SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify the minimum CPPN output required to produce a non-zero parameter value.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_TYPES"><B>SUBSTRATE_NEURON_MODEL_TYPES</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify allowable neuron model types.
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#SUBSTRATE_SIMULATION_RESOLUTION"><B>SUBSTRATE_SIMULATION_RESOLUTION</B></A> - 
Static variable in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL"><B>SUBSTRATE_SYNAPSE_MODEL</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify which synapse model to use.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM"><B>SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify which parameter in the synapse model will be set to 0 if the connection should not be
 enabled or plastic.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS"><B>SUBSTRATE_SYNAPSE_MODEL_PARAMS</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify CPPN outputs that set the parameters for each synapse.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX"><B>SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify maximum values for the parameters for each synapse.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN"><B>SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify minimum values for the parameters for each synapse.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD"><B>SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify the minimum CPPN output required to produce a non-zero parameter value.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_TYPES"><B>SUBSTRATE_SYNAPSE_MODEL_TYPES</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Property to specify allowable synapse model types.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_WIDTH"><B>SUBSTRATE_WIDTH</B></A> - 
Static variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>Comma separated list of the width of each layer in the network, including input and output layers and starting
 with the input layer.
<DT><A HREF="../com/ojcoleman/ahni/util/ArrayUtil.html#sum(double[])"><B>sum(double[])</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/ArrayUtil.html" title="class in com.ojcoleman.ahni.util">ArrayUtil</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/util/ArrayUtil.html#sum(int[])"><B>sum(int[])</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/ArrayUtil.html" title="class in com.ojcoleman.ahni.util">ArrayUtil</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html#sumErrors()"><B>sumErrors()</B></A> - 
Method in enum com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html" title="enum in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator.ErrorType</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#SURVIVAL_RATE_KEY"><B>SURVIVAL_RATE_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, survival rate
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseDisableParamName"><B>synapseDisableParamName</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsMax"><B>synapseModelParamsMax</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The maximum values for each synapse model parameter.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsMin"><B>synapseModelParamsMin</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The minimum values for each synapse model parameter.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsRange"><B>synapseModelParamsRange</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The value range for each synapse model parameter (synapseModelParamsMax[i] - synapseModelParamsMin[i]).
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsThreshold"><B>synapseModelParamsThreshold</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>the minimum (possibly CPPN) output required to produce a non-zero parameter value.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeCount"><B>synapseModelTypeCount</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The number of available synapse types.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam"><B>synapseModelTypeParam</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The name of the parameter in the synapse model that specifies the type of a synapse.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypes"><B>synapseModelTypes</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The allowable values for the <A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam"><CODE>TranscriberAdaptor.synapseModelTypeParam</CODE></A> model parameter that specify the type of a synapse.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamNames"><B>synapseParamNames</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The names or labels for parameters in the synapse model.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamsEnabled"><B>synapseParamsEnabled</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Subclasses may use this to know whether synapse model parameters are enabled and available.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseTypesEnabled"><B>synapseTypesEnabled</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>Subclasses may use this to know whether multiple synapse types are enabled and available.
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-17.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-19.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-18.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-18.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">K</A> <A HREF="index-11.html">L</A> <A HREF="index-12.html">M</A> <A HREF="index-13.html">N</A> <A HREF="index-14.html">O</A> <A HREF="index-15.html">P</A> <A HREF="index-16.html">Q</A> <A HREF="index-17.html">R</A> <A HREF="index-18.html">S</A> <A HREF="index-19.html">T</A> <A HREF="index-20.html">U</A> <A HREF="index-21.html">V</A> <A HREF="index-22.html">W</A> <A HREF="index-23.html">X</A> <A HREF="index-24.html">Y</A> <A HREF="index-25.html">Z</A> <HR>

</BODY>
</HTML>
