<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Mar 22 11:50:04 EST 2013 -->
<TITLE>
C-Index
</TITLE>

<META NAME="date" CONTENT="2013-03-22">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="C-Index";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-2.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-4.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-3.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-3.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">K</A> <A HREF="index-11.html">L</A> <A HREF="index-12.html">M</A> <A HREF="index-13.html">N</A> <A HREF="index-14.html">O</A> <A HREF="index-15.html">P</A> <A HREF="index-16.html">Q</A> <A HREF="index-17.html">R</A> <A HREF="index-18.html">S</A> <A HREF="index-19.html">T</A> <A HREF="index-20.html">U</A> <A HREF="index-21.html">V</A> <A HREF="index-22.html">W</A> <A HREF="index-23.html">X</A> <A HREF="index-24.html">Y</A> <A HREF="index-25.html">Z</A> <HR>
<A NAME="_C_"><!-- --></A><H2>
<B>C</B></H2>
<DL>
<DT><A HREF="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn"><B>CacheNeuronConnection</B></A> - Class in <A HREF="../com/anji/nn/package-summary.html">com.anji.nn</A><DD>A connection between neurons that caches its value from previous timestep.<DT><A HREF="../com/anji/nn/CacheNeuronConnection.html#CacheNeuronConnection(com.anji.nn.Neuron)"><B>CacheNeuronConnection(Neuron)</B></A> - 
Constructor for class com.anji.nn.<A HREF="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/CacheNeuronConnection.html#CacheNeuronConnection(com.anji.nn.Neuron, double)"><B>CacheNeuronConnection(Neuron, double)</B></A> - 
Constructor for class com.anji.nn.<A HREF="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#calculateBoundsUnitsToIntegerUnitsRatio()"><B>calculateBoundsUnitsToIntegerUnitsRatio()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Calculates and sets the m_boundsUnitsToIntegerUnits field based
 
 on the current lower and upper bounds of this IntegerGene.
<DT><A HREF="../com/anji/integration/ErrorFunction.html#calculateError(double[][], double[][], boolean)"><B>calculateError(double[][], double[][], boolean)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/ErrorFunction.html" title="class in com.anji.integration">ErrorFunction</A>
<DD>Calculate the sum of differences between <code>responses</code> and target values.
<DT><A HREF="../com/anji/integration/ErrorFunction.html#calculateError(double[][][], double[][][], boolean)"><B>calculateError(double[][][], double[][][], boolean)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/ErrorFunction.html" title="class in com.anji.integration">ErrorFunction</A>
<DD>Calculate the sum of differences between <code>responses</code> and target values.
<DT><A HREF="../com/anji/integration/ExponentialTargetFitnessFunction.html#calculateErrorFitness(double[][], double, double)"><B>calculateErrorFitness(double[][], double, double)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/ExponentialTargetFitnessFunction.html" title="class in com.anji.integration">ExponentialTargetFitnessFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/TargetFitnessFunction.html#calculateErrorFitness(double[][], double, double)"><B>calculateErrorFitness(double[][], double, double)</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/TargetFitnessFunction.html" title="class in com.anji.integration">TargetFitnessFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatTargetFitnessFunction.html#calculateErrorFitness(double[][], double, double)"><B>calculateErrorFitness(double[][], double, double)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/NeatTargetFitnessFunction.html" title="class in com.anji.neat">NeatTargetFitnessFunction</A>
<DD>Subtract <code>responses</code> from targets, sum all differences, subtract from max fitness, and square result.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html#children"><B>children</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.QuadPoint</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#CHROM_COMPAT_COMMON_COEFF_KEY"><B>CHROM_COMPAT_COMMON_COEFF_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, speciation chromosome compatibility common coefficient
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#CHROM_COMPAT_DISJOINT_COEFF_KEY"><B>CHROM_COMPAT_DISJOINT_COEFF_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, speciation chromosome compatibility disjoint coefficient
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#CHROM_COMPAT_EXCESS_COEFF_KEY"><B>CHROM_COMPAT_EXCESS_COEFF_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, speciation chromosome compatibility excess coefficient
<DT><A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised"><B>Chromosome</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>Chromosomes represent potential solutions and consist of a fixed-length collection of genes.<DT><A HREF="../org/jgapcustomised/Chromosome.html#Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long)"><B>Chromosome(ChromosomeMaterial, Long)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.
<DT><A HREF="../org/jgapcustomised/Species.html#CHROMOSOME_TAG"><B>CHROMOSOME_TAG</B></A> - 
Static variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>XML chromosome tag
<DT><A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><B>ChromosomeFitnessComparator</B></A>&lt;<A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="type parameter in ChromosomeFitnessComparator">T</A>&gt; - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>Enables sorting of chromosomes by their fitness.<DT><A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html#ChromosomeFitnessComparator()"><B>ChromosomeFitnessComparator()</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised">ChromosomeFitnessComparator</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html#ChromosomeFitnessComparator(boolean, boolean)"><B>ChromosomeFitnessComparator(boolean, boolean)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised">ChromosomeFitnessComparator</A>
<DD>Enables sorting of chromosomes in order of fitness.
<DT><A HREF="../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, java.lang.String)"><B>chromosomeFromXml(Configuration, String)</B></A> - 
Static method in class com.anji.persistence.<A HREF="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, org.w3c.dom.Node)"><B>chromosomeFromXml(Configuration, Node)</B></A> - 
Static method in class com.anji.persistence.<A HREF="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised"><B>ChromosomeMaterial</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>This is the guts of the original Chromosome object, pulled out so the genes can be modified by genetic operators
 before creating the Chromosome object.<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)"><B>ChromosomeMaterial(Collection&lt;Allele&gt;, Long, Long)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>Create chromosome with two parents.
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long)"><B>ChromosomeMaterial(Collection&lt;Allele&gt;, Long)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>Create chromosome with one parents.
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection)"><B>ChromosomeMaterial(Collection&lt;Allele&gt;)</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>Create chromosome with no parents.
<DT><A HREF="../org/jgapcustomised/impl/ChromosomePool.html" title="class in org.jgapcustomised.impl"><B>ChromosomePool</B></A> - Class in <A HREF="../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A><DD>Provides a pooling mechanism for Chromosome instances so that discarded Chromosome instances can be recycled, thus
 saving memory and the overhead of constructing new ones from scratch each time.<DT><A HREF="../org/jgapcustomised/impl/ChromosomePool.html#ChromosomePool()"><B>ChromosomePool()</B></A> - 
Constructor for class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/ChromosomePool.html" title="class in org.jgapcustomised.impl">ChromosomePool</A>
<DD>Constructor.
<DT><A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#chromosomesIterator"><B>chromosomesIterator</B></A> - 
Variable in class com.ojcoleman.ahni.evaluation.<A HREF="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/ChromosomeToNetworkXml.html" title="class in com.anji.integration"><B>ChromosomeToNetworkXml</B></A> - Class in <A HREF="../com/anji/integration/package-summary.html">com.anji.integration</A><DD>Converts chromosome to activator, which it then uses to write network XML according to <a
 href="http://nevt.sourceforge.net/">NEVT </a> data model.<DT><A HREF="../com/anji/integration/ChromosomeToNetworkXml.html#ChromosomeToNetworkXml()"><B>ChromosomeToNetworkXml()</B></A> - 
Constructor for class com.anji.integration.<A HREF="../com/anji/integration/ChromosomeToNetworkXml.html" title="class in com.anji.integration">ChromosomeToNetworkXml</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="class in com.ojcoleman.ahni.util"><B>CircularFifoBuffer</B></A>&lt;<A HREF="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="type parameter in CircularFifoBuffer">E</A>&gt; - Class in <A HREF="../com/ojcoleman/ahni/util/package-summary.html">com.ojcoleman.ahni.util</A><DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/util/CircularFifoBuffer.html#CircularFifoBuffer(int)"><B>CircularFifoBuffer(int)</B></A> - 
Constructor for class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="class in com.ojcoleman.ahni.util">CircularFifoBuffer</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/ClampedAbsoluteActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>ClampedAbsoluteActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>Absolute activation function.<DT><A HREF="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>ClampedLinearActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>&nbsp;<DT><A HREF="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html#ClampedLinearActivationFunction()"><B>ClampedLinearActivationFunction()</B></A> - 
Constructor for class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">ClampedLinearActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/util/Properties.html#CLASS_SUFFIX"><B>CLASS_SUFFIX</B></A> - 
Static variable in class com.anji.util.<A HREF="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>
<DD>made public for unit tests
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#cleanup()"><B>cleanup()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>Executed by the genetic engine when this Gene instance is no
 
 longer needed and should perform any necessary resource cleanup.
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#cleanup()"><B>cleanup()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>Executed by the genetic engine when this Gene instance is no
 
 longer needed and should perform any necessary resource cleanup.
<DT><A HREF="../com/anji/nn/Pattern.html#clear()"><B>clear()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/Pattern.html" title="class in com.anji.nn">Pattern</A>
<DD>set values to 0
<DT><A HREF="../org/jgapcustomised/impl/Pool.html#clear()"><B>clear()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/Pool.html" title="class in org.jgapcustomised.impl">Pool</A>
<DD>Empties out this pool of all objects.
<DT><A HREF="../org/jgapcustomised/Species.html#clear()"><B>clear()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>Remove all chromosomes from this species.
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#clone(java.lang.Long)"><B>clone(Long)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>Returns a copy of this ChromosomeMaterial.
<DT><A HREF="../com/anji/neat/ConnectionAllele.html#cloneAllele()"><B>cloneAllele()</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeuronAllele.html#cloneAllele()"><B>cloneAllele()</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Allele.html#cloneAllele()"><B>cloneAllele()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/impl/BooleanAllele.html#cloneAllele()"><B>cloneAllele()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/impl/IntegerAllele.html#cloneAllele()"><B>cloneAllele()</B></A> - 
Method in class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Chromosome.html#cloneMaterial()"><B>cloneMaterial()</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#cloneOper"><B>cloneOper</B></A> - 
Variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/impl/CloneReproductionOperator.html" title="class in org.jgapcustomised.impl"><B>CloneReproductionOperator</B></A> - Class in <A HREF="../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</A><DD>Produces offspring be creating clones of parents.<DT><A HREF="../org/jgapcustomised/impl/CloneReproductionOperator.html#CloneReproductionOperator()"><B>CloneReproductionOperator()</B></A> - 
Constructor for class org.jgapcustomised.impl.<A HREF="../org/jgapcustomised/impl/CloneReproductionOperator.html" title="class in org.jgapcustomised.impl">CloneReproductionOperator</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/package-summary.html"><B>com.anji</B></A> - package com.anji<DD>&nbsp;<DT><A HREF="../com/anji/integration/package-summary.html"><B>com.anji.integration</B></A> - package com.anji.integration<DD><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.<DT><A HREF="../com/anji/neat/package-summary.html"><B>com.anji.neat</B></A> - package com.anji.neat<DD><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.<DT><A HREF="../com/anji/nn/package-summary.html"><B>com.anji.nn</B></A> - package com.anji.nn<DD><code>com.anji.nn</code> contains classes implementing a simple artificial neural net framework.<DT><A HREF="../com/anji/nn/activationfunction/package-summary.html"><B>com.anji.nn.activationfunction</B></A> - package com.anji.nn.activationfunction<DD>&nbsp;<DT><A HREF="../com/anji/persistence/package-summary.html"><B>com.anji.persistence</B></A> - package com.anji.persistence<DD><code>com.anji.persistence</code> contains classes implementing a simple persistence interface.<DT><A HREF="../com/anji/run/package-summary.html"><B>com.anji.run</B></A> - package com.anji.run<DD><code>com.anji.run</code> contains classes concerned with an evolutionary run.<DT><A HREF="../com/anji/util/package-summary.html"><B>com.anji.util</B></A> - package com.anji.util<DD><code>com.anji.util</code> contains utility classes used in the ANJI framework.<DT><A HREF="../com/ojcoleman/ahni/package-summary.html"><B>com.ojcoleman.ahni</B></A> - package com.ojcoleman.ahni<DD><strong>AHNI - Another HyperNEAT Implementation.</strong><DT><A HREF="../com/ojcoleman/ahni/evaluation/package-summary.html"><B>com.ojcoleman.ahni.evaluation</B></A> - package com.ojcoleman.ahni.evaluation<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html"><B>com.ojcoleman.ahni.evaluation.mocostfunctions</B></A> - package com.ojcoleman.ahni.evaluation.mocostfunctions<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/event/package-summary.html"><B>com.ojcoleman.ahni.event</B></A> - package com.ojcoleman.ahni.event<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/experiments/package-summary.html"><B>com.ojcoleman.ahni.experiments</B></A> - package com.ojcoleman.ahni.experiments<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/experiments/objectrecognition/package-summary.html"><B>com.ojcoleman.ahni.experiments.objectrecognition</B></A> - package com.ojcoleman.ahni.experiments.objectrecognition<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/hyperneat/package-summary.html"><B>com.ojcoleman.ahni.hyperneat</B></A> - package com.ojcoleman.ahni.hyperneat<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/nn/package-summary.html"><B>com.ojcoleman.ahni.nn</B></A> - package com.ojcoleman.ahni.nn<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/transcriber/package-summary.html"><B>com.ojcoleman.ahni.transcriber</B></A> - package com.ojcoleman.ahni.transcriber<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/util/package-summary.html"><B>com.ojcoleman.ahni.util</B></A> - package com.ojcoleman.ahni.util<DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/util/PostProcess.html#combineResults(java.lang.String, java.io.BufferedWriter)"><B>combineResults(String, BufferedWriter)</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</A>
<DD>Combine the results from multiple runs into one file.
<DT><A HREF="../com/ojcoleman/ahni/util/PostProcess.html#combineResults(java.lang.String, boolean)"><B>combineResults(String, boolean)</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</A>
<DD>Combine the results from multiple runs into one Results object.
<DT><A HREF="../com/anji/neat/WeightMagnitudeComparator.html#compare(java.lang.Object, java.lang.Object)"><B>compare(Object, Object)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/WeightMagnitudeComparator.html" title="class in com.anji.neat">WeightMagnitudeComparator</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html#compare(T, T)"><B>compare(T, T)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised">ChromosomeFitnessComparator</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Allele.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Chromosome.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>
<DD>Compares the given Chromosome to this Chromosome.
<DT><A HREF="../org/jgapcustomised/ChromosomeMaterial.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>
<DD>Compares the given Chromosome to this Chromosome.
<DT><A HREF="../org/jgapcustomised/Gene.html#compareTo(java.lang.Object)"><B>compareTo(Object)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised">Gene</A>
<DD>Comparison based on innovation ID.
<DT><A HREF="../com/ojcoleman/ahni/util/PostProcess.html#compressAverage(java.io.BufferedReader, java.io.BufferedWriter, int)"><B>compressAverage(BufferedReader, BufferedWriter, int)</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</A>
<DD>Given a result file from a set of runs, for each run computes averages over some number of generations (the
 window size) within the run.
<DT><A HREF="../com/ojcoleman/ahni/util/PostProcess.html#compressAverage(com.ojcoleman.ahni.util.Results, int)"><B>compressAverage(Results, int)</B></A> - 
Static method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</A>
<DD>Given a result file from a set of runs, for each run computes averages over some number of generations (the
 window size) within the run.
<DT><A HREF="../com/anji/util/Configurable.html" title="interface in com.anji.util"><B>Configurable</B></A> - Interface in <A HREF="../com/anji/util/package-summary.html">com.anji.util</A><DD>All classes implementing this interface must have a public no-argument constructor.<DT><A HREF="../com/ojcoleman/ahni/hyperneat/Configurable.html" title="interface in com.ojcoleman.ahni.hyperneat"><B>Configurable</B></A> - Interface in <A HREF="../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</A><DD>All classes implementing this interface must have a public no-argument constructor.<DT><A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised"><B>Configuration</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>The Configuration class represents the current configuration of plugins and flags necessary to execute the genetic
 algorithm (such as fitness function, natural selector, genetic operators, and so on).<DT><A HREF="../org/jgapcustomised/Configuration.html#Configuration()"><B>Configuration()</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/util/Properties.html#configureLogger()"><B>configureLogger()</B></A> - 
Method in class com.anji.util.<A HREF="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/XmlPersistableAllele.html#CONN_XML_TAG"><B>CONN_XML_TAG</B></A> - 
Static variable in class com.anji.integration.<A HREF="../com/anji/integration/XmlPersistableAllele.html" title="class in com.anji.integration">XmlPersistableAllele</A>
<DD>connection XML tag
<DT><A HREF="../com/anji/nn/Connection.html" title="interface in com.anji.nn"><B>Connection</B></A> - Interface in <A HREF="../com/anji/nn/package-summary.html">com.anji.nn</A><DD>Abstract connection to carry data into neurons.<DT><A HREF="../com/anji/neat/NeatIdMap.html#CONNECTION_ID_MAP_XML_TAG"><B>CONNECTION_ID_MAP_XML_TAG</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</A>
<DD>XML tag for NEAT ID mapping,connection ID
<DT><A HREF="../com/anji/neat/NeatIdMap.html#CONNECTION_ID_XML_TAG"><B>CONNECTION_ID_XML_TAG</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</A>
<DD>XML tag for NEAT ID mapping, connection ID
<DT><A HREF="../com/anji/neat/NeatIdMap.html#CONNECTION_TO_CONNECTION_XML_TAG"><B>CONNECTION_TO_CONNECTION_XML_TAG</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</A>
<DD>XML tag for NEAT ID mapping, connection
<DT><A HREF="../com/anji/neat/NeatIdMap.html#CONNECTION_TO_NEURON_XML_TAG"><B>CONNECTION_TO_NEURON_XML_TAG</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</A>
<DD>XML tag for NEAT ID mapping, neuron
<DT><A HREF="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat"><B>ConnectionAllele</B></A> - Class in <A HREF="../com/anji/neat/package-summary.html">com.anji.neat</A><DD>Gene corresponding to NEAT connection gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a><DT><A HREF="../com/anji/neat/ConnectionAllele.html#ConnectionAllele(com.anji.neat.ConnectionGene)"><B>ConnectionAllele(ConnectionGene)</B></A> - 
Constructor for class com.anji.neat.<A HREF="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/AddConnectionMutationOperator.html#connectionAllowed(com.anji.neat.NeuronAllele, com.anji.neat.NeuronAllele, java.util.Collection)"><B>connectionAllowed(NeuronAllele, NeuronAllele, Collection&lt;ConnectionAllele&gt;)</B></A> - 
Method in class com.anji.neat.<A HREF="../com/anji/neat/AddConnectionMutationOperator.html" title="class in com.anji.neat">AddConnectionMutationOperator</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#connectionExprThresh"><B>connectionExprThresh</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>The minimum output from the CPPN required to create a non-zero weight value in the substrate.
<DT><A HREF="../com/anji/integration/XmlPersistableAllele.html#connectionFromXml(org.w3c.dom.Node)"><B>connectionFromXml(Node)</B></A> - 
Static method in class com.anji.integration.<A HREF="../com/anji/integration/XmlPersistableAllele.html" title="class in com.anji.integration">XmlPersistableAllele</A>
<DD>Convert from XML to <code>ConnectionGene</code> object
<DT><A HREF="../com/anji/neat/ConnectionGene.html" title="class in com.anji.neat"><B>ConnectionGene</B></A> - Class in <A HREF="../com/anji/neat/package-summary.html">com.anji.neat</A><DD>Gene corresponding to NEAT connection gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a><DT><A HREF="../com/anji/neat/ConnectionGene.html#ConnectionGene(java.lang.Long, java.lang.Long, java.lang.Long)"><B>ConnectionGene(Long, Long, Long)</B></A> - 
Constructor for class com.anji.neat.<A HREF="../com/anji/neat/ConnectionGene.html" title="class in com.anji.neat">ConnectionGene</A>
<DD>Construct new ConnectionGene with given src, destination, and ID.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#connectionRange"><B>connectionRange</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>Limits the incoming connections to a target neuron to include those from source neurons within the specified
 range of the target neuron.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#connectionWeightMax"><B>connectionWeightMax</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The maximum connection weight in the substrate.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#connectionWeightMin"><B>connectionWeightMin</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</A>
<DD>The minimum connection weight in the substrate.
<DT><A HREF="../com/anji/integration/ConsoleLogEventListener.html" title="class in com.anji.integration"><B>ConsoleLogEventListener</B></A> - Class in <A HREF="../com/anji/integration/package-summary.html">com.anji.integration</A><DD>Writes log events to stdout.<DT><A HREF="../com/anji/integration/ConsoleLogEventListener.html#ConsoleLogEventListener(org.jgapcustomised.Configuration)"><B>ConsoleLogEventListener(Configuration)</B></A> - 
Constructor for class com.anji.integration.<A HREF="../com/anji/integration/ConsoleLogEventListener.html" title="class in com.anji.integration">ConsoleLogEventListener</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/ConsoleLogEventListener.html#ConsoleLogEventListener(org.jgapcustomised.Configuration, java.io.PrintStream)"><B>ConsoleLogEventListener(Configuration, PrintStream)</B></A> - 
Constructor for class com.anji.integration.<A HREF="../com/anji/integration/ConsoleLogEventListener.html" title="class in com.anji.integration">ConsoleLogEventListener</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Species.html#containsFittest"><B>containsFittest</B></A> - 
Variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>True iff this species contains the fittest individual from the entire population.
<DT><A HREF="../com/ojcoleman/ahni/util/PropertiesConverter.html#convert(java.lang.String)"><B>convert(String)</B></A> - 
Method in class com.ojcoleman.ahni.util.<A HREF="../com/ojcoleman/ahni/util/PropertiesConverter.html" title="class in com.ojcoleman.ahni.util">PropertiesConverter</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>ConvertToSignedActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>&nbsp;<DT><A HREF="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html#ConvertToSignedActivationFunction()"><B>ConvertToSignedActivationFunction()</B></A> - 
Constructor for class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html" title="class in com.anji.nn.activationfunction">ConvertToSignedActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html#coords"><B>coords</B></A> - 
Variable in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html#coordsEnabled()"><B>coordsEnabled()</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</A>
<DD>Returns true iff storing coordinates for neurons has been enabled.
<DT><A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html#coordsMin"><B>coordsMin</B></A> - 
Variable in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html#coordsRange"><B>coordsRange</B></A> - 
Variable in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/Copyright.html" title="class in com.anji"><B>Copyright</B></A> - Class in <A HREF="../com/anji/package-summary.html">com.anji</A><DD>Copyright<DT><A HREF="../com/anji/nn/activationfunction/CosineActivationFunction.html" title="class in com.anji.nn.activationfunction"><B>CosineActivationFunction</B></A> - Class in <A HREF="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</A><DD>Cosine activation function.<DT><A HREF="../com/anji/nn/activationfunction/AbsoluteActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/AbsoluteActivationFunction.html" title="class in com.anji.nn.activationfunction">AbsoluteActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/ActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in interface com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/ActivationFunction.html" title="interface in com.anji.nn.activationfunction">ActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/BipolarSigmoidActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/BipolarSigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">BipolarSigmoidActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/ClampedAbsoluteActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/ClampedAbsoluteActivationFunction.html" title="class in com.anji.nn.activationfunction">ClampedAbsoluteActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">ClampedLinearActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html" title="class in com.anji.nn.activationfunction">ConvertToSignedActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/CosineActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/CosineActivationFunction.html" title="class in com.anji.nn.activationfunction">CosineActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/DivideActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/DivideActivationFunction.html" title="class in com.anji.nn.activationfunction">DivideActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/EvSailSigmoidActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/EvSailSigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">EvSailSigmoidActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/GaussianActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/GaussianActivationFunction.html" title="class in com.anji.nn.activationfunction">GaussianActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/InverseAbsActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/InverseAbsActivationFunction.html" title="class in com.anji.nn.activationfunction">InverseAbsActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/LinearActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/LinearActivationFunction.html" title="class in com.anji.nn.activationfunction">LinearActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/LogicActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/LogicActivationFunction.html" title="class in com.anji.nn.activationfunction">LogicActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/NegatedLinearActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/NegatedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">NegatedLinearActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/RampActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/RampActivationFunction.html" title="class in com.anji.nn.activationfunction">RampActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/RecipriocalActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/RecipriocalActivationFunction.html" title="class in com.anji.nn.activationfunction">RecipriocalActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SigmoidActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/SigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">SigmoidActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">SignedClampedLinearActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SignedStepActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/SignedStepActivationFunction.html" title="class in com.anji.nn.activationfunction">SignedStepActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/SineActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/SineActivationFunction.html" title="class in com.anji.nn.activationfunction">SineActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/StepActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/StepActivationFunction.html" title="class in com.anji.nn.activationfunction">StepActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/TanhActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/TanhActivationFunction.html" title="class in com.anji.nn.activationfunction">TanhActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/activationfunction/TanhCubicActivationFunction.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.activationfunction.<A HREF="../com/anji/nn/activationfunction/TanhCubicActivationFunction.html" title="class in com.anji.nn.activationfunction">TanhCubicActivationFunction</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/AnjiNet.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/BiasConnection.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/BiasConnection.html" title="class in com.anji.nn">BiasConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/CacheNeuronConnection.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/Connection.html#cost()"><B>cost()</B></A> - 
Method in interface com.anji.nn.<A HREF="../com/anji/nn/Connection.html" title="interface in com.anji.nn">Connection</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/Neuron.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/NeuronConnection.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/NeuronConnection.html" title="class in com.anji.nn">NeuronConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/RandomConnection.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/RandomConnection.html" title="class in com.anji.nn">RandomConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/nn/StepHourglassConnection.html#cost()"><B>cost()</B></A> - 
Method in class com.anji.nn.<A HREF="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn">StepHourglassConnection</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/GridNet.html#cost()"><B>cost()</B></A> - 
Method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Species.html#COUNT_TAG"><B>COUNT_TAG</B></A> - 
Static variable in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>XML count tag
<DT><A HREF="../com/anji/integration/ErrorRateCounter.html#countErrors(double[][], double[][])"><B>countErrors(double[][], double[][])</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/ErrorRateCounter.html" title="class in com.anji.integration">ErrorRateCounter</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/integration/ErrorRateCounter.html#countErrors(java.lang.String, double[][], double[][])"><B>countErrors(String, double[][], double[][])</B></A> - 
Method in class com.anji.integration.<A HREF="../com/anji/integration/ErrorRateCounter.html" title="class in com.anji.integration">ErrorRateCounter</A>
<DD>Calculate the sum of differences between <code>responses</code> and target values.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnActivator"><B>cppnActivator</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnInput"><B>cppnInput</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cppnInputCount"><B>cppnInputCount</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>The number of inputs to the CPPN.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnMax"><B>cppnMax</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnMin"><B>cppnMin</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnOutput"><B>cppnOutput</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cppnOutputCount"><B>cppnOutputCount</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>The number of outputs from the CPPN.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnOutputPlusMinusUnitBounded"><B>cppnOutputPlusMinusUnitBounded</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnOutputUnitBounded"><B>cppnOutputUnitBounded</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnRange"><B>cppnRange</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html" title="class in com.ojcoleman.ahni.evaluation.mocostfunctions"><B>CPPNSizeCost</B></A> - Class in <A HREF="../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</A><DD>&nbsp;<DT><A HREF="../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html#CPPNSizeCost()"><B>CPPNSizeCost()</B></A> - 
Constructor for class com.ojcoleman.ahni.evaluation.mocostfunctions.<A HREF="../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html" title="class in com.ojcoleman.ahni.evaluation.mocostfunctions">CPPNSizeCost</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cppnTranscriber"><B>cppnTranscriber</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>Transcriber for network for use as a CPPN.
<DT><A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html#cppnValue"><B>cppnValue</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.QuadPoint</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#createNeuronCollection(java.lang.String, int, boolean, boolean, boolean)"><B>createNeuronCollection(String, int, boolean, boolean, boolean)</B></A> - 
Static method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>&nbsp;
<DT><A HREF="../com/ojcoleman/ahni/nn/BainNN.html#createSynapseCollection(java.lang.String, int, boolean, boolean, double, double)"><B>createSynapseCollection(String, int, boolean, boolean, double, double)</B></A> - 
Static method in class com.ojcoleman.ahni.nn.<A HREF="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</A>
<DD>&nbsp;
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#CROSSOVER_MUTATE_RATE_KEY"><B>CROSSOVER_MUTATE_RATE_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, the probability that an individual produced by the crossover operator will be a candidate for having mutations applied to it (independent of other mutation probabilities).
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#CROSSOVER_PROPORTION_KEY"><B>CROSSOVER_PROPORTION_KEY</B></A> - 
Static variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>properties key, proportion of new population that is generated via crossover from two parents (as opposed to mutation of a single parent), [0, 1].
<DT><A HREF="../com/anji/neat/NeatConfiguration.html#crossoverOper"><B>crossoverOper</B></A> - 
Variable in class com.anji.neat.<A HREF="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/CrossoverReproductionOperator.html" title="class in org.jgapcustomised"><B>CrossoverReproductionOperator</B></A> - Class in <A HREF="../org/jgapcustomised/package-summary.html">org.jgapcustomised</A><DD>Abstract crossover reporduction operator handles iteration over population to determine pairs of parent chromosomes.<DT><A HREF="../org/jgapcustomised/CrossoverReproductionOperator.html#CrossoverReproductionOperator()"><B>CrossoverReproductionOperator()</B></A> - 
Constructor for class org.jgapcustomised.<A HREF="../org/jgapcustomised/CrossoverReproductionOperator.html" title="class in org.jgapcustomised">CrossoverReproductionOperator</A>
<DD>&nbsp;
<DT><A HREF="../org/jgapcustomised/Species.html#cull(java.util.Collection)"><B>cull(Collection&lt;Chromosome&gt;)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>Remove all chromosomes from this specie except <code>keepers</code> NOTE: this method does not update the species
 field in the removed Chromosomes.
<DT><A HREF="../org/jgapcustomised/Species.html#cullToElites(org.jgapcustomised.Chromosome)"><B>cullToElites(Chromosome)</B></A> - 
Method in class org.jgapcustomised.<A HREF="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>
<DD>remove all non-elite chromosomes from this species, except for population-wide fittest
<DT><A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cyclesPerStep"><B>cyclesPerStep</B></A> - 
Variable in class com.ojcoleman.ahni.transcriber.<A HREF="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</A>
<DD>For substrate networks with recurrent connections, the number of activation cycles to perform each time the
 substrate network is presented with new input and queried for its output.
</DL>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="index-2.html"><B>PREV LETTER</B></A>&nbsp;
&nbsp;<A HREF="index-4.html"><B>NEXT LETTER</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?index-filesindex-3.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-3.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<A HREF="index-1.html">A</A> <A HREF="index-2.html">B</A> <A HREF="index-3.html">C</A> <A HREF="index-4.html">D</A> <A HREF="index-5.html">E</A> <A HREF="index-6.html">F</A> <A HREF="index-7.html">G</A> <A HREF="index-8.html">H</A> <A HREF="index-9.html">I</A> <A HREF="index-10.html">K</A> <A HREF="index-11.html">L</A> <A HREF="index-12.html">M</A> <A HREF="index-13.html">N</A> <A HREF="index-14.html">O</A> <A HREF="index-15.html">P</A> <A HREF="index-16.html">Q</A> <A HREF="index-17.html">R</A> <A HREF="index-18.html">S</A> <A HREF="index-19.html">T</A> <A HREF="index-20.html">U</A> <A HREF="index-21.html">V</A> <A HREF="index-22.html">W</A> <A HREF="index-23.html">X</A> <A HREF="index-24.html">Y</A> <A HREF="index-25.html">Z</A> <HR>

</BODY>
</HTML>
