<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_75) on Sun Mar 15 11:01:18 AEDT 2015 -->
<title>S-Index</title>
<meta name="date" content="2015-03-15">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="S-Index";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="index-18.html">Prev Letter</a></li>
<li><a href="index-20.html">Next Letter</a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?index-filesindex-19.html" target="_top">Frames</a></li>
<li><a href="index-19.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Y</a>&nbsp;<a href="index-26.html">Z</a>&nbsp;<a href="index-27.html">_</a>&nbsp;<a name="_S_">
<!--   -->
</a>
<h2 class="title">S</h2>
<dl>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#sampleCount">sampleCount</a></span> - Variable in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html" title="class in com.ojcoleman.ahni.evaluation.novelty">GenericBehaviourEvaluator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#SAMPLES">SAMPLES</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html" title="class in com.ojcoleman.ahni.evaluation.novelty">GenericBehaviourEvaluator</a></dt>
<dd>
<div class="block">The number of output samples to record for each sequence.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#SAMPLING_INTERVAL">SAMPLING_INTERVAL</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html" title="class in com.ojcoleman.ahni.evaluation.novelty">GenericBehaviourEvaluator</a></dt>
<dd>
<div class="block">Output samples will be taken every [fitness.function.generic_novelty.sampling_interval]th step in the sequence.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#samplingInterval">samplingInterval</a></span> - Variable in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html" title="class in com.ojcoleman.ahni.evaluation.novelty">GenericBehaviourEvaluator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#SAVE_IMAGES_KEY">SAVE_IMAGES_KEY</a></span> - Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#SAVE_IMAGES_KEY">SAVE_IMAGES_KEY</a></span> - Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#scale(int,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)">scale(int, int, HyperNEATTranscriber)</a></span> - Method in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>
<div class="block">Allow a sub-class to make the necessary changes when a substrate scale occurs.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html#scale(int,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)">scale(int, int, HyperNEATTranscriber)</a></span> - Method in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction2.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction2</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#scale(int,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)">scale(int, int, HyperNEATTranscriber)</a></span> - Method in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#scale(int,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)">scale(int, int, HyperNEATTranscriber)</a></span> - Method in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/TestTargetFitnessFunction.html#scale(int,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber)">scale(int, int, HyperNEATTranscriber)</a></span> - Method in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/TestTargetFitnessFunction.html" title="class in com.ojcoleman.ahni.experiments">TestTargetFitnessFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_COUNT_KEY">SCALE_COUNT_KEY</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>
<div class="block">Property key for the maximum number of rescalings to perform.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_FACTOR_KEY">SCALE_FACTOR_KEY</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>
<div class="block">Property key for multiplicative amount to scale substrate by when the performance of the best individual in the
 population has reached the specified level.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_PERFORMANCE_KEY">SCALE_PERFORMANCE_KEY</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>
<div class="block">Property key for the performance level required before a scaling is performed.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#SCALE_RIP_KEY">SCALE_RIP_KEY</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>
<div class="block">Property key for whether the performance values should be recorded before the final scaling has been performed.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#scaleCount">scaleCount</a></span> - Variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/ArrayUtil.html#scaleFromUnit(double[],%20com.ojcoleman.ahni.util.Range)">scaleFromUnit(double[], Range)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/ArrayUtil.html" title="class in com.ojcoleman.ahni.util">ArrayUtil</a></dt>
<dd>
<div class="block">Scales each value in the array to the given.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.TransformType.html#scaleOutput()">scaleOutput()</a></span> - Method in enum com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.TransformType.html" title="enum in com.ojcoleman.ahni.experiments.csb">StaticTransformEnvironment.TransformType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html#scalePerformance">scalePerformance</a></span> - Variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATFitnessFunction</a></dt>
<dd>
<div class="block">The performance level required before a scaling is performed.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/ArrayUtil.html#scaleToUnit(double[],%20com.ojcoleman.ahni.util.Range)">scaleToUnit(double[], Range)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/ArrayUtil.html" title="class in com.ojcoleman.ahni.util">ArrayUtil</a></dt>
<dd>
<div class="block">Scales each value to the range [0 ,1], using the given Range.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#SEED">SEED</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html" title="class in com.ojcoleman.ahni.experiments">RLRecurrentNetworkBased</a></dt>
<dd>
<div class="block">The seed to use for the RNG to generate the network.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/hyperneat/HyperNEATConfiguration.html#SEED_CHROMOSOME_KEY">SEED_CHROMOSOME_KEY</a></span> - Static variable in class com.ojcoleman.ahni.hyperneat.<a href="../com/ojcoleman/ahni/hyperneat/HyperNEATConfiguration.html" title="class in com.ojcoleman.ahni.hyperneat">HyperNEATConfiguration</a></dt>
<dd>
<div class="block">The file name of a seed chromosome in XML format.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration)">select(Configuration)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration">SimpleSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/SimpleSelector.html#select(org.jgapcustomised.Configuration,%20int)">select(Configuration, int)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration">SimpleSelector</a></dt>
<dd>
<div class="block">Returns the <code>a_howManyToSelect</code> chromosomes with highest fitness.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration)">select(Configuration)</a></span> - Method in class com.ojcoleman.ahni.misc.<a href="../com/ojcoleman/ahni/misc/NSGAIISelector.html" title="class in com.ojcoleman.ahni.misc">NSGAIISelector</a></dt>
<dd>
<div class="block">Modified version of <a href="../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)"><code>NaturalSelector.select(Configuration)</code></a> that selects a number of
 elites and parents for each species based on the species fitness and the non-dominated sorting.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration,%20int)">select(Configuration, int)</a></span> - Method in class com.ojcoleman.ahni.misc.<a href="../com/ojcoleman/ahni/misc/NSGAIISelector.html" title="class in com.ojcoleman.ahni.misc">NSGAIISelector</a></dt>
<dd>
<div class="block">This method should not be used, only the <a href="../com/ojcoleman/ahni/misc/NSGAIISelector.html#select(org.jgapcustomised.Configuration)"><code>NSGAIISelector.select(Configuration)</code></a> method should be used.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/WeightedRouletteSelector.html#select(org.jgapcustomised.Configuration,%20int)">select(Configuration, int)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/WeightedRouletteSelector.html" title="class in org.jgapcustomised.impl">WeightedRouletteSelector</a></dt>
<dd>
<div class="block">Select a given number of Chromosomes from the pool that will move on to the next generation population.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration)">select(Configuration)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>
<div class="block">Select a given number of Chromosomes from the pool that will move on to the next generation population.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#select(org.jgapcustomised.Configuration,%20int)">select(Configuration, int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SELECTOR_CLASS_KEY">SELECTOR_CLASS_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, the NaturalSelector to use to perform the selection process.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#sequenceCount">sequenceCount</a></span> - Variable in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html" title="class in com.ojcoleman.ahni.evaluation.novelty">GenericBehaviourEvaluator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html#SEQUENCES">SEQUENCES</a></span> - Static variable in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/GenericBehaviourEvaluator.html" title="class in com.ojcoleman.ahni.evaluation.novelty">GenericBehaviourEvaluator</a></dt>
<dd>
<div class="block">The number of sequences to test individuals on.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/DoubleVector.html#set(int,%20double)">set(int, double)</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/DoubleVector.html" title="class in com.ojcoleman.ahni.util">DoubleVector</a></dt>
<dd>
<div class="block">Replaces the element at the specified 
 position in me with the specified element
 (optional operation).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Range.html#set(double,%20double)">set(double, double)</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Range.html" title="class in com.ojcoleman.ahni.util">Range</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#setAlleles(java.util.SortedSet)">setAlleles(SortedSet&lt;Allele&gt;)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">for hibernate</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeuronAllele.html#setBias(double)">setBias(double)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/Neuron.html#setBias(double)">setBias(double)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setBulkFitnessFunction(org.jgapcustomised.BulkFitnessFunction)">setBulkFitnessFunction(BulkFitnessFunction)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">Sets the bulk fitness function to be used for this genetic algorithm.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Point.html#setCoordinates(double,%20double,%20double)">setCoordinates(double, double, double)</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Point.html" title="class in com.ojcoleman.ahni.util">Point</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/NNAdaptor.html#setCoords(int,%20double,%20double)">setCoords(int, double, double)</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</a></dt>
<dd>
<div class="block">Sets the coordinate of the specified neuron.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/NNAdaptor.html#setCoords(int,%20double,%20double,%20double)">setCoords(int, double, double, double)</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</a></dt>
<dd>
<div class="block">Sets the coordinate of the specified neuron.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/novelty/NoveltySearch.html#setCurrentPopulation(java.util.List)">setCurrentPopulation(List&lt;Behaviour&gt;)</a></span> - Method in class com.ojcoleman.ahni.evaluation.novelty.<a href="../com/ojcoleman/ahni/evaluation/novelty/NoveltySearch.html" title="class in com.ojcoleman.ahni.evaluation.novelty">NoveltySearch</a></dt>
<dd>
<div class="block">This may be called before evaluating individuals from a population to allow determining novelty based on the
 archive and the current population.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Parallel.html#setDefaultThreads(int)">setDefaultThreads(int)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Parallel.html" title="class in com.ojcoleman.ahni.util">Parallel</a></dt>
<dd>
<div class="block">Sets the default number of threads that will be used.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#setElites(java.util.List)">setElites(List&lt;Chromosome&gt;)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setElitismMinSpeciesSize(int)">setElitismMinSpeciesSize(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setElitismMinToSelect(int)">setElitismMinToSelect(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setElitismProportion(double)">setElitismProportion(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setEvaluationDataStable()">setEvaluationDataStable()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Indicate that the evaluation data (e.g.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setEventManager(org.jgapcustomised.event.EventManager)">setEventManager(EventManager)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">Sets the EventManager that is to be associated with this configuration.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/hyperneat/Properties.html#setEvolver(com.ojcoleman.ahni.hyperneat.HyperNEATEvolver)">setEvolver(HyperNEATEvolver)</a></span> - Method in class com.ojcoleman.ahni.hyperneat.<a href="../com/ojcoleman/ahni/hyperneat/Properties.html" title="class in com.ojcoleman.ahni.hyperneat">Properties</a></dt>
<dd>
<div class="block">Set the HyperNEATEvolver that is handling this run.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setFitnessFunction(org.jgapcustomised.FitnessFunction)">setFitnessFunction(FitnessFunction)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">Sets the fitness function to be used for this genetic algorithm.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setFitnessValue(double)">setFitnessValue(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Sets the overall fitness value of this Chromosome.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setFitnessValue(double,%20int)">setFitnessValue(double, int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Sets the fitness value of this Chromosome for the given objective.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setFitnessValues(double[])">setFitnessValues(double[])</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Sets all the fitness values of this Chromosome.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html#setGeneration(int)">setGeneration(int)</a></span> - Method in class com.ojcoleman.ahni.hyperneat.<a href="../com/ojcoleman/ahni/hyperneat/HyperNEATEvolver.html" title="class in com.ojcoleman.ahni.hyperneat">HyperNEATEvolver</a></dt>
<dd>
<div class="block">Sets the current generation number, used by <a href="../com/ojcoleman/ahni/evaluation/Minion.html" title="class in com.ojcoleman.ahni.evaluation"><code>Minion</code></a>.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/Neuron.html#setId(long)">setId(long)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/NeuronConnection.html#setId(long)">setId(long)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/NeuronConnection.html" title="class in com.anji.nn">NeuronConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setIdFactory(org.jgapcustomised.IdFactory)">setIdFactory(IdFactory)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Gene.html#setInnovationId(java.lang.Long)">setInnovationId(Long)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised">Gene</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/AnjiActivator.html#setInputPattern(com.anji.nn.Pattern)">setInputPattern(Pattern)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/GridNet.html#setInputs(double[][])">setInputs(double[][])</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></dt>
<dd>
<div class="block">Set array to use as the input layer.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#setIsSelectedForNextGeneration(boolean)">setIsSelectedForNextGeneration(boolean)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>
<div class="block">Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Point2DImmut.html#setLocation(double,%20double)">setLocation(double, double)</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Point2DImmut.html" title="class in com.ojcoleman.ahni.util">Point2DImmut</a></dt>
<dd>
<div class="block">
 Throws an IllegalStateException as the coordinates of a Point2DImmut are immutable and may not be modified.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/TargetFitnessFunction.html#setMaxFitnessValue(int)">setMaxFitnessValue(int)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/TargetFitnessFunction.html" title="class in com.anji.integration">TargetFitnessFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/FitnessFunction.html#setMaxFitnessValue(int)">setMaxFitnessValue(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/FitnessFunction.html" title="class in org.jgapcustomised">FitnessFunction</a></dt>
<dd>
<div class="block"><span class="strong">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setMaxStagnantGenerations(int)">setMaxStagnantGenerations(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setMinAge(int)">setMinAge(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/Environment.html#setMinimumStepsToSolve(int)">setMinimumStepsToSolve(int)</a></span> - Method in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/Environment.html" title="class in com.ojcoleman.ahni.experiments.csb">Environment</a></dt>
<dd>
<div class="block">Set the minimum required steps to solve this environment instance (generally only for internal use).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html#setMinimumStepsToSolve(int)">setMinimumStepsToSolve(int)</a></span> - Method in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb">SimpleNavigationEnvironment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ReproductionOperator.html#setMutateRate(double)">setMutateRate(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ReproductionOperator.html" title="class in org.jgapcustomised">ReproductionOperator</a></dt>
<dd>
<div class="block">Set the probability that individuals produced by this operator will be a candidate for having mutations applied to them (independent of the mutation rates of MutationOperators).</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/PruneMutationOperator.html#setMutationRate(double)">setMutationRate(double)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/PruneMutationOperator.html" title="class in com.anji.neat">PruneMutationOperator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/MutationOperator.html#setMutationRate(double)">setMutationRate(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/MutationOperator.html" title="class in org.jgapcustomised">MutationOperator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/AnjiActivator.html#setName(java.lang.String)">setName(String)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/AnjiNet.html#setName(java.lang.String)">setName(String)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/util/Properties.html#setName(java.lang.String)">setName(String)</a></span> - Method in class com.anji.util.<a href="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#setName(java.lang.String)">setName(String)</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/GridNet.html#setName(java.lang.String)">setName(String)</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setNaturalSelector(org.jgapcustomised.NaturalSelector)">setNaturalSelector(NaturalSelector)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">set selector</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#setNeuronDisabled(int,%20boolean)">setNeuronDisabled(int, boolean)</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>
<div class="block">Allows keeping a record of which neurons are disabled.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(double,%20double,%20double,%20com.ojcoleman.bain.base.NeuronCollection,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN,%20boolean)">setNeuronParameters(double, double, double, NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</a></dt>
<dd>
<div class="block">Set the parameters for a neuron, specifying the neuron coordinates directly (see
 <a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(double,%20double,%20double)"><code>HyperNEATTranscriber.CPPN.setTargetCoordinates(double, double, double)</code></a>).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(int,%20int,%20int,%20com.ojcoleman.bain.base.NeuronCollection,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN,%20boolean)">setNeuronParameters(int, int, int, NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</a></dt>
<dd>
<div class="block">Set the parameters for a neuron, specifying the neuron coordinates with grid indices into the substrate (see
 <a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinatesFromGridIndices(int,%20int,%20int)"><code>HyperNEATTranscriber.CPPN.setTargetCoordinatesFromGridIndices(int, int, int)</code></a>).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(com.ojcoleman.ahni.util.Point,%20com.ojcoleman.bain.base.NeuronCollection,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN,%20boolean)">setNeuronParameters(Point, NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</a></dt>
<dd>
<div class="block">Set the parameters for a neuron, specifying the neuron coordinates with a Point (see
 <a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(com.ojcoleman.ahni.util.Point)"><code>HyperNEATTranscriber.CPPN.setTargetCoordinates(Point)</code></a>).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setNeuronParameters(com.ojcoleman.bain.base.NeuronCollection,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN,%20boolean)">setNeuronParameters(NeuronCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</a></dt>
<dd>
<div class="block">Set the parameters for a neuron.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/AnjiActivator.html#setNumCycles(int)">setNumCycles(int)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/AnjiActivator.html" title="class in com.anji.integration">AnjiActivator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html#setPatterns(double[][][],%20double[][][],%20double,%20double)">setPatterns(double[][][], double[][][], double, double)</a></span> - Method in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/HyperNEATTargetFitnessFunction.html" title="class in com.ojcoleman.ahni.evaluation">HyperNEATTargetFitnessFunction</a></dt>
<dd>
<div class="block">Set the input and target output pattern pairs to use for evaluations.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html#setPatterns(double[][],%20double[][],%20double,%20double)">setPatterns(double[][], double[][], double, double)</a></span> - Method in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/TargetFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">TargetFitnessFunctionMT</a></dt>
<dd>
<div class="block">Set the input and target output pattern pairs to use for evaluations.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setPerformanceValue(double)">setPerformanceValue(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Sets the primary performance value of this Chromosome.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setPerformanceValue(java.lang.String,%20double)">setPerformanceValue(String, double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Sets the specified performance value of this Chromosome.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/EvaluationData.html#setPerformanceValues(java.util.Map)">setPerformanceValues(Map&lt;String, Double&gt;)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/EvaluationData.html" title="class in org.jgapcustomised">EvaluationData</a></dt>
<dd>
<div class="block">Sets the specified performance values of this Chromosome.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setPopulationSize(int)">setPopulationSize(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">Sets the population size to be used for this genetic algorithm.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#setPreviousBestPerforming(org.jgapcustomised.Chromosome)">setPreviousBestPerforming(Chromosome)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/RandomConnection.html#setRand(java.util.Random)">setRand(Random)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/RandomConnection.html" title="class in com.anji.nn">RandomConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setRandomGenerator(java.util.Random)">setRandomGenerator(Random)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">Sets the random generator to be used for this genetic algorithm.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#setRepresentative(org.jgapcustomised.ChromosomeMaterial)">setRepresentative(ChromosomeMaterial)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">Sets the representative material.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#setSampleChromosomeMaterial(org.jgapcustomised.ChromosomeMaterial)">setSampleChromosomeMaterial(ChromosomeMaterial)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>
<div class="block">Sets sample ChromosomeMaterial that is to be used as a guide for the construction of other Chromosomes.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#setSecondaryParentId(java.lang.Long)">setSecondaryParentId(Long)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#setShouldMutate(boolean)">setShouldMutate(boolean)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">May be used by reproduction operators to indicate that this individual is a candidate for mutations.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ReproductionOperator.html#setSlice(double)">setSlice(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ReproductionOperator.html" title="class in org.jgapcustomised">ReproductionOperator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinates(double,%20double)">setSourceCoordinates(double, double)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the source neuron in a two-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinates(double,%20double,%20double)">setSourceCoordinates(double, double, double)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the source neuron in a three-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinates(com.ojcoleman.ahni.util.Point)">setSourceCoordinates(Point)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the source neuron in a substrate with dimensions with range [0, 1] (coordinates are
 translated to a user-specified range if necessary, e.g.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setSourceCoordinatesFromGridIndices(int,%20int,%20int)">setSourceCoordinatesFromGridIndices(int, int, int)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the source neuron specified by the given indices into a grid-based substrate.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setSpeciatedFitness(boolean)">setSpeciatedFitness(boolean)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpeciationTarget(int)">setSpeciationTarget(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpeciationThreshold(double)">setSpeciationThreshold(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpeciationThresholdMax(double)">setSpeciationThresholdMax(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpeciationThresholdMin(double)">setSpeciationThresholdMin(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#setSpecie(org.jgapcustomised.Species)">setSpecie(Species)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>
<div class="block">should only be called from Species; assigns this chromosome to <code>aSpecie</code>; throws exception if
 chromosome is added to a specie twice</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatCommonCoeff(double)">setSpecieCompatCommonCoeff(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatDisjointCoeff(double)">setSpecieCompatDisjointCoeff(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatExcessCoeff(double)">setSpecieCompatExcessCoeff(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatMismatchUseValues(boolean)">setSpecieCompatMismatchUseValues(boolean)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#setSpecieCompatNormalise(boolean)">setSpecieCompatNormalise(boolean)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#setStagnantGenerationsCount(int)">setStagnantGenerationsCount(int)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#setStepsPerStepForNonLayeredFF()">setStepsPerStepForNonLayeredFF()</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>
<div class="block">(Re)calculates the number of steps required to fully activate a non-layered feed-forward network.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#setSumOfSquaredConnectionLengths(double)">setSumOfSquaredConnectionLengths(double)</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#setSurvivalRate(double)">setSurvivalRate(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setSynapseParameters(com.ojcoleman.bain.base.SynapseCollection,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN,%20boolean,%20boolean)">setSynapseParameters(SynapseCollection, int, HyperNEATTranscriber&lt;BainNN&gt;.CPPN, boolean, boolean)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</a></dt>
<dd>
<div class="block">Set the parameters for a synapse.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(double,%20double)">setTargetCoordinates(double, double)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the source neuron in a two-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(double,%20double,%20double)">setTargetCoordinates(double, double, double)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the source neuron in a three-dimensional substrate with dimensions with range [0, 1]
 (coordinates are translated to a user-specified range if necessary, e.g.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinates(com.ojcoleman.ahni.util.Point)">setTargetCoordinates(Point)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the target neuron in a substrate with dimensions with range [0, 1] (coordinates are
 translated to a user-specified range if necessary, e.g.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#setTargetCoordinatesFromGridIndices(int,%20int,%20int)">setTargetCoordinatesFromGridIndices(int, int, int)</a></span> - Method in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Set the coordinates of the target neuron specified by the given indices into a grid-based substrate.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/ConnectionAllele.html#setToRandomValue(java.util.Random,%20boolean)">setToRandomValue(Random, boolean)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a></dt>
<dd>
<div class="block">Set weight to random value from a Gaussian distribution determined by <a href="../com/anji/neat/ConnectionAllele.html#RANDOM_STD_DEV"><code>ConnectionAllele.RANDOM_STD_DEV</code></a></div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeuronAllele.html#setToRandomValue(java.util.Random,%20boolean)">setToRandomValue(Random, boolean)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a></dt>
<dd>
<div class="block">Set bias to random value from a Gaussian distribution determined by <a href="../com/anji/neat/ConnectionAllele.html#RANDOM_STD_DEV"><code>ConnectionAllele.RANDOM_STD_DEV</code></a></div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/integration/ParamAllele.html#setToRandomValue(java.util.Random,%20boolean)">setToRandomValue(Random, boolean)</a></span> - Method in class com.ojcoleman.ahni.integration.<a href="../com/ojcoleman/ahni/integration/ParamAllele.html" title="class in com.ojcoleman.ahni.integration">ParamAllele</a></dt>
<dd>
<div class="block">Set value to random value from a Gaussian distribution determined by <a href="../com/ojcoleman/ahni/integration/ParamAllele.html#RANDOM_STD_DEV"><code>ParamAllele.RANDOM_STD_DEV</code></a></div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Allele.html#setToRandomValue(java.util.Random,%20boolean)">setToRandomValue(Random, boolean)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></dt>
<dd>
<div class="block">Sets the value of this Gene to a random legal value for the implementation.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/BooleanAllele.html#setToRandomValue(java.util.Random,%20boolean)">setToRandomValue(Random, boolean)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></dt>
<dd>
<div class="block">Sets the value (allele) of this Gene to a random legal value.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#setToRandomValue(java.util.Random,%20boolean)">setToRandomValue(Random, boolean)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>
<div class="block">Sets the value (allele) of this Gene to a random Integer value between
 
 the lower and upper bounds (if any) of this Gene.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/Environment.html#setUp(int)">setUp(int)</a></span> - Method in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/Environment.html" title="class in com.ojcoleman.ahni.experiments.csb">Environment</a></dt>
<dd>
<div class="block">Set-up the environment configuration.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html#setUp(int)">setUp(int)</a></span> - Method in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb">SimpleNavigationEnvironment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.html#setUp(int)">setUp(int)</a></span> - Method in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb">StaticTransformEnvironment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/util/Reset.html#setUserInteraction(boolean)">setUserInteraction(boolean)</a></span> - Method in class com.anji.util.<a href="../com/anji/util/Reset.html" title="class in com.anji.util">Reset</a></dt>
<dd>
<div class="block">default is false; set to true if you want it to delete everything without prompting user</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/ConnectionAllele.html#setValue(double)">setValue(double)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a></dt>
<dd>
<div class="block">Sets the weight value.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeuronAllele.html#setValue(double)">setValue(double)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a></dt>
<dd>
<div class="block">Sets the bias value.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/Pattern.html#setValue(int,%20double)">setValue(int, double)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/Pattern.html" title="class in com.anji.nn">Pattern</a></dt>
<dd>
<div class="block">Set stimulus value at <code>idx</code> to <code>value</code>.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/integration/ParamAllele.html#setValue(double)">setValue(double)</a></span> - Method in class com.ojcoleman.ahni.integration.<a href="../com/ojcoleman/ahni/integration/ParamAllele.html" title="class in com.ojcoleman.ahni.integration">ParamAllele</a></dt>
<dd>
<div class="block">Sets the value value.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Allele.html#setValue(double)">setValue(double)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></dt>
<dd>
<div class="block">Generic method for setting the value of an allele.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/BooleanAllele.html#setValue(java.lang.Object)">setValue(Object)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></dt>
<dd>
<div class="block">Sets the value of this Gene to the new given value.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/BooleanAllele.html#setValue(double)">setValue(double)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#setValue(java.lang.Integer)">setValue(Integer)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>
<div class="block">Sets the value (allele) of this Gene to the new given value.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#setValue(double)">setValue(double)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/BooleanAllele.html#setValueFromPersistentRepresentation(java.lang.String)">setValueFromPersistentRepresentation(String)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></dt>
<dd>
<div class="block">Sets the value and internal state of this Gene from the string
 
 representation returned by a previous invocation of the
 
 getPersistentRepresentation() method.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#setValueFromPersistentRepresentation(java.lang.String)">setValueFromPersistentRepresentation(String)</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>
<div class="block">Sets the value and internal state of this Gene from the string
 
 representation returned by a previous invocation of the
 
 getPersistentRepresentation() method.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/Pattern.html#setValues(double[])">setValues(double[])</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/Pattern.html" title="class in com.anji.nn">Pattern</a></dt>
<dd>
<div class="block">Set stimulus values.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/ConnectionAllele.html#setWeight(double)">setWeight(double)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/NeuronConnection.html#setWeight(double)">setWeight(double)</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/NeuronConnection.html" title="class in com.anji.nn">NeuronConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html#SeventyTwoDegrees">SeventyTwoDegrees</a></span> - Static variable in class com.ojcoleman.ahni.experiments.doublepolebalancing.<a href="../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html" title="class in com.ojcoleman.ahni.experiments.doublepolebalancing">DoublePoleBalancing</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#SHAPE_SIZE_KEY">SHAPE_SIZE_KEY</a></span> - Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#SHAPE_SIZE_KEY">SHAPE_SIZE_KEY</a></span> - Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html#SHAPE_TYPE_KEY">SHAPE_TYPE_KEY</a></span> - Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction3.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction3</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html#SHAPE_TYPE_KEY">SHAPE_TYPE_KEY</a></span> - Static variable in class com.ojcoleman.ahni.experiments.objectrecognition.<a href="../com/ojcoleman/ahni/experiments/objectrecognition/ObjectRecognitionFitnessFunction4.html" title="class in com.ojcoleman.ahni.experiments.objectrecognition">ObjectRecognitionFitnessFunction4</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#shouldMutate()">shouldMutate()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Indicates whether this individual is a candidate for mutations (after reproduction).</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/SigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SigmoidActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Classic Sigmoid.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SignedClampedLinearActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html#SignedClampedLinearActivationFunction()">SignedClampedLinearActivationFunction()</a></span> - Constructor for class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">SignedClampedLinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/nn/activationfunction/SignedStepActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SignedStepActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Step activation function.</div>
</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb"><span class="strong">SimpleNavigationEnvironment</span></a> - Class in <a href="../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a></dt>
<dd>
<div class="block">
 Implements an environment that essentially represents a simple navigation like task.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html#SimpleNavigationEnvironment()">SimpleNavigationEnvironment()</a></span> - Constructor for class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb">SimpleNavigationEnvironment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.Obstacle.html" title="class in com.ojcoleman.ahni.experiments.csb"><span class="strong">SimpleNavigationEnvironment.Obstacle</span></a> - Class in <a href="../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.Obstacle.html#SimpleNavigationEnvironment.Obstacle(double)">SimpleNavigationEnvironment.Obstacle(double)</a></span> - Constructor for class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.Obstacle.html" title="class in com.ojcoleman.ahni.experiments.csb">SimpleNavigationEnvironment.Obstacle</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration"><span class="strong">SimpleSelector</span></a> - Class in <a href="../com/anji/integration/package-summary.html">com.anji.integration</a></dt>
<dd>
<div class="block">Selects chromosomes based directly on fitness value, as opposed to a statistical probability.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/SimpleSelector.html#SimpleSelector()">SimpleSelector()</a></span> - Constructor for class com.anji.integration.<a href="../com/anji/integration/SimpleSelector.html" title="class in com.anji.integration">SimpleSelector</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/nn/activationfunction/SineActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SineActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Sine activation function.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/MDP.html#SINGLE_REWARD_STATE">SINGLE_REWARD_STATE</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/MDP.html" title="class in com.ojcoleman.ahni.experiments">MDP</a></dt>
<dd>
<div class="block">Whether there should be a single state which when reached yields a reward value of 1 (rather than any transition
 to any state yielding a randomly generated reward value according to
 fitness.function.mdp.transition.reward.ratio.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/util/Properties.html#singletonObjectProperty(java.lang.String)">singletonObjectProperty(String)</a></span> - Method in class com.anji.util.<a href="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/util/Properties.html#singletonObjectProperty(java.lang.Class)">singletonObjectProperty(Class&lt;T&gt;)</a></span> - Method in class com.anji.util.<a href="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat"><span class="strong">SingleTopologicalMutationOperator</span></a> - Class in <a href="../com/anji/neat/package-summary.html">com.anji.neat</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/SingleTopologicalMutationOperator.html#SingleTopologicalMutationOperator()">SingleTopologicalMutationOperator()</a></span> - Constructor for class com.anji.neat.<a href="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat">SingleTopologicalMutationOperator</a></dt>
<dd>
<div class="block">should call <code>init()</code> after this constructor</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/SingleTopologicalMutationOperator.html#SingleTopologicalMutationOperator(double,%20double,%20com.anji.nn.RecurrencyPolicy)">SingleTopologicalMutationOperator(double, double, RecurrencyPolicy)</a></span> - Constructor for class com.anji.neat.<a href="../com/anji/neat/SingleTopologicalMutationOperator.html" title="class in com.anji.neat">SingleTopologicalMutationOperator</a></dt>
<dd>
<div class="block">ctor</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html#SixDegrees">SixDegrees</a></span> - Static variable in class com.ojcoleman.ahni.experiments.doublepolebalancing.<a href="../com/ojcoleman/ahni/experiments/doublepolebalancing/DoublePoleBalancing.html" title="class in com.ojcoleman.ahni.experiments.doublepolebalancing">DoublePoleBalancing</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/Environment.html#size">size</a></span> - Variable in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/Environment.html" title="class in com.ojcoleman.ahni.experiments.csb">Environment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html#SIZE">SIZE</a></span> - Static variable in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/RLContinuousStateBased.html" title="class in com.ojcoleman.ahni.experiments.csb">RLContinuousStateBased</a></dt>
<dd>
<div class="block">The number of variables defining an environment state.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html#SIZE">SIZE</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/RLRecurrentNetworkBased.html" title="class in com.ojcoleman.ahni.experiments">RLRecurrentNetworkBased</a></dt>
<dd>
<div class="block">The number of nodes in the network.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/bellmanfordsp/Bag.html#size()">size()</a></span> - Method in class com.ojcoleman.ahni.util.bellmanfordsp.<a href="../com/ojcoleman/ahni/util/bellmanfordsp/Bag.html" title="class in com.ojcoleman.ahni.util.bellmanfordsp">Bag</a></dt>
<dd>
<div class="block">Returns the number of items in this bag.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/bellmanfordsp/Queue.html#size()">size()</a></span> - Method in class com.ojcoleman.ahni.util.bellmanfordsp.<a href="../com/ojcoleman/ahni/util/bellmanfordsp/Queue.html" title="class in com.ojcoleman.ahni.util.bellmanfordsp">Queue</a></dt>
<dd>
<div class="block">Returns the number of items in this queue.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/CircularFifoBuffer.html#size()">size()</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="class in com.ojcoleman.ahni.util">CircularFifoBuffer</a></dt>
<dd>
<div class="block">Returns the number of elements in the buffer.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/DoubleVector.html#size()">size()</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/DoubleVector.html" title="class in com.ojcoleman.ahni.util">DoubleVector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#size()">size()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>
<div class="block">Returns the size of this Chromosome (the number of alleles it contains).</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#size()">size()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Returns the size of this ChromosomeMaterial (the number of alleles it contains).</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/Pool.html#size()">size()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/Pool.html" title="class in org.jgapcustomised.impl">Pool</a></dt>
<dd>
<div class="block">Retrieves the number of objects currently available in this pool.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#size()">size()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html#SKEWED">SKEWED</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html" title="class in com.anji.neat">RemoveConnectionMutationOperator.Strategy</a></dt>
<dd>
<div class="block">likelihood of connection being removed is inversely proprtional to weight</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html#SMALL">SMALL</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/RemoveConnectionMutationOperator.Strategy.html" title="class in com.anji.neat">RemoveConnectionMutationOperator.Strategy</a></dt>
<dd>
<div class="block">number of connections removed (n) is calculated based on total number of connections, and the n smallest
 connections are removed, excepting any whose weight is larger than <code>maxWeightRemoved</code></div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/DoubleVector.html#sort()">sort()</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/DoubleVector.html" title="class in com.ojcoleman.ahni.util">DoubleVector</a></dt>
<dd>
<div class="block">Sorts the elements in this vector into ascending numerical order.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.Connection.html#source">source</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.Connection.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.Connection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.TempConnection.html#sourcePoint">sourcePoint</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.TempConnection.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.TempConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/SpeciationStrategyKMeans.html#speciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">speciate(List&lt;Chromosome&gt;, List&lt;Species&gt;, Genotype)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/SpeciationStrategyKMeans.html" title="class in com.anji.neat">SpeciationStrategyKMeans</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/SpeciationStrategyOriginal.html#speciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">speciate(List&lt;Chromosome&gt;, List&lt;Species&gt;, Genotype)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/SpeciationStrategyOriginal.html" title="class in com.anji.neat">SpeciationStrategyOriginal</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationStrategy.html#speciate(java.util.List,%20java.util.List,%20org.jgapcustomised.Genotype)">speciate(List&lt;Chromosome&gt;, List&lt;Species&gt;, Genotype)</a></span> - Method in interface org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationStrategy.html" title="interface in org.jgapcustomised">SpeciationStrategy</a></dt>
<dd>
<div class="block">Speciates the genomes in genomeList into the provided species.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SPECIATED_FITNESS_KEY">SPECIATED_FITNESS_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#speciatedFitness">speciatedFitness</a></span> - Variable in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Genotype.html#SPECIATION_STRATEGY_CLASS_KEY">SPECIATION_STRATEGY_CLASS_KEY</a></span> - Static variable in class org.jgapcustomised.<a href="../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SPECIATION_TARGET_KEY">SPECIATION_TARGET_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, target number of species, default is popul.size / 15.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SPECIATION_THRESHOLD_CHANGE_KEY">SPECIATION_THRESHOLD_CHANGE_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block"><span class="strong">Deprecated.</span>
<div class="block"><i>Adjustment amounts are now calculated as a ratio of the target and current species counts. Properties key, amount to change speciation threshold to maintain speciation target</i></div>
</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SPECIATION_THRESHOLD_KEY">SPECIATION_THRESHOLD_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, speciation threshold</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SPECIATION_THRESHOLD_MAX_KEY">SPECIATION_THRESHOLD_MAX_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, speciation threshold maximum value for auto-adjustment</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SPECIATION_THRESHOLD_MIN_KEY">SPECIATION_THRESHOLD_MIN_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, speciation threshold minimum value for auto-adjustment</div>
</dd>
<dt><a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised"><span class="strong">SpeciationParms</span></a> - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#SpeciationParms()">SpeciationParms()</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>
<div class="block">default ctor</div>
</dd>
<dt><a href="../org/jgapcustomised/SpeciationStrategy.html" title="interface in org.jgapcustomised"><span class="strong">SpeciationStrategy</span></a> - Interface in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">An interface for classes that implement a strategy for dividing genomes into distinct <a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised"><code>Species</code></a>.</div>
</dd>
<dt><a href="../com/anji/neat/SpeciationStrategyKMeans.html" title="class in com.anji.neat"><span class="strong">SpeciationStrategyKMeans</span></a> - Class in <a href="../com/anji/neat/package-summary.html">com.anji.neat</a></dt>
<dd>
<div class="block">Implements speciation using k-means clustering, as described in <a
 href="https://sites.google.com/site/sharpneat/speciation/speciation-by-k-means-clustering"
 >https://sites.google.com/site/sharpneat/speciation/speciation-by-k-means-clustering</a>
 
 This code was adapted from SharpNEAT by Colin Green (see http://sharpneat.sourceforge.net/).</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/SpeciationStrategyKMeans.html#SpeciationStrategyKMeans()">SpeciationStrategyKMeans()</a></span> - Constructor for class com.anji.neat.<a href="../com/anji/neat/SpeciationStrategyKMeans.html" title="class in com.anji.neat">SpeciationStrategyKMeans</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/neat/SpeciationStrategyOriginal.html" title="class in com.anji.neat"><span class="strong">SpeciationStrategyOriginal</span></a> - Class in <a href="../com/anji/neat/package-summary.html">com.anji.neat</a></dt>
<dd>
<div class="block">Implements the original speciation strategy for NEAT as described in:<br />
 EVOLVING NEURAL NETWORKS THROUGH AUGMENTING TOPOLOGIES<br />
 Kenneth O.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/SpeciationStrategyOriginal.html#SpeciationStrategyOriginal()">SpeciationStrategyOriginal()</a></span> - Constructor for class com.anji.neat.<a href="../com/anji/neat/SpeciationStrategyOriginal.html" title="class in com.anji.neat">SpeciationStrategyOriginal</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#SPECIE_TAG">SPECIE_TAG</a></span> - Static variable in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">XML base tag</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#specieCompatMismatchUseValues()">specieCompatMismatchUseValues()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/SpeciationParms.html#specieCompatNormalise()">specieCompatNormalise()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#species">species</a></span> - Variable in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised"><span class="strong">Species</span></a> - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">Species are reproductively isolated segments of a population.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#Species(org.jgapcustomised.SpeciationParms,%20org.jgapcustomised.ChromosomeMaterial)">Species(SpeciationParms, ChromosomeMaterial)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">Create new species defined by given representative material.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#Species(org.jgapcustomised.SpeciationParms,%20org.jgapcustomised.Chromosome)">Species(SpeciationParms, Chromosome)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">Create new species with given Chromosome as first member of species.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/SqrtActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SqrtActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Square-root function.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/SqrtAndLinearActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SqrtAndLinearActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Square-root function for values with magnitude > 1, otherwise linear.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html#squareErrors()">squareErrors()</a></span> - Method in enum com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html" title="enum in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator.ErrorType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html#squareTotalError()">squareTotalError()</a></span> - Method in enum com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html" title="enum in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator.ErrorType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatIdMap.html#SRC_NEURON_ID_XML_TAG">SRC_NEURON_ID_XML_TAG</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</a></dt>
<dd>
<div class="block">XML tag for NEAT ID mapping, source neuron ID</div>
</dd>
<dt><span class="strong"><a href="../com/anji/persistence/FilePersistence.html#startRun(java.lang.String)">startRun(String)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/HibernatePersistence.html#startRun(java.lang.String)">startRun(String)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/Persistence.html#startRun(java.lang.String)">startRun(String)</a></span> - Method in interface com.anji.persistence.<a href="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</a></dt>
<dd>
<div class="block">Begin run.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/Environment.html#startState">startState</a></span> - Variable in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/Environment.html" title="class in com.ojcoleman.ahni.experiments.csb">Environment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/MDP.html#STATE_COUNT_INCREASE_DELTA">STATE_COUNT_INCREASE_DELTA</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/MDP.html" title="class in com.ojcoleman.ahni.experiments">MDP</a></dt>
<dd>
<div class="block">The amount to increase the number of states in the generated environments when the current size has been
 sufficiently mastered (see <a href="../com/ojcoleman/ahni/experiments/MDP.html#DIFFICULTY_INCREASE_PERFORMANCE"><code>MDP.DIFFICULTY_INCREASE_PERFORMANCE</code></a>.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/MDP.html#STATE_COUNT_INITIAL">STATE_COUNT_INITIAL</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/MDP.html" title="class in com.ojcoleman.ahni.experiments">MDP</a></dt>
<dd>
<div class="block">The initial number of states in the generated environments.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/MDP.html#STATE_COUNT_MAX">STATE_COUNT_MAX</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/MDP.html" title="class in com.ojcoleman.ahni.experiments">MDP</a></dt>
<dd>
<div class="block">The maximum amount to increase the number of states in the generated environments to.</div>
</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb"><span class="strong">StaticTransformEnvironment</span></a> - Class in <a href="../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a></dt>
<dd>
<div class="block">
 An environment which uses arbitrary functions to transform signals to and from the environment state.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.html#StaticTransformEnvironment()">StaticTransformEnvironment()</a></span> - Constructor for class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb">StaticTransformEnvironment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.Transform.html" title="class in com.ojcoleman.ahni.experiments.csb"><span class="strong">StaticTransformEnvironment.Transform</span></a> - Class in <a href="../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.Transform.html#StaticTransformEnvironment.Transform(com.ojcoleman.ahni.experiments.csb.StaticTransformEnvironment.TransformType,%20int,%20int,%20int,%20int,%20com.ojcoleman.ahni.util.Range)">StaticTransformEnvironment.Transform(StaticTransformEnvironment.TransformType, int, int, int, int, Range)</a></span> - Constructor for class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.Transform.html" title="class in com.ojcoleman.ahni.experiments.csb">StaticTransformEnvironment.Transform</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/csb/StaticTransformEnvironment.TransformType.html" title="enum in com.ojcoleman.ahni.experiments.csb"><span class="strong">StaticTransformEnvironment.TransformType</span></a> - Enum in <a href="../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/util/Statistics.html" title="class in com.ojcoleman.ahni.util"><span class="strong">Statistics</span></a> - Class in <a href="../com/ojcoleman/ahni/util/package-summary.html">com.ojcoleman.ahni.util</a></dt>
<dd>
<div class="block">Provides methods to perform statistical analysis on a set of results.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Statistics.html#Statistics(int)">Statistics(int)</a></span> - Constructor for class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Statistics.html" title="class in com.ojcoleman.ahni.util">Statistics</a></dt>
<dd>
<div class="block">Creates a new empty Statistics for run results consisting of the given number of generations.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Statistics.html#Statistics(com.ojcoleman.ahni.util.Results)">Statistics(Results)</a></span> - Constructor for class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Statistics.html" title="class in com.ojcoleman.ahni.util">Statistics</a></dt>
<dd>
<div class="block">Creates a new Statistics based on the given results.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/SteepSigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">SteepSigmoidActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Steepened Sigmoid with slope of 4.9.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/AnjiNet.html#step()">step()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</a></dt>
<dd>
<div class="block">indicates a time step has passed</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/CacheNeuronConnection.html#step()">step()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</a></dt>
<dd>
<div class="block">update value</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/Neuron.html#step()">step()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</a></dt>
<dd>
<div class="block">indicates a time step has passed</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/StepHourglassConnection.html#step()">step()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn">StepHourglassConnection</a></dt>
<dd>
<div class="block">proceed one step closer to completion</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/GridNet.html#step()">step()</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></dt>
<dd>
<div class="block">Perform one activation step, consisting of cyclesPerStep steps, calculating new activation for all neurons.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/StepActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">StepActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Step activation function.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/GridNet.html#stepFF()">stepFF()</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></dt>
<dd>
<div class="block">Perform one complete cycle for a feed forward network, propogating signal from input layer to output layer, only
 activating each layer once in sequence (connections can only exist from layer n to layer n+1).</div>
</dd>
<dt><a href="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn"><span class="strong">StepHourglassConnection</span></a> - Class in <a href="../com/anji/nn/package-summary.html">com.anji.nn</a></dt>
<dd>
<div class="block">StepHourglassConnection Connection returning a value between 0 and 1, where 1 indicates the fully allotted amount of
 time remaining and 0 indicates no time remainnig.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/StepHourglassConnection.html#StepHourglassConnection()">StepHourglassConnection()</a></span> - Constructor for class com.anji.nn.<a href="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn">StepHourglassConnection</a></dt>
<dd>
<div class="block">ctor</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/MDP.html#STEPS_PER_TRIAL">STEPS_PER_TRIAL</a></span> - Static variable in class com.ojcoleman.ahni.experiments.<a href="../com/ojcoleman/ahni/experiments/MDP.html" title="class in com.ojcoleman.ahni.experiments">MDP</a></dt>
<dd>
<div class="block">The number of steps in the environment per trial.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/TargetFitnessFunction.html#STIMULI_FILE_NAME_KEY">STIMULI_FILE_NAME_KEY</a></span> - Static variable in class com.anji.integration.<a href="../com/anji/integration/TargetFitnessFunction.html" title="class in com.anji.integration">TargetFitnessFunction</a></dt>
<dd>
<div class="block">properties key, file containing strimuli</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#STIMULUS_SIZE_KEY">STIMULUS_SIZE_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, dimension of neural net stimulus</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#store()">store()</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">Store to persistence.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatIdMap.html#store()">store()</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</a></dt>
<dd>
<div class="block">Persist ID factories and maps.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/persistence/FilePersistence.html#store(org.jgapcustomised.Chromosome)">store(Chromosome)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/FilePersistence.html#store(com.anji.integration.Activator)">store(Activator)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/FilePersistence.html#store(com.anji.run.Run)">store(Run)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/HibernatePersistence.html#store(org.jgapcustomised.Chromosome)">store(Chromosome)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/HibernatePersistence.html#store(com.anji.integration.Activator)">store(Activator)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/HibernatePersistence.html#store(com.anji.run.Run)">store(Run)</a></span> - Method in class com.anji.persistence.<a href="../com/anji/persistence/HibernatePersistence.html" title="class in com.anji.persistence">HibernatePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/Persistence.html#store(org.jgapcustomised.Chromosome)">store(Chromosome)</a></span> - Method in interface com.anji.persistence.<a href="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/Persistence.html#store(com.anji.integration.Activator)">store(Activator)</a></span> - Method in interface com.anji.persistence.<a href="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/Persistence.html#store(com.anji.run.Run)">store(Run)</a></span> - Method in interface com.anji.persistence.<a href="../com/anji/persistence/Persistence.html" title="interface in com.anji.persistence">Persistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/IdFactory.html#store()">store()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/IdFactory.html" title="class in org.jgapcustomised">IdFactory</a></dt>
<dd>
<div class="block">persist object to file</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/PresentationEventListener.html#storeRun(boolean)">storeRun(boolean)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/PresentationEventListener.html" title="class in com.anji.integration">PresentationEventListener</a></dt>
<dd>
<div class="block">Store/update run presentation data based on <code>genotype</code>.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/Copyright.html#STRING">STRING</a></span> - Static variable in class com.anji.<a href="../com/anji/Copyright.html" title="class in com.anji">Copyright</a></dt>
<dd>
<div class="block">copyright string</div>
</dd>
<dt><span class="strong"><a href="../com/anji/util/Properties.html#SUBSTITUTION_ENABLE_KEY">SUBSTITUTION_ENABLE_KEY</a></span> - Static variable in class com.anji.util.<a href="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a></dt>
<dd>
<div class="block">Optional property key.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_CYCLES_PER_STEP">SUBSTRATE_CYCLES_PER_STEP</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">For recurrent networks, the number of activation cycles to perform each time the substrate network is presented
 with new input and queried for its output.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_DEPTH">SUBSTRATE_DEPTH</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block"><span class="strong">Deprecated.</span>
<div class="block"><i>The number of layers in the substrate network. This is no longer used, the depth is now determined by the number of elements in ann.hyperneat.height and ann.hyperneat.width.</i></div>
</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#SUBSTRATE_EXECUTION_MODE">SUBSTRATE_EXECUTION_MODE</a></span> - Static variable in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_HEIGHT">SUBSTRATE_HEIGHT</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">Comma separated list of the height of each layer in the network, including input and output layers and starting
 with the input layer.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html#SUBSTRATE_MAX_RECURRENT_CYCLE">SUBSTRATE_MAX_RECURRENT_CYCLE</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBain.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBain</a></dt>
<dd>
<div class="block">When determining if the substrate network is recurrent, the maximum cycle length to search for before the network
 is considered recurrent.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL">SUBSTRATE_NEURON_MODEL</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify which neuron model to use.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS">SUBSTRATE_NEURON_MODEL_PARAMS</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify the available parameters for neurons.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES">SUBSTRATE_NEURON_MODEL_PARAMS_CLASSES</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Optional property to allow specifying that instead of parameter values being defined individually for each
 neuron, there will be a set of parameter value collections which can be thought of as defining a class of neuron
 as determined by the specific parameter values (which will determine the behaviour of that class of neuron).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_MAX">SUBSTRATE_NEURON_MODEL_PARAMS_MAX</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify maximum values for the parameters for each neuron.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_MIN">SUBSTRATE_NEURON_MODEL_PARAMS_MIN</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify minimum values for the parameters for each neuron.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD">SUBSTRATE_NEURON_MODEL_PARAMS_THRESHOLD</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify the minimum CPPN output required to produce a non-zero parameter value.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_NEURON_MODEL_TYPES">SUBSTRATE_NEURON_MODEL_TYPES</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify allowable neuron model types.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#SUBSTRATE_SIMULATION_RESOLUTION">SUBSTRATE_SIMULATION_RESOLUTION</a></span> - Static variable in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL">SUBSTRATE_SYNAPSE_MODEL</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify which synapse model to use.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM">SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify which parameter in the synapse model will be set to 0 if the connection should not be enabled
 or plastic.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS">SUBSTRATE_SYNAPSE_MODEL_PARAMS</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify the available parameters for synapses.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES">SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Optional property to allow specifying that instead of parameter values being defined individually for each
 synapse, there will be a set of parameter value collections which can be thought of as defining a class of
 synapse as determined by the specific parameter values (which will determine the behaviour of that class of
 synapse).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX">SUBSTRATE_SYNAPSE_MODEL_PARAMS_MAX</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify maximum values for the parameters for each synapse.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN">SUBSTRATE_SYNAPSE_MODEL_PARAMS_MIN</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify minimum values for the parameters for each synapse.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD">SUBSTRATE_SYNAPSE_MODEL_PARAMS_THRESHOLD</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify the minimum CPPN output required to produce a non-zero parameter value.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_TYPES">SUBSTRATE_SYNAPSE_MODEL_TYPES</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Property to specify allowable synapse model types.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#SUBSTRATE_WIDTH">SUBSTRATE_WIDTH</a></span> - Static variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">Comma separated list of the width of each layer in the network, including input and output layers and starting
 with the input layer.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/ArrayUtil.html#sum(double[])">sum(double[])</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/ArrayUtil.html" title="class in com.ojcoleman.ahni.util">ArrayUtil</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/ArrayUtil.html#sum(int[])">sum(int[])</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/ArrayUtil.html" title="class in com.ojcoleman.ahni.util">ArrayUtil</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html#sumErrors()">sumErrors()</a></span> - Method in enum com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/TargetFitnessCalculator.ErrorType.html" title="enum in com.ojcoleman.ahni.evaluation">TargetFitnessCalculator.ErrorType</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#SURVIVAL_RATE_KEY">SURVIVAL_RATE_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, survival rate</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/NaturalSelector.html#survivalRate">survivalRate</a></span> - Variable in class org.jgapcustomised.<a href="../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised">NaturalSelector</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#synapseDisableParamClassIndex">synapseDisableParamClassIndex</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriberBainBase</a></dt>
<dd>
<div class="block">If synapse parameter classes are being used (see <a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES"><code>TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_PARAMS_CLASSES</code></a>, <a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseDisableParamName"><code>TranscriberAdaptor.synapseDisableParamName</code></a>) and 
 a synapse disable parameter has been specified (see <a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM"><code>TranscriberAdaptor.SUBSTRATE_SYNAPSE_MODEL_DISABLE_PARAM</code></a>), then this should
 be set to the index of a <code>SynapseConfiguration</code> in the <code>SynapseCollection</code> passed to <a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriberBainBase.html#setSynapseParameters(com.ojcoleman.bain.base.SynapseCollection,%20int,%20com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN,%20boolean,%20boolean)"><code>HyperNEATTranscriberBainBase.setSynapseParameters(SynapseCollection, int, com.ojcoleman.ahni.transcriber.HyperNEATTranscriber.CPPN, boolean, boolean)</code></a>
 that has the disable parameter set to 0, otherwise it should be left with the value -1.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseDisableParamName">synapseDisableParamName</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#synapseLength">synapseLength</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>
<div class="block">Squared length of the current synapse (set when query() called).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamClassCount">synapseModelParamClassCount</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The number of parameter collections defining synapse model classes/types.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsMax">synapseModelParamsMax</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The maximum values for each synapse model parameter.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsMin">synapseModelParamsMin</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The minimum values for each synapse model parameter.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsRange">synapseModelParamsRange</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The value range for each synapse model parameter (synapseModelParamsMax[i] - synapseModelParamsMin[i]).</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelParamsThreshold">synapseModelParamsThreshold</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">the minimum (possibly CPPN) output required to produce a non-zero parameter value.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeCount">synapseModelTypeCount</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The number of available synapse types.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam">synapseModelTypeParam</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The name of the parameter in the synapse model that specifies the type of a synapse.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypes">synapseModelTypes</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The allowable values for the <a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseModelTypeParam"><code>TranscriberAdaptor.synapseModelTypeParam</code></a> model parameter that specify the type of a synapse.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamNames">synapseParamNames</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The names or labels for parameters in the synapse model.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseParamsEnabled">synapseParamsEnabled</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Subclasses may use this to know whether synapse model parameters are enabled and available.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#synapseTypesEnabled">synapseTypesEnabled</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">Subclasses may use this to know whether multiple synapse types are enabled and available.</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">J</a>&nbsp;<a href="index-11.html">K</a>&nbsp;<a href="index-12.html">L</a>&nbsp;<a href="index-13.html">M</a>&nbsp;<a href="index-14.html">N</a>&nbsp;<a href="index-15.html">O</a>&nbsp;<a href="index-16.html">P</a>&nbsp;<a href="index-17.html">Q</a>&nbsp;<a href="index-18.html">R</a>&nbsp;<a href="index-19.html">S</a>&nbsp;<a href="index-20.html">T</a>&nbsp;<a href="index-21.html">U</a>&nbsp;<a href="index-22.html">V</a>&nbsp;<a href="index-23.html">W</a>&nbsp;<a href="index-24.html">X</a>&nbsp;<a href="index-25.html">Y</a>&nbsp;<a href="index-26.html">Z</a>&nbsp;<a href="index-27.html">_</a>&nbsp;</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="index-18.html">Prev Letter</a></li>
<li><a href="index-20.html">Next Letter</a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?index-filesindex-19.html" target="_top">Frames</a></li>
<li><a href="index-19.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
