<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.6.0_27) on Tue Aug 06 11:43:00 EST 2013 -->
<title>C-Index</title>
<meta name="date" content="2013-08-06">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="C-Index";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="index-2.html">PREV LETTER</a></li>
<li><a href="index-4.html">NEXT LETTER</a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?index-filesindex-3.html" target="_top">FRAMES</a></li>
<li><a href="index-3.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="contentContainer"><a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">K</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;<a name="_C_">
<!--   -->
</a>
<h2 class="title">C</h2>
<dl>
<dt><a href="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn"><span class="strong">CacheNeuronConnection</span></a> - Class in <a href="../com/anji/nn/package-summary.html">com.anji.nn</a></dt>
<dd>
<div class="block">A connection between neurons that caches its value from previous timestep.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/CacheNeuronConnection.html#CacheNeuronConnection(com.anji.nn.Neuron)">CacheNeuronConnection(Neuron)</a></span> - Constructor for class com.anji.nn.<a href="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/CacheNeuronConnection.html#CacheNeuronConnection(com.anji.nn.Neuron, double)">CacheNeuronConnection(Neuron, double)</a></span> - Constructor for class com.anji.nn.<a href="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#calculateAverageFitness()">calculateAverageFitness()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#calculateBoundsUnitsToIntegerUnitsRatio()">calculateBoundsUnitsToIntegerUnitsRatio()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>
<div class="block">Calculates and sets the m_boundsUnitsToIntegerUnits field based
 
 on the current lower and upper bounds of this IntegerGene.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/ErrorFunction.html#calculateError(double[][], double[][], boolean)">calculateError(double[][], double[][], boolean)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/ErrorFunction.html" title="class in com.anji.integration">ErrorFunction</a></dt>
<dd>
<div class="block">Calculate the sum of differences between <code>responses</code> and target values.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/ErrorFunction.html#calculateError(double[][][], double[][][], boolean)">calculateError(double[][][], double[][][], boolean)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/ErrorFunction.html" title="class in com.anji.integration">ErrorFunction</a></dt>
<dd>
<div class="block">Calculate the sum of differences between <code>responses</code> and target values.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/ExponentialTargetFitnessFunction.html#calculateErrorFitness(double[][], double, double)">calculateErrorFitness(double[][], double, double)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/ExponentialTargetFitnessFunction.html" title="class in com.anji.integration">ExponentialTargetFitnessFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/TargetFitnessFunction.html#calculateErrorFitness(double[][], double, double)">calculateErrorFitness(double[][], double, double)</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/TargetFitnessFunction.html" title="class in com.anji.integration">TargetFitnessFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatTargetFitnessFunction.html#calculateErrorFitness(double[][], double, double)">calculateErrorFitness(double[][], double, double)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeatTargetFitnessFunction.html" title="class in com.anji.neat">NeatTargetFitnessFunction</a></dt>
<dd>
<div class="block">Subtract <code>responses</code> from targets, sum all differences, subtract from max fitness, and square result.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/Range.html#checkUnitRange(double, java.lang.String)">checkUnitRange(double, String)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/Range.html" title="class in com.ojcoleman.ahni.util">Range</a></dt>
<dd>
<div class="block">Ensure the given value is in the unit range.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html#children">children</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.QuadPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#CHROM_COMPAT_COMMON_COEFF_KEY">CHROM_COMPAT_COMMON_COEFF_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, speciation chromosome compatibility common coefficient</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#CHROM_COMPAT_DISJOINT_COEFF_KEY">CHROM_COMPAT_DISJOINT_COEFF_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, speciation chromosome compatibility disjoint coefficient</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#CHROM_COMPAT_EXCESS_COEFF_KEY">CHROM_COMPAT_EXCESS_COEFF_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, speciation chromosome compatibility excess coefficient</div>
</dd>
<dt><a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised"><span class="strong">Chromosome</span></a> - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">Chromosomes represent potential solutions and consist of a fixed-length collection of genes.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long, int)">Chromosome(ChromosomeMaterial, Long, int)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>
<div class="block">this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#CHROMOSOME_TAG">CHROMOSOME_TAG</a></span> - Static variable in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">XML chromosome tag</div>
</dd>
<dt><a href="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><span class="strong">ChromosomeFitnessComparator</span></a>&lt;<a href="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="type parameter in ChromosomeFitnessComparator">T</a>&gt; - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">Enables sorting of chromosomes by their fitness.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeFitnessComparator.html#ChromosomeFitnessComparator()">ChromosomeFitnessComparator()</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised">ChromosomeFitnessComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeFitnessComparator.html#ChromosomeFitnessComparator(boolean, boolean)">ChromosomeFitnessComparator(boolean, boolean)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised">ChromosomeFitnessComparator</a></dt>
<dd>
<div class="block">Enables sorting of chromosomes in order of fitness.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, java.lang.String)">chromosomeFromXml(Configuration, String)</a></span> - Static method in class com.anji.persistence.<a href="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/persistence/FilePersistence.html#chromosomeFromXml(org.jgapcustomised.Configuration, org.w3c.dom.Node)">chromosomeFromXml(Configuration, Node)</a></span> - Static method in class com.anji.persistence.<a href="../com/anji/persistence/FilePersistence.html" title="class in com.anji.persistence">FilePersistence</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised"><span class="strong">ChromosomeMaterial</span></a> - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">This is the guts of the original Chromosome object, pulled out so the genes can be modified by genetic operators
 before creating the Chromosome object.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)">ChromosomeMaterial(Collection&lt;Allele&gt;, Long, Long)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Create chromosome with two parents.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long)">ChromosomeMaterial(Collection&lt;Allele&gt;, Long)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Create chromosome with one parents.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection)">ChromosomeMaterial(Collection&lt;Allele&gt;)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Create chromosome with no parents.</div>
</dd>
<dt><a href="../org/jgapcustomised/ChromosomePerformanceComparator.html" title="class in org.jgapcustomised"><span class="strong">ChromosomePerformanceComparator</span></a>&lt;<a href="../org/jgapcustomised/ChromosomePerformanceComparator.html" title="type parameter in ChromosomePerformanceComparator">T</a>&gt; - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">Enables sorting of chromosomes by their performance.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomePerformanceComparator.html#ChromosomePerformanceComparator()">ChromosomePerformanceComparator()</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomePerformanceComparator.html" title="class in org.jgapcustomised">ChromosomePerformanceComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomePerformanceComparator.html#ChromosomePerformanceComparator(boolean)">ChromosomePerformanceComparator(boolean)</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomePerformanceComparator.html" title="class in org.jgapcustomised">ChromosomePerformanceComparator</a></dt>
<dd>
<div class="block">Enables sorting of chromosomes in order of fitness.</div>
</dd>
<dt><a href="../org/jgapcustomised/impl/ChromosomePool.html" title="class in org.jgapcustomised.impl"><span class="strong">ChromosomePool</span></a> - Class in <a href="../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a></dt>
<dd>
<div class="block">Provides a pooling mechanism for Chromosome instances so that discarded Chromosome instances can be recycled, thus
 saving memory and the overhead of constructing new ones from scratch each time.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/ChromosomePool.html#ChromosomePool()">ChromosomePool()</a></span> - Constructor for class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/ChromosomePool.html" title="class in org.jgapcustomised.impl">ChromosomePool</a></dt>
<dd>
<div class="block">Constructor.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html#chromosomesIterator">chromosomesIterator</a></span> - Variable in class com.ojcoleman.ahni.evaluation.<a href="../com/ojcoleman/ahni/evaluation/BulkFitnessFunctionMT.html" title="class in com.ojcoleman.ahni.evaluation">BulkFitnessFunctionMT</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/integration/ChromosomeToNetworkXml.html" title="class in com.anji.integration"><span class="strong">ChromosomeToNetworkXml</span></a> - Class in <a href="../com/anji/integration/package-summary.html">com.anji.integration</a></dt>
<dd>
<div class="block">Converts chromosome to activator, which it then uses to write network XML according to <a
 href="http://nevt.sourceforge.net/">NEVT </a> data model.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/ChromosomeToNetworkXml.html#ChromosomeToNetworkXml()">ChromosomeToNetworkXml()</a></span> - Constructor for class com.anji.integration.<a href="../com/anji/integration/ChromosomeToNetworkXml.html" title="class in com.anji.integration">ChromosomeToNetworkXml</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="class in com.ojcoleman.ahni.util"><span class="strong">CircularFifoBuffer</span></a>&lt;<a href="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="type parameter in CircularFifoBuffer">E</a>&gt; - Class in <a href="../com/ojcoleman/ahni/util/package-summary.html">com.ojcoleman.ahni.util</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/CircularFifoBuffer.html#CircularFifoBuffer(int)">CircularFifoBuffer(int)</a></span> - Constructor for class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/CircularFifoBuffer.html" title="class in com.ojcoleman.ahni.util">CircularFifoBuffer</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/nn/activationfunction/ClampedAbsoluteActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">ClampedAbsoluteActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Absolute activation function.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">ClampedLinearActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html#ClampedLinearActivationFunction()">ClampedLinearActivationFunction()</a></span> - Constructor for class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">ClampedLinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/util/Properties.html#CLASS_SUFFIX">CLASS_SUFFIX</a></span> - Static variable in class com.anji.util.<a href="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a></dt>
<dd>
<div class="block">made public for unit tests</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/BooleanAllele.html#cleanup()">cleanup()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></dt>
<dd>
<div class="block">Executed by the genetic engine when this Gene instance is no
 
 longer needed and should perform any necessary resource cleanup.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#cleanup()">cleanup()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>
<div class="block">Executed by the genetic engine when this Gene instance is no
 
 longer needed and should perform any necessary resource cleanup.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/Pool.html#clear()">clear()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/Pool.html" title="class in org.jgapcustomised.impl">Pool</a></dt>
<dd>
<div class="block">Empties out this pool of all objects.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#clear()">clear()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">Remove all chromosomes from this species.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#clone(java.lang.Long)">clone(Long)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Returns a copy of this ChromosomeMaterial.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/ConnectionAllele.html#cloneAllele()">cloneAllele()</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeuronAllele.html#cloneAllele()">cloneAllele()</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Allele.html#cloneAllele()">cloneAllele()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/BooleanAllele.html#cloneAllele()">cloneAllele()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/BooleanAllele.html" title="class in org.jgapcustomised.impl">BooleanAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/IntegerAllele.html#cloneAllele()">cloneAllele()</a></span> - Method in class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/IntegerAllele.html" title="class in org.jgapcustomised.impl">IntegerAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#cloneMaterial()">cloneMaterial()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#cloneOper">cloneOper</a></span> - Variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/jgapcustomised/impl/CloneReproductionOperator.html" title="class in org.jgapcustomised.impl"><span class="strong">CloneReproductionOperator</span></a> - Class in <a href="../org/jgapcustomised/impl/package-summary.html">org.jgapcustomised.impl</a></dt>
<dd>
<div class="block">Produces offspring be creating clones of parents.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/impl/CloneReproductionOperator.html#CloneReproductionOperator()">CloneReproductionOperator()</a></span> - Constructor for class org.jgapcustomised.impl.<a href="../org/jgapcustomised/impl/CloneReproductionOperator.html" title="class in org.jgapcustomised.impl">CloneReproductionOperator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html#collision(int, org.apache.commons.math3.linear.ArrayRealVector)">collision(int, ArrayRealVector)</a></span> - Method in class com.ojcoleman.ahni.experiments.csb.<a href="../com/ojcoleman/ahni/experiments/csb/SimpleNavigationEnvironment.html" title="class in com.ojcoleman.ahni.experiments.csb">SimpleNavigationEnvironment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/package-summary.html">com.anji</a> - package com.anji</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/integration/package-summary.html">com.anji.integration</a> - package com.anji.integration</dt>
<dd>
<div class="block"><code>com.anji.integration</code> contains classes that provide the glue between the various components of the system, including artificial neural net implementations, <a href="http://jgap.sourceforge.net/">JGAP</a>, <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a>, Log4J, and persistence and presentation layers.</div>
</dd>
<dt><a href="../com/anji/neat/package-summary.html">com.anji.neat</a> - package com.anji.neat</dt>
<dd>
<div class="block"><code>com.anji.neat</code> contains classes implementing <a href="http://www.cs.utexas.edu/users/kstanley/neat.html">NEAT</a> framework, including genes, mutation operators, and fitness function used for XOR.</div>
</dd>
<dt><a href="../com/anji/nn/package-summary.html">com.anji.nn</a> - package com.anji.nn</dt>
<dd>
<div class="block"><code>com.anji.nn</code> contains classes implementing a simple artificial neural net framework.</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a> - package com.anji.nn.activationfunction</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/persistence/package-summary.html">com.anji.persistence</a> - package com.anji.persistence</dt>
<dd>
<div class="block"><code>com.anji.persistence</code> contains classes implementing a simple persistence interface.</div>
</dd>
<dt><a href="../com/anji/run/package-summary.html">com.anji.run</a> - package com.anji.run</dt>
<dd>
<div class="block"><code>com.anji.run</code> contains classes concerned with an evolutionary run.</div>
</dd>
<dt><a href="../com/anji/util/package-summary.html">com.anji.util</a> - package com.anji.util</dt>
<dd>
<div class="block"><code>com.anji.util</code> contains utility classes used in the ANJI framework.</div>
</dd>
<dt><a href="../com/ojcoleman/ahni/package-summary.html">com.ojcoleman.ahni</a> - package com.ojcoleman.ahni</dt>
<dd>
<div class="block"><strong>AHNI - Another HyperNEAT Implementation.</strong></div>
</dd>
<dt><a href="../com/ojcoleman/ahni/evaluation/package-summary.html">com.ojcoleman.ahni.evaluation</a> - package com.ojcoleman.ahni.evaluation</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</a> - package com.ojcoleman.ahni.evaluation.mocostfunctions</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/evaluation/novelty/package-summary.html">com.ojcoleman.ahni.evaluation.novelty</a> - package com.ojcoleman.ahni.evaluation.novelty</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/event/package-summary.html">com.ojcoleman.ahni.event</a> - package com.ojcoleman.ahni.event</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/package-summary.html">com.ojcoleman.ahni.experiments</a> - package com.ojcoleman.ahni.experiments</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/csb/package-summary.html">com.ojcoleman.ahni.experiments.csb</a> - package com.ojcoleman.ahni.experiments.csb</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/experiments/objectrecognition/package-summary.html">com.ojcoleman.ahni.experiments.objectrecognition</a> - package com.ojcoleman.ahni.experiments.objectrecognition</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</a> - package com.ojcoleman.ahni.hyperneat</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/misc/package-summary.html">com.ojcoleman.ahni.misc</a> - package com.ojcoleman.ahni.misc</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/nn/package-summary.html">com.ojcoleman.ahni.nn</a> - package com.ojcoleman.ahni.nn</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/transcriber/package-summary.html">com.ojcoleman.ahni.transcriber</a> - package com.ojcoleman.ahni.transcriber</dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/util/package-summary.html">com.ojcoleman.ahni.util</a> - package com.ojcoleman.ahni.util</dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/PostProcess.html#combineResults(java.lang.String, java.io.BufferedWriter)">combineResults(String, BufferedWriter)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</a></dt>
<dd>
<div class="block">Combine the results from multiple runs into one file.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/PostProcess.html#combineResults(java.lang.String, boolean)">combineResults(String, boolean)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</a></dt>
<dd>
<div class="block">Combine the results from multiple runs into one Results object.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/WeightMagnitudeComparator.html#compare(java.lang.Object, java.lang.Object)">compare(Object, Object)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/WeightMagnitudeComparator.html" title="class in com.anji.neat">WeightMagnitudeComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeFitnessComparator.html#compare(T, T)">compare(T, T)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised">ChromosomeFitnessComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomePerformanceComparator.html#compare(T, T)">compare(T, T)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomePerformanceComparator.html" title="class in org.jgapcustomised">ChromosomePerformanceComparator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Allele.html#compareTo(java.lang.Object)">compareTo(Object)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#compareTo(java.lang.Object)">compareTo(Object)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>
<div class="block">Compares the given Chromosome to this Chromosome by their ID.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/ChromosomeMaterial.html#compareTo(java.lang.Object)">compareTo(Object)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></dt>
<dd>
<div class="block">Compares the given Chromosome to this Chromosome.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Gene.html#compareTo(java.lang.Object)">compareTo(Object)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised">Gene</a></dt>
<dd>
<div class="block">Comparison based on innovation ID.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/PostProcess.html#compressAverage(java.io.BufferedReader, java.io.BufferedWriter, int)">compressAverage(BufferedReader, BufferedWriter, int)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</a></dt>
<dd>
<div class="block">Given a result file from a set of runs, for each run computes averages over some number of generations (the
 window size) within the run.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/PostProcess.html#compressAverage(com.ojcoleman.ahni.util.Results, int)">compressAverage(Results, int)</a></span> - Static method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/PostProcess.html" title="class in com.ojcoleman.ahni.util">PostProcess</a></dt>
<dd>
<div class="block">Given a result file from a set of runs, for each run computes averages over some number of generations (the
 window size) within the run.</div>
</dd>
<dt><a href="../com/anji/util/Configurable.html" title="interface in com.anji.util"><span class="strong">Configurable</span></a> - Interface in <a href="../com/anji/util/package-summary.html">com.anji.util</a></dt>
<dd>
<div class="block">All classes implementing this interface must have a public no-argument constructor.</div>
</dd>
<dt><a href="../com/ojcoleman/ahni/hyperneat/Configurable.html" title="interface in com.ojcoleman.ahni.hyperneat"><span class="strong">Configurable</span></a> - Interface in <a href="../com/ojcoleman/ahni/hyperneat/package-summary.html">com.ojcoleman.ahni.hyperneat</a></dt>
<dd>
<div class="block">All classes implementing this interface must have a public no-argument constructor.</div>
</dd>
<dt><a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised"><span class="strong">Configuration</span></a> - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">The Configuration class represents the current configuration of plugins and flags necessary to execute the genetic
 algorithm (such as fitness function, natural selector, genetic operators, and so on).</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Configuration.html#Configuration()">Configuration()</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/util/Properties.html#configureLogger()">configureLogger()</a></span> - Method in class com.anji.util.<a href="../com/anji/util/Properties.html" title="class in com.anji.util">Properties</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/XmlPersistableAllele.html#CONN_XML_TAG">CONN_XML_TAG</a></span> - Static variable in class com.anji.integration.<a href="../com/anji/integration/XmlPersistableAllele.html" title="class in com.anji.integration">XmlPersistableAllele</a></dt>
<dd>
<div class="block">connection XML tag</div>
</dd>
<dt><a href="../com/anji/nn/Connection.html" title="interface in com.anji.nn"><span class="strong">Connection</span></a> - Interface in <a href="../com/anji/nn/package-summary.html">com.anji.nn</a></dt>
<dd>
<div class="block">Abstract connection to carry data into neurons.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatIdMap.html#CONNECTION_ID_MAP_XML_TAG">CONNECTION_ID_MAP_XML_TAG</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</a></dt>
<dd>
<div class="block">XML tag for NEAT ID mapping,connection ID</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatIdMap.html#CONNECTION_ID_XML_TAG">CONNECTION_ID_XML_TAG</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</a></dt>
<dd>
<div class="block">XML tag for NEAT ID mapping, connection ID</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatIdMap.html#CONNECTION_TO_CONNECTION_XML_TAG">CONNECTION_TO_CONNECTION_XML_TAG</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</a></dt>
<dd>
<div class="block">XML tag for NEAT ID mapping, connection</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatIdMap.html#CONNECTION_TO_NEURON_XML_TAG">CONNECTION_TO_NEURON_XML_TAG</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatIdMap.html" title="class in com.anji.neat">NeatIdMap</a></dt>
<dd>
<div class="block">XML tag for NEAT ID mapping, neuron</div>
</dd>
<dt><a href="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat"><span class="strong">ConnectionAllele</span></a> - Class in <a href="../com/anji/neat/package-summary.html">com.anji.neat</a></dt>
<dd>
<div class="block">Gene corresponding to NEAT connection gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a></div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/ConnectionAllele.html#ConnectionAllele(com.anji.neat.ConnectionGene)">ConnectionAllele(ConnectionGene)</a></span> - Constructor for class com.anji.neat.<a href="../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/AddConnectionMutationOperator.html#connectionAllowed(com.anji.neat.NeuronAllele, com.anji.neat.NeuronAllele, java.util.Collection)">connectionAllowed(NeuronAllele, NeuronAllele, Collection&lt;ConnectionAllele&gt;)</a></span> - Method in class com.anji.neat.<a href="../com/anji/neat/AddConnectionMutationOperator.html" title="class in com.anji.neat">AddConnectionMutationOperator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#connectionExprThresh">connectionExprThresh</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">The minimum output from the CPPN required to create a non-zero weight value in the substrate.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/XmlPersistableAllele.html#connectionFromXml(org.w3c.dom.Node)">connectionFromXml(Node)</a></span> - Static method in class com.anji.integration.<a href="../com/anji/integration/XmlPersistableAllele.html" title="class in com.anji.integration">XmlPersistableAllele</a></dt>
<dd>
<div class="block">Convert from XML to <code>ConnectionGene</code> object</div>
</dd>
<dt><a href="../com/anji/neat/ConnectionGene.html" title="class in com.anji.neat"><span class="strong">ConnectionGene</span></a> - Class in <a href="../com/anji/neat/package-summary.html">com.anji.neat</a></dt>
<dd>
<div class="block">Gene corresponding to NEAT connection gene according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf"> Evolving Neural Networks through Augmenting
 Topologies </a></div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/ConnectionGene.html#ConnectionGene(java.lang.Long, java.lang.Long, java.lang.Long)">ConnectionGene(Long, Long, Long)</a></span> - Constructor for class com.anji.neat.<a href="../com/anji/neat/ConnectionGene.html" title="class in com.anji.neat">ConnectionGene</a></dt>
<dd>
<div class="block">Construct new ConnectionGene with given src, destination, and ID.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#connectionRange">connectionRange</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">Limits the incoming connections to a target neuron to include those from source neurons within the specified
 range of the target neuron.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#connectionWeightMax">connectionWeightMax</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The maximum connection weight in the substrate.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html#connectionWeightMin">connectionWeightMin</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/TranscriberAdaptor.html" title="class in com.ojcoleman.ahni.transcriber">TranscriberAdaptor</a></dt>
<dd>
<div class="block">The minimum connection weight in the substrate.</div>
</dd>
<dt><a href="../com/anji/integration/ConsoleLogEventListener.html" title="class in com.anji.integration"><span class="strong">ConsoleLogEventListener</span></a> - Class in <a href="../com/anji/integration/package-summary.html">com.anji.integration</a></dt>
<dd>
<div class="block">Writes log events to stdout.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/ConsoleLogEventListener.html#ConsoleLogEventListener(org.jgapcustomised.Configuration)">ConsoleLogEventListener(Configuration)</a></span> - Constructor for class com.anji.integration.<a href="../com/anji/integration/ConsoleLogEventListener.html" title="class in com.anji.integration">ConsoleLogEventListener</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/ConsoleLogEventListener.html#ConsoleLogEventListener(org.jgapcustomised.Configuration, java.io.PrintStream)">ConsoleLogEventListener(Configuration, PrintStream)</a></span> - Constructor for class com.anji.integration.<a href="../com/anji/integration/ConsoleLogEventListener.html" title="class in com.anji.integration">ConsoleLogEventListener</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#containsBestPerforming">containsBestPerforming</a></span> - Variable in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">True iff this species contains the bestPerforming individual from the entire population.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/util/PropertiesConverter.html#convert(java.lang.String)">convert(String)</a></span> - Method in class com.ojcoleman.ahni.util.<a href="../com/ojcoleman/ahni/util/PropertiesConverter.html" title="class in com.ojcoleman.ahni.util">PropertiesConverter</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">ConvertToSignedActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html#ConvertToSignedActivationFunction()">ConvertToSignedActivationFunction()</a></span> - Constructor for class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html" title="class in com.anji.nn.activationfunction">ConvertToSignedActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/NNAdaptor.html#coords">coords</a></span> - Variable in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/NNAdaptor.html#coordsEnabled()">coordsEnabled()</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</a></dt>
<dd>
<div class="block">Returns true iff storing coordinates for neurons has been enabled.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/NNAdaptor.html#coordsMin">coordsMin</a></span> - Variable in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/NNAdaptor.html#coordsRange">coordsRange</a></span> - Variable in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/NNAdaptor.html" title="class in com.ojcoleman.ahni.nn">NNAdaptor</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/anji/Copyright.html" title="class in com.anji"><span class="strong">Copyright</span></a> - Class in <a href="../com/anji/package-summary.html">com.anji</a></dt>
<dd>
<div class="block">Copyright</div>
</dd>
<dt><a href="../com/anji/nn/activationfunction/CosineActivationFunction.html" title="class in com.anji.nn.activationfunction"><span class="strong">CosineActivationFunction</span></a> - Class in <a href="../com/anji/nn/activationfunction/package-summary.html">com.anji.nn.activationfunction</a></dt>
<dd>
<div class="block">Cosine activation function.</div>
</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/AbsoluteActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/AbsoluteActivationFunction.html" title="class in com.anji.nn.activationfunction">AbsoluteActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/ActivationFunction.html#cost()">cost()</a></span> - Method in interface com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/ActivationFunction.html" title="interface in com.anji.nn.activationfunction">ActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/BipolarSigmoidActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/BipolarSigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">BipolarSigmoidActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/ClampedAbsoluteActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/ClampedAbsoluteActivationFunction.html" title="class in com.anji.nn.activationfunction">ClampedAbsoluteActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/ClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">ClampedLinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/ConvertToSignedActivationFunction.html" title="class in com.anji.nn.activationfunction">ConvertToSignedActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/CosineActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/CosineActivationFunction.html" title="class in com.anji.nn.activationfunction">CosineActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/DivideActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/DivideActivationFunction.html" title="class in com.anji.nn.activationfunction">DivideActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/EvSailSigmoidActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/EvSailSigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">EvSailSigmoidActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/GaussianActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/GaussianActivationFunction.html" title="class in com.anji.nn.activationfunction">GaussianActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/InverseAbsActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/InverseAbsActivationFunction.html" title="class in com.anji.nn.activationfunction">InverseAbsActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/LinearActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/LinearActivationFunction.html" title="class in com.anji.nn.activationfunction">LinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/LogicActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/LogicActivationFunction.html" title="class in com.anji.nn.activationfunction">LogicActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/MultiplyActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/MultiplyActivationFunction.html" title="class in com.anji.nn.activationfunction">MultiplyActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/NegatedLinearActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/NegatedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">NegatedLinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/RampActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/RampActivationFunction.html" title="class in com.anji.nn.activationfunction">RampActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/RecipriocalActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/RecipriocalActivationFunction.html" title="class in com.anji.nn.activationfunction">RecipriocalActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SigmoidActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SigmoidActivationFunction.html" title="class in com.anji.nn.activationfunction">SigmoidActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SignedClampedLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">SignedClampedLinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SignedStepActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SignedStepActivationFunction.html" title="class in com.anji.nn.activationfunction">SignedStepActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SineActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SineActivationFunction.html" title="class in com.anji.nn.activationfunction">SineActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SqrtActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SqrtActivationFunction.html" title="class in com.anji.nn.activationfunction">SqrtActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/SqrtAndLinearActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/SqrtAndLinearActivationFunction.html" title="class in com.anji.nn.activationfunction">SqrtAndLinearActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/StepActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/StepActivationFunction.html" title="class in com.anji.nn.activationfunction">StepActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/TanhActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/TanhActivationFunction.html" title="class in com.anji.nn.activationfunction">TanhActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/activationfunction/TanhCubicActivationFunction.html#cost()">cost()</a></span> - Method in class com.anji.nn.activationfunction.<a href="../com/anji/nn/activationfunction/TanhCubicActivationFunction.html" title="class in com.anji.nn.activationfunction">TanhCubicActivationFunction</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/AnjiNet.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/AnjiNet.html" title="class in com.anji.nn">AnjiNet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/BiasConnection.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/BiasConnection.html" title="class in com.anji.nn">BiasConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/CacheNeuronConnection.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/CacheNeuronConnection.html" title="class in com.anji.nn">CacheNeuronConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/Connection.html#cost()">cost()</a></span> - Method in interface com.anji.nn.<a href="../com/anji/nn/Connection.html" title="interface in com.anji.nn">Connection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/Neuron.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/Neuron.html" title="class in com.anji.nn">Neuron</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/NeuronConnection.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/NeuronConnection.html" title="class in com.anji.nn">NeuronConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/RandomConnection.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/RandomConnection.html" title="class in com.anji.nn">RandomConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/nn/StepHourglassConnection.html#cost()">cost()</a></span> - Method in class com.anji.nn.<a href="../com/anji/nn/StepHourglassConnection.html" title="class in com.anji.nn">StepHourglassConnection</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/GridNet.html#cost()">cost()</a></span> - Method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/GridNet.html" title="class in com.ojcoleman.ahni.nn">GridNet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#COUNT_TAG">COUNT_TAG</a></span> - Static variable in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">XML count tag</div>
</dd>
<dt><span class="strong"><a href="../com/anji/integration/ErrorRateCounter.html#countErrors(double[][], double[][])">countErrors(double[][], double[][])</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/ErrorRateCounter.html" title="class in com.anji.integration">ErrorRateCounter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/integration/ErrorRateCounter.html#countErrors(java.lang.String, double[][], double[][])">countErrors(String, double[][], double[][])</a></span> - Method in class com.anji.integration.<a href="../com/anji/integration/ErrorRateCounter.html" title="class in com.anji.integration">ErrorRateCounter</a></dt>
<dd>
<div class="block">Calculate the sum of differences between <code>responses</code> and target values.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnActivator">cppnActivator</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnInput">cppnInput</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cppnInputCount">cppnInputCount</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">The number of inputs to the CPPN.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnMax">cppnMax</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnMin">cppnMin</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnOutput">cppnOutput</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cppnOutputCount">cppnOutputCount</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">The number of outputs from the CPPN.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnOutputPlusMinusUnitBounded">cppnOutputPlusMinusUnitBounded</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnOutputUnitBounded">cppnOutputUnitBounded</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html#cppnRange">cppnRange</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.CPPN.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber.CPPN</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html" title="class in com.ojcoleman.ahni.evaluation.mocostfunctions"><span class="strong">CPPNSizeCost</span></a> - Class in <a href="../com/ojcoleman/ahni/evaluation/mocostfunctions/package-summary.html">com.ojcoleman.ahni.evaluation.mocostfunctions</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html#CPPNSizeCost()">CPPNSizeCost()</a></span> - Constructor for class com.ojcoleman.ahni.evaluation.mocostfunctions.<a href="../com/ojcoleman/ahni/evaluation/mocostfunctions/CPPNSizeCost.html" title="class in com.ojcoleman.ahni.evaluation.mocostfunctions">CPPNSizeCost</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cppnTranscriber">cppnTranscriber</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">Transcriber for network for use as a CPPN.</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html#cppnValue">cppnValue</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/ESHyperNEATTranscriberBain.QuadPoint.html" title="class in com.ojcoleman.ahni.transcriber">ESHyperNEATTranscriberBain.QuadPoint</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#createNeuronCollection(java.lang.String, int, boolean, boolean, boolean)">createNeuronCollection(String, int, boolean, boolean, boolean)</a></span> - Static method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/nn/BainNN.html#createSynapseCollection(java.lang.String, int, boolean, boolean, double, double)">createSynapseCollection(String, int, boolean, boolean, double, double)</a></span> - Static method in class com.ojcoleman.ahni.nn.<a href="../com/ojcoleman/ahni/nn/BainNN.html" title="class in com.ojcoleman.ahni.nn">BainNN</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#CROSSOVER_MUTATE_RATE_KEY">CROSSOVER_MUTATE_RATE_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, the probability that an individual produced by the crossover operator will be a candidate for having mutations applied to it (independent of other mutation probabilities).</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#CROSSOVER_PROPORTION_KEY">CROSSOVER_PROPORTION_KEY</a></span> - Static variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>
<div class="block">properties key, proportion of new population that is generated via crossover from two parents (as opposed to mutation of a single parent), [0, 1].</div>
</dd>
<dt><span class="strong"><a href="../com/anji/neat/NeatConfiguration.html#crossoverOper">crossoverOper</a></span> - Variable in class com.anji.neat.<a href="../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a></dt>
<dd>&nbsp;</dd>
<dt><a href="../org/jgapcustomised/CrossoverReproductionOperator.html" title="class in org.jgapcustomised"><span class="strong">CrossoverReproductionOperator</span></a> - Class in <a href="../org/jgapcustomised/package-summary.html">org.jgapcustomised</a></dt>
<dd>
<div class="block">Abstract crossover reporduction operator handles iteration over population to determine pairs of parent chromosomes.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/CrossoverReproductionOperator.html#CrossoverReproductionOperator()">CrossoverReproductionOperator()</a></span> - Constructor for class org.jgapcustomised.<a href="../org/jgapcustomised/CrossoverReproductionOperator.html" title="class in org.jgapcustomised">CrossoverReproductionOperator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Chromosome.html#crowdingDistance">crowdingDistance</a></span> - Variable in class org.jgapcustomised.<a href="../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a></dt>
<dd>
<div class="block">Used by selection algorithms to store a value representing how crowded the fitness space is relative to other
 individuals with the same rank (on the same Pareto front), most useful for multi-objective selection algorithms
 such as NSGA-II.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#cull(java.util.Collection)">cull(Collection&lt;Chromosome&gt;)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">Remove all chromosomes from this specie except <code>keepers</code> NOTE: this method does not update the species
 field in the removed Chromosomes.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#cullClones()">cullClones()</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">Remove the clones from this species, as determined by <a href="../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)"><code>Chromosome.isEquivalent(Chromosome)</code></a>.</div>
</dd>
<dt><span class="strong"><a href="../org/jgapcustomised/Species.html#cullToElites(org.jgapcustomised.Chromosome)">cullToElites(Chromosome)</a></span> - Method in class org.jgapcustomised.<a href="../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></dt>
<dd>
<div class="block">remove all non-elite chromosomes from this species, except for population-wide bestPerforming</div>
</dd>
<dt><span class="strong"><a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html#cyclesPerStep">cyclesPerStep</a></span> - Variable in class com.ojcoleman.ahni.transcriber.<a href="../com/ojcoleman/ahni/transcriber/HyperNEATTranscriber.html" title="class in com.ojcoleman.ahni.transcriber">HyperNEATTranscriber</a></dt>
<dd>
<div class="block">For substrate networks with recurrent connections, the number of activation cycles to perform each time the
 substrate network is presented with new input and queried for its output.</div>
</dd>
</dl>
<a href="index-1.html">A</a>&nbsp;<a href="index-2.html">B</a>&nbsp;<a href="index-3.html">C</a>&nbsp;<a href="index-4.html">D</a>&nbsp;<a href="index-5.html">E</a>&nbsp;<a href="index-6.html">F</a>&nbsp;<a href="index-7.html">G</a>&nbsp;<a href="index-8.html">H</a>&nbsp;<a href="index-9.html">I</a>&nbsp;<a href="index-10.html">K</a>&nbsp;<a href="index-11.html">L</a>&nbsp;<a href="index-12.html">M</a>&nbsp;<a href="index-13.html">N</a>&nbsp;<a href="index-14.html">O</a>&nbsp;<a href="index-15.html">P</a>&nbsp;<a href="index-16.html">Q</a>&nbsp;<a href="index-17.html">R</a>&nbsp;<a href="index-18.html">S</a>&nbsp;<a href="index-19.html">T</a>&nbsp;<a href="index-20.html">U</a>&nbsp;<a href="index-21.html">V</a>&nbsp;<a href="index-22.html">W</a>&nbsp;<a href="index-23.html">X</a>&nbsp;<a href="index-24.html">Y</a>&nbsp;<a href="index-25.html">Z</a>&nbsp;</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="../overview-tree.html">Tree</a></li>
<li><a href="../deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="index-2.html">PREV LETTER</a></li>
<li><a href="index-4.html">NEXT LETTER</a></li>
</ul>
<ul class="navList">
<li><a href="../index.html?index-filesindex-3.html" target="_top">FRAMES</a></li>
<li><a href="index-3.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
