<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Jul 05 16:56:46 EST 2013 -->
<TITLE>
ChromosomeMaterial
</TITLE>

<META NAME="date" CONTENT="2013-07-05">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ChromosomeMaterial";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ChromosomeMaterial.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgapcustomised/ChromosomePerformanceComparator.html" title="class in org.jgapcustomised"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html?org/jgapcustomised/ChromosomeMaterial.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ChromosomeMaterial.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.jgapcustomised</FONT>
<BR>
Class ChromosomeMaterial</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../resources/inherit.gif" ALT="extended by "><B>org.jgapcustomised.ChromosomeMaterial</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.io.Serializable, java.lang.Comparable</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>ChromosomeMaterial</B><DT>extends java.lang.Object<DT>implements java.lang.Comparable, java.io.Serializable</DL>
</PRE>

<P>
This is the guts of the original Chromosome object, pulled out so the genes can be modified by genetic operators
 before creating the Chromosome object. Also enables us to handle special cases, like sample chromosome, where you
 don't need a Configuration or fitness value. Also, made methods not synchronized, since only Genotype.evolve() should
 be modifying this object.
<P>

<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../serialized-form.html#org.jgapcustomised.ChromosomeMaterial">Serialized Form</A></DL>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection)">ChromosomeMaterial</A></B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialAlleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create chromosome with no parents.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long)">ChromosomeMaterial</A></B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialGenes,
                   java.lang.Long&nbsp;aPrimaryParentId)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create chromosome with one parents.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)">ChromosomeMaterial</A></B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialAlleles,
                   java.lang.Long&nbsp;aPrimaryParentId,
                   java.lang.Long&nbsp;aSecondaryParentId)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create chromosome with two parents.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#clone(java.lang.Long)">clone</A></B>(java.lang.Long&nbsp;parentId)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a copy of this ChromosomeMaterial.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#compareTo(java.lang.Object)">compareTo</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares the given Chromosome to this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#distance(org.jgapcustomised.ChromosomeMaterial, org.jgapcustomised.SpeciationParms)">distance</A></B>(<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
         <A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>&nbsp;speciationParms)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Chromosome against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.SortedSet&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#getAlleles()">getAlleles</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the set of genes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#getMaxInnovationID()">getMaxInnovationID</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#getMinInnovationID()">getMinInnovationID</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#getPrimaryParentId()">getPrimaryParentId</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#getSecondaryParentId()">getSecondaryParentId</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#isEquivalent(org.jgapcustomised.ChromosomeMaterial)">isEquivalent</A></B>(<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Chromosome against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#randomInitialChromosomeMaterial(org.jgapcustomised.Configuration)">randomInitialChromosomeMaterial</A></B>(<A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convenience method that returns a new Chromosome instance with its genes values (alleles) randomized.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#setAlleles(java.util.SortedSet)">setAlleles</A></B>(java.util.SortedSet&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;aAlleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;for hibernate</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#setSecondaryParentId(java.lang.Long)">setSecondaryParentId</A></B>(java.lang.Long&nbsp;id)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#setShouldMutate(boolean)">setShouldMutate</A></B>(boolean&nbsp;mutate)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;May be used by reproduction operators to indicate that this individual is a candidate for mutations.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#shouldMutate()">shouldMutate</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Indicates whether this individual is a candidate for mutations (after reproduction).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a string representation of this Chromosome, useful for some display purposes.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)"><!-- --></A><H3>
ChromosomeMaterial</H3>
<PRE>
public <B>ChromosomeMaterial</B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialAlleles,
                          java.lang.Long&nbsp;aPrimaryParentId,
                          java.lang.Long&nbsp;aSecondaryParentId)</PRE>
<DL>
<DD>Create chromosome with two parents. Used for crossover.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>a_initialAlleles</CODE> - <DD><CODE>aPrimaryParentId</CODE> - <DD><CODE>aSecondaryParentId</CODE> - </DL>
</DL>
<HR>

<A NAME="ChromosomeMaterial(java.util.Collection, java.lang.Long)"><!-- --></A><H3>
ChromosomeMaterial</H3>
<PRE>
public <B>ChromosomeMaterial</B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialGenes,
                          java.lang.Long&nbsp;aPrimaryParentId)</PRE>
<DL>
<DD>Create chromosome with one parents. Used for cloning.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>a_initialGenes</CODE> - <DD><CODE>aPrimaryParentId</CODE> - <DT><B>See Also:</B><DD><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)"><CODE>ChromosomeMaterial(Collection, Long, Long)</CODE></A></DL>
</DL>
<HR>

<A NAME="ChromosomeMaterial(java.util.Collection)"><!-- --></A><H3>
ChromosomeMaterial</H3>
<PRE>
public <B>ChromosomeMaterial</B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;a_initialAlleles)</PRE>
<DL>
<DD>Create chromosome with no parents. Used for startup sample chromosome material.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>a_initialAlleles</CODE> - <DT><B>See Also:</B><DD><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#ChromosomeMaterial(java.util.Collection, java.lang.Long, java.lang.Long)"><CODE>ChromosomeMaterial(Collection, Long, Long)</CODE></A></DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="clone(java.lang.Long)"><!-- --></A><H3>
clone</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A> <B>clone</B>(java.lang.Long&nbsp;parentId)</PRE>
<DL>
<DD>Returns a copy of this ChromosomeMaterial. The returned instance can evolve independently of this instance.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>parentId</CODE> - represents ID of chromosome that was cloned. If this is initial chromosome material without a
            parent, or is the clone of material only (e.g., before the it has become a Chromosome), parentId ==
            null.
<DT><B>Returns:</B><DD>copy of this object</DL>
</DD>
</DL>
<HR>

<A NAME="getAlleles()"><!-- --></A><H3>
getAlleles</H3>
<PRE>
public java.util.SortedSet&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt; <B>getAlleles</B>()</PRE>
<DL>
<DD>Retrieves the set of genes. This method exists primarily for the benefit of GeneticOperators that require the
 ability to manipulate Chromosomes at a low level.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>an array of the Genes contained within this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD>Returns a string representation of this Chromosome, useful for some display purposes.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A string representation of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="randomInitialChromosomeMaterial(org.jgapcustomised.Configuration)"><!-- --></A><H3>
randomInitialChromosomeMaterial</H3>
<PRE>
public static <A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A> <B>randomInitialChromosomeMaterial</B>(<A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration)
                                                          throws <A HREF="../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</A></PRE>
<DL>
<DD>Convenience method that returns a new Chromosome instance with its genes values (alleles) randomized. Note that,
 if possible, this method will acquire a Chromosome instance from the active ChromosomePool (if any) and then
 randomize its gene values before returning it. If a Chromosome cannot be acquired from the pool, then a new
 instance will be constructed and its gene values randomized before returning it.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current active configuration.
<DT><B>Returns:</B><DD>new <code>ChromosomeMaterial</code>
<DT><B>Throws:</B>
<DD><CODE><A HREF="../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</A></CODE> - if the given Configuration instance is invalid.
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if the given Configuration instance is null.</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has a set of genes equal to this one.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="isEquivalent(org.jgapcustomised.ChromosomeMaterial)"><!-- --></A><H3>
isEquivalent</H3>
<PRE>
public boolean <B>isEquivalent</B>(<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;other)</PRE>
<DL>
<DD>Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has a set of genes with equal values (eg connection weight, activation type) to this one.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="compareTo(java.lang.Object)"><!-- --></A><H3>
compareTo</H3>
<PRE>
public int <B>compareTo</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares the given Chromosome to this Chromosome. This chromosome is considered to be "less than" the given
 chromosome if it has a fewer number of genes or if any of its gene values (alleles) are less than their
 corresponding gene values in the other chromosome.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>compareTo</CODE> in interface <CODE>java.lang.Comparable</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The Chromosome against which to compare this chromosome.
<DT><B>Returns:</B><DD>a negative number if this chromosome is "less than" the given chromosome, zero if they are equal to each
         other, and a positive number if this chromosome is "greater than" the given chromosome.<DT><B>See Also:</B><DD><CODE>Comparable.compareTo(java.lang.Object)</CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getPrimaryParentId()"><!-- --></A><H3>
getPrimaryParentId</H3>
<PRE>
public java.lang.Long <B>getPrimaryParentId</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>primary parent ID; dominant parent if chromosome spawned by crossover</DL>
</DD>
</DL>
<HR>

<A NAME="getSecondaryParentId()"><!-- --></A><H3>
getSecondaryParentId</H3>
<PRE>
public java.lang.Long <B>getSecondaryParentId</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>primary parent ID; recessive parent if chromosome spawned by crossover</DL>
</DD>
</DL>
<HR>

<A NAME="setSecondaryParentId(java.lang.Long)"><!-- --></A><H3>
setSecondaryParentId</H3>
<PRE>
public void <B>setSecondaryParentId</B>(java.lang.Long&nbsp;id)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>id</CODE> - ID of recessive parent</DL>
</DD>
</DL>
<HR>

<A NAME="getMinInnovationID()"><!-- --></A><H3>
getMinInnovationID</H3>
<PRE>
public long <B>getMinInnovationID</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getMaxInnovationID()"><!-- --></A><H3>
getMaxInnovationID</H3>
<PRE>
public long <B>getMaxInnovationID</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="distance(org.jgapcustomised.ChromosomeMaterial, org.jgapcustomised.SpeciationParms)"><!-- --></A><H3>
distance</H3>
<PRE>
public double <B>distance</B>(<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;target,
                       <A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>&nbsp;speciationParms)</PRE>
<DL>
<DD>Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology. Made it
 generic enough that the genes do not have to be nodes and connections.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>target</CODE> - <DD><CODE>speciationParms</CODE> - 
<DT><B>Returns:</B><DD>distance between this object and <code>target</code><DT><B>See Also:</B><DD><A HREF="../../org/jgapcustomised/Allele.html#distance(org.jgapcustomised.Allele)"><CODE>Allele.distance(Allele)</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="setAlleles(java.util.SortedSet)"><!-- --></A><H3>
setAlleles</H3>
<PRE>
public void <B>setAlleles</B>(java.util.SortedSet&lt;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;aAlleles)</PRE>
<DL>
<DD>for hibernate
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>aAlleles</CODE> - </DL>
</DD>
</DL>
<HR>

<A NAME="shouldMutate()"><!-- --></A><H3>
shouldMutate</H3>
<PRE>
public boolean <B>shouldMutate</B>()</PRE>
<DL>
<DD>Indicates whether this individual is a candidate for mutations (after reproduction). Default value is true.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the mutateProbability</DL>
</DD>
</DL>
<HR>

<A NAME="setShouldMutate(boolean)"><!-- --></A><H3>
setShouldMutate</H3>
<PRE>
public void <B>setShouldMutate</B>(boolean&nbsp;mutate)</PRE>
<DL>
<DD>May be used by reproduction operators to indicate that this individual is a candidate for mutations.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/ChromosomeMaterial.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgapcustomised/ChromosomePerformanceComparator.html" title="class in org.jgapcustomised"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html?org/jgapcustomised/ChromosomeMaterial.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="ChromosomeMaterial.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
