<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.6.0_27) on Fri Oct 11 09:14:24 EST 2013 -->
<title>Chromosome</title>
<meta name="date" content="2013-10-11">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Chromosome";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromosome.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-files/index-1.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../org/jgapcustomised/BulkFitnessFunction.html" title="class in org.jgapcustomised"><span class="strong">PREV CLASS</span></a></li>
<li><a href="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><span class="strong">NEXT CLASS</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?org/jgapcustomised/Chromosome.html" target="_top">FRAMES</a></li>
<li><a href="Chromosome.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>SUMMARY:&nbsp;</li>
<li>NESTED&nbsp;|&nbsp;</li>
<li><a href="#field_summary">FIELD</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">METHOD</a></li>
</ul>
<ul class="subNavList">
<li>DETAIL:&nbsp;</li>
<li><a href="#field_detail">FIELD</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">METHOD</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<p class="subTitle">org.jgapcustomised</p>
<h2 title="Class Chromosome" class="title">Class Chromosome</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>org.jgapcustomised.Chromosome</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd>java.io.Serializable, java.lang.Comparable</dd>
</dl>
<hr>
<br>
<pre>public class <strong>Chromosome</strong>
extends java.lang.Object
implements java.lang.Comparable, java.io.Serializable</pre>
<div class="block">Chromosomes represent potential solutions and consist of a fixed-length collection of genes. Each gene represents a
 discrete part of the solution. Each gene in the Chromosome may be backed by a different concrete implementation of
 the Gene interface, but all genes with the same innovation ID must share the same concrete implementation across
 Chromosomes within a single population (genotype).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../serialized-form.html#org.jgapcustomised.Chromosome">Serialized Form</a></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#behaviours">behaviours</a></strong></code>
<div class="block">May be used by implementations of novelty search.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#crowdingDistance">crowdingDistance</a></strong></code>
<div class="block">Used by selection algorithms to store a value representing how crowded the fitness space is relative to other
 individuals with the same rank (on the same Pareto front), most useful for multi-objective selection algorithms
 such as NSGA-II.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#DEFAULT_ID">DEFAULT_ID</a></strong></code>
<div class="block">default ID</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#evaluationDataStable">evaluationDataStable</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#isElite">isElite</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double[]</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_fitnessValue">m_fitnessValue</a></strong></code>
<div class="block">Stores the fitness value(s) of this Chromosome as determined by the active fitness function(s).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_isSelectedForNextGeneration">m_isSelectedForNextGeneration</a></strong></code>
<div class="block">Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_overallFitnessValue">m_overallFitnessValue</a></strong></code>
<div class="block">Stores the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <a href="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><code>NaturalSelector</code></a> based on the fitness values over all objectives.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_performanceValue">m_performanceValue</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_specie">m_specie</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#novelty">novelty</a></strong></code>
<div class="block">May be used by implementations of novelty search.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#rank">rank</a></strong></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long, int, int)">Chromosome</a></strong>(<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;a_material,
          java.lang.Long&nbsp;an_id,
          int&nbsp;objectiveCount,
          int&nbsp;behaviourCount)</code>
<div class="block">this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#cloneMaterial()">cloneMaterial</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#compareTo(java.lang.Object)">compareTo</a></strong>(java.lang.Object&nbsp;o)</code>
<div class="block">Compares the given Chromosome to this Chromosome by their ID.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">distance</a></strong>(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;target,
        <a href="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;parms)</code>
<div class="block">Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#dominates(org.jgapcustomised.Chromosome)">dominates</a></strong>(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;otherChromosome)</code>
<div class="block">Checks whether this individual dominates the specified other individual, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#equals(java.lang.Object)">equals</a></strong>(java.lang.Object&nbsp;other)</code>
<div class="block">Compares this Chromosome against the specified object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</a></strong>(<a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;alleleToMatch)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.SortedSet</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getAlleles()">getAlleles</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getFitnessValue()">getFitnessValue</a></strong>()</code>
<div class="block">Returns the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <a href="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><code>NaturalSelector</code></a> based on the fitness values over all objectives.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getFitnessValue(int)">getFitnessValue</a></strong>(int&nbsp;objective)</code>
<div class="block">Returns the fitness value of this Chromosome for the given objective, as determined by the active fitness
 function.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getFitnessValues()">getFitnessValues</a></strong>()</code>
<div class="block">Returns a reference to the array of fitness values for this Chromosome for each objective, as determined by the
 active fitness function.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getId()">getId</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getMaterial()">getMaterial</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getNoveltyObjectiveCount()">getNoveltyObjectiveCount</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getObjectiveCount()">getObjectiveCount</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getPerformanceValue()">getPerformanceValue</a></strong>()</code>
<div class="block">Returns the performance value of this Chromosome as determined by the active fitness function.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getPrimaryParentId()">getPrimaryParentId</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getSecondaryParentId()">getSecondaryParentId</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getSpeciatedFitnessValue()">getSpeciatedFitnessValue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getSpecie()">getSpecie</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#hashCode()">hashCode</a></strong>()</code>
<div class="block">Retrieve a hash code for this Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)">isEquivalent</a></strong>(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c2)</code>
<div class="block">Compares this Chromosome against the specified object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#isEvaluationDataStable()">isEvaluationDataStable</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#isSelectedForNextGeneration()">isSelectedForNextGeneration</a></strong>()</code>
<div class="block">Retrieves whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#resetEvaluationData()">resetEvaluationData</a></strong>()</code>
<div class="block">Resets the performance, fitness value(s) and recorded novelty behaviours(s) of this Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#resetFitnessValues()">resetFitnessValues</a></strong>()</code>
<div class="block">Resets the fitness value(s) of this Chromosome to Double.NaN.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#resetPerformanceValue()">resetPerformanceValue</a></strong>()</code>
<div class="block">Resets the performance value of this Chromosome to Double.NaN.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#resetSpecie()">resetSpecie</a></strong>()</code>
<div class="block">Resets (clears) the specie this chromose belongs to.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setEvaluationDataStable()">setEvaluationDataStable</a></strong>()</code>
<div class="block">Indicate that the evaluation data (e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setFitnessValue(double)">setFitnessValue</a></strong>(double&nbsp;a_newFitnessValue)</code>
<div class="block">Sets the overall fitness value of this Chromosome.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setFitnessValue(double, int)">setFitnessValue</a></strong>(double&nbsp;a_newFitnessValue,
               int&nbsp;objective)</code>
<div class="block">Sets the fitness value of this Chromosome for the given objective.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setIsSelectedForNextGeneration(boolean)">setIsSelectedForNextGeneration</a></strong>(boolean&nbsp;a_isSelected)</code>
<div class="block">Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setPerformanceValue(double)">setPerformanceValue</a></strong>(double&nbsp;aPerformanceValue)</code>
<div class="block">Sets the performance value of this Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setSpecie(org.jgapcustomised.Species)">setSpecie</a></strong>(<a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&nbsp;aSpecie)</code>
<div class="block">should only be called from Species; assigns this chromosome to <code>aSpecie</code>; throws exception if
 chromosome is added to a specie twice</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#size()">size</a></strong>()</code>
<div class="block">Returns the size of this Chromosome (the number of alleles it contains).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#toString()">toString</a></strong>()</code>
<div class="block">Returns a string representation of this Chromosome, useful for some display purposes.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="DEFAULT_ID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_ID</h4>
<pre>public static final&nbsp;java.lang.Long DEFAULT_ID</pre>
<div class="block">default ID</div>
</li>
</ul>
<a name="m_isSelectedForNextGeneration">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_isSelectedForNextGeneration</h4>
<pre>protected&nbsp;boolean m_isSelectedForNextGeneration</pre>
<div class="block">Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.</div>
</li>
</ul>
<a name="m_fitnessValue">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_fitnessValue</h4>
<pre>protected&nbsp;double[] m_fitnessValue</pre>
<div class="block">Stores the fitness value(s) of this Chromosome as determined by the active fitness function(s). A value of
 Double.NaN indicates that a fitness value has not yet been set.</div>
</li>
</ul>
<a name="m_overallFitnessValue">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_overallFitnessValue</h4>
<pre>protected&nbsp;double m_overallFitnessValue</pre>
<div class="block">Stores the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <a href="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><code>NaturalSelector</code></a> based on the fitness values over all objectives. A
 value of Double.NaN indicates that a fitness value has not yet been set.</div>
</li>
</ul>
<a name="m_performanceValue">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_performanceValue</h4>
<pre>protected&nbsp;double m_performanceValue</pre>
</li>
</ul>
<a name="evaluationDataStable">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>evaluationDataStable</h4>
<pre>protected&nbsp;boolean evaluationDataStable</pre>
</li>
</ul>
<a name="m_specie">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_specie</h4>
<pre>protected&nbsp;<a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a> m_specie</pre>
</li>
</ul>
<a name="isElite">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isElite</h4>
<pre>public&nbsp;boolean isElite</pre>
</li>
</ul>
<a name="rank">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>rank</h4>
<pre>public&nbsp;int rank</pre>
</li>
</ul>
<a name="novelty">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>novelty</h4>
<pre>public&nbsp;double novelty</pre>
<div class="block">May be used by implementations of novelty search.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../com/ojcoleman/ahni/evaluation/novelty/NoveltySearch.html" title="class in com.ojcoleman.ahni.evaluation.novelty"><code>NoveltySearch</code></a></dd></dl>
</li>
</ul>
<a name="behaviours">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>behaviours</h4>
<pre>public&nbsp;<a href="../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</a>[] behaviours</pre>
<div class="block">May be used by implementations of novelty search. An array is used so that multiple behaviours may be defined.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../com/ojcoleman/ahni/evaluation/novelty/NoveltySearch.html" title="class in com.ojcoleman.ahni.evaluation.novelty"><code>NoveltySearch</code></a></dd></dl>
</li>
</ul>
<a name="crowdingDistance">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>crowdingDistance</h4>
<pre>public&nbsp;double crowdingDistance</pre>
<div class="block">Used by selection algorithms to store a value representing how crowded the fitness space is relative to other
 individuals with the same rank (on the same Pareto front), most useful for multi-objective selection algorithms
 such as NSGA-II.</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long, int, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Chromosome</h4>
<pre>public&nbsp;Chromosome(<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;a_material,
          java.lang.Long&nbsp;an_id,
          int&nbsp;objectiveCount,
          int&nbsp;behaviourCount)</pre>
<div class="block">this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_material</code> - Genetic material to be contained within this Chromosome instance.</dd><dd><code>an_id</code> - unique ID of new chromosome</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>distance</h4>
<pre>public&nbsp;double&nbsp;distance(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;target,
              <a href="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;parms)</pre>
<div class="block">Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology. It is generic
 enough that the alleles do not have to be nodes and connections.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>target</code> - </dd><dd><code>parms</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>distance between this object and <code>target</code></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../org/jgapcustomised/ChromosomeMaterial.html#distance(org.jgapcustomised.ChromosomeMaterial, org.jgapcustomised.SpeciationParms)"><code>ChromosomeMaterial.distance(ChromosomeMaterial, SpeciationParms)</code></a></dd></dl>
</li>
</ul>
<a name="getId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getId</h4>
<pre>public&nbsp;java.lang.Long&nbsp;getId()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>Long unique identifier for chromosome; useful for <code>hashCode()</code> and persistence</dd></dl>
</li>
</ul>
<a name="size()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>size</h4>
<pre>public&nbsp;int&nbsp;size()</pre>
<div class="block">Returns the size of this Chromosome (the number of alleles it contains). A Chromosome's size is constant and will
 never change.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The number of alleles contained within this Chromosome instance.</dd></dl>
</li>
</ul>
<a name="cloneMaterial()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>cloneMaterial</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;cloneMaterial()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>clone with primary parent ID of this chromosome and the same genetic material.</dd></dl>
</li>
</ul>
<a name="getAlleles()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlleles</h4>
<pre>public&nbsp;java.util.SortedSet&nbsp;getAlleles()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>SortedSet alleles, sorted by innovation ID</dd></dl>
</li>
</ul>
<a name="findMatchingGene(org.jgapcustomised.Allele)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findMatchingGene</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;findMatchingGene(<a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;alleleToMatch)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleleToMatch</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>Gene gene with same innovation ID as <code>geneToMatch</code, or <code>null</code> if none match</dd></dl>
</li>
</ul>
<a name="getFitnessValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFitnessValue</h4>
<pre>public&nbsp;double&nbsp;getFitnessValue()</pre>
<div class="block">Returns the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <a href="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><code>NaturalSelector</code></a> based on the fitness values over all objectives.
 Fitness values are in the range [0, 1], however the value Double.NaN indicates that a fitness hasn't been set
 yet.</div>
</li>
</ul>
<a name="getFitnessValue(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFitnessValue</h4>
<pre>public&nbsp;double&nbsp;getFitnessValue(int&nbsp;objective)</pre>
<div class="block">Returns the fitness value of this Chromosome for the given objective, as determined by the active fitness
 function. Fitness values are in the range [0, 1], however the value Double.NaN indicates that a fitness hasn't
 been set yet.</div>
</li>
</ul>
<a name="getFitnessValues()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFitnessValues</h4>
<pre>public&nbsp;double[]&nbsp;getFitnessValues()</pre>
<div class="block">Returns a reference to the array of fitness values for this Chromosome for each objective, as determined by the
 active fitness function. Fitness values are in the range [0, 1], however the value Double.NaN indicates that a
 fitness hasn't been set yet.</div>
</li>
</ul>
<a name="getSpeciatedFitnessValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSpeciatedFitnessValue</h4>
<pre>public&nbsp;double&nbsp;getSpeciatedFitnessValue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>int fitness value adjusted for fitness sharing according to <a
         href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf>NEAT </a> paradigm.</dd></dl>
</li>
</ul>
<a name="setFitnessValue(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFitnessValue</h4>
<pre>public&nbsp;void&nbsp;setFitnessValue(double&nbsp;a_newFitnessValue)</pre>
<div class="block">Sets the overall fitness value of this Chromosome. This method is for use by bulk fitness functions and should
 not be invoked from anything else. This is the raw fitness value, before species fitness sharing. Fitness values
 must be in the range [0, 1].</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_newFitnessValue</code> - the fitness of this Chromosome.</dd></dl>
</li>
</ul>
<a name="setFitnessValue(double, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFitnessValue</h4>
<pre>public&nbsp;void&nbsp;setFitnessValue(double&nbsp;a_newFitnessValue,
                   int&nbsp;objective)</pre>
<div class="block">Sets the fitness value of this Chromosome for the given objective. This method is for use by bulk fitness
 functions and should not be invoked from anything else. Fitness values must be in the range [0, 1].</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_newFitnessValue</code> - the fitness of this Chromosome.</dd></dl>
</li>
</ul>
<a name="getPerformanceValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPerformanceValue</h4>
<pre>public&nbsp;double&nbsp;getPerformanceValue()</pre>
<div class="block">Returns the performance value of this Chromosome as determined by the active fitness function. Performance values
 are in the range [0, 1], however the value Double.NaN indicates that a fitness hasn't been set yet.</div>
</li>
</ul>
<a name="setPerformanceValue(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setPerformanceValue</h4>
<pre>public&nbsp;void&nbsp;setPerformanceValue(double&nbsp;aPerformanceValue)</pre>
<div class="block">Sets the performance value of this Chromosome. This method is for use by bulk fitness functions and should not be
 invoked from anything else. Performance values must be in the range [0, 1].</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>aPerformanceValue</code> - the fitness of this Chromosome.</dd></dl>
</li>
</ul>
<a name="resetPerformanceValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>resetPerformanceValue</h4>
<pre>public&nbsp;void&nbsp;resetPerformanceValue()</pre>
<div class="block">Resets the performance value of this Chromosome to Double.NaN.</div>
</li>
</ul>
<a name="resetFitnessValues()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>resetFitnessValues</h4>
<pre>public&nbsp;void&nbsp;resetFitnessValues()</pre>
<div class="block">Resets the fitness value(s) of this Chromosome to Double.NaN.</div>
</li>
</ul>
<a name="resetEvaluationData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>resetEvaluationData</h4>
<pre>public&nbsp;void&nbsp;resetEvaluationData()</pre>
<div class="block">Resets the performance, fitness value(s) and recorded novelty behaviours(s) of this Chromosome.</div>
</li>
</ul>
<a name="setEvaluationDataStable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setEvaluationDataStable</h4>
<pre>public&nbsp;void&nbsp;setEvaluationDataStable()</pre>
<div class="block">Indicate that the evaluation data (e.g. fitness, performance, behaviours) is stable: it won't change in future evaluations.
 This only has an effect if the fitness function pays attention to this value, and the effect that it has may vary between implementations.</div>
</li>
</ul>
<a name="isEvaluationDataStable()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEvaluationDataStable</h4>
<pre>public&nbsp;boolean&nbsp;isEvaluationDataStable()</pre>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;java.lang.String&nbsp;toString()</pre>
<div class="block">Returns a string representation of this Chromosome, useful for some display purposes.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
<dt><span class="strong">Returns:</span></dt><dd>A string representation of this Chromosome.</dd></dl>
</li>
</ul>
<a name="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public&nbsp;boolean&nbsp;equals(java.lang.Object&nbsp;other)</pre>
<div class="block">Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has the same ID.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code>equals</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>other</code> - The object to compare against.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the objects are the same, false otherwise.</dd></dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<div class="block">Retrieve a hash code for this Chromosome.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code>hashCode</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the hash code of this Chromosome.</dd></dl>
</li>
</ul>
<a name="compareTo(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compareTo</h4>
<pre>public&nbsp;int&nbsp;compareTo(java.lang.Object&nbsp;o)</pre>
<div class="block">Compares the given Chromosome to this Chromosome by their ID.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code>compareTo</code>&nbsp;in interface&nbsp;<code>java.lang.Comparable</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>o</code> - The Chromosome against which to compare this chromosome.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a negative number if this chromosome is "less than" the given chromosome, zero if they are equal to each
         other, and a positive number if this chromosome is "greater than" the given chromosome.</dd></dl>
</li>
</ul>
<a name="isEquivalent(org.jgapcustomised.Chromosome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEquivalent</h4>
<pre>public&nbsp;boolean&nbsp;isEquivalent(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;c2)</pre>
<div class="block">Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has a set of genes with equal values (eg connection weight, activation type) to this
 one.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>c2</code> - The object to compare against.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the objects are the same, false otherwise.</dd></dl>
</li>
</ul>
<a name="setIsSelectedForNextGeneration(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setIsSelectedForNextGeneration</h4>
<pre>public&nbsp;void&nbsp;setIsSelectedForNextGeneration(boolean&nbsp;a_isSelected)</pre>
<div class="block">Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_isSelected</code> - true if this Chromosome has been selected, false otherwise.</dd></dl>
</li>
</ul>
<a name="isSelectedForNextGeneration()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSelectedForNextGeneration</h4>
<pre>public&nbsp;boolean&nbsp;isSelectedForNextGeneration()</pre>
<div class="block">Retrieves whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this Chromosome has been selected, false otherwise.</dd></dl>
</li>
</ul>
<a name="setSpecie(org.jgapcustomised.Species)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSpecie</h4>
<pre>public&nbsp;void&nbsp;setSpecie(<a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&nbsp;aSpecie)</pre>
<div class="block">should only be called from Species; assigns this chromosome to <code>aSpecie</code>; throws exception if
 chromosome is added to a specie twice</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>aSpecie</code> - </dd></dl>
</li>
</ul>
<a name="resetSpecie()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>resetSpecie</h4>
<pre>public&nbsp;void&nbsp;resetSpecie()</pre>
<div class="block">Resets (clears) the specie this chromose belongs to.</div>
</li>
</ul>
<a name="getSpecie()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSpecie</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&nbsp;getSpecie()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>this chromosome's specie</dd></dl>
</li>
</ul>
<a name="getPrimaryParentId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPrimaryParentId</h4>
<pre>public&nbsp;java.lang.Long&nbsp;getPrimaryParentId()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>primary parent ID; this is the dominant parent for chromosomes spawned by crossover, and the only parent
         for chromosomes spawned by cloning</dd></dl>
</li>
</ul>
<a name="getSecondaryParentId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSecondaryParentId</h4>
<pre>public&nbsp;java.lang.Long&nbsp;getSecondaryParentId()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>secondary parent ID; this is the recessive parent for chromosomes spawned by crossover, and null for
         chromosomes spawned by cloning</dd></dl>
</li>
</ul>
<a name="getObjectiveCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getObjectiveCount</h4>
<pre>public&nbsp;int&nbsp;getObjectiveCount()</pre>
</li>
</ul>
<a name="getNoveltyObjectiveCount()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNoveltyObjectiveCount</h4>
<pre>public&nbsp;int&nbsp;getNoveltyObjectiveCount()</pre>
</li>
</ul>
<a name="dominates(org.jgapcustomised.Chromosome)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>dominates</h4>
<pre>public&nbsp;boolean&nbsp;dominates(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;otherChromosome)</pre>
<div class="block">Checks whether this individual dominates the specified other individual, i.e. it is at least as good as the other
 one in all objectives and for at least one objective it is better (i.e. higher fitness value).
 
 This code is based on JNSGA2 by Joachim Melcher, Institut AIFB, Universitaet Karlsruhe (TH), Germany
 http://sourceforge.net/projects/jnsga2</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>otherChromosome</code> - other individual</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>true</code> iff this individual dominates the specified one</dd></dl>
</li>
</ul>
<a name="getMaterial()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getMaterial</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;getMaterial()</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromosome.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-files/index-1.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../org/jgapcustomised/BulkFitnessFunction.html" title="class in org.jgapcustomised"><span class="strong">PREV CLASS</span></a></li>
<li><a href="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><span class="strong">NEXT CLASS</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?org/jgapcustomised/Chromosome.html" target="_top">FRAMES</a></li>
<li><a href="Chromosome.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>SUMMARY:&nbsp;</li>
<li>NESTED&nbsp;|&nbsp;</li>
<li><a href="#field_summary">FIELD</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">METHOD</a></li>
</ul>
<ul class="subNavList">
<li>DETAIL:&nbsp;</li>
<li><a href="#field_detail">FIELD</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">METHOD</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
