<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Jul 05 16:56:46 EST 2013 -->
<TITLE>
Chromosome
</TITLE>

<META NAME="date" CONTENT="2013-07-05">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Chromosome";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/Chromosome.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgapcustomised/BulkFitnessFunction.html" title="class in org.jgapcustomised"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html?org/jgapcustomised/Chromosome.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Chromosome.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.jgapcustomised</FONT>
<BR>
Class Chromosome</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../resources/inherit.gif" ALT="extended by "><B>org.jgapcustomised.Chromosome</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.io.Serializable, java.lang.Comparable</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>Chromosome</B><DT>extends java.lang.Object<DT>implements java.lang.Comparable, java.io.Serializable</DL>
</PRE>

<P>
Chromosomes represent potential solutions and consist of a fixed-length collection of genes. Each gene represents a
 discrete part of the solution. Each gene in the Chromosome may be backed by a different concrete implementation of
 the Gene interface, but all genes with the same innovation ID must share the same concrete implementation across
 Chromosomes within a single population (genotype).
<P>

<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../serialized-form.html#org.jgapcustomised.Chromosome">Serialized Form</A></DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.ArrayList&lt;<A HREF="../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#behaviour">behaviour</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;May be used by implementations of novelty search.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#crowdingDistance">crowdingDistance</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Used by selection algorithms to store a value representing how crowded the fitness space is relative to other
 individuals with the same rank (on the same Pareto front), most useful for multi-objective selection algorithms
 such as NSGA-II.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.lang.Long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#DEFAULT_ID">DEFAULT_ID</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;default ID</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#isElite">isElite</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#m_fitnessValue">m_fitnessValue</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores the fitness value(s) of this Chromosome as determined by the active fitness function(s).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#m_isSelectedForNextGeneration">m_isSelectedForNextGeneration</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#m_overallFitnessValue">m_overallFitnessValue</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Stores the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <A HREF="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><CODE>NaturalSelector</CODE></A> based on the fitness values over all objectives.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#m_performanceValue">m_performanceValue</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#m_specie">m_specie</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#novelty">novelty</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;May be used by implementations of novelty search.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long, int)">Chromosome</A></B>(<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;a_material,
           java.lang.Long&nbsp;an_id,
           int&nbsp;objectiveCount)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#cloneMaterial()">cloneMaterial</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#compareTo(java.lang.Object)">compareTo</A></B>(java.lang.Object&nbsp;o)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares the given Chromosome to this Chromosome by their ID.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">distance</A></B>(<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;target,
         <A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>&nbsp;parms)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#dominates(org.jgapcustomised.Chromosome)">dominates</A></B>(<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;otherChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Checks whether this individual dominates the specified other individual, i.e.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Chromosome against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</A></B>(<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;alleleToMatch)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.SortedSet</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getAlleles()">getAlleles</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getFitnessValue()">getFitnessValue</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <A HREF="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><CODE>NaturalSelector</CODE></A> based on the fitness values over all objectives.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getFitnessValue(int)">getFitnessValue</A></B>(int&nbsp;objective)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the fitness value of this Chromosome for the given objective, as determined by the active fitness
 function.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double[]</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getFitnessValues()">getFitnessValues</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a reference to the array of fitness values for this Chromosome for each objective, as determined by the
 active fitness function.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getId()">getId</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getObjectiveCount()">getObjectiveCount</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getPerformanceValue()">getPerformanceValue</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the performance value of this Chromosome as determined by the active fitness function.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getPrimaryParentId()">getPrimaryParentId</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.Long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getSecondaryParentId()">getSecondaryParentId</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getSpeciatedFitnessValue()">getSpeciatedFitnessValue</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#getSpecie()">getSpecie</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#hashCode()">hashCode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieve a hash code for this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#isEquivalent(org.jgapcustomised.Chromosome)">isEquivalent</A></B>(<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c2)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Chromosome against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#isSelectedForNextGeneration()">isSelectedForNextGeneration</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves whether this Chromosome has been selected by the natural selector to continue to the next generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#resetEvaluationData()">resetEvaluationData</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Resets the performance, fitness value(s) and recorded novelty behaviour(s) of this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#resetFitnessValues()">resetFitnessValues</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Resets the fitness value(s) of this Chromosome to Double.NaN.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#resetPerformanceValue()">resetPerformanceValue</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Resets the performance value of this Chromosome to Double.NaN.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#setCrowdingDistance(double)">setCrowdingDistance</A></B>(double&nbsp;positiveInfinity)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#setFitnessValue(double)">setFitnessValue</A></B>(double&nbsp;a_newFitnessValue)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the overall fitness value of this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#setFitnessValue(double, int)">setFitnessValue</A></B>(double&nbsp;a_newFitnessValue,
                int&nbsp;objective)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the fitness value of this Chromosome for the given objective.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#setIsSelectedForNextGeneration(boolean)">setIsSelectedForNextGeneration</A></B>(boolean&nbsp;a_isSelected)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#setPerformanceValue(double)">setPerformanceValue</A></B>(double&nbsp;aPerformanceValue)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sets the performance value of this Chromosome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#size()">size</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the size of this Chromosome (the number of alleles it contains).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Chromosome.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a string representation of this Chromosome, useful for some display purposes.</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="DEFAULT_ID"><!-- --></A><H3>
DEFAULT_ID</H3>
<PRE>
public static final java.lang.Long <B>DEFAULT_ID</B></PRE>
<DL>
<DD>default ID
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_isSelectedForNextGeneration"><!-- --></A><H3>
m_isSelectedForNextGeneration</H3>
<PRE>
protected boolean <B>m_isSelectedForNextGeneration</B></PRE>
<DL>
<DD>Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_fitnessValue"><!-- --></A><H3>
m_fitnessValue</H3>
<PRE>
protected double[] <B>m_fitnessValue</B></PRE>
<DL>
<DD>Stores the fitness value(s) of this Chromosome as determined by the active fitness function(s). A value of
 Double.NaN indicates that a fitness value has not yet been set.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_overallFitnessValue"><!-- --></A><H3>
m_overallFitnessValue</H3>
<PRE>
protected double <B>m_overallFitnessValue</B></PRE>
<DL>
<DD>Stores the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <A HREF="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><CODE>NaturalSelector</CODE></A> based on the fitness values over all objectives. A
 value of Double.NaN indicates that a fitness value has not yet been set.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_performanceValue"><!-- --></A><H3>
m_performanceValue</H3>
<PRE>
protected double <B>m_performanceValue</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_specie"><!-- --></A><H3>
m_specie</H3>
<PRE>
protected <A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A> <B>m_specie</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="isElite"><!-- --></A><H3>
isElite</H3>
<PRE>
public boolean <B>isElite</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="novelty"><!-- --></A><H3>
novelty</H3>
<PRE>
public double <B>novelty</B></PRE>
<DL>
<DD>May be used by implementations of novelty search.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../com/ojcoleman/ahni/evaluation/novelty/NoveltySearch.html" title="class in com.ojcoleman.ahni.evaluation.novelty"><CODE>NoveltySearch</CODE></A></DL>
</DL>
<HR>

<A NAME="behaviour"><!-- --></A><H3>
behaviour</H3>
<PRE>
public java.util.ArrayList&lt;<A HREF="../../com/ojcoleman/ahni/evaluation/novelty/Behaviour.html" title="class in com.ojcoleman.ahni.evaluation.novelty">Behaviour</A>&gt; <B>behaviour</B></PRE>
<DL>
<DD>May be used by implementations of novelty search. A list is used so that one or more behaviours may be stored.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../com/ojcoleman/ahni/evaluation/novelty/NoveltySearch.html" title="class in com.ojcoleman.ahni.evaluation.novelty"><CODE>NoveltySearch</CODE></A></DL>
</DL>
<HR>

<A NAME="crowdingDistance"><!-- --></A><H3>
crowdingDistance</H3>
<PRE>
public double <B>crowdingDistance</B></PRE>
<DL>
<DD>Used by selection algorithms to store a value representing how crowded the fitness space is relative to other
 individuals with the same rank (on the same Pareto front), most useful for multi-objective selection algorithms
 such as NSGA-II.
<P>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long, int)"><!-- --></A><H3>
Chromosome</H3>
<PRE>
public <B>Chromosome</B>(<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&nbsp;a_material,
                  java.lang.Long&nbsp;an_id,
                  int&nbsp;objectiveCount)</PRE>
<DL>
<DD>this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>a_material</CODE> - Genetic material to be contained within this Chromosome instance.<DD><CODE>an_id</CODE> - unique ID of new chromosome</DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)"><!-- --></A><H3>
distance</H3>
<PRE>
public double <B>distance</B>(<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;target,
                       <A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A>&nbsp;parms)</PRE>
<DL>
<DD>Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology. It is generic
 enough that the alleles do not have to be nodes and connections.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>target</CODE> - <DD><CODE>parms</CODE> - 
<DT><B>Returns:</B><DD>distance between this object and <code>target</code><DT><B>See Also:</B><DD><A HREF="../../org/jgapcustomised/ChromosomeMaterial.html#distance(org.jgapcustomised.ChromosomeMaterial, org.jgapcustomised.SpeciationParms)"><CODE>ChromosomeMaterial.distance(ChromosomeMaterial, SpeciationParms)</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="getId()"><!-- --></A><H3>
getId</H3>
<PRE>
public java.lang.Long <B>getId</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>Long unique identifier for chromosome; useful for <code>hashCode()</code> and persistence</DL>
</DD>
</DL>
<HR>

<A NAME="size()"><!-- --></A><H3>
size</H3>
<PRE>
public int <B>size</B>()</PRE>
<DL>
<DD>Returns the size of this Chromosome (the number of alleles it contains). A Chromosome's size is constant and will
 never change.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The number of alleles contained within this Chromosome instance.</DL>
</DD>
</DL>
<HR>

<A NAME="cloneMaterial()"><!-- --></A><H3>
cloneMaterial</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A> <B>cloneMaterial</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>clone with primary parent ID of this chromosome and the same genetic material.</DL>
</DD>
</DL>
<HR>

<A NAME="getAlleles()"><!-- --></A><H3>
getAlleles</H3>
<PRE>
public java.util.SortedSet <B>getAlleles</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>SortedSet alleles, sorted by innovation ID</DL>
</DD>
</DL>
<HR>

<A NAME="findMatchingGene(org.jgapcustomised.Allele)"><!-- --></A><H3>
findMatchingGene</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A> <B>findMatchingGene</B>(<A HREF="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&nbsp;alleleToMatch)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleleToMatch</CODE> - 
<DT><B>Returns:</B><DD>Gene gene with same innovation ID as <code>geneToMatch</code, or <code>null</code> if none match</DL>
</DD>
</DL>
<HR>

<A NAME="getFitnessValue()"><!-- --></A><H3>
getFitnessValue</H3>
<PRE>
public double <B>getFitnessValue</B>()</PRE>
<DL>
<DD>Returns the overall fitness value of this Chromosome, either as determined by the active fitness function or as
 determined by the <A HREF="../../org/jgapcustomised/NaturalSelector.html" title="class in org.jgapcustomised"><CODE>NaturalSelector</CODE></A> based on the fitness values over all objectives.
 Fitness values are in the range [0, 1], however the value Double.NaN indicates that a fitness hasn't been set
 yet.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getFitnessValue(int)"><!-- --></A><H3>
getFitnessValue</H3>
<PRE>
public double <B>getFitnessValue</B>(int&nbsp;objective)</PRE>
<DL>
<DD>Returns the fitness value of this Chromosome for the given objective, as determined by the active fitness
 function. Fitness values are in the range [0, 1], however the value Double.NaN indicates that a fitness hasn't
 been set yet.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getFitnessValues()"><!-- --></A><H3>
getFitnessValues</H3>
<PRE>
public double[] <B>getFitnessValues</B>()</PRE>
<DL>
<DD>Returns a reference to the array of fitness values for this Chromosome for each objective, as determined by the
 active fitness function. Fitness values are in the range [0, 1], however the value Double.NaN indicates that a
 fitness hasn't been set yet.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getSpeciatedFitnessValue()"><!-- --></A><H3>
getSpeciatedFitnessValue</H3>
<PRE>
public double <B>getSpeciatedFitnessValue</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>int fitness value adjusted for fitness sharing according to <a
         href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf>NEAT </a> paradigm.</DL>
</DD>
</DL>
<HR>

<A NAME="setFitnessValue(double)"><!-- --></A><H3>
setFitnessValue</H3>
<PRE>
public void <B>setFitnessValue</B>(double&nbsp;a_newFitnessValue)</PRE>
<DL>
<DD>Sets the overall fitness value of this Chromosome. This method is for use by bulk fitness functions and should
 not be invoked from anything else. This is the raw fitness value, before species fitness sharing. Fitness values
 must be in the range [0, 1].
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_newFitnessValue</CODE> - the fitness of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="setFitnessValue(double, int)"><!-- --></A><H3>
setFitnessValue</H3>
<PRE>
public void <B>setFitnessValue</B>(double&nbsp;a_newFitnessValue,
                            int&nbsp;objective)</PRE>
<DL>
<DD>Sets the fitness value of this Chromosome for the given objective. This method is for use by bulk fitness
 functions and should not be invoked from anything else. Fitness values must be in the range [0, 1].
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_newFitnessValue</CODE> - the fitness of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="getPerformanceValue()"><!-- --></A><H3>
getPerformanceValue</H3>
<PRE>
public double <B>getPerformanceValue</B>()</PRE>
<DL>
<DD>Returns the performance value of this Chromosome as determined by the active fitness function. Performance values
 are in the range [0, 1], however the value Double.NaN indicates that a fitness hasn't been set yet.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setPerformanceValue(double)"><!-- --></A><H3>
setPerformanceValue</H3>
<PRE>
public void <B>setPerformanceValue</B>(double&nbsp;aPerformanceValue)</PRE>
<DL>
<DD>Sets the performance value of this Chromosome. This method is for use by bulk fitness functions and should not be
 invoked from anything else. Performance values must be in the range [0, 1].
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>aPerformanceValue</CODE> - the fitness of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="resetPerformanceValue()"><!-- --></A><H3>
resetPerformanceValue</H3>
<PRE>
public void <B>resetPerformanceValue</B>()</PRE>
<DL>
<DD>Resets the performance value of this Chromosome to Double.NaN.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="resetFitnessValues()"><!-- --></A><H3>
resetFitnessValues</H3>
<PRE>
public void <B>resetFitnessValues</B>()</PRE>
<DL>
<DD>Resets the fitness value(s) of this Chromosome to Double.NaN.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="resetEvaluationData()"><!-- --></A><H3>
resetEvaluationData</H3>
<PRE>
public void <B>resetEvaluationData</B>()</PRE>
<DL>
<DD>Resets the performance, fitness value(s) and recorded novelty behaviour(s) of this Chromosome.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD>Returns a string representation of this Chromosome, useful for some display purposes.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A string representation of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has the same ID.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="hashCode()"><!-- --></A><H3>
hashCode</H3>
<PRE>
public int <B>hashCode</B>()</PRE>
<DL>
<DD>Retrieve a hash code for this Chromosome.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>hashCode</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the hash code of this Chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="compareTo(java.lang.Object)"><!-- --></A><H3>
compareTo</H3>
<PRE>
public int <B>compareTo</B>(java.lang.Object&nbsp;o)</PRE>
<DL>
<DD>Compares the given Chromosome to this Chromosome by their ID.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE>compareTo</CODE> in interface <CODE>java.lang.Comparable</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>o</CODE> - The Chromosome against which to compare this chromosome.
<DT><B>Returns:</B><DD>a negative number if this chromosome is "less than" the given chromosome, zero if they are equal to each
         other, and a positive number if this chromosome is "greater than" the given chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="isEquivalent(org.jgapcustomised.Chromosome)"><!-- --></A><H3>
isEquivalent</H3>
<PRE>
public boolean <B>isEquivalent</B>(<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;c2)</PRE>
<DL>
<DD>Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has a set of genes with equal values (eg connection weight, activation type) to this
 one.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>c2</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="setIsSelectedForNextGeneration(boolean)"><!-- --></A><H3>
setIsSelectedForNextGeneration</H3>
<PRE>
public void <B>setIsSelectedForNextGeneration</B>(boolean&nbsp;a_isSelected)</PRE>
<DL>
<DD>Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_isSelected</CODE> - true if this Chromosome has been selected, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="isSelectedForNextGeneration()"><!-- --></A><H3>
isSelectedForNextGeneration</H3>
<PRE>
public boolean <B>isSelectedForNextGeneration</B>()</PRE>
<DL>
<DD>Retrieves whether this Chromosome has been selected by the natural selector to continue to the next generation.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>true if this Chromosome has been selected, false otherwise.</DL>
</DD>
</DL>
<HR>

<A NAME="getSpecie()"><!-- --></A><H3>
getSpecie</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A> <B>getSpecie</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>this chromosome's specie</DL>
</DD>
</DL>
<HR>

<A NAME="getPrimaryParentId()"><!-- --></A><H3>
getPrimaryParentId</H3>
<PRE>
public java.lang.Long <B>getPrimaryParentId</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>primary parent ID; this is the dominant parent for chromosomes spawned by crossover, and the only parent
         for chromosomes spawned by cloning</DL>
</DD>
</DL>
<HR>

<A NAME="getSecondaryParentId()"><!-- --></A><H3>
getSecondaryParentId</H3>
<PRE>
public java.lang.Long <B>getSecondaryParentId</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>secondary parent ID; this is the recessive parent for chromosomes spawned by crossover, and null for
         chromosomes spawned by cloning</DL>
</DD>
</DL>
<HR>

<A NAME="getObjectiveCount()"><!-- --></A><H3>
getObjectiveCount</H3>
<PRE>
public int <B>getObjectiveCount</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="dominates(org.jgapcustomised.Chromosome)"><!-- --></A><H3>
dominates</H3>
<PRE>
public boolean <B>dominates</B>(<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&nbsp;otherChromosome)</PRE>
<DL>
<DD>Checks whether this individual dominates the specified other individual, i.e. it is at least as good as the other
 one in all objectives and for at least one objective it is better (i.e. higher fitness value).
 
 This code is based on JNSGA2 by Joachim Melcher, Institut AIFB, Universitaet Karlsruhe (TH), Germany
 http://sourceforge.net/projects/jnsga2
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>otherChromosome</CODE> - other individual
<DT><B>Returns:</B><DD><code>true</code> iff this individual dominates the specified one</DL>
</DD>
</DL>
<HR>

<A NAME="setCrowdingDistance(double)"><!-- --></A><H3>
setCrowdingDistance</H3>
<PRE>
public void <B>setCrowdingDistance</B>(double&nbsp;positiveInfinity)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/Chromosome.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgapcustomised/BulkFitnessFunction.html" title="class in org.jgapcustomised"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html?org/jgapcustomised/Chromosome.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Chromosome.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
