<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_09) on Fri Nov 30 12:10:36 EST 2012 -->
<title>Chromosome</title>
<meta name="date" content="2012-11-30">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Chromosome";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromosome.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-files/index-1.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../org/jgapcustomised/BulkFitnessFunction.html" title="interface in org.jgapcustomised"><span class="strong">Prev Class</span></a></li>
<li><a href="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?org/jgapcustomised/Chromosome.html" target="_top">Frames</a></li>
<li><a href="Chromosome.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.jgapcustomised</div>
<h2 title="Class Chromosome" class="title">Class Chromosome</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>org.jgapcustomised.Chromosome</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd>java.io.Serializable, java.lang.Comparable</dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">Chromosome</span>
extends java.lang.Object
implements java.lang.Comparable, java.io.Serializable</pre>
<div class="block">Chromosomes represent potential solutions and consist of a fixed-length collection of genes. Each gene represents a
 discrete part of the solution. Each gene in the Chromosome may be backed by a different concrete implementation of
 the Gene interface, but all genes with the same innovation ID must share the same concrete implementation across
 Chromosomes within a single population (genotype).</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../serialized-form.html#org.jgapcustomised.Chromosome">Serialized Form</a></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#DEFAULT_ID">DEFAULT_ID</a></strong></code>
<div class="block">default ID</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#isElite">isElite</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_fitnessValue">m_fitnessValue</a></strong></code>
<div class="block">Stores the fitness value of this Chromosome as determined by the active fitness function.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_isSelectedForNextGeneration">m_isSelectedForNextGeneration</a></strong></code>
<div class="block">Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#m_performanceValue">m_performanceValue</a></strong></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long)">Chromosome</a></strong>(<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;a_material,
          java.lang.Long&nbsp;an_id)</code>
<div class="block">this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#cloneMaterial()">cloneMaterial</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#compareTo(java.lang.Object)">compareTo</a></strong>(java.lang.Object&nbsp;o)</code>
<div class="block">Compares the given Chromosome to this Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">distance</a></strong>(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;target,
        <a href="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;parms)</code>
<div class="block">Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#equals(java.lang.Object)">equals</a></strong>(java.lang.Object&nbsp;other)</code>
<div class="block">Compares this Chromosome against the specified object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#findMatchingGene(org.jgapcustomised.Allele)">findMatchingGene</a></strong>(<a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;alleleToMatch)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.SortedSet</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getAlleles()">getAlleles</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getFitnessValue()">getFitnessValue</a></strong>()</code>
<div class="block">Retrieves the fitness value of this Chromosome, as determined by the active fitness function.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getId()">getId</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getPerformanceValue()">getPerformanceValue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getPrimaryParentId()">getPrimaryParentId</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.Long</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getSecondaryParentId()">getSecondaryParentId</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getSpeciatedFitnessValue()">getSpeciatedFitnessValue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a></code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#getSpecie()">getSpecie</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#hashCode()">hashCode</a></strong>()</code>
<div class="block">Retrieve a hash code for this Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#isSelectedForNextGeneration()">isSelectedForNextGeneration</a></strong>()</code>
<div class="block">Retrieves whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setFitnessValue(int)">setFitnessValue</a></strong>(int&nbsp;a_newFitnessValue)</code>
<div class="block">Sets the fitness value of this Chromosome.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setIsSelectedForNextGeneration(boolean)">setIsSelectedForNextGeneration</a></strong>(boolean&nbsp;a_isSelected)</code>
<div class="block">Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#setPerformanceValue(double)">setPerformanceValue</a></strong>(double&nbsp;mPerformanceValue)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#size()">size</a></strong>()</code>
<div class="block">Returns the size of this Chromosome (the number of alleles it contains).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../org/jgapcustomised/Chromosome.html#toString()">toString</a></strong>()</code>
<div class="block">Returns a string representation of this Chromosome, useful for some display purposes.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="DEFAULT_ID">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DEFAULT_ID</h4>
<pre>public static final&nbsp;java.lang.Long DEFAULT_ID</pre>
<div class="block">default ID</div>
</li>
</ul>
<a name="m_isSelectedForNextGeneration">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_isSelectedForNextGeneration</h4>
<pre>protected&nbsp;boolean m_isSelectedForNextGeneration</pre>
<div class="block">Keeps track of whether or not this Chromosome has been selected by the natural selector to move on to the next
 generation.</div>
</li>
</ul>
<a name="m_fitnessValue">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_fitnessValue</h4>
<pre>protected&nbsp;int m_fitnessValue</pre>
<div class="block">Stores the fitness value of this Chromosome as determined by the active fitness function. A value of -1 indicates
 that this field has not yet been set with this Chromosome's fitness values (valid fitness values are always
 positive).</div>
</li>
</ul>
<a name="m_performanceValue">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_performanceValue</h4>
<pre>protected&nbsp;double m_performanceValue</pre>
</li>
</ul>
<a name="isElite">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>isElite</h4>
<pre>public&nbsp;boolean isElite</pre>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="Chromosome(org.jgapcustomised.ChromosomeMaterial, java.lang.Long)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Chromosome</h4>
<pre>public&nbsp;Chromosome(<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;a_material,
          java.lang.Long&nbsp;an_id)</pre>
<div class="block">this should only be called when a chromosome is being created from persistence; otherwise, the ID should be
 generated by <code>a_activeConfiguration</code>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_material</code> - Genetic material to be contained within this Chromosome instance.</dd><dd><code>an_id</code> - unique ID of new chromosome</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="distance(org.jgapcustomised.Chromosome, org.jgapcustomised.SpeciationParms)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>distance</h4>
<pre>public&nbsp;double&nbsp;distance(<a href="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</a>&nbsp;target,
              <a href="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</a>&nbsp;parms)</pre>
<div class="block">Calculates compatibility distance between this and <code>target</code> according to <a
 href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf">NEAT </a> speciation methodology. It is generic
 enough that the alleles do not have to be nodes and connections.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>target</code> - </dd><dd><code>parms</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>distance between this object and <code>target</code></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../org/jgapcustomised/ChromosomeMaterial.html#distance(org.jgapcustomised.ChromosomeMaterial, org.jgapcustomised.SpeciationParms)"><code>ChromosomeMaterial.distance(ChromosomeMaterial, SpeciationParms)</code></a></dd></dl>
</li>
</ul>
<a name="getId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getId</h4>
<pre>public&nbsp;java.lang.Long&nbsp;getId()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>Long unique identifier for chromosome; useful for <code>hashCode()</code> and persistence</dd></dl>
</li>
</ul>
<a name="size()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>size</h4>
<pre>public&nbsp;int&nbsp;size()</pre>
<div class="block">Returns the size of this Chromosome (the number of alleles it contains). A Chromosome's size is constant and will
 never change.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The number of alleles contained within this Chromosome instance.</dd></dl>
</li>
</ul>
<a name="cloneMaterial()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>cloneMaterial</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;cloneMaterial()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>clone with primary parent ID of this chromosome and the same genetic material.</dd></dl>
</li>
</ul>
<a name="getAlleles()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlleles</h4>
<pre>public&nbsp;java.util.SortedSet&nbsp;getAlleles()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>SortedSet alleles, sorted by innovation ID</dd></dl>
</li>
</ul>
<a name="findMatchingGene(org.jgapcustomised.Allele)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>findMatchingGene</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;findMatchingGene(<a href="../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;alleleToMatch)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleleToMatch</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>Gene gene with same innovation ID as <code>geneToMatch</code, or <code>null</code> if none match</dd></dl>
</li>
</ul>
<a name="getFitnessValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFitnessValue</h4>
<pre>public&nbsp;int&nbsp;getFitnessValue()</pre>
<div class="block">Retrieves the fitness value of this Chromosome, as determined by the active fitness function.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a positive integer value representing the fitness of this Chromosome, or -1 if fitness function has not
         yet assigned a fitness value to this Chromosome.</dd></dl>
</li>
</ul>
<a name="getSpeciatedFitnessValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSpeciatedFitnessValue</h4>
<pre>public&nbsp;int&nbsp;getSpeciatedFitnessValue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>int fitness value adjusted for fitness sharing according to <a
         href="http://nn.cs.utexas.edu/downloads/papers/stanley.ec02.pdf>NEAT </a> paradigm.</dd></dl>
</li>
</ul>
<a name="setFitnessValue(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFitnessValue</h4>
<pre>public&nbsp;void&nbsp;setFitnessValue(int&nbsp;a_newFitnessValue)</pre>
<div class="block">Sets the fitness value of this Chromosome. This method is for use by bulk fitness functions and should not be
 invoked from anything else. This is the raw fitness value, before species fitness sharing. Setting the fitness
 value will reset the performance value to -1, thus the performance value should always be set after setting the
 fitness.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_newFitnessValue</code> - a positive integer representing the fitness of this Chromosome. if 0, fitness is set as
            1.</dd></dl>
</li>
</ul>
<a name="getPerformanceValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPerformanceValue</h4>
<pre>public&nbsp;double&nbsp;getPerformanceValue()</pre>
</li>
</ul>
<a name="setPerformanceValue(double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setPerformanceValue</h4>
<pre>public&nbsp;void&nbsp;setPerformanceValue(double&nbsp;mPerformanceValue)</pre>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;java.lang.String&nbsp;toString()</pre>
<div class="block">Returns a string representation of this Chromosome, useful for some display purposes.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
<dt><span class="strong">Returns:</span></dt><dd>A string representation of this Chromosome.</dd></dl>
</li>
</ul>
<a name="equals(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public&nbsp;boolean&nbsp;equals(java.lang.Object&nbsp;other)</pre>
<div class="block">Compares this Chromosome against the specified object. The result is true if and the argument is an instance of
 the Chromosome class and has a set of genes equal to this one.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code>equals</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>other</code> - The object to compare against.</dd>
<dt><span class="strong">Returns:</span></dt><dd>true if the objects are the same, false otherwise.</dd></dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<div class="block">Retrieve a hash code for this Chromosome.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code>hashCode</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the hash code of this Chromosome.</dd></dl>
</li>
</ul>
<a name="compareTo(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>compareTo</h4>
<pre>public&nbsp;int&nbsp;compareTo(java.lang.Object&nbsp;o)</pre>
<div class="block">Compares the given Chromosome to this Chromosome. This chromosome is considered to be "less than" the given
 chromosome if it has a fewer number of genes or if any of its gene values (alleles) are less than their
 corresponding gene values in the other chromosome.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code>compareTo</code>&nbsp;in interface&nbsp;<code>java.lang.Comparable</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>o</code> - The Chromosome against which to compare this chromosome.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a negative number if this chromosome is "less than" the given chromosome, zero if they are equal to each
         other, and a positive number if this chromosome is "greater than" the given chromosome.</dd></dl>
</li>
</ul>
<a name="setIsSelectedForNextGeneration(boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setIsSelectedForNextGeneration</h4>
<pre>public&nbsp;void&nbsp;setIsSelectedForNextGeneration(boolean&nbsp;a_isSelected)</pre>
<div class="block">Sets whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_isSelected</code> - true if this Chromosome has been selected, false otherwise.</dd></dl>
</li>
</ul>
<a name="isSelectedForNextGeneration()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isSelectedForNextGeneration</h4>
<pre>public&nbsp;boolean&nbsp;isSelectedForNextGeneration()</pre>
<div class="block">Retrieves whether this Chromosome has been selected by the natural selector to continue to the next generation.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>true if this Chromosome has been selected, false otherwise.</dd></dl>
</li>
</ul>
<a name="getSpecie()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSpecie</h4>
<pre>public&nbsp;<a href="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</a>&nbsp;getSpecie()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>this chromosome's specie</dd></dl>
</li>
</ul>
<a name="getPrimaryParentId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPrimaryParentId</h4>
<pre>public&nbsp;java.lang.Long&nbsp;getPrimaryParentId()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>primary parent ID; this is the dominant parent for chromosomes spawned by crossover, and the only parent
         for chromosomes spawned by cloning</dd></dl>
</li>
</ul>
<a name="getSecondaryParentId()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSecondaryParentId</h4>
<pre>public&nbsp;java.lang.Long&nbsp;getSecondaryParentId()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>secondary parent ID; this is the recessive parent for chromosomes spawned by crossover, and null for
         chromosomes spawned by cloning</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromosome.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-files/index-1.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../org/jgapcustomised/BulkFitnessFunction.html" title="interface in org.jgapcustomised"><span class="strong">Prev Class</span></a></li>
<li><a href="../../org/jgapcustomised/ChromosomeFitnessComparator.html" title="class in org.jgapcustomised"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?org/jgapcustomised/Chromosome.html" target="_top">Frames</a></li>
<li><a href="Chromosome.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
