<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Mar 22 08:30:59 EST 2013 -->
<TITLE>
Genotype
</TITLE>

<META NAME="date" CONTENT="2013-03-22">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Genotype";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/Genotype.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgapcustomised/IdFactory.html" title="class in org.jgapcustomised"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html?org/jgapcustomised/Genotype.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Genotype.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.jgapcustomised</FONT>
<BR>
Class Genotype</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../resources/inherit.gif" ALT="extended by "><B>org.jgapcustomised.Genotype</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD>java.io.Serializable</DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>Genotype</B><DT>extends java.lang.Object<DT>implements java.io.Serializable</DL>
</PRE>

<P>
Genotypes are fixed-length populations of chromosomes. As an instance of a <code>Genotype</code> is evolved, all of
 its <code>Chromosome</code> objects are also evolved. A <code>Genotype</code> may be constructed normally, whereby an
 array of <code>Chromosome</code> objects must be provided, or the static <code>randomInitialGenotype()</code> method
 can be used to generate a <code>Genotype</code> with a randomized <code>Chromosome</code> population. Changes made by
 Tucker and James for <a href="http://anji.sourceforge.net/">ANJI </a>:
 <ul>
 <li>added species</li>
 <li>modified order of operations in <code>evolve()</code></li>
 <li>added <code>addChromosome*()</code> methods</li>
 </ul>
<P>

<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../serialized-form.html#org.jgapcustomised.Genotype">Serialized Form</A></DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#bestPerforming">bestPerforming</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#eliteCount">eliteCount</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#fittest">fittest</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#generation">generation</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#lastGenChangedSpeciesCompatThreshold">lastGenChangedSpeciesCompatThreshold</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#m_activeConfiguration">m_activeConfiguration</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The current active Configuration instance.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#m_chromosomes">m_chromosomes</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Chromosomes that makeup thie Genotype's population.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;java.util.List&lt;<A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#m_species">m_species</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Species that makeup this Genotype's population.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#m_specParms">m_specParms</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#maxSpeciesSize">maxSpeciesSize</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#minSpeciesSize">minSpeciesSize</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#props">props</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#targetPerformanceType">targetPerformanceType</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#zeroFitnessCount">zeroFitnessCount</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#Genotype(com.anji.util.Properties, org.jgapcustomised.Configuration, java.util.List)">Genotype</A></B>(<A HREF="../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>&nbsp;props,
         <A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration,
         java.util.List&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;a_initialChromosomes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This constructor is used for random initial Genotypes.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#addChromosomes(java.util.Collection)">addChromosomes</A></B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chromosomes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add the specified chromosomes to this Genotype.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#addChromosomesFromMaterial(java.util.Collection)">addChromosomesFromMaterial</A></B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&gt;&nbsp;chromosomeMaterial)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Add Chromosomes to this Genotype described by the given ChromosomeMaterial objects.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#equals(java.lang.Object)">equals</A></B>(java.lang.Object&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Compares this Genotype against the specified object.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#evolve()">evolve</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Performs one generation cycle, evaluating fitness, selecting survivors, repopulting with offspring, and mutating
 new population.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getAveragePopulationFitness()">getAveragePopulationFitness</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getBestPerforming()">getBestPerforming</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getChromosomes()">getChromosomes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getFittest()">getFittest</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getFittestChromosome()">getFittestChromosome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Retrieves the Chromosome in the population with the highest fitness value.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getMaxSpeciesSize()">getMaxSpeciesSize</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getMinSpeciesSize()">getMinSpeciesSize</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getNumberOfChromosomesWithZeroFitnessFromLastGen()">getNumberOfChromosomesWithZeroFitnessFromLastGen</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getParameters()">getParameters</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.util.List&lt;<A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#getSpecies()">getSpecies</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#randomInitialGenotype(com.anji.util.Properties, org.jgapcustomised.Configuration)">randomInitialGenotype</A></B>(<A HREF="../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>&nbsp;props,
                      <A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convenience method that returns a newly constructed Genotype instance configured according to the given
 Configuration instance.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#respeciate()">respeciate</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Clears all species and creates a fresh speciation.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#speciate()">speciate</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(Re)assigns all chromosomes to a species.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;java.lang.String</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../org/jgapcustomised/Genotype.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="m_activeConfiguration"><!-- --></A><H3>
m_activeConfiguration</H3>
<PRE>
protected <A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A> <B>m_activeConfiguration</B></PRE>
<DL>
<DD>The current active Configuration instance.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="props"><!-- --></A><H3>
props</H3>
<PRE>
protected <A HREF="../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A> <B>props</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_specParms"><!-- --></A><H3>
m_specParms</H3>
<PRE>
protected <A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A> <B>m_specParms</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_species"><!-- --></A><H3>
m_species</H3>
<PRE>
protected java.util.List&lt;<A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt; <B>m_species</B></PRE>
<DL>
<DD>Species that makeup this Genotype's population.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="m_chromosomes"><!-- --></A><H3>
m_chromosomes</H3>
<PRE>
protected java.util.List&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt; <B>m_chromosomes</B></PRE>
<DL>
<DD>Chromosomes that makeup thie Genotype's population.
<P>
<DL>
</DL>
</DL>
<HR>

<A NAME="generation"><!-- --></A><H3>
generation</H3>
<PRE>
protected int <B>generation</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="lastGenChangedSpeciesCompatThreshold"><!-- --></A><H3>
lastGenChangedSpeciesCompatThreshold</H3>
<PRE>
protected int <B>lastGenChangedSpeciesCompatThreshold</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="targetPerformanceType"><!-- --></A><H3>
targetPerformanceType</H3>
<PRE>
protected int <B>targetPerformanceType</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="fittest"><!-- --></A><H3>
fittest</H3>
<PRE>
protected <A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> <B>fittest</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="bestPerforming"><!-- --></A><H3>
bestPerforming</H3>
<PRE>
protected <A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> <B>bestPerforming</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="zeroFitnessCount"><!-- --></A><H3>
zeroFitnessCount</H3>
<PRE>
protected int <B>zeroFitnessCount</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="eliteCount"><!-- --></A><H3>
eliteCount</H3>
<PRE>
protected int <B>eliteCount</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="maxSpeciesSize"><!-- --></A><H3>
maxSpeciesSize</H3>
<PRE>
protected int <B>maxSpeciesSize</B></PRE>
<DL>
<DL>
</DL>
</DL>
<HR>

<A NAME="minSpeciesSize"><!-- --></A><H3>
minSpeciesSize</H3>
<PRE>
protected int <B>minSpeciesSize</B></PRE>
<DL>
<DL>
</DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="Genotype(com.anji.util.Properties, org.jgapcustomised.Configuration, java.util.List)"><!-- --></A><H3>
Genotype</H3>
<PRE>
public <B>Genotype</B>(<A HREF="../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>&nbsp;props,
                <A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration,
                java.util.List&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;a_initialChromosomes)
         throws <A HREF="../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</A></PRE>
<DL>
<DD>This constructor is used for random initial Genotypes. Note that the Configuration object must be in a valid
 state when this method is invoked, or a InvalidconfigurationException will be thrown.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - The current active Configuration object.<DD><CODE>a_initialChromosomes</CODE> - <code>List</code> contains Chromosome objects: The Chromosome population to be
            managed by this Genotype instance.
<DT><B>Throws:</B>
<DD><CODE>java.lang.IllegalArgumentException</CODE> - if either the given Configuration object or the array of Chromosomes is null, or
             if any of the Genes in the array of Chromosomes is null.
<DD><CODE><A HREF="../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</A></CODE> - if the given Configuration object is in an invalid state.</DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="addChromosomes(java.util.Collection)"><!-- --></A><H3>
addChromosomes</H3>
<PRE>
protected void <B>addChromosomes</B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt;&nbsp;chromosomes)</PRE>
<DL>
<DD>Add the specified chromosomes to this Genotype.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chromosomes</CODE> - A collection of Chromosome objects.</DL>
</DD>
</DL>
<HR>

<A NAME="addChromosomesFromMaterial(java.util.Collection)"><!-- --></A><H3>
addChromosomesFromMaterial</H3>
<PRE>
protected void <B>addChromosomesFromMaterial</B>(java.util.Collection&lt;<A HREF="../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A>&gt;&nbsp;chromosomeMaterial)</PRE>
<DL>
<DD>Add Chromosomes to this Genotype described by the given ChromosomeMaterial objects.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>chromosomeMaterial</CODE> - A collection of ChromosomeMaterial objects.</DL>
</DD>
</DL>
<HR>

<A NAME="respeciate()"><!-- --></A><H3>
respeciate</H3>
<PRE>
protected void <B>respeciate</B>()</PRE>
<DL>
<DD>Clears all species and creates a fresh speciation.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="speciate()"><!-- --></A><H3>
speciate</H3>
<PRE>
protected void <B>speciate</B>()</PRE>
<DL>
<DD>(Re)assigns all chromosomes to a species. If no existing species are compatible with a chromosome a new species
 is created.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getChromosomes()"><!-- --></A><H3>
getChromosomes</H3>
<PRE>
public java.util.List&lt;<A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A>&gt; <B>getChromosomes</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>List contains Chromosome objects, the population of Chromosomes.</DL>
</DD>
</DL>
<HR>

<A NAME="getSpecies()"><!-- --></A><H3>
getSpecies</H3>
<PRE>
public java.util.List&lt;<A HREF="../../org/jgapcustomised/Species.html" title="class in org.jgapcustomised">Species</A>&gt; <B>getSpecies</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>List contains Species objects</DL>
</DD>
</DL>
<HR>

<A NAME="getFittestChromosome()"><!-- --></A><H3>
getFittestChromosome</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> <B>getFittestChromosome</B>()</PRE>
<DL>
<DD>Retrieves the Chromosome in the population with the highest fitness value.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The Chromosome with the highest fitness value, or null if there are no chromosomes in this Genotype.</DL>
</DD>
</DL>
<HR>

<A NAME="evolve()"><!-- --></A><H3>
evolve</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> <B>evolve</B>()</PRE>
<DL>
<DD>Performs one generation cycle, evaluating fitness, selecting survivors, repopulting with offspring, and mutating
 new population. This is a modified version of original JGAP method which changes order of operations and splits
 <code>GeneticOperator</code> into <code>ReproductionOperator</code> and <code>MutationOperator</code>. New order
 of operations:
 <ol>
 <li>assign <b>fitness </b> to all members of population with <code>BulkFitnessFunction</code> or
 <code>FitnessFunction</code></li>
 <li><b>select </b> survivors and remove casualties from population</li>
 <li>re-fill population with offspring via <b>reproduction </b> operators</li>
 <li><b>mutate </b> offspring (note, survivors are passed on un-mutated)</li>
 </ol>
 Genetic event <code>GeneticEvent.GENOTYPE_EVALUATED_EVENT</code> is fired between steps 2 and 3. Genetic event
 <code>GeneticEvent.GENOTYPE_EVOLVED_EVENT</code> is fired after step 4.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getFittest()"><!-- --></A><H3>
getFittest</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> <B>getFittest</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getBestPerforming()"><!-- --></A><H3>
getBestPerforming</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/Chromosome.html" title="class in org.jgapcustomised">Chromosome</A> <B>getBestPerforming</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getNumberOfChromosomesWithZeroFitnessFromLastGen()"><!-- --></A><H3>
getNumberOfChromosomesWithZeroFitnessFromLastGen</H3>
<PRE>
public int <B>getNumberOfChromosomesWithZeroFitnessFromLastGen</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getParameters()"><!-- --></A><H3>
getParameters</H3>
<PRE>
public <A HREF="../../org/jgapcustomised/SpeciationParms.html" title="class in org.jgapcustomised">SpeciationParms</A> <B>getParameters</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getMaxSpeciesSize()"><!-- --></A><H3>
getMaxSpeciesSize</H3>
<PRE>
public int <B>getMaxSpeciesSize</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getMinSpeciesSize()"><!-- --></A><H3>
getMinSpeciesSize</H3>
<PRE>
public int <B>getMinSpeciesSize</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public java.lang.String <B>toString</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>toString</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD><code>String</code> representation of this <code>Genotype</code> instance.</DL>
</DD>
</DL>
<HR>

<A NAME="randomInitialGenotype(com.anji.util.Properties, org.jgapcustomised.Configuration)"><!-- --></A><H3>
randomInitialGenotype</H3>
<PRE>
public static <A HREF="../../org/jgapcustomised/Genotype.html" title="class in org.jgapcustomised">Genotype</A> <B>randomInitialGenotype</B>(<A HREF="../../com/anji/util/Properties.html" title="class in com.anji.util">Properties</A>&nbsp;props,
                                             <A HREF="../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</A>&nbsp;a_activeConfiguration)
                                      throws <A HREF="../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</A></PRE>
<DL>
<DD>Convenience method that returns a newly constructed Genotype instance configured according to the given
 Configuration instance. The population of Chromosomes will created according to the setup of the sample
 Chromosome in the Configuration object, but the gene values (alleles) will be set to random legal values.
 <p>
 Note that the given Configuration instance must be in a valid state at the time this method is invoked, or an
 InvalidConfigurationException will be thrown.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>a_activeConfiguration</CODE> - 
<DT><B>Returns:</B><DD>A newly constructed Genotype instance.
<DT><B>Throws:</B>
<DD><CODE><A HREF="../../org/jgapcustomised/InvalidConfigurationException.html" title="class in org.jgapcustomised">InvalidConfigurationException</A></CODE> - if the given Configuration instance not in a valid state.</DL>
</DD>
</DL>
<HR>

<A NAME="getAveragePopulationFitness()"><!-- --></A><H3>
getAveragePopulationFitness</H3>
<PRE>
public double <B>getAveragePopulationFitness</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="equals(java.lang.Object)"><!-- --></A><H3>
equals</H3>
<PRE>
public boolean <B>equals</B>(java.lang.Object&nbsp;other)</PRE>
<DL>
<DD>Compares this Genotype against the specified object. The result is true if the argument is an instance of the
 Genotype class, has exactly the same number of chromosomes as the given Genotype, and, for each Chromosome in
 this Genotype, there is an equal chromosome in the given Genotype. The chromosomes do not need to appear in the
 same order within the populations.
<P>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE>equals</CODE> in class <CODE>java.lang.Object</CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The object to compare against.
<DT><B>Returns:</B><DD>true if the objects are the same, false otherwise.</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/Genotype.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../org/jgapcustomised/Gene.html" title="class in org.jgapcustomised"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../org/jgapcustomised/IdFactory.html" title="class in org.jgapcustomised"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../index.html?org/jgapcustomised/Genotype.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Genotype.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
