<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_09) on Tue Jan 29 12:34:55 EST 2013 -->
<title>IntegerAllele</title>
<meta name="date" content="2013-01-29">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="IntegerAllele";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/IntegerAllele.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../org/jgapcustomised/impl/DefaultConfiguration.html" title="class in org.jgapcustomised.impl"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../org/jgapcustomised/impl/Pool.html" title="class in org.jgapcustomised.impl"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised/impl/IntegerAllele.html" target="_top">Frames</a></li>
<li><a href="IntegerAllele.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.jgapcustomised.impl</div>
<h2 title="Class IntegerAllele" class="title">Class IntegerAllele</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">org.jgapcustomised.Allele</a></li>
<li>
<ul class="inheritance">
<li>org.jgapcustomised.impl.IntegerAllele</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd>java.lang.Comparable</dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">IntegerAllele</span>
extends <a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></pre>
<div class="block">A Gene implementation that supports a integer values for its allele. Upper and lower bounds may optionally be
 provided to restrict the range of legal values allowed by this Gene instance.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static long</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#INTEGER_RANGE">INTEGER_RANGE</a></strong></code>
<div class="block">Represents the constant range of values supported by integers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a></code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#m_activeConfiguration">m_activeConfiguration</a></strong></code>
<div class="block">The current active configuration that is in use.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected long</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#m_boundsUnitsToIntegerUnits">m_boundsUnitsToIntegerUnits</a></strong></code>
<div class="block">Stores the number of integer range units that a single bounds-range unit represents.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#m_lowerBounds">m_lowerBounds</a></strong></code>
<div class="block">The lower bounds of values represented by this Gene.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected int</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#m_upperBounds">m_upperBounds</a></strong></code>
<div class="block">The upper bounds of values represented by this Gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.lang.Integer</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#m_value">m_value</a></strong></code>
<div class="block">References the internal integer value (allele) of this Gene.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#PERSISTENT_FIELD_DELIMITER">PERSISTENT_FIELD_DELIMITER</a></strong></code>
<div class="block">Represents the delimiter that is used to separate fields in the persistent representation of IntegerGene
 instances.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#IntegerAllele()">IntegerAllele</a></strong>()</code>
<div class="block">Constructs a new IntegerGene with default settings.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#IntegerAllele(org.jgapcustomised.Configuration)">IntegerAllele</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration)</code>
<div class="block">Constructs a new IntegerGene according to the given active configuration.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#IntegerAllele(org.jgapcustomised.Configuration, int, int)">IntegerAllele</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
             int&nbsp;a_lowerBounds,
             int&nbsp;a_upperBounds)</code>
<div class="block">Constructs a new IntegerGene with the given active configuration and the specified lower and upper bounds for
 values represented by this Gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#IntegerAllele(int, int)">IntegerAllele</a></strong>(int&nbsp;a_lowerBounds,
             int&nbsp;a_upperBounds)</code>
<div class="block">Constructs a new IntegerGene with the specified lower and upper bounds for values (alleles) of this Gene
 instance.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#calculateBoundsUnitsToIntegerUnitsRatio()">calculateBoundsUnitsToIntegerUnitsRatio</a></strong>()</code>
<div class="block">Calculates and sets the m_boundsUnitsToIntegerUnits field based
 
 on the current lower and upper bounds of this IntegerGene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#cleanup()">cleanup</a></strong>()</code>
<div class="block">Executed by the genetic engine when this Gene instance is no
 
 longer needed and should perform any necessary resource cleanup.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#cloneAllele()">cloneAllele</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#distance(org.jgapcustomised.Allele)">distance</a></strong>(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;target)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#getPersistentRepresentation()">getPersistentRepresentation</a></strong>()</code>
<div class="block">Retrieves a string representation of this Gene that includes any
 
 information required to reconstruct it at a later time, such as its
 
 value and internal state.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.Integer</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#getValue()">getValue</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#hashCode()">hashCode</a></strong>()</code>
<div class="block">Retrieves the hash code value for this IntegerGene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#intValue()">intValue</a></strong>()</code>
<div class="block">Retrieves the int value of this Gene, which may be more convenient in
 
 some cases than the more general getAllele() method.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#mapValueToWithinBounds()">mapValueToWithinBounds</a></strong>()</code>
<div class="block">Maps the value of this IntegerGene to within the bounds specified by
 
 the m_upperBounds and m_lowerBounds instance variables.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#newAllele(org.jgapcustomised.Configuration)">newAllele</a></strong>(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration)</code>
<div class="block">Provides an implementation-independent means for creating new Gene instances.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#setToRandomValue(java.util.Random, boolean)">setToRandomValue</a></strong>(java.util.Random&nbsp;a_numberGenerator,
                boolean&nbsp;onlyPerturbFromCurrentValue)</code>
<div class="block">Sets the value (allele) of this Gene to a random Integer value between
 
 the lower and upper bounds (if any) of this Gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#setValue(java.lang.Integer)">setValue</a></strong>(java.lang.Integer&nbsp;a_newValue)</code>
<div class="block">Sets the value (allele) of this Gene to the new given value.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#setValueFromPersistentRepresentation(java.lang.String)">setValueFromPersistentRepresentation</a></strong>(java.lang.String&nbsp;a_representation)</code>
<div class="block">Sets the value and internal state of this Gene from the string
 
 representation returned by a previous invocation of the
 
 getPersistentRepresentation() method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#toString()">toString</a></strong>()</code>
<div class="block">Retrieves a string representation of this IntegerGene's value that
 
 may be useful for display purposes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../org/jgapcustomised/impl/IntegerAllele.html#toXml()">toXml</a></strong>()</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.jgapcustomised.Allele">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.jgapcustomised.<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></h3>
<code><a href="../../../org/jgapcustomised/Allele.html#compareTo(java.lang.Object)">compareTo</a>, <a href="../../../org/jgapcustomised/Allele.html#equals(java.lang.Object)">equals</a>, <a href="../../../org/jgapcustomised/Allele.html#getGene()">getGene</a>, <a href="../../../org/jgapcustomised/Allele.html#getGenes(java.util.Collection)">getGenes</a>, <a href="../../../org/jgapcustomised/Allele.html#getInnovationId()">getInnovationId</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="INTEGER_RANGE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>INTEGER_RANGE</h4>
<pre>protected static final&nbsp;long INTEGER_RANGE</pre>
<div class="block">Represents the constant range of values supported by integers.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../constant-values.html#org.jgapcustomised.impl.IntegerAllele.INTEGER_RANGE">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="PERSISTENT_FIELD_DELIMITER">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PERSISTENT_FIELD_DELIMITER</h4>
<pre>protected static final&nbsp;java.lang.String PERSISTENT_FIELD_DELIMITER</pre>
<div class="block">Represents the delimiter that is used to separate fields in the persistent representation of IntegerGene
 instances.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../constant-values.html#org.jgapcustomised.impl.IntegerAllele.PERSISTENT_FIELD_DELIMITER">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="m_value">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_value</h4>
<pre>protected&nbsp;java.lang.Integer m_value</pre>
<div class="block">References the internal integer value (allele) of this Gene.</div>
</li>
</ul>
<a name="m_upperBounds">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_upperBounds</h4>
<pre>protected&nbsp;int m_upperBounds</pre>
<div class="block">The upper bounds of values represented by this Gene. If not explicitly provided by the user, this should be set
 to Integer.MAX_VALUE.</div>
</li>
</ul>
<a name="m_lowerBounds">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_lowerBounds</h4>
<pre>protected&nbsp;int m_lowerBounds</pre>
<div class="block">The lower bounds of values represented by this Gene. If not explicitly provided by the user, this should be set
 to Integer.MIN_VALUE</div>
</li>
</ul>
<a name="m_boundsUnitsToIntegerUnits">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>m_boundsUnitsToIntegerUnits</h4>
<pre>protected&nbsp;long m_boundsUnitsToIntegerUnits</pre>
<div class="block">Stores the number of integer range units that a single bounds-range unit represents. For example, if the integer
 range is -2 billion to +2 billion and the bounds range is -1 billion to +1 billion, then each unit in the bounds
 range would map to 2 units in the integer range. The value of this variable would therefore be 2. This mapping
 unit is used to map illegal allele values that are outside of the bounds to legal allele values that are within
 the bounds.</div>
</li>
</ul>
<a name="m_activeConfiguration">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>m_activeConfiguration</h4>
<pre>protected&nbsp;<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a> m_activeConfiguration</pre>
<div class="block">The current active configuration that is in use.</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="IntegerAllele()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>IntegerAllele</h4>
<pre>public&nbsp;IntegerAllele()</pre>
<div class="block">Constructs a new IntegerGene with default settings. No bounds will be put into effect for values (alleles) of
 this Gene instance, other than the standard range of integer values.</div>
</li>
</ul>
<a name="IntegerAllele(int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>IntegerAllele</h4>
<pre>public&nbsp;IntegerAllele(int&nbsp;a_lowerBounds,
             int&nbsp;a_upperBounds)</pre>
<div class="block">Constructs a new IntegerGene with the specified lower and upper bounds for values (alleles) of this Gene
 instance.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_lowerBounds</code> - The lowest value that this Gene may possess, inclusive.</dd><dd><code>a_upperBounds</code> - The highest value that this Gene may possess, inclusive.</dd></dl>
</li>
</ul>
<a name="IntegerAllele(org.jgapcustomised.Configuration)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>IntegerAllele</h4>
<pre>public&nbsp;IntegerAllele(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration)</pre>
<div class="block">Constructs a new IntegerGene according to the given active configuration.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_activeConfiguration</code> - The current active configuration.</dd></dl>
</li>
</ul>
<a name="IntegerAllele(org.jgapcustomised.Configuration, int, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>IntegerAllele</h4>
<pre>public&nbsp;IntegerAllele(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration,
             int&nbsp;a_lowerBounds,
             int&nbsp;a_upperBounds)</pre>
<div class="block">Constructs a new IntegerGene with the given active configuration and the specified lower and upper bounds for
 values represented by this Gene.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_activeConfiguration</code> - The current active configuration.</dd><dd><code>a_lowerBounds</code> - The lowest value that this Gene may represent, inclusive.</dd><dd><code>a_upperBounds</code> - The highest value that this Gene may represent, inclusive.</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="toXml()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toXml</h4>
<pre>public&nbsp;java.lang.String&nbsp;toXml()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>XML representation of object</dd></dl>
</li>
</ul>
<a name="distance(org.jgapcustomised.Allele)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>distance</h4>
<pre>public&nbsp;double&nbsp;distance(<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;target)</pre>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../org/jgapcustomised/Allele.html#distance(org.jgapcustomised.Allele)">distance</a></code>&nbsp;in class&nbsp;<code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>target</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>compatibility distance based on intValue; always positive</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../org/jgapcustomised/Allele.html#distance(org.jgapcustomised.Allele)"><code>Allele.distance(Allele)</code></a></dd></dl>
</li>
</ul>
<a name="newAllele(org.jgapcustomised.Configuration)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>newAllele</h4>
<pre>public&nbsp;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;newAllele(<a href="../../../org/jgapcustomised/Configuration.html" title="class in org.jgapcustomised">Configuration</a>&nbsp;a_activeConfiguration)</pre>
<div class="block">Provides an implementation-independent means for creating new Gene instances. The new instance that is created
 and returned should be setup with any implementation-dependent configuration that this Gene instance is setup
 with (aside from the actual value, of course). For example, if this Gene were setup with bounds on its value,
 then the Gene instance returned from this method should also be setup with those same bounds. This is important,
 as the JGAP core will invoke this method on each Gene in the sample Chromosome in order to create each new Gene
 in the same respective gene position for a new Chromosome.
 <p>
 It should be noted that nothing is guaranteed about the actual value of the returned Gene and it should therefore
 be considered to be undefined.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_activeConfiguration</code> - The current active configuration.</dd>
<dt><span class="strong">Returns:</span></dt><dd>A new Gene instance of the same type and with the same setup as this concrete Gene.</dd></dl>
</li>
</ul>
<a name="cloneAllele()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>cloneAllele</h4>
<pre>public&nbsp;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&nbsp;cloneAllele()</pre>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../org/jgapcustomised/Allele.html#cloneAllele()">cloneAllele</a></code>&nbsp;in class&nbsp;<code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>clone</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../org/jgapcustomised/Allele.html#cloneAllele()"><code>Allele.cloneAllele()</code></a></dd></dl>
</li>
</ul>
<a name="setValue(java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setValue</h4>
<pre>public&nbsp;void&nbsp;setValue(java.lang.Integer&nbsp;a_newValue)</pre>
<div class="block">Sets the value (allele) of this Gene to the new given value. This class expects the value to be an Integer
 instance. If the value is above or below the upper or lower bounds, it will be mappped to within the allowable
 range.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_newValue</code> - the new value of this Gene instance.</dd></dl>
</li>
</ul>
<a name="getValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getValue</h4>
<pre>public&nbsp;java.lang.Integer&nbsp;getValue()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>value</dd></dl>
</li>
</ul>
<a name="getPersistentRepresentation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getPersistentRepresentation</h4>
<pre>public&nbsp;java.lang.String&nbsp;getPersistentRepresentation()
                                             throws java.lang.UnsupportedOperationException</pre>
<div class="block">Retrieves a string representation of this Gene that includes any
 
 information required to reconstruct it at a later time, such as its
 
 value and internal state. This string will be used to represent this
 
 Gene in XML persistence. This is an optional method but, if not
 
 implemented, XML persistence and possibly other features will not be
 
 available. An UnsupportedOperationException should be thrown if no
 
 implementation is provided.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>A string representation of this Gene's current state.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.UnsupportedOperationException</code> - to indicate that no implementation
 
             is provided for this method.</dd></dl>
</li>
</ul>
<a name="setValueFromPersistentRepresentation(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setValueFromPersistentRepresentation</h4>
<pre>public&nbsp;void&nbsp;setValueFromPersistentRepresentation(java.lang.String&nbsp;a_representation)
                                          throws <a href="../../../org/jgapcustomised/UnsupportedRepresentationException.html" title="class in org.jgapcustomised">UnsupportedRepresentationException</a></pre>
<div class="block">Sets the value and internal state of this Gene from the string
 
 representation returned by a previous invocation of the
 
 getPersistentRepresentation() method. This is an optional method but,
 
 if not implemented, XML persistence and possibly other features will not
 
 be available. An UnsupportedOperationException should be thrown if no
 
 implementation is provided.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>a_representation</code> - the string representation retrieved from a
 
            prior call to the getPersistentRepresentation()
 
            method.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.UnsupportedOperationException</code> - to indicate that no implementation
 
             is provided for this method.</dd>
<dd><code><a href="../../../org/jgapcustomised/UnsupportedRepresentationException.html" title="class in org.jgapcustomised">UnsupportedRepresentationException</a></code> - if this Gene implementation
 
             does not support the given string representation.</dd></dl>
</li>
</ul>
<a name="intValue()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>intValue</h4>
<pre>public&nbsp;int&nbsp;intValue()</pre>
<div class="block">Retrieves the int value of this Gene, which may be more convenient in
 
 some cases than the more general getAllele() method.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the int value of this Gene.</dd></dl>
</li>
</ul>
<a name="setToRandomValue(java.util.Random, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setToRandomValue</h4>
<pre>public&nbsp;void&nbsp;setToRandomValue(java.util.Random&nbsp;a_numberGenerator,
                    boolean&nbsp;onlyPerturbFromCurrentValue)</pre>
<div class="block">Sets the value (allele) of this Gene to a random Integer value between
 
 the lower and upper bounds (if any) of this Gene.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../org/jgapcustomised/Allele.html#setToRandomValue(java.util.Random, boolean)">setToRandomValue</a></code>&nbsp;in class&nbsp;<code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>a_numberGenerator</code> - The random number generator that should be
 
            used to create any random values. It's important
 
            to use this generator to maintain the user's
 
            flexibility to configure the genetic engine
 
            to use the random number generator of their
 
            choice.</dd><dd><code>onlyPerturbFromCurrentValue</code> - If true then the value should only be perturbed some small amount from the
            current value. If false then the value should be set to a completely new value irrespective of the
            current value.</dd></dl>
</li>
</ul>
<a name="hashCode()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hashCode</h4>
<pre>public&nbsp;int&nbsp;hashCode()</pre>
<div class="block">Retrieves the hash code value for this IntegerGene.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../org/jgapcustomised/Allele.html#hashCode()">hashCode</a></code>&nbsp;in class&nbsp;<code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>this IntegerGene's hash code.</dd><dt><span class="strong">See Also:</span></dt><dd><code>Object.hashCode()</code></dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;java.lang.String&nbsp;toString()</pre>
<div class="block">Retrieves a string representation of this IntegerGene's value that
 
 may be useful for display purposes.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../org/jgapcustomised/Allele.html#toString()">toString</a></code>&nbsp;in class&nbsp;<code><a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a string representation of this IntegerGene's value.</dd><dt><span class="strong">See Also:</span></dt><dd><code>Object.toString()</code></dd></dl>
</li>
</ul>
<a name="cleanup()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>cleanup</h4>
<pre>public&nbsp;void&nbsp;cleanup()</pre>
<div class="block">Executed by the genetic engine when this Gene instance is no
 
 longer needed and should perform any necessary resource cleanup.</div>
</li>
</ul>
<a name="mapValueToWithinBounds()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>mapValueToWithinBounds</h4>
<pre>protected&nbsp;void&nbsp;mapValueToWithinBounds()</pre>
<div class="block">Maps the value of this IntegerGene to within the bounds specified by
 
 the m_upperBounds and m_lowerBounds instance variables. The value's
 
 relative position within the integer range will be preserved within the
 
 bounds range (in other words, if the value is about halfway between the
 
 integer max and min, then the resulting value will be about halfway
 
 between the upper bounds and lower bounds). If the value is null or
 
 is already within the bounds, it will be left unchanged.</div>
</li>
</ul>
<a name="calculateBoundsUnitsToIntegerUnitsRatio()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>calculateBoundsUnitsToIntegerUnitsRatio</h4>
<pre>protected&nbsp;void&nbsp;calculateBoundsUnitsToIntegerUnitsRatio()</pre>
<div class="block">Calculates and sets the m_boundsUnitsToIntegerUnits field based
 
 on the current lower and upper bounds of this IntegerGene. For example,
 
 if the integer range is -2 billion to +2 billion and the bounds range
 
 is -1 billion to +1 billion, then each unit in the bounds range would
 
 map to 2 units in the integer range. The m_boundsUnitsToIntegerUnits
 
 field would therefore be 2. This mapping unit is used to map illegal
 
 allele values that are outside of the bounds to legal allele values that
 
 are within the bounds.</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/IntegerAllele.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../org/jgapcustomised/impl/DefaultConfiguration.html" title="class in org.jgapcustomised.impl"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../org/jgapcustomised/impl/Pool.html" title="class in org.jgapcustomised.impl"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?org/jgapcustomised/impl/IntegerAllele.html" target="_top">Frames</a></li>
<li><a href="IntegerAllele.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
