<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Fri Mar 22 08:30:59 EST 2013 -->
<TITLE>
NeatChromosomeUtility
</TITLE>

<META NAME="date" CONTENT="2013-03-22">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="NeatChromosomeUtility";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/NeatChromosomeUtility.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../com/anji/neat/NeatActivator.html" title="class in com.anji.neat"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?com/anji/neat/NeatChromosomeUtility.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NeatChromosomeUtility.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
com.anji.neat</FONT>
<BR>
Class NeatChromosomeUtility</H2>
<PRE>
java.lang.Object
  <IMG SRC="../../../resources/inherit.gif" ALT="extended by "><B>com.anji.neat.NeatChromosomeUtility</B>
</PRE>
<HR>
<DL>
<DT><PRE>public class <B>NeatChromosomeUtility</B><DT>extends java.lang.Object</DL>
</PRE>

<P>
Utility class capturing functionality pertaining to NEAT neuron and connection genes.
<P>

<P>
<DL>
<DT><B>Author:</B></DT>
  <DD>Philip Tucker</DD>
</DL>
<HR>

<P>

<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#NeatChromosomeUtility()">NeatChromosomeUtility</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForDestNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForDestNeurons</A></B>(java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connAlleles,
                                       java.util.Collection&nbsp;destNeuronInnovationIds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForSrcNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForSrcNeurons</A></B>(java.util.Collection&nbsp;connAlleles,
                                      java.util.Collection&nbsp;srcNeuronInnovationIds)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionList(java.util.Collection)">getConnectionList</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns all connection genes in <code>genes</code> as <code>List</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionMap(java.util.Set)">getConnectionMap</A></B>(java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;returns all connections in <code>alleles</code> as <code>SortedMap</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection)">getNeuronList</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return all neurons in <code>genes</code> as <code>List</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection, com.anji.neat.NeuronType)">getNeuronList</A></B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles,
              <A HREF="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</A>&nbsp;type)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronMap(java.util.Collection)">getNeuronMap</A></B>(java.util.Collection&nbsp;alleles)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return all neurons in <code>alleles</code> as <code>SortedMap</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronMap(java.util.Collection, com.anji.neat.NeuronType)">getNeuronMap</A></B>(java.util.Collection&nbsp;alleles,
             <A HREF="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</A>&nbsp;type)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if type == null, returns all neurons in <code>alleles</code>; otherwise, returns only neurons of
 <code>type</code></TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;java.util.List&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#initAlleles(short, short, short, com.anji.neat.NeatConfiguration, boolean)">initAlleles</A></B>(short&nbsp;numInputs,
            short&nbsp;numHidden,
            short&nbsp;numOutputs,
            <A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
            boolean&nbsp;fullyConnected)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;constructs genes for neural net with specified input and output dimension, JGAP/NEAT configuration, and amount of
 connectivity</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)">neuronsAreConnected</A></B>(java.lang.Long&nbsp;srcNeuronId,
                    java.lang.Long&nbsp;destNeuronId,
                    java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connGenes)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;non-recursive starting point for recursive search</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#newSampleChromosomeMaterial(short, short, short, com.anji.neat.NeatConfiguration, boolean)">newSampleChromosomeMaterial</A></B>(short&nbsp;newNumInputs,
                            short&nbsp;newNumHidden,
                            short&nbsp;newNumOutputs,
                            <A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
                            boolean&nbsp;fullyConnected)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;factory method to construct chromosome material for neural net with specified input and output dimension,
 JGAP/NEAT configuration, and amount of connectivity</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.Object</B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="NeatChromosomeUtility()"><!-- --></A><H3>
NeatChromosomeUtility</H3>
<PRE>
public <B>NeatChromosomeUtility</B>()</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="newSampleChromosomeMaterial(short, short, short, com.anji.neat.NeatConfiguration, boolean)"><!-- --></A><H3>
newSampleChromosomeMaterial</H3>
<PRE>
public static <A HREF="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</A> <B>newSampleChromosomeMaterial</B>(short&nbsp;newNumInputs,
                                                             short&nbsp;newNumHidden,
                                                             short&nbsp;newNumOutputs,
                                                             <A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
                                                             boolean&nbsp;fullyConnected)</PRE>
<DL>
<DD>factory method to construct chromosome material for neural net with specified input and output dimension,
 JGAP/NEAT configuration, and amount of connectivity
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>newNumInputs</CODE> - <DD><CODE>newNumHidden</CODE> - <DD><CODE>newNumOutputs</CODE> - <DD><CODE>config</CODE> - <DD><CODE>fullyConnected</CODE> - all layers fully connected if true, not connected at all otherwise
<DT><B>Returns:</B><DD>ChromosomeMaterial</DL>
</DD>
</DL>
<HR>

<A NAME="extractConnectionAllelesForDestNeurons(java.util.Collection, java.util.Collection)"><!-- --></A><H3>
extractConnectionAllelesForDestNeurons</H3>
<PRE>
public static java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt; <B>extractConnectionAllelesForDestNeurons</B>(java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connAlleles,
                                                                                            java.util.Collection&nbsp;destNeuronInnovationIds)</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>connAlleles</CODE> - <code>Collection</code> contains <code>ConnectionA</code> llele objects<DD><CODE>destNeuronInnovationIds</CODE> - <code>Collection</code> contains Long objects
<DT><B>Returns:</B><DD><code>Collection</code> containing <code>ConnectionAllele</code> objects, those in
         <code>connAlleles</code> whose destination neuron is in <code>destNeuronInnovationIds</code></DL>
</DD>
</DL>
<HR>

<A NAME="extractConnectionAllelesForSrcNeurons(java.util.Collection, java.util.Collection)"><!-- --></A><H3>
extractConnectionAllelesForSrcNeurons</H3>
<PRE>
public static java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt; <B>extractConnectionAllelesForSrcNeurons</B>(java.util.Collection&nbsp;connAlleles,
                                                                                           java.util.Collection&nbsp;srcNeuronInnovationIds)</PRE>
<DL>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>connAlleles</CODE> - <code>Collection</code> contains ConnectionGene objects<DD><CODE>srcNeuronInnovationIds</CODE> - <code>Collection</code> contains Long objects
<DT><B>Returns:</B><DD><code>Collection</code> containing ConnectionGene objects, those in connGenes whose source neuron is
         srcNeuronGene</DL>
</DD>
</DL>
<HR>

<A NAME="initAlleles(short, short, short, com.anji.neat.NeatConfiguration, boolean)"><!-- --></A><H3>
initAlleles</H3>
<PRE>
public static java.util.List&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt; <B>initAlleles</B>(short&nbsp;numInputs,
                                                 short&nbsp;numHidden,
                                                 short&nbsp;numOutputs,
                                                 <A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</A>&nbsp;config,
                                                 boolean&nbsp;fullyConnected)</PRE>
<DL>
<DD>constructs genes for neural net with specified input and output dimension, JGAP/NEAT configuration, and amount of
 connectivity
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>numInputs</CODE> - <DD><CODE>numHidden</CODE> - <DD><CODE>numOutputs</CODE> - <DD><CODE>config</CODE> - <DD><CODE>fullyConnected</CODE> - all layers fully connected if true, not connected at all otherwise
<DT><B>Returns:</B><DD>List contains Allele objects</DL>
</DD>
</DL>
<HR>

<A NAME="getNeuronMap(java.util.Collection)"><!-- --></A><H3>
getNeuronMap</H3>
<PRE>
public static java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt; <B>getNeuronMap</B>(java.util.Collection&nbsp;alleles)</PRE>
<DL>
<DD>return all neurons in <code>alleles</code> as <code>SortedMap</code>
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleles</CODE> - <code>Collection</code> contains <code>Allele</code> objects
<DT><B>Returns:</B><DD><code>SortedMap</code> containing key <code>Long</code> innovation id, value <code>NeuronGene</code>
         objects<DT><B>See Also:</B><DD><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronMap(java.util.Collection, com.anji.neat.NeuronType)"><CODE>getNeuronMap(Collection, NeuronType)</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="getNeuronList(java.util.Collection)"><!-- --></A><H3>
getNeuronList</H3>
<PRE>
public static java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt; <B>getNeuronList</B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</PRE>
<DL>
<DD>return all neurons in <code>genes</code> as <code>List</code>
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleles</CODE> - <code>Collection</code> contains <code>Allele</code> objects
<DT><B>Returns:</B><DD><code>List</code> containing <code>NeuronGene</code> objects<DT><B>See Also:</B><DD><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection, com.anji.neat.NeuronType)"><CODE>getNeuronList(Collection, NeuronType)</CODE></A></DL>
</DD>
</DL>
<HR>

<A NAME="getNeuronMap(java.util.Collection, com.anji.neat.NeuronType)"><!-- --></A><H3>
getNeuronMap</H3>
<PRE>
public static java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt; <B>getNeuronMap</B>(java.util.Collection&nbsp;alleles,
                                                                            <A HREF="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</A>&nbsp;type)</PRE>
<DL>
<DD>if type == null, returns all neurons in <code>alleles</code>; otherwise, returns only neurons of
 <code>type</code>
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleles</CODE> - <code>Collection</code> contains <code>Allele</code> objects<DD><CODE>type</CODE> - 
<DT><B>Returns:</B><DD>SortedMap contains key Long innovation id, value NeuronGene objects</DL>
</DD>
</DL>
<HR>

<A NAME="getNeuronList(java.util.Collection, com.anji.neat.NeuronType)"><!-- --></A><H3>
getNeuronList</H3>
<PRE>
public static java.util.List&lt;<A HREF="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</A>&gt; <B>getNeuronList</B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles,
                                                         <A HREF="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</A>&nbsp;type)</PRE>
<DL>
<DD>if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleles</CODE> - <code>Collection</code> contains gene objects<DD><CODE>type</CODE> - 
<DT><B>Returns:</B><DD><code>List</code> contains <code>NeuronAllele</code> objects</DL>
</DD>
</DL>
<HR>

<A NAME="getConnectionMap(java.util.Set)"><!-- --></A><H3>
getConnectionMap</H3>
<PRE>
public static java.util.SortedMap&lt;java.lang.Long,<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt; <B>getConnectionMap</B>(java.util.Set&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</PRE>
<DL>
<DD>returns all connections in <code>alleles</code> as <code>SortedMap</code>
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleles</CODE> - <code>SortedSet</code> contains <code>Allele</code> objects
<DT><B>Returns:</B><DD><code>SortedMap</code> containing key <code>Long</code> innovation id, value
         <code>ConnectionAllele</code> objects</DL>
</DD>
</DL>
<HR>

<A NAME="getConnectionList(java.util.Collection)"><!-- --></A><H3>
getConnectionList</H3>
<PRE>
public static java.util.List&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt; <B>getConnectionList</B>(java.util.Collection&lt;<A HREF="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</A>&gt;&nbsp;alleles)</PRE>
<DL>
<DD>returns all connection genes in <code>genes</code> as <code>List</code>
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>alleles</CODE> - <code>Collection</code> contains gene objects
<DT><B>Returns:</B><DD><code>List</code> containing <code>ConnectionGene</code> objects</DL>
</DD>
</DL>
<HR>

<A NAME="neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)"><!-- --></A><H3>
neuronsAreConnected</H3>
<PRE>
public static boolean <B>neuronsAreConnected</B>(java.lang.Long&nbsp;srcNeuronId,
                                          java.lang.Long&nbsp;destNeuronId,
                                          java.util.Collection&lt;<A HREF="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</A>&gt;&nbsp;connGenes)</PRE>
<DL>
<DD>non-recursive starting point for recursive search
<P>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>srcNeuronId</CODE> - <DD><CODE>destNeuronId</CODE> - <DD><CODE>connGenes</CODE> - 
<DT><B>Returns:</B><DD>true if <code>srcNeuronId</code> and <code>destNeuronId</code> are connected<DT><B>See Also:</B><DD><A HREF="../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection, java.util.Set)"><CODE>neuronsAreConnected(Long, Long, Collection, Set)</CODE></A></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/NeatChromosomeUtility.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../index-files/index-1.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../com/anji/neat/NeatActivator.html" title="class in com.anji.neat"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../index.html?com/anji/neat/NeatChromosomeUtility.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NeatChromosomeUtility.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>

</BODY>
</HTML>
