<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.6.0_27) on Mon Sep 30 15:20:30 EST 2013 -->
<title>NeatChromosomeUtility</title>
<meta name="date" content="2013-09-30">
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="NeatChromosomeUtility";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NeatChromosomeUtility.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../com/anji/neat/NeatActivator.html" title="class in com.anji.neat"><span class="strong">PREV CLASS</span></a></li>
<li><a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat"><span class="strong">NEXT CLASS</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?com/anji/neat/NeatChromosomeUtility.html" target="_top">FRAMES</a></li>
<li><a href="NeatChromosomeUtility.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>SUMMARY:&nbsp;</li>
<li>NESTED&nbsp;|&nbsp;</li>
<li>FIELD&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">METHOD</a></li>
</ul>
<ul class="subNavList">
<li>DETAIL:&nbsp;</li>
<li>FIELD&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">METHOD</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<p class="subTitle">com.anji.neat</p>
<h2 title="Class NeatChromosomeUtility" class="title">Class NeatChromosomeUtility</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>com.anji.neat.NeatChromosomeUtility</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <strong>NeatChromosomeUtility</strong>
extends java.lang.Object</pre>
<div class="block">Utility class capturing functionality pertaining to NEAT neuron and connection genes.</div>
<dl><dt><span class="strong">Author:</span></dt>
  <dd>Philip Tucker</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#NeatChromosomeUtility()">NeatChromosomeUtility</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#connectionAllowed(com.anji.neat.NeatConfiguration, java.lang.Long, java.lang.Long, java.util.Collection)">connectionAllowed</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                 java.lang.Long&nbsp;srcID,
                 java.lang.Long&nbsp;destID,
                 java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;conns)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#connectionAllowed(com.anji.neat.NeatConfiguration, com.anji.neat.NeuronAllele, com.anji.neat.NeuronAllele, java.util.Collection)">connectionAllowed</a></strong>(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                 <a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&nbsp;src,
                 <a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&nbsp;dest,
                 java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;conns)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForDestNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForDestNeurons</a></strong>(java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;connAlleles,
                                      java.util.Collection&nbsp;destNeuronInnovationIds)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#extractConnectionAllelesForSrcNeurons(java.util.Collection, java.util.Collection)">extractConnectionAllelesForSrcNeurons</a></strong>(java.util.Collection&nbsp;connAlleles,
                                     java.util.Collection&nbsp;srcNeuronInnovationIds)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionList(java.util.Collection)">getConnectionList</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</code>
<div class="block">returns all connection genes in <code>genes</code> as <code>List</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getConnectionMap(java.util.Set)">getConnectionMap</a></strong>(java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</code>
<div class="block">returns all connections in <code>alleles</code> as <code>SortedMap</code></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection)">getNeuronList</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</code>
<div class="block">return all neurons in <code>genes</code> as <code>List</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection, com.anji.neat.NeuronType)">getNeuronList</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles,
             <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>&nbsp;type)</code>
<div class="block">if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronListForTypes(java.util.Collection, com.anji.neat.NeuronType[])">getNeuronListForTypes</a></strong>(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles,
                     <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>[]&nbsp;types)</code>
<div class="block">if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronMap(java.util.Collection)">getNeuronMap</a></strong>(java.util.Collection&nbsp;alleles)</code>
<div class="block">return all neurons in <code>alleles</code> as <code>SortedMap</code></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronMap(java.util.Collection, com.anji.neat.NeuronType)">getNeuronMap</a></strong>(java.util.Collection&nbsp;alleles,
            <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>&nbsp;type)</code>
<div class="block">if type == null, returns all neurons in <code>alleles</code>; otherwise, returns only neurons of
 <code>type</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#initAlleles(short, short, short, com.anji.neat.NeatConfiguration, boolean)">initAlleles</a></strong>(short&nbsp;numInputs,
           short&nbsp;numHidden,
           short&nbsp;numOutputs,
           <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
           boolean&nbsp;fullyConnected)</code>
<div class="block">constructs genes for neural net with specified input and output dimension, JGAP/NEAT configuration, and amount of
 connectivity</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)">neuronsAreConnected</a></strong>(java.lang.Long&nbsp;srcNeuronId,
                   java.lang.Long&nbsp;destNeuronId,
                   java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;connGenes)</code>
<div class="block">non-recursive starting point for recursive search</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a></code></td>
<td class="colLast"><code><strong><a href="../../../com/anji/neat/NeatChromosomeUtility.html#newSampleChromosomeMaterial(short, short, short, com.anji.neat.NeatConfiguration, boolean)">newSampleChromosomeMaterial</a></strong>(short&nbsp;newNumInputs,
                           short&nbsp;newNumHidden,
                           short&nbsp;newNumOutputs,
                           <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                           boolean&nbsp;fullyConnected)</code>
<div class="block">factory method to construct chromosome material for neural net with specified input and output dimension,
 JGAP/NEAT configuration, and amount of connectivity</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="NeatChromosomeUtility()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>NeatChromosomeUtility</h4>
<pre>public&nbsp;NeatChromosomeUtility()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="newSampleChromosomeMaterial(short, short, short, com.anji.neat.NeatConfiguration, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>newSampleChromosomeMaterial</h4>
<pre>public static&nbsp;<a href="../../../org/jgapcustomised/ChromosomeMaterial.html" title="class in org.jgapcustomised">ChromosomeMaterial</a>&nbsp;newSampleChromosomeMaterial(short&nbsp;newNumInputs,
                                             short&nbsp;newNumHidden,
                                             short&nbsp;newNumOutputs,
                                             <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                                             boolean&nbsp;fullyConnected)</pre>
<div class="block">factory method to construct chromosome material for neural net with specified input and output dimension,
 JGAP/NEAT configuration, and amount of connectivity</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>newNumInputs</code> - </dd><dd><code>newNumHidden</code> - </dd><dd><code>newNumOutputs</code> - </dd><dd><code>config</code> - </dd><dd><code>fullyConnected</code> - all layers fully connected if true, not connected at all otherwise</dd>
<dt><span class="strong">Returns:</span></dt><dd>ChromosomeMaterial</dd></dl>
</li>
</ul>
<a name="extractConnectionAllelesForDestNeurons(java.util.Collection, java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractConnectionAllelesForDestNeurons</h4>
<pre>public static&nbsp;java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;extractConnectionAllelesForDestNeurons(java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;connAlleles,
                                                                            java.util.Collection&nbsp;destNeuronInnovationIds)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connAlleles</code> - <code>Collection</code> contains <code>ConnectionA</code> llele objects</dd><dd><code>destNeuronInnovationIds</code> - <code>Collection</code> contains Long objects</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>Collection</code> containing <code>ConnectionAllele</code> objects, those in
         <code>connAlleles</code> whose destination neuron is in <code>destNeuronInnovationIds</code></dd></dl>
</li>
</ul>
<a name="extractConnectionAllelesForSrcNeurons(java.util.Collection, java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extractConnectionAllelesForSrcNeurons</h4>
<pre>public static&nbsp;java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;extractConnectionAllelesForSrcNeurons(java.util.Collection&nbsp;connAlleles,
                                                                           java.util.Collection&nbsp;srcNeuronInnovationIds)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connAlleles</code> - <code>Collection</code> contains ConnectionGene objects</dd><dd><code>srcNeuronInnovationIds</code> - <code>Collection</code> contains Long objects</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>Collection</code> containing ConnectionGene objects, those in connGenes whose source neuron is
         srcNeuronGene</dd></dl>
</li>
</ul>
<a name="initAlleles(short, short, short, com.anji.neat.NeatConfiguration, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>initAlleles</h4>
<pre>public static&nbsp;java.util.List&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;initAlleles(short&nbsp;numInputs,
                                 short&nbsp;numHidden,
                                 short&nbsp;numOutputs,
                                 <a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                                 boolean&nbsp;fullyConnected)</pre>
<div class="block">constructs genes for neural net with specified input and output dimension, JGAP/NEAT configuration, and amount of
 connectivity</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>numInputs</code> - </dd><dd><code>numHidden</code> - </dd><dd><code>numOutputs</code> - </dd><dd><code>config</code> - </dd><dd><code>fullyConnected</code> - all layers fully connected if true, not connected at all otherwise</dd>
<dt><span class="strong">Returns:</span></dt><dd>List contains Allele objects</dd></dl>
</li>
</ul>
<a name="getNeuronMap(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronMap</h4>
<pre>public static&nbsp;java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;getNeuronMap(java.util.Collection&nbsp;alleles)</pre>
<div class="block">return all neurons in <code>alleles</code> as <code>SortedMap</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>Collection</code> contains <code>Allele</code> objects</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>SortedMap</code> containing key <code>Long</code> innovation id, value <code>NeuronGene</code>
         objects</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronMap(java.util.Collection, com.anji.neat.NeuronType)"><code>getNeuronMap(Collection, NeuronType)</code></a></dd></dl>
</li>
</ul>
<a name="getNeuronList(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronList</h4>
<pre>public static&nbsp;java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;getNeuronList(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</pre>
<div class="block">return all neurons in <code>genes</code> as <code>List</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>Collection</code> contains <code>Allele</code> objects</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>List</code> containing <code>NeuronGene</code> objects</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../com/anji/neat/NeatChromosomeUtility.html#getNeuronList(java.util.Collection, com.anji.neat.NeuronType)"><code>getNeuronList(Collection, NeuronType)</code></a></dd></dl>
</li>
</ul>
<a name="getNeuronMap(java.util.Collection, com.anji.neat.NeuronType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronMap</h4>
<pre>public static&nbsp;java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;getNeuronMap(java.util.Collection&nbsp;alleles,
                                                            <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>&nbsp;type)</pre>
<div class="block">if type == null, returns all neurons in <code>alleles</code>; otherwise, returns only neurons of
 <code>type</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>Collection</code> contains <code>Allele</code> objects</dd><dd><code>type</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>SortedMap contains key Long innovation id, value NeuronGene objects</dd></dl>
</li>
</ul>
<a name="getNeuronList(java.util.Collection, com.anji.neat.NeuronType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronList</h4>
<pre>public static&nbsp;java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;getNeuronList(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles,
                                         <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>&nbsp;type)</pre>
<div class="block">if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>Collection</code> contains gene objects</dd><dd><code>type</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd><code>List</code> contains <code>NeuronAllele</code> objects</dd></dl>
</li>
</ul>
<a name="getNeuronListForTypes(java.util.Collection, com.anji.neat.NeuronType[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNeuronListForTypes</h4>
<pre>public static&nbsp;java.util.List&lt;<a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&gt;&nbsp;getNeuronListForTypes(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles,
                                                 <a href="../../../com/anji/neat/NeuronType.html" title="class in com.anji.neat">NeuronType</a>[]&nbsp;types)</pre>
<div class="block">if type == null, returns all neuron genes in <code>genes</code>; otherwise, returns only neuron genes of type</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>Collection</code> contains gene objects</dd><dd><code>types</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd><code>List</code> contains <code>NeuronAllele</code> objects</dd></dl>
</li>
</ul>
<a name="getConnectionMap(java.util.Set)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getConnectionMap</h4>
<pre>public static&nbsp;java.util.SortedMap&lt;java.lang.Long,<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;getConnectionMap(java.util.Set&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</pre>
<div class="block">returns all connections in <code>alleles</code> as <code>SortedMap</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>SortedSet</code> contains <code>Allele</code> objects</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>SortedMap</code> containing key <code>Long</code> innovation id, value
         <code>ConnectionAllele</code> objects</dd></dl>
</li>
</ul>
<a name="getConnectionList(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getConnectionList</h4>
<pre>public static&nbsp;java.util.List&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;getConnectionList(java.util.Collection&lt;<a href="../../../org/jgapcustomised/Allele.html" title="class in org.jgapcustomised">Allele</a>&gt;&nbsp;alleles)</pre>
<div class="block">returns all connection genes in <code>genes</code> as <code>List</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alleles</code> - <code>Collection</code> contains gene objects</dd>
<dt><span class="strong">Returns:</span></dt><dd><code>List</code> containing <code>ConnectionGene</code> objects</dd></dl>
</li>
</ul>
<a name="neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>neuronsAreConnected</h4>
<pre>public static&nbsp;boolean&nbsp;neuronsAreConnected(java.lang.Long&nbsp;srcNeuronId,
                          java.lang.Long&nbsp;destNeuronId,
                          java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;connGenes)</pre>
<div class="block">non-recursive starting point for recursive search</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>srcNeuronId</code> - </dd><dd><code>destNeuronId</code> - </dd><dd><code>connGenes</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>true if <code>srcNeuronId</code> and <code>destNeuronId</code> are connected</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection, java.util.Set)"><code>neuronsAreConnected(Long, Long, Collection, Set)</code></a></dd></dl>
</li>
</ul>
<a name="connectionAllowed(com.anji.neat.NeatConfiguration, com.anji.neat.NeuronAllele, com.anji.neat.NeuronAllele, java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>connectionAllowed</h4>
<pre>public static&nbsp;boolean&nbsp;connectionAllowed(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                        <a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&nbsp;src,
                        <a href="../../../com/anji/neat/NeuronAllele.html" title="class in com.anji.neat">NeuronAllele</a>&nbsp;dest,
                        java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;conns)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>config</code> - </dd><dd><code>src</code> - </dd><dd><code>dest</code> - </dd><dd><code>conns</code> - <code>SortedMap</code> contains key <code>Long</code> id, value <code>ConnectionAllele</code>
            objects</dd>
<dt><span class="strong">Returns:</span></dt><dd>true of connection between <code>src</code> and <code>dest</code> is allowed according to recurrency
         policy; false otherwise.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)"><code>neuronsAreConnected(Long, Long, Collection)</code></a></dd></dl>
</li>
</ul>
<a name="connectionAllowed(com.anji.neat.NeatConfiguration, java.lang.Long, java.lang.Long, java.util.Collection)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>connectionAllowed</h4>
<pre>public static&nbsp;boolean&nbsp;connectionAllowed(<a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat">NeatConfiguration</a>&nbsp;config,
                        java.lang.Long&nbsp;srcID,
                        java.lang.Long&nbsp;destID,
                        java.util.Collection&lt;<a href="../../../com/anji/neat/ConnectionAllele.html" title="class in com.anji.neat">ConnectionAllele</a>&gt;&nbsp;conns)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>config</code> - </dd><dd><code>srcID</code> - innovation ID</dd><dd><code>destID</code> - innovation ID</dd><dd><code>conns</code> - <code>SortedMap</code> contains key <code>Long</code> id, value <code>ConnectionAllele</code>
            objects</dd>
<dt><span class="strong">Returns:</span></dt><dd>true of connection between <code>src</code> and <code>dest</code> is allowed according to recurrency
         policy; false otherwise.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../com/anji/neat/NeatChromosomeUtility.html#neuronsAreConnected(java.lang.Long, java.lang.Long, java.util.Collection)"><code>neuronsAreConnected(Long, Long, Collection)</code></a></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/NeatChromosomeUtility.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../com/anji/neat/NeatActivator.html" title="class in com.anji.neat"><span class="strong">PREV CLASS</span></a></li>
<li><a href="../../../com/anji/neat/NeatConfiguration.html" title="class in com.anji.neat"><span class="strong">NEXT CLASS</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?com/anji/neat/NeatChromosomeUtility.html" target="_top">FRAMES</a></li>
<li><a href="NeatChromosomeUtility.html" target="_top">NO FRAMES</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>SUMMARY:&nbsp;</li>
<li>NESTED&nbsp;|&nbsp;</li>
<li>FIELD&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">METHOD</a></li>
</ul>
<ul class="subNavList">
<li>DETAIL:&nbsp;</li>
<li>FIELD&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">CONSTR</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">METHOD</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
